<div class="container">

<table style="width: 100%;"><tr>
<td>vcf2txt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Covert VCF File to A Data Frame
</h2>

<h3>Description</h3>

<p>It parses a VCF file and extract necessary information for CNV analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">vcf2txt(vcf.file, normal.col = 10, tumor.col = 11, MQ.cutoff = 30)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf.file</code></td>
<td>
<p>vcf file name.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normal.col</code></td>
<td>
<p>the number of the column in which the genotype
and read depth information of normal tissue are located in the vcf file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tumor.col</code></td>
<td>
<p>the number of the column in which the genotype
and read depth information of tumor tissue are located in the vcf file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MQ.cutoff</code></td>
<td>
<p>the minimum criterion of mapping quality.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that the first 9 columns in vcf file are mandatory, followed by the
information for called variant starting from the 10th column.
</p>


<h3>Value</h3>

<p>A data frame of detailed information about each variant, including 
chrosome position, reference and alternative alleles, genotype and read depth 
carrying reference and alternative alleles for normal and tumor respectively.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Danecek, P., Auton, A., Abecasis, G., Albers, C. A., Banks, E., DePristo, M. A., 
Handsaker, R. E., Lunter, G., Marth, G. T., Sherry, S. T., et al. (2011) 
The variant call format and VCFtools. <em>Bioinformatics</em>, <b>27</b>:2156â€“2158.
</p>
<p>http://www.1000genomes.org/node/101
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## an example VCF file from WES
## download WES_example.vcf.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/WES_example.vcf.gz"
tryCatch({download.file(url=url, destfile="WES_example.vcf.gz")
         }, error = function(e) {
          download.file(url=url, destfile="WES_example.vcf.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

## convert VCf file to a data frame
vcf_table &lt;- vcf2txt(vcf.file="WES_example.vcf.gz", normal.col=9+1, tumor.col=9+2)

## see how vcf_table looks like
## download vcf_table.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/vcf_table.txt.gz"
tryCatch({download.file(url=url, destfile="vcf_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="vcf_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

vcf_table &lt;- read.delim(file="vcf_table.txt.gz", as.is=TRUE)
head(vcf_table)

## End(Not run)

</code></pre>


</div>