<div class="container">

<table style="width: 100%;"><tr>
<td>SNP.CNV</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
CNV Analysis Pipeline for SNP array Data
</h2>

<h3>Description</h3>

<p>All analysis steps are integrate into a pipeline. The results, 
including visualization plots are placed in a directory as specified by user.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SNP.CNV(snp, output.dir, sample.id, 
    do.GC.adjust = FALSE,
    gc.file = system.file("extdata","GC_1kb_hg19.txt.gz",package="saasCNV"),
    min.chr.probe = 100, min.snps = 10, 
    joint.segmentation.pvalue.cutoff = 1e-04, max.chpts = 30, 
    do.merge = TRUE, use.null.data = TRUE, 
    num.perm = 1000, maxL = NULL, 
    merge.pvalue.cutoff = 0.05, 
    do.cnvcall.on.merge = TRUE, 
    cnvcall.pvalue.cutoff = 0.05, 
    do.boundary.refine = FALSE, 
    do.plot = TRUE, cex = 0.3, 
    ref.num.probe = NULL, 
    do.gene.anno = FALSE, 
    gene.anno.file = NULL, 
    seed = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snp</code></td>
<td>
<p>a data frame constructed from a text file with LRR and BAF information. 

</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.dir</code></td>
<td>
<p>the directory to which all the results will be located.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.id</code></td>
<td>
<p>sample ID to be displayed in the data frame of the results and 
the title of some diagnosis plots.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.GC.adjust</code></td>
<td>
<p>logical. If GC content adjustment on <code>log2ratio</code> to be carried out. 
Default is <code>FALSE</code>. See <code>GC.adjust</code> for details.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gc.file</code></td>
<td>
<p>the location of tab-delimit file with GC content (averaged per 1kb window) 
information. See <code>GC.adjust</code> for details.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.chr.probe</code></td>
<td>
<p>the minimum number of probes tagging a chromosome for it 
to be passed to the subsequent analysis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.snps</code></td>
<td>
<p>the minimum number of probes a segment needs to span.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>joint.segmentation.pvalue.cutoff</code></td>
<td>
<p>the p-value cut-off one (or a pair) of change points to
be determined as significant in each cycle of joint segmentation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.chpts</code></td>
<td>
<p>the maximum number of change points to be detected for each 
chromosome.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.merge</code></td>
<td>
<p>logical. If segments merging step to be carried out. 
Default is <code>TRUE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.null.data</code></td>
<td>
<p>logical. If only data for probes located in normal copy 
segments to be used for bootstrapping. Default is <code>TRUE</code>. If a more 
aggressive merging is needed, it can be switched to <code>FALSE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.perm</code></td>
<td>
<p>the number of replicates drawn by bootstrap.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxL</code></td>
<td>
<p>integer. The maximum length in terms of number of probes a bootstrapped segment
may span. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically specified 
as 1/100 of the number of data points.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>merge.pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for merging. If the empirical p-value is 
greater than the cut-off value, the two adjacent segments under consideration will 
be merged.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.cnvcall.on.merge</code></td>
<td>
<p>logical. If CNV call to be done for the segments after 
merging step. Default is <code>TRUE</code>. If <code>TRUE</code>, CNV call will be done on the 
segments resulting directly from joint segmentation without merging step.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cnvcall.pvalue.cutoff</code></td>
<td>
<p>a p-value cut-off for CNV calling.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.boundary.refine</code></td>
<td>
<p>logical. If the segment boundaries based on the grid of 
heterozygous probes to be refined by all probes with LRR data. Default is <code>FALSE</code>. 
We do not recommend to perform this step except in the case that the segment 
boundaries need to be aligned well on the same grid of probes for downstream analysis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.plot</code></td>
<td>
<p>logical. If diagnosis plots to be output. Default is <code>TRUE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>a numerical value giving the amount by which plotting text and 
symbols should be magnified relative to the default. It can be adjusted 
in order to make the plot legible.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.num.probe</code></td>
<td>
<p>integer. The reference number of probes against which 
a segment is compared in order to determine the cex of the segment 
to be displayed. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically 
specified as 1/100 of the number of data points.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.gene.anno</code></td>
<td>
<p>logical. If gene annotation step to be performed. Default is <code>FALSE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.anno.file</code></td>
<td>
<p>a tab-delimited file containing gene annotation information. 
For example, RefSeq annotation file which can be found at UCSC genome browser.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>integer. Random seed can be set for reproducibility of results.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>TRUE</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See the vignettes of the package for more details.
</p>


<h3>Value</h3>

<p>The results, including visualization plots are placed in 
subdirectories of the output directory <code>output.dir</code>
as specified by user.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Zhongyang Zhang and Ke Hao. (2015) SAAS-CNV: A Joint Segmentation Approach on Aggregated and Allele Specific Signals 
for the Identification of Somatic Copy Number Alterations with Next-Generation Sequencing Data. 
PLoS Computational Biology, 11(11):e1004618.
</p>


<h3>See Also</h3>

<p><code>NGS.CNV</code>, <code>snp.cnv.data</code>,
<code>joint.segmentation</code>, <code>merging.segments</code>
<code>cnv.call</code>, <code>diagnosis.seg.plot.chr</code>,
<code>genome.wide.plot</code>, <code>diagnosis.cluster.plot</code>, 
<code>snp.refine.boundary</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## the pipeline for SNP array analysis
## download snp_table.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/snp_table.txt.gz"
tryCatch({download.file(url=url, destfile="snp_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="snp_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

snp_table &lt;- read.delim(file="snp_table.txt.gz", as.is=TRUE)

## download refGene_hg19.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/refGene_hg19.txt.gz"
tryCatch({download.file(url=url, destfile="refGene_hg19.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="refGene_hg19.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

sample.id &lt;- "SNP_0116"
output.dir &lt;- file.path(getwd(), "test_saasCNV")
	
SNP.CNV(snp=snp_table, output.dir=output.dir, sample.id=sample.id,
        min.chr.probe=100,
        min.snps=10,
        joint.segmentation.pvalue.cutoff=1e-4,
        max.chpts=30,
        do.merge=TRUE, use.null.data=TRUE, num.perm=1000, maxL=5000, 
        merge.pvalue.cutoff=0.05,
        do.cnvcall.on.merge=TRUE, 
        cnvcall.pvalue.cutoff=0.05,
        do.boundary.refine=TRUE,
        do.plot=TRUE, cex=0.3, ref.num.probe=5000,
        do.gene.anno=TRUE,
        gene.anno.file="refGene_hg19.txt.gz",
        seed=123456789,
        verbose=TRUE)

## End(Not run)

</code></pre>


</div>