<div class="container">

<table style="width: 100%;"><tr>
<td>diagnosis.cluster.plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Visualize Genome-Wide SCNA Profile in 2D Cluster Plot
</h2>

<h3>Description</h3>

<p>An optional function to visualize genome-wide SCNA Profile in 
log2mBAF-log2ratio 2D cluster plot.
</p>


<h3>Usage</h3>

<pre><code class="language-R">diagnosis.cluster.plot(segs, chrs, min.snps, max.cex = 3, ref.num.probe = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>segs</code></td>
<td>
<p>a data frame containing segment location, summary statistics 
and SCNA status resulting from <code>cnv.call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrs</code></td>
<td>
<p>the chromosomes to be visualized. For example, 1:22.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.snps</code></td>
<td>
<p>the minimum number of probes a segment span.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cex</code></td>
<td>
<p>the maximum of cex a circle is associated with. See details.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.num.probe</code></td>
<td>
<p>integer. The reference number of probes against which 
a segment is compared in order to determine the cex of the segment 
to be displayed. Default is <code>NULL</code>. If <code>NULL</code>, It will be automatically 
specified as 1/100 of the number of data points.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>on the main log2mBAF-log2ratio panel, each circle corresponds to a segment, 
with the size reflecting the length of the segment; the color code is specified 
in legend; the dashed gray lines indicate the adjusted baselines. 
The side panels, corresponding to log2ratio and log2mBAF dimension respectively, 
show the distribution of median values of each segment weighted by its length.
</p>


<h3>Value</h3>

<p>An R plot will be generated.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>See Also</h3>

<p><code>joint.segmentation</code>, <code>cnv.call</code>, 
<code>diagnosis.seg.plot.chr</code>, <code>genome.wide.plot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(seq.data)
data(seq.cnv)

diagnosis.cluster.plot(segs=seq.cnv, 
                       chrs=sub("^chr","",unique(seq.cnv$chr)), 
                       min.snps=10, max.cex=3, ref.num.probe=1000)
</code></pre>


</div>