<div class="container">

<table style="width: 100%;"><tr>
<td>cnv.data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Construct Data Frame for CNV Inference with NGS Data
</h2>

<h3>Description</h3>

<p>Transform read depth information into log2ratio and log2mBAF
that we use for joint segmentation and CNV calling.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cnv.data(vcf, min.chr.probe = 100, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>a data frame constructed from a vcf file. See <code>vcf2txt</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.chr.probe</code></td>
<td>
<p>the minimum number of probes tagging a chromosome for it 
to be passed to the subsequent analysis.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical. If more details to be output. Default is <code>FALSE</code>.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame containing the log2raio and log2mBAF values for each probe site.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Staaf, J., Vallon-Christersson, J., Lindgren, D., Juliusson, G., Rosenquist, R., 
Hoglund, M., Borg, A., Ringner, M. (2008) Normalization of Illumina Infinium 
whole-genome SNP data improves copy number estimates and allelic intensity ratios. 
<em>BMC bioinformatics</em>, <b>9</b>:409.
</p>


<h3>See Also</h3>

<p><code>vcf2txt</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## load a data frame constructed from a vcf file with vcf2txt

## Not run: 
## download vcf_table.txt.gz
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/vcf_table.txt.gz"
tryCatch({download.file(url=url, destfile="vcf_table.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="vcf_table.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

vcf_table &lt;- read.delim(file="vcf_table.txt.gz", as.is=TRUE)
seq.data &lt;- cnv.data(vcf=vcf_table, min.chr.probe=100, verbose=TRUE)

## End(Not run)

## see how seq.data looks like
data(seq.data)
head(seq.data)
</code></pre>


</div>