<div class="container">

<table style="width: 100%;"><tr>
<td>GC.adjust</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
GC Content Adjustment
</h2>

<h3>Description</h3>

<p>This function adjusts log2ratio by GC content using LOESS.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GC.adjust(data, gc, maxNumDataPoints = 10000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>A data frame generated by <code>cnv.data</code> 
or <code>snp.cnv.data</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gc</code></td>
<td>
<p>A data frame containing three columns: <code>chr</code>, <code>position</code> 
and <code>GC</code>. See the example data below for details.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxNumDataPoints</code></td>
<td>
<p>The maximum number of data points used for loess fit.
Default is 10000.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The method for GC content adjustment was adopted from CNAnorm (Gusnato et al. 2012).
</p>


<h3>Value</h3>

<p>A data frame containing the log2ratio (GC adjusted) and log2mBAF values 
for each probe site in the same format as generated by <code>cnv.data</code> 
or <code>snp.cnv.data</code>. The original log2ratio is renamed as 
<code>log2ratio.woGCAdj</code>. The GC-adjusted log2ratio is nameed as <code>log2ratio</code>.
</p>


<h3>Note</h3>

<p>This function is optional in the analysis pipeline and is now in beta version.
</p>


<h3>Author(s)</h3>

<p>Zhongyang Zhang &lt;zhongyang.zhang@mssm.edu&gt;
</p>


<h3>References</h3>

<p>Gusnanto, A, Wood HM, Pawitan Y, Rabbitts P, Berri S (2012) Correcting for 
cancer genome size and tumour cell content enables better estimation of 
copy number alterations from next-generation sequence data. 
<em>Bioinformatics</em>, <b>28</b>:40-47.
</p>


<h3>See Also</h3>

<p><code>cnv.data</code>, <code>snp.cnv.data</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## CNV data generated by cnv.data
data(seq.data)
head(seq.data)

## Not run: 
## an example GC content file
url &lt;- "https://zhangz05.u.hpc.mssm.edu/saasCNV/data/GC_1kb_hg19.txt.gz"
tryCatch({download.file(url=url, destfile="GC_1kb_hg19.txt.gz")
         }, error = function(e) {
          download.file(url=url, destfile="GC_1kb_hg19.txt.gz", method="curl")
         })
## If download.file fails to download the data, please manually download it from the url.

gc &lt;- read.delim(file = "GC_1kb_hg19.txt.gz", as.is=TRUE)
head(gc)

## GC content adjustment
seq.data &lt;- GC.adjust(data = seq.data, gc = gc, maxNumDataPoints = 10000)
head(seq.data)

## End(Not run)

</code></pre>


</div>