<div class="container">

<table style="width: 100%;"><tr>
<td>SAMseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Significance analysis of sequencing data - simple user interface</h2>

<h3>Description</h3>

<p>Correlates a large number of features (eg. genes) with an outcome
variable, such as a group indicator, quantitative variable or survival time.
This is a simple user interface for the samr function applied to sequencing data.
For array data applications, see the function SAM.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SAMseq(x, y, censoring.status = NULL, 
resp.type = c("Quantitative", "Two class unpaired", 
"Survival", "Multiclass", "Two class paired"), 
geneid = NULL, genenames = NULL, nperms = 100, 
random.seed = NULL, nresamp = 20, fdr.output = 0.20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Feature matrix: p (number of features) by n (number of samples),
one observation per column (missing values allowed)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>n-vector of outcome measurements</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>censoring.status</code></td>
<td>
<p>n-vector of censoring censoring.status (1=died or event occurred, 0=survived, or event was censored), needed for a censored survival outcome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resp.type</code></td>
<td>
<p>Problem type:
"Quantitative" for a continuous parameter;
"Two class unpaired" for two classes with unpaired observations; 
"Survival" for censored survival outcome; 
"Multiclass": more than 2 groups;  
"Two class paired" for two classes with paired observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneid</code></td>
<td>
<p>Optional character vector of geneids for output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genenames</code></td>
<td>
<p>Optional character vector of genenames for output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperms</code></td>
<td>
<p>Number of permutations used to estimate false discovery rates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random.seed</code></td>
<td>
<p>Optional initial seed for random number generator (integer)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nresamp</code></td>
<td>
<p>Number of resamples used to construct test statistic. Default 20.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr.output</code></td>
<td>
<p>(Approximate) False Discovery Rate cutoff for output in significant genes table</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is a simple, user-friendly interface to the samr package used on sequencing data.
It automatically disables arguments/features that do not apply to sequencing data.
It calls samr, samr.compute.delta.table and samr.compute.siggenes.table.
samr detects differential expression for microarray data,
and sequencing data,
and other data with a large number of features. samr is the R package
that is called by the "official" SAM Excel Addin.
The format of the response vector y and the calling sequence
is illustrated in the examples below. A more complete description
is given in the SAM manual
at http://www-stat.stanford.edu/~tibs/SAM</p>


<h3>Value</h3>

<p>A list with components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>samr.obj</code></td>
<td>
<p>Output of samr. See documentation for samr for details </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>siggenes.table</code></td>
<td>
<p>Table of significant genes, output of  samr.compute.siggenes.table.
This has components: genes.up—matrix of significant genes having positive correlation with the outcome and
genes.lo—matrix of significant genes having negative correlation with the outcome.
For survival data, genes.up are those genes having positive correlation with risk-
that is, increased expression corresponds to higher risk (shorter survival)
genes.lo are those whose increased expression corresponds to lower risk (longer survival).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta.table</code></td>
<td>
<p>Output of  samr.compute.delta.table.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>del</code></td>
<td>
<p>Value of delta (distance from 45 degree line in SAM plot) for 
used for creating delta.table and siggenes.table. Changing the input value fdr.output
will change the resulting del.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p>The calling sequence</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Jun Li and Balasubrimanian Narasimhan and Robert Tibshirani</p>


<h3>References</h3>

<p>Tusher, V.,  Tibshirani, R.  and Chu, G. (2001): 
Significance analysis of microarrays applied to the ionizing radiation response  PNAS 2001 98: 5116-5121, (Apr 24). 
http://www-stat.stanford.edu/~tibs/SAM
</p>
<p>Li, Jun and Tibshirani, R. (2011). Finding consistent patterns: a nonparametric
approach for identifying differential expression in
RNA-Seq data. To appear, Statistical Methods in Medical Research.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
######### two class unpaired comparison
set.seed(100)
mu &lt;- matrix(100, 1000, 20)
mu[1:100, 11:20] &lt;- 200
mu &lt;- scale(mu, center=FALSE, scale=runif(20, 0.5, 1.5))
x &lt;- matrix(rpois(length(mu), mu), 1000, 20)
y &lt;- c(rep(1, 10), rep(2, 10))

samfit &lt;- SAMseq(x, y, resp.type = "Two class unpaired") 

# examine significant gene list
print(samfit)

# plot results
plot(samfit)

######### two class paired comparison
set.seed(100)
mu &lt;- matrix(100, 1000, 20)
mu[1:100, 11:20] &lt;- 200
mu &lt;- scale(mu, center=FALSE, scale=runif(20, 0.5, 1.5))
x &lt;- matrix(rpois(length(mu), mu), 1000, 20)
y &lt;- c(-(1:10), 1:10)

samfit &lt;- SAMseq(x, y, resp.type = "Two class paired") 

# examine significant gene list
print(samfit)

# plot results
plot(samfit)

######### Multiclass comparison
set.seed(100)
mu &lt;- matrix(100, 1000, 20)
mu[1:20, 1:5] &lt;- 120
mu[21:50, 6:10] &lt;- 80
mu[51:70, 11:15] &lt;- 150
mu[71:100, 16:20] &lt;- 60
mu &lt;- scale(mu, center=FALSE, scale=runif(20, 0.5, 1.5))
x &lt;- matrix(rpois(length(mu), mu), 1000, 20)
y &lt;- c(rep(1:4, rep(5, 4)))

samfit &lt;- SAMseq(x, y, resp.type = "Multiclass") 

# examine significant gene list
print(samfit)

# plot results
plot(samfit)

######### Quantitative comparison
set.seed(100)
mu &lt;- matrix(100, 1000, 20)
y &lt;- runif(20, 1, 3)
mu[1 : 100, ] &lt;- matrix(rep(100 * y, 100), ncol=20, byrow=TRUE)
mu &lt;- scale(mu, center=FALSE, scale=runif(20, 0.5, 1.5))
x &lt;- matrix(rpois(length(mu), mu), 1000, 20)
samfit &lt;- SAMseq(x, y, resp.type = "Quantitative")

# examine significant gene list
print(samfit)

# plot results
plot(samfit)

######### Survival comparison
set.seed(100)
mu &lt;- matrix(100, 1000, 20)
y &lt;- runif(20, 1, 3)
mu[1 : 100, ] &lt;- matrix(rep(100 * y, 100), ncol=20, byrow=TRUE)
mu &lt;- scale(mu, center=FALSE, scale=runif(20, 0.5, 1.5))
x &lt;- matrix(rpois(length(mu), mu), 1000, 20)
y &lt;- y + runif(20, -0.5, 0.5)
censoring.status &lt;- as.numeric(y &lt; 2.3)
y[y &gt;= 2.3] &lt;- 2.3
samfit &lt;- SAMseq(x, y, censoring.status = censoring.status, 
resp.type = "Survival")

# examine significant gene list
print(samfit)

# plot results
plot(samfit)
</code></pre>


</div>