<div class="container">

<table style="width: 100%;"><tr>
<td>sblgwas</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sparse Bayesian Learning for QTL Mapping and Genome-Wide Association Studies</h2>

<h3>Description</h3>

<p>The sparse Bayesian learning (SBL) method for quantitative trait
locus (QTL) mapping and genome-wide association studies (GWAS)
deals with a linear mixed model. This is also
a multiple locus model that includes all markers (random effects) in a single
model and detect significant markers simultaneously. SBL method adopts
coordinate descent algorithm to update parameters by estimating one parameter
at a time conditional on the posterior modes of all other parameters. The parameter
estimation process requires multiple iterations and the final estimated
parameters take the values when the entire program converges.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sblgwas(x, y, z, t = -1, max.iter = 200, min.err = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a design matrix for fixed effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a vector of response variables</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>z</code></td>
<td>
<p>a design matrix for random effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t</code></td>
<td>
<p>a number between [-2,0] to control sparseness of the model,
default is -1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.iter</code></td>
<td>
<p>maximum number of iterations set by user to stop the program,
default is 200.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.err</code></td>
<td>
<p>minimum threshold of mean squared error of random effects estimated
from the current and the previous iteration to stop the program, default is 1e-6.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The multiple locus hierarchical linear mixed model of SBL is
</p>
<p style="text-align: center;"><code class="reqn">y=X\beta+Z\gamma+\epsilon</code>
</p>

<p>where y is an <code class="reqn">n*1</code> vector of response variables (observations of the trait);
X is an <code class="reqn">n*p</code> design matrix for fixed effects; <code class="reqn">\beta</code> is a <code class="reqn">p*1</code>
vector of fixed effect; Z is an <code class="reqn">n*m</code> genotype indicator matrix;
<code class="reqn">\gamma</code> is an <code class="reqn">m*1</code> vector of marker effects and <code class="reqn">\epsilon</code> is
an <code class="reqn">n*1</code> vector of residual errors with an aassumed
<code class="reqn">\epsilon~N(0,\Sigma)</code> distribution. Each marker effect, <code class="reqn">\gamma[k]</code>
for marker <em>k</em>, is treated as a random variable following <code class="reqn">N(0,\Phi[k])</code>
distribution, where <code class="reqn">\Phi[k]</code> is the prior variance. The estimate of
<code class="reqn">\gamma[k]</code> is best linear unbiased prediciton (BLUP). The estimate of
<code class="reqn">\Phi[k]</code> is miximum likelihood estimate (MLE).
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>iteration</code></td>
<td>
<p>a matrix storing intermediate results of each iteration before
the entire program converges, including <br></p>
<p>"<code>iter</code>" iteration indicator <br></p>
<p>"<code>error</code>" mean squared error of random effects estimated from the current and
the previous iteration <br></p>
<p>"<code>s2</code>" estimated variance of residual error <br></p>
<p>"<code class="reqn">beta[1]\dots beta[p]</code>" estimates of fixed effects <br></p>
<p>"<code class="reqn">gamma[1]\dots gamma[m]</code>" estimates of random effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parm</code></td>
<td>
<p>a vector containing 5 elements: "<code>iter</code>", "<code>error</code>",
"<code>s2</code>", "<code>beta</code>" and "<code>df</code>" <br></p>
<p>"<code>iter</code>" the number of iterations required by program to stop <br></p>
<p>"<code>error</code>" mean square error of random effect estimated from the last iteration
before the program stops <br></p>
<p>"<code>s2</code>" estimated variance of residual error <br></p>
<p>"<code>beta</code>" estimate of fixed effect <br></p>
<p>"<code>df</code>" the effective degree of freedom from total random effects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blup</code></td>
<td>
<p>a matrix containing 4 columns: "<code>gamma</code>", "<code>vg</code>",
"<code>wald</code>" and "<code>p_wald</code>" <br></p>
<p>"<code>gamma</code>" estimate of random effect <br></p>
<p>"<code>vg</code>" estimated variance of random effect <br></p>
<p>"<code>wald</code>" Wald statistic calculated as <code class="reqn">\gamma^2</code>/<code class="reqn">\Phi</code>
</p>
<p>"<code>p_wald</code>" the <em>p</em>-value of Wald statistic following Chi-squared
distribution with 1 degree of freedom</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Meiyue Wang and Shizhong Xu <br>
Maintainer: Meiyue Wang <a href="mailto:mwang024@ucr.edu">mwang024@ucr.edu</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Load example data from sbl package
data(gen)
data(phe)
data(intercept)

# Run sblgwas() to perform association study of example data
# setting t = 0 leads to the most sparse model
fit&lt;-sblgwas(x=intercept, y=phe, z=gen, t=0)
my.blup&lt;-fit$blup

# setting t = -2 leads to the least sparse model
fit&lt;-sblgwas(x=intercept, y=phe, z=gen, t=-2)
my.blup&lt;-fit$blup
</code></pre>


</div>