<div class="container">

<table style="width: 100%;"><tr>
<td>spatialPreprocess</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Preprocess a spatial dataset for SC-MEB</h2>

<h3>Description</h3>

<p>Adds metadata required for downstream analyses, and (optionally) performs PCA
on log-normalized expression of top HVGs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">spatialPreprocess(
  sce,
  platform = c("Visium", "ST"),
  n.PCs = 15,
  n.HVGs = 2000,
  skip.PCA = FALSE,
  log.normalize = TRUE,
  assay.type = "logcounts",
  BSPARAM = BiocSingular::ExactParam()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sce</code></td>
<td>
<p>SingleCellExperiment to preprocess</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>platform</code></td>
<td>
<p>Spatial sequencing platform. Used to determine spot layout
and neighborhood structure (Visium = hex, ST = square).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.PCs</code></td>
<td>
<p>Number of principal components to compute. We suggest using the
top 15 PCs in most cases.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.HVGs</code></td>
<td>
<p>Number of highly variable genes to run PCA upon.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip.PCA</code></td>
<td>
<p>Skip PCA (if dimensionality reduction was previously
computed.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.normalize</code></td>
<td>
<p>Whether to log-normalize the input data with scater. May
be omitted if log-normalization previously computed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay.type</code></td>
<td>
<p>Name of assay in <code>sce</code> containing normalized counts.
Leave as "logcounts" unless you explicitly pre-computed a different
normalization and added it to <code>sce</code> under another assay. Note that we
do not recommend running BayesSpace on PCs computed from raw counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BSPARAM</code></td>
<td>
<p>A BiocSingularParam object specifying which
algorithm should be used to perform the PCA. By default, an exact PCA is
performed, as current spatial datasets are generally small (&lt;10,000 spots).
To perform a faster approximate PCA, please specify
<code>FastAutoParam()</code> and set a random seed to ensure
reproducibility.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>SingleCellExperiment with PCA and SC.MEB metadata
</p>


<h3>Examples</h3>

<pre><code class="language-R">## read the simulated data 
data(sce)
platform = "ST"
out = find_neighbors2(sce, platform)
</code></pre>


</div>