<div class="container">

<table style="width: 100%;"><tr>
<td>lung73</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Clustering of 73 Lung Tumors</h2>

<h3>Description</h3>

<p>Bootstrapping hierarchical clustering of the DNA microarray data set
of 73 lung tissue samples each containing 916 observed genes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(lung73)
</code></pre>


<h3>Format</h3>

<p><code>lung73.pvclust</code> and <code>lung.pvclust</code> are objects of class <code>"pvclust"</code>
defined in <span class="pkg">pvclust</span> of Suzuki and Shimodaira (2006).
</p>
<p><code>lung73.sb</code> and <code>lung.sb</code> are an object of class <code>"scalebootv"</code> of length
72.
</p>


<h3>Details</h3>

<p>The microarray dataset of Garber et al. (2001) is reanalyzed in Suzuki
and Shimodaira (2006), and is found in <code>data(lung)</code> of
the <span class="pkg">pvclust</span> package. We reanalyze it, again, by the script shown in
Examples. The result of <code>pvclust</code> is stored in
<code>lung73.pvclust</code> and <code>lung.pvclust</code>, and model fitting to bootstrap probabilities
by the <span class="pkg">scaleboot</span> package
is stored in <code>lung73.sb</code> and <code>lung.sb</code>.
A wide scale range is used in lung73.pvclust and lung73.sb, and the default scale range of pvclust is used in lung.pvclust and lung.sb.
The microarray
dataset is not included in <code>data(lung73)</code>, but it is found in
<code>data(lung)</code> of the <span class="pkg">pvclust</span> package.
</p>


<h3>Source</h3>

<p>Garber, M. E. et al. (2001)
Diversity of gene expression in adenocarcinoma of the lung,
<em>Proceedings of the National Academy of Sciences</em>,
98, 13784-13789 (dataset is available from
<a href="http://genome-www.stanford.edu/lung_cancer/adeno/">http://genome-www.stanford.edu/lung_cancer/adeno/</a>).
</p>


<h3>References</h3>

<p>Suzuki, R. and Shimodaira, H. (2006).
pvclust: An R package for hierarchical clustering with p-values,
<em>Bioinformatics</em>, 22, 1540-1542 (software is available from
CRAN or
<a href="http://stat.sys.i.kyoto-u.ac.jp/prog/pvclust/">http://stat.sys.i.kyoto-u.ac.jp/prog/pvclust/</a>).
</p>


<h3>See Also</h3>

<p><code>sbpvclust</code>, <code>sbfit.pvclust</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## Parallel setup
library(parallel)
length(cl &lt;- makeCluster(detectCores()))
## script to create lung73.pvclust and lung73.sb
## multiscale bootstrap resampling of hierarchical clustering
library(pvclust)
data(lung)
### default pvclust scales
lung.pvclust &lt;- pvclust(lung, nboot=10000, parallel=cl)
lung.sb &lt;- sbfit(lung.pvclust,cluster=cl) # model fitting
### wider range of scales than pvclust default
sa &lt;- 9^seq(-1,1,length=13) 
lung73.pvclust &lt;- pvclust(lung,r=1/sa,nboot=10000,parallel=cl) 
lung73.sb &lt;- sbfit(lung73.pvclust,cluster=cl) # model fitting

## End(Not run)

## replace si/au/bp entries in pvclust object
library(pvclust)
data(lung73) # loading the previously computed bootstrap

### the original pvclust result
plot(lung.pvclust, print.pv = c("si", "au", "bp"), cex=0.5, cex.pv=0.5)
pvrect(lung.pvclust, pv="si") #  (defualt pvclust uses pv="au")

### default pvclust scales with p-values of k=2
lung.k2 &lt;- sbpvclust(lung73.pvclust,lung73.sb, k=2)
plot(lung.k2, print.pv = c("si", "au", "bp"), cex=0.5, cex.pv=0.5)
pvrect(lung.k2, pv="si")

### wider scales with p-values of k=3 (default of scaleboot)
lung73.k3 &lt;- sbpvclust(lung73.pvclust,lung73.sb)
plot(lung73.k3, print.pv = c("si", "au", "bp"), cex=0.5, cex.pv=0.5)
pvrect(lung73.k3, pv="si")

## diagnostics of fitting

### diagnose edges 61,...,69
lung73.sb[61:69] # print fitting details
plot(lung73.sb[61:69]) # plot curve fitting
summary(lung73.sb[61:69]) # print raw(=bp)/si/au p-values

### diagnose edge 67
lung73.sb[[67]] # print fitting
plot(lung73.sb[[67]],legend="topleft") # plot curve fitting
summary(lung73.sb[[67]]) # print au p-values

</code></pre>


</div>