<div class="container">

<table style="width: 100%;"><tr>
<td>pbmc2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>pbmc2</h2>

<h3>Description</h3>

<p>A subset of human PBMC scRNA-seq data that was sequenced using inDrops platform.
The Seurat(version 4.0.5) package was used for normalized using the NormalizeData
function with the "LogNormalize" method and a scale factor of 10,000. After modeling
the mean-variance relationship with the  FindVariableFeautre function within "vst"
methods, we selected the top 2,000 highly variable genes and only used this selection
going forward. The dataframe of the cell type label and a gene expression matrix of
644 cells in the row and 2,000 genes in the column.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(pbmc2, package="scAnnotate")
</code></pre>


<h3>Format</h3>

<p>a data frame
</p>


<h3>References</h3>

<p>Ding,  J.et  al.(2019).   Systematic  comparative  analysis  of  single  cellrna-sequencing methods.bioRxiv
</p>


</div>