<div class="container">

<table style="width: 100%;"><tr>
<td>InteNet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Integrate network data from single-cell RNA-seq and ATAC-seq</h2>

<h3>Description</h3>

<p>For the SNARE-seq dataset, a droplet-based method to simultaneously profile gene expression and chromatin accessibility in each of thousands of single nuclei,
the <code>InteNet</code> function can integrate network data of scRNA-seq data and scATAC-seq data (results of the <code>ConNetGNN</code> function) to into a gene-cell network.
</p>


<h3>Usage</h3>

<pre><code class="language-R">InteNet(RNA_net, ATAC_net, parallel.cores = 2, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>RNA_net</code></td>
<td>
<p>Network data for RNA datasets. Produced by the <code>ConNetGNN</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ATAC_net</code></td>
<td>
<p>Network data for ATAC datasets. Produced by the <code>ConNetGNN</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel.cores</code></td>
<td>
<p>Number of processors to use when doing the calculations in parallel (default: <code>2</code>). If <code>parallel.cores=0</code>, then it will use all available core processors unless we set this argument with a smaller number.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Gives information about each step. Default: <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>InteNet
</p>
<p>The scATAC-seq dataset needs to be converted into a gene activity matrix according to the process of <code>Signac</code>(<code>https://satijalab.org/signac/articles/snareseq.html</code>).
The subsequent process is consistent with the scRNA-seq dataset. The <code>InteNet</code> function then integrates the network data of RNA-seq data and ATAC-seq data into a gene-cell network.
With integrated network data as input, <code>scPathway</code> and <code>cpGModule</code> functions will infer pathway activity score matrix and gene modules supported by single-cell multi-omics.
</p>


<h3>Value</h3>

<p>A list.
</p>


<h3>Examples</h3>

<pre><code class="language-R">require(ActivePathways)
require(parallel)
data(RNA_net)
data(ATAC_net)
## Not run: 
RNA_ATAC_IntNet&lt;-InteNet(RNA_net,ATAC_net,parallel.cores=1)

## End(Not run)

# View data
data(RNA_ATAC_IntNet)
summary(RNA_ATAC_IntNet)
</code></pre>


</div>