<div class="container">

<table style="width: 100%;"><tr>
<td>scCAN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>scCAN</h2>

<h3>Description</h3>

<p>This is the main function to perform sc-RNA seq data clustering clustering. scCAN is fully unsupervised scRNA-seq
clustering framework that uses deep neural network and network fusion-based clustering algorithm.
First, scCAN applies a non-negative autoencoder to filter scRNA-seq data.
Second, the filtered data is passed to stacked Bayesian autoencoder to get multiple low-dimensional representations of input data.
Subsequently, scCAN converts these compressed data into networks and unify those networks to a single graph.
Then, scCAN uses a spectral clustering algorithm to obtain final clusters assignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">scCAN(
  data,
  sparse = FALSE,
  n.neighbors = 30,
  alpha = 0.5,
  n.iters = 10,
  ncores = 10,
  r.seed = 1,
  subsamp = TRUE,
  k = 2:15,
  samp.size = 5000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Gene expression matrix, with rows represent samples and columns represent genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sparse</code></td>
<td>
<p>Boolen variable indicating whether data is a sparse matrix. The input must be a non negative sparse matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.neighbors</code></td>
<td>
<p>Number of neighboring cells that are used to caculate the edge's weight. The number of neighbors are set <code>n.neighbors = 30</code> by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>A hyper parameter that control the weight of graph. This values is set to <code>alpha = 0.5</code> by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.iters</code></td>
<td>
<p>A hyper-parameter to set the number of network fusion iterations. It is set to <code>n.iters = 10</code> by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of processor cores to use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.seed</code></td>
<td>
<p>A parameter to set a seed for reproducibility. This values is set to <code>r.seed = 1</code> by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subsamp</code></td>
<td>
<p>Enable subsampling process for big data. This values is set to <code>subsamp = T</code> by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>A vector to search for optimal number of cluster.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samp.size</code></td>
<td>
<p>A parameter to control number of sub-sampled cells.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List with the following keys:
</p>

<ul>
<li>
<p> cluster - A numeric vector containing cluster assignment for each sample.
</p>
</li>
<li>
<p> k  - The optimal number of cluster.
</p>
</li>
<li>
<p> latent - The latent data generated from autoencoders.
</p>
</li>
</ul>
<h3>References</h3>

<p>1. Duc Tran, Hung Nguyen, Bang Tran, Carlo La Vecchia, Hung N. Luu, Tin Nguyen (2021). Fast and precise single-cell data analysis using a hierarchical autoencoder. Nature Communications, 12, 1029. doi: 10.1038/s41467-021-21312-2
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Not run if scDHA has not installed yet.
# Load the package and the example data (SCE dataset)
library(scCAN)
#Load example data
data("SCE")

#Get data matrix and label
data &lt;- t(SCE$data); label &lt;- as.character(SCE$cell_type1)

#Generate clustering result, the input matrix has rows as samples and columns as genes
result &lt;- scCAN(data, r.seed = 1)

#Get the clustering result
cluster &lt;- result$cluster

#Calculate adjusted Rand Index
ari &lt;- round(scCAN::adjustedRandIndex(cluster,label), 2)
message(paste0("ARI = ", ari))


## End(Not run)
</code></pre>


</div>