<div class="container">

<table style="width: 100%;"><tr>
<td>Stacked_VlnPlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Stacked Violin Plot</h2>

<h3>Description</h3>

<p>Code for creating stacked violin plot gene expression.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Stacked_VlnPlot(
  seurat_object,
  features,
  group.by = NULL,
  split.by = NULL,
  idents = NULL,
  x_lab_rotate = FALSE,
  plot_legend = FALSE,
  colors_use = NULL,
  color_seed = 123,
  ggplot_default_colors = FALSE,
  plot_spacing = 0.15,
  spacing_unit = "cm",
  vln_linewidth = NULL,
  pt.size = NULL,
  raster = NULL,
  add.noise = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seurat_object</code></td>
<td>
<p>Seurat object name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>Features to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group.by</code></td>
<td>
<p>Group (color) cells in different ways (for example, orig.ident).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split.by</code></td>
<td>
<p>A variable to split the violin plots by,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idents</code></td>
<td>
<p>Which classes to include in the plot (default is all).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x_lab_rotate</code></td>
<td>
<p>logical or numeric.  If logical whether to rotate x-axis labels 45 degrees
(Default is FALSE).  If numeric must be either 45 or 90.  Setting 45 is equivalent to setting TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_legend</code></td>
<td>
<p>logical.  Adds plot legend containing <code>idents</code> to the returned plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colors_use</code></td>
<td>
<p>specify color palette to used in <code>VlnPlot</code>.  By default if
number of levels plotted is less than or equal to 36 it will use "polychrome" and if greater than 36
will use "varibow" with shuffle = TRUE both from <code>DiscretePalette_scCustomize</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color_seed</code></td>
<td>
<p>random seed for the "varibow" palette shuffle if <code>colors_use = NULL</code> and number of
groups plotted is greater than 36.  Default = 123.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ggplot_default_colors</code></td>
<td>
<p>logical.  If <code>colors_use = NULL</code>, Whether or not to return plot using
default ggplot2 "hue" palette instead of default "polychrome" or "varibow" palettes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot_spacing</code></td>
<td>
<p>Numerical value specifying the vertical spacing between each plot in the stack.
Default is 0.15 ("cm").  Spacing dependent on unit provided to <code>spacing_unit</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spacing_unit</code></td>
<td>
<p>Unit to use in specifying vertical spacing between plots.  Default is "cm".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vln_linewidth</code></td>
<td>
<p>Adjust the linewidth of violin outline.  Must be numeric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pt.size</code></td>
<td>
<p>Adjust point size for plotting.  Default for <code>Stacked_VlnPlot</code> is 0 to avoid issues with
rendering so many points in vector form.  Alternatively, see <code>raster</code> parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>raster</code></td>
<td>
<p>Convert points to raster format.  Default is NULL which will rasterize by default if
greater than 100,000 total points plotted (# Cells x # of features).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.noise</code></td>
<td>
<p>logical, determine if adding a small noise for plotting (Default is TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Extra parameters passed to <code>VlnPlot</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Author(s)</h3>

<p>Ming Tang (Original Code), Sam Marsh (Wrap single function, added/modified functionality)
</p>


<h3>References</h3>

<p><a href="https://divingintogeneticsandgenomics.rbind.io/post/stacked-violin-plot-for-visualizing-single-cell-data-in-seurat/">https://divingintogeneticsandgenomics.rbind.io/post/stacked-violin-plot-for-visualizing-single-cell-data-in-seurat/</a>
</p>


<h3>See Also</h3>

<p><a href="https://twitter.com/tangming2005">https://twitter.com/tangming2005</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(Seurat)
Stacked_VlnPlot(seurat_object = pbmc_small, features = c("CD3E", "CD8", "GZMB", "MS4A1"),
x_lab_rotate = TRUE)

</code></pre>


</div>