<div class="container">

<table style="width: 100%;"><tr>
<td>Add_Top_Gene_Pct_Seurat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add Percent of High Abundance Genes</h2>

<h3>Description</h3>

<p>Add the percentage of counts occupied by the top XX most highly expressed genes in each cell.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Add_Top_Gene_Pct_Seurat(
  seurat_object,
  num_top_genes = 50,
  meta_col_name = NULL,
  assay = "RNA",
  overwrite = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seurat_object</code></td>
<td>
<p>object name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_top_genes</code></td>
<td>
<p>An integer vector specifying the size(s) of the top set of high-abundance genes.
Used to compute the percentage of library size occupied by the most highly expressed genes in each cell.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>meta_col_name</code></td>
<td>
<p>name to use for new meta data column.  Default is "percent_topXX", where XX is
equal to the value provided to <code>num_top_genes</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>assay to use in calculation.  Default is "RNA".  <em>Note</em> This should only be changed if
storing corrected and uncorrected assays in same object (e.g. outputs of both Cell Ranger and Cell Bender).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overwrite</code></td>
<td>
<p>Logical.  Whether to overwrite existing an meta.data column.  Default is FALSE meaning that
function will abort if column with name provided to <code>meta_col_name</code> is present in meta.data slot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical, whether to print messages with status updates, default is TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A Seurat Object
</p>


<h3>References</h3>

<p>This function uses scuttle package (license: GPL-3) to calculate the percent of expression
coming from top XX genes in each cell.  Parameter description for <code>num_top_genes</code> also from scuttle.
If using this function in analysis, in addition to citing scCustomize, please cite scuttle:
McCarthy DJ, Campbell KR, Lun ATL, Willis QF (2017). “Scater: pre-processing, quality control,
normalisation and visualisation of single-cell RNA-seq data in R.” Bioinformatics, 33, 1179-1186.
<a href="doi:10.1093/bioinformatics/btw777">doi:10.1093/bioinformatics/btw777</a>.
</p>


<h3>See Also</h3>

<p><a href="https://bioconductor.org/packages/release/bioc/html/scuttle.html">https://bioconductor.org/packages/release/bioc/html/scuttle.html</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
library(Seurat)
pbmc_small &lt;- Add_Top_Gene_Pct_Seurat(seurat_object = pbmc_small, num_top_genes = 50)

## End(Not run)

</code></pre>


</div>