<div class="container">

<table style="width: 100%;"><tr>
<td>celltype_expression</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute cell type specific gene expression</h2>

<h3>Description</h3>

<p>Compute cell type specific gene expression based on predicted cell proportions and reference data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">celltype_expression(bulk, ref, phenodata, prop, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bulk</code></td>
<td>
<p>a matrix-like object of bulk RNA-seq data with rows representing genes, columns representing samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref</code></td>
<td>
<p>a matrix-like object of scRNA-seq data with rows representing genes, columns representing cells.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenodata</code></td>
<td>
<p>a data.frame with rows representing cells, columns representing cell attributes. It should at least contain the first two
columns as:
</p>

<ol>
<li>
<p> cell barcodes
</p>
</li>
<li>
<p> cell types
</p>
</li>
</ol>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop</code></td>
<td>
<p>a matrix-like object of cell proportion values with rows representing cell types, columns representing samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional parameters passed to <code>create.RCTD</code> from <code>spacexr</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>this function is inspired by cell-type specific gene expression estimation for doublet mode in <code>spacexr</code>. See examples from <code>run_de</code>.
</p>


<h3>Value</h3>

<p>a list with length equal to number of unique cell types in phenodata. Each element in the list represents gene expression matrix for each unique cell type.
</p>


</div>