<div class="container">

<table style="width: 100%;"><tr>
<td>gsea_heatmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Heatmap to demonstrate enrichment of selected gene-sets</h2>

<h3>Description</h3>

<p>Heatmap to demonstrate enrichment of selected gene-sets.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gsea_heatmap(
  normdata,
  teststats,
  gmtfile,
  numgenes,
  gsname_up,
  gsname_down,
  anncol,
  color,
  anncolors = NULL,
  rankcol = TRUE,
  zscore_range = c(-3, 3)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>normdata</code></td>
<td>
<p>a matrix-like object of normalized &amp; untransformed bulk RNA-seq data, with rows representing genes and columns representing samples.
The first element of the output from function <code>run_de</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>teststats</code></td>
<td>
<p>a data.frame containing differential expression results with five columns: "Gene name", "log2 fold change", "log2 average expression",
"p value", "adjusted p value". The second element of the output from function <code>run_de</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gmtfile</code></td>
<td>
<p>path to gmt file used for GSEA analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numgenes</code></td>
<td>
<p>Number of genes to include in the heatmap. Will choose <code>numgenes</code> # of top up-regulated genes, as well as <code>numgenes</code> # of down-regulated genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gsname_up</code></td>
<td>
<p>a character value indicating selected up-regulated gene-set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gsname_down</code></td>
<td>
<p>a character value indicating selected down-regulated gene-set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anncol</code></td>
<td>
<p>a data.frame of sample meta information to include as column annotation bars. See option annCol from <code>aheatmap</code> for more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>color used for heatmap. See option color option from <code>aheatmap</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anncolors</code></td>
<td>
<p>optional data.frame to define colors for column annotations in <code>anncol</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rankcol</code></td>
<td>
<p>a logical value indicating whether to sort samples based on correlation between fold change &amp; gene expression for better visualization. Default to TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zscore_range</code></td>
<td>
<p>a vector of length two indicating the desired range of z-score transformed data. Default to c(-3, 3).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>this function does not support output from GSEA <a href="https://github.com/GSEA-MSigDB/GSEA_R">R implementation</a>
</p>


<h3>Value</h3>

<p>a heatmap outputted from <code>aheatmap</code> function from <code>NMF</code> package
</p>


</div>