<div class="container">

<table style="width: 100%;"><tr>
<td>run_scINSIGHT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform scINSIGHT on normalized datasets</h2>

<h3>Description</h3>

<p>Perform INterpreting single cell gene expresSIon bioloGically Heterogeneous daTa (scINSIGHT) to return factorized <code class="reqn">W_{\ell1}</code>, <code class="reqn">W_{\ell2}</code>, <code class="reqn">H</code> and <code class="reqn">V</code> matrices.
</p>
<p>This factorization produces a <code class="reqn">W_{\ell1}</code> matrix (cells by <code class="reqn">K_j</code>), a <code class="reqn">W_{\ell2}</code> matrix (cells by <code class="reqn">K</code>), a shared <code class="reqn">V</code> matrix (<code class="reqn">K</code> by genes)
for each sample, and a <code class="reqn">H</code> (<code class="reqn">K_j</code> by genes) matrix for each condition. <code class="reqn">W_{\ell2}</code> are the expression matrices of <code class="reqn">K</code> common gene modules for all samples,
<code class="reqn">V</code> is the membership matrix of <code class="reqn">K</code> common gene modules, and it's shared by all samples.
<code class="reqn">W_{\ell1}</code> are the expression matrices of <code class="reqn">K_j</code> condition-specific gene modules for all samples,
and <code class="reqn">H</code> are the membership matrices of <code class="reqn">K_j</code> condition-specific gene modules for all conditions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_scINSIGHT(
  object,
  K = seq(5, 15, 2),
  K_j = 2,
  LDA = c(0.001, 0.01, 0.1, 1, 10),
  thre.niter = 500,
  thre.delta = 0.01,
  num.cores = 1,
  B = 5,
  out.dir = NULL,
  method = "increase"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>scINSIGHT</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>Number of common gene modules. (default <code>c(5, 7, 9, 11, 13, 15)</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K_j</code></td>
<td>
<p>Number of dataset-specific gene modules. (default 2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LDA</code></td>
<td>
<p>Regularization parameters. (default <code>c(0.001, 0.01, 0.1, 1, 10)</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thre.niter</code></td>
<td>
<p>Maximum number of block coordinate descent iterations to perform. (default 500)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thre.delta</code></td>
<td>
<p>Stop iteration when the reduction of objective function is less than the threshold. (default 0.01)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.cores</code></td>
<td>
<p>Number of cores used for optimizing factorizations in parallel (default 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p>Number of repeats with random seed from 1 to B. (default 5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.dir</code></td>
<td>
<p>Output directory of scINSIGHT results. (default NULL)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Method of updating the factorization (default "increase"). If provide multiple <code class="reqn">K</code>, user can choose method between "increase" and "decrease".
</p>
<p>For "increase", the algorithm will first perform factorization with the least <code class="reqn">K=K_1</code>. Then initialize <code class="reqn">K_2-K_1</code> facotrs,
where <code class="reqn">K_2</code> is the <code class="reqn">K</code> sightly larger than <code class="reqn">K_1</code>, and perform facotrization with these new facotrs. Continue this process until the largest <code class="reqn">K</code>.
</p>
<p>For "increase", the algorithm will first perform factorization with the largest <code class="reqn">K=K_1</code>. Then choose <code class="reqn">K_2</code> facotrs,
where <code class="reqn">K_2</code> is the <code class="reqn">K</code> sightly less than <code class="reqn">K_1</code>, and perform facotrization with these new facotrs. Continue this process until the least <code class="reqn">K</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>scINSIGHT</code> object with <code class="reqn">W_1</code>, <code class="reqn">W_2</code>, <code class="reqn">H</code>, <code class="reqn">V</code> and parameters slots set.
</p>


</div>