<div class="container">

<table style="width: 100%;"><tr>
<td>scISR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>scISR: Single-cell Imputation using Subspace Regression</h2>

<h3>Description</h3>

<p>Perform single-cell Imputation using Subspace Regression
</p>


<h3>Usage</h3>

<pre><code class="language-R">scISR(
  data,
  ncores = 1,
  force_impute = FALSE,
  do_fast = TRUE,
  preprocessing = TRUE,
  batch_impute = FALSE,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Input matrix or data frame. Rows represent genes while columns represent samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores that the algorithm should use. Default value is <code>1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>force_impute</code></td>
<td>
<p>Always perform imputation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do_fast</code></td>
<td>
<p>Use fast imputation implementation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>preprocessing</code></td>
<td>
<p>Perform preprocessing on original data to filter out low quality features.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch_impute</code></td>
<td>
<p>Perform imputation in batches to reduce memory consumption.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>Seed for reproducibility. Default value is <code>1</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>scISR performs imputation for single-cell sequencing data. scISR identifies the true dropout values in the scRNA-seq dataset using
hyper-geomtric testing approach. Based on the result obtained from hyper-geometric testing, the original dataset is segregated into two
subsets including training data and imputable data. Next, training data is used for constructing a generalize linear regression model that
is used for imputation on the imputable data.
</p>


<h3>Value</h3>

<p><code>scISR</code> returns an imputed single-cell expression matrix where rows represent genes while columns represent samples.
</p>


<h3>Examples</h3>

<pre><code class="language-R">{
# Load the package
library(scISR)
# Load Goolam dataset
data('Goolam');
# Use only 500 random genes for example
set.seed(1)
raw &lt;- Goolam$data[sample(seq_len(nrow(Goolam$data)), 500), ]
label &lt;- Goolam$label

# Perform the imputation
imputed &lt;- scISR(data = raw)

if(requireNamespace('mclust'))
{
  library(mclust)
  # Perform PCA and k-means clustering on raw data
  set.seed(1)
  # Filter genes that have only zeros from raw data
  raw_filer &lt;- raw[rowSums(raw != 0) &gt; 0, ]
  pca_raw &lt;- irlba::prcomp_irlba(t(raw_filer), n = 50)$x
  cluster_raw &lt;- kmeans(pca_raw, length(unique(label)),
                        nstart = 2000, iter.max = 2000)$cluster
  print(paste('ARI of clusters using raw data:',
              round(adjustedRandIndex(cluster_raw, label),3)))

  # Perform PCA and k-means clustering on imputed data
  set.seed(1)
  pca_imputed &lt;- irlba::prcomp_irlba(t(imputed), n = 50)$x
  cluster_imputed &lt;- kmeans(pca_imputed, length(unique(label)),
                            nstart = 2000, iter.max = 2000)$cluster
  print(paste('ARI of clusters using imputed data:',
              round(adjustedRandIndex(cluster_imputed, label),3)))
}
}
</code></pre>


</div>