<div class="container">

<table style="width: 100%;"><tr>
<td>spectralClones</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Spectral clustering method for clonal partitioning</h2>

<h3>Description</h3>

<p><code>spectralClones</code> provides an unsupervised spectral clustering 
approach to infer clonal relationships in high-throughput Adaptive Immune Receptor 
Repertoire sequencing (AIRR-seq) data. This approach clusters B or T cell receptor 
sequences based on junction region sequence similarity and shared mutations within 
partitions that share the same V gene, J gene, and junction length, allowing for 
ambiguous V or J gene annotations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">spectralClones(
  db,
  method = c("novj", "vj"),
  germline = "germline_alignment",
  sequence = "sequence_alignment",
  junction = "junction",
  v_call = "v_call",
  j_call = "j_call",
  clone = "clone_id",
  fields = NULL,
  cell_id = NULL,
  locus = "locus",
  only_heavy = TRUE,
  split_light = TRUE,
  targeting_model = NULL,
  len_limit = NULL,
  first = FALSE,
  cdr3 = FALSE,
  mod3 = FALSE,
  max_n = 0,
  threshold = NULL,
  base_sim = 0.95,
  iter_max = 1000,
  nstart = 1000,
  nproc = 1,
  verbose = FALSE,
  log = NULL,
  summarize_clones = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>data.frame containing sequence data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>one of the <code>"novj"</code> or <code>"vj"</code>. See Details for description.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>germline</code></td>
<td>
<p>character name of the column containing the germline or reference sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>character name of the column containing input sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction</code></td>
<td>
<p>character name of the column containing junction sequences.
Also used to determine sequence length for grouping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>name of the column containing the V-segment allele calls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_call</code></td>
<td>
<p>name of the column containing the J-segment allele calls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clone</code></td>
<td>
<p>output column name containing the clonal cluster identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fields</code></td>
<td>
<p>character vector of additional columns to use for grouping. 
Sequences with disjoint values in the specified fields will be classified 
as separate clones.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell_id</code></td>
<td>
<p>name of the column containing cell identifiers or barcodes. 
If specified, grouping will be performed in single-cell mode
with the behavior governed by the <code>locus</code> and 
<code>only_heavy</code> arguments. If set to <code>NULL</code> then the 
bulk sequencing data is assumed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>name of the column containing locus information. 
Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only_heavy</code></td>
<td>
<p>use only the IGH (BCR) or TRB/TRD (TCR) sequences 
for grouping. Only applicable to single-cell data.
Ignored if <code>cell_id=NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split_light</code></td>
<td>
<p>split clones by light chains. Ignored if <code>cell_id=NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targeting_model</code></td>
<td>
<p>TargetingModel object. Only applicable if 
<code>method="vj"</code>. See Details for description.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>len_limit</code></td>
<td>
<p>IMGT_V object defining the regions and boundaries of the Ig 
sequences. If NULL, mutations are counted for entire sequence. Only 
applicable if <code>method</code> = <code>"vj"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>first</code></td>
<td>
<p>specifies how to handle multiple V(D)J assignments for initial grouping. 
If <code>TRUE</code> only the first call of the gene assignments is used. 
If <code>FALSE</code> the union of ambiguous gene assignments is used to 
group all sequences with any overlapping gene calls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cdr3</code></td>
<td>
<p>if <code>TRUE</code> removes 3 nucleotides from both ends of <code>"junction"</code> 
prior to clustering (converts IMGT junction to CDR3 region). 
If <code>TRUE</code> this will also remove records with a junction length 
less than 7 nucleotides.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mod3</code></td>
<td>
<p>if <code>TRUE</code> removes records with a <code>junction</code> length that is not divisible by 
3 in nucleotide space.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_n</code></td>
<td>
<p>the maximum number of degenerate characters to permit in the junction sequence before excluding the 
record from clonal assignment. Default is set to be zero. Set it as <code>"NULL"</code> 
for no action.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>the supervising cut-off to enforce an upper-limit distance for clonal grouping.
A numeric value between (0,1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base_sim</code></td>
<td>
<p>required similarity cut-off for sequences in equal distances from each other.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>iter_max</code></td>
<td>
<p>the maximum number of iterations allowed for kmean clustering step.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nstart</code></td>
<td>
<p>the number of random sets chosen for kmean clustering initialization.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>number of cores to distribute the function over.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>if <code>TRUE</code> prints out a summary of each step cloning process.
if <code>FALSE</code> (default) process cloning silently.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log</code></td>
<td>
<p>output path and filename to save the <code>verbose</code> log. 
The input file directory is used if path is not specified.
The default is <code>NULL</code> for no action.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>summarize_clones</code></td>
<td>
<p>if <code>TRUE</code> performs a series of analysis to assess the clonal landscape
and returns a ScoperClones object. If <code>FALSE</code> then
a modified input <code>db</code> is returned. When grouping by <code>fields</code>, 
<code>summarize_clones</code> should be <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If <code>method="novj"</code>, then clonal relationships are inferred using an adaptive 
threshold that indicates the level of similarity among junction sequences in a local neighborhood. 
</p>
<p>If <code>method="vj"</code>, then clonal relationships are inferred not only on 
junction region homology, but also taking into account the mutation profiles in the V 
and J segments. Mutation counts are determined by comparing the input sequences (in the 
column specified by <code>sequence</code>) to the effective germline sequence (IUPAC representation 
of sequences in the column specified by <code>germline</code>). 
</p>
<p>While not mandatory, the influence of SHM hot-/cold-spot biases in the clonal inference 
process will be noted if a SHM targeting model is provided through the <code>targeting_model</code> 
argument. See TargetingModel for more technical details.
</p>
<p>If the <code>threshold</code> argument is specified, then an upper limit for clonal grouping will 
be imposed to prevent sequences with dissimilarity above the threshold from grouping together. 
Any sequence with a distance greater than the <code>threshold</code> value from the other sequences, 
will be assigned to a singleton group.
</p>


<h3>Value</h3>

<p>If <code>summarize_clones=TRUE</code> (default) a ScoperClones object is returned that includes the 
clonal assignment summary information and a modified input <code>db</code> in the <code>db</code> slot that 
contains clonal identifiers in the specified <code>clone</code> column.
If <code>summarize_clones=FALSE</code> modified <code>data.frame</code> is returned with clone identifiers in the 
specified <code>clone</code> column.
</p>


<h3>Single-cell data</h3>

<p>To invoke single-cell mode the <code>cell_id</code> argument must be specified and the <code>locus</code> 
column must be correct. Otherwise, clustering will be performed with bulk sequencing assumptions, 
using all input sequences regardless of the values in the <code>locus</code> column.
</p>
<p>Values in the <code>locus</code> column must be one of <code>c("IGH", "IGI", "IGK", "IGL")</code> for BCR 
or <code>c("TRA", "TRB", "TRD", "TRG")</code> for TCR sequences. Otherwise, the operation will exit and 
return an error message.
</p>
<p>Under single-cell mode with paired-chain sequences, there is a choice of whether 
grouping should be done by (a) using IGH (BCR) or TRB/TRD (TCR) sequences only or
(b) using IGH plus IGK/IGL (BCR) or TRB/TRD plus TRA/TRG (TCR) sequences. 
This is governed by the <code>only_heavy</code> argument. There is also choice as to whether 
inferred clones should be split by the light/short chain (IGK, IGL, TRA, TRG) following 
heavy/long chain clustering, which is governed by the <code>split_light</code> argument.
</p>
<p>In single-cell mode, clonal clustering will not be performed on data were cells are 
assigned multiple heavy/long chain sequences (IGH, TRB, TRD). If observed, the operation 
will exit and return an error message. Cells that lack a heavy/long chain sequence (i.e., cells with 
light/short chains only) will be assigned a <code>clone_id</code> of <code>NA</code>.
</p>


<h3>See Also</h3>

<p>See plotCloneSummary for plotting summary results. See groupGenes for 
more details about grouping requirements.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Subset example data
db &lt;- subset(ExampleDb, c_call == "IGHG")

# Find clonal groups
results &lt;- spectralClones(db, method="novj", germline="germline_alignment_d_mask")

# Retrieve modified input data with clonal clustering identifiers
df &lt;- as.data.frame(results)
  
# Plot clonal summaries
plot(results, binwidth=0.02)

</code></pre>


</div>