<div class="container">

<table style="width: 100%;"><tr>
<td>fragmentoverlapcount</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Count Overlap of ATAC-seq Fragments</h2>

<h3>Description</h3>

<p>Count Overlap of ATAC-seq Fragments
</p>


<h3>Usage</h3>

<pre><code class="language-R">fragmentoverlapcount(
  file,
  targetregions,
  excluderegions = NULL,
  targetbarcodes = NULL,
  Tn5offset = c(1, 0),
  barcodesuffix = NULL,
  dobptonext = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Filename of the file for ATAC-seq fragments.
The file must be block gzipped (using the <code>bgzip</code> command)
and accompanied with the index file (made using the <code>tabix</code> command).
The uncompressed file must be a tab delimited file,
where each row represents one fragment.
The first four columns are chromosome name, start position, end position,
and barcode (i.e., name) of the cell including the fragment.
The remaining columns are ignored.
See vignette for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targetregions</code></td>
<td>
<p>GRanges object for the regions where overlaps are counted.
Usually all of the autosomes.
If there is memory problem, split a chromosome into smaller chunks,
for example by 10 Mb.
The function loads each element of <code>targetregions</code> sequentially,
and smaller elements require less memory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>excluderegions</code></td>
<td>
<p>GRanges object for the regions to be excluded.
Simple repeats in the genome should be listed here,
because repeats can cause false overlaps.
A fragment is discarded if its 5' or 3' end is located in <code>excluderegions</code>.
If <code>NULL</code>, fragments are not excluded by this criterion.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targetbarcodes</code></td>
<td>
<p>Character vector for the barcodes of cells to be analyzed,
such as those passing quality control.
If <code>NULL</code>, all barcodes in the input file are analyzed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Tn5offset</code></td>
<td>
<p>Numeric vector of length two.
The enzyme for ATAC-seq is a homodimer of Tn5.
The transposition sites of two Tn5 proteins are 9 bp apart,
and the (representative) site of accessibility is in between.
If the start and end position of your input file is taken from BAM file,
set the paramater to <code>c(4, -5)</code> to adjust the offset.
Alternatively, values such as <code>c(0, -9)</code> could generate similar results;
what matters the most is the difference between the two numbers.
The fragments.tsv.gz file generated by 10x Cell Ranger already adjusts the shift
but is recorded as a BED file. In this case, use <code>c(1, 0)</code> (default value).
If unsure, set to <code>"guess"</code>,
in which case the program returns a guess.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>barcodesuffix</code></td>
<td>
<p>Add suffix to barcodes per targetregions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dobptonext</code></td>
<td>
<p>(experimental feature) Whether to compute smoothed distance
to the next fragment (irrelevant to BC) as bptonext,
which is the inverse of chromatin accessibility, and append as 9th to 14th columns.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A tibble with each row corresponding to a cell.
For each cell, its barcode, the total count of the fragments <code>nfrag</code>,
and the count distinguished by overlap depth are given.
</p>


</div>