<div class="container">

<table style="width: 100%;"><tr>
<td>SCRIPsimu</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SCRIP simulation</h2>

<h3>Description</h3>

<p>Simulate count data for single cell RNA-sequencing using SCIRP method
</p>


<h3>Usage</h3>

<pre><code class="language-R">SCRIPsimu(
  data,
  params,
  method = "single",
  base_allcellmeans_SC = NULL,
  pre.bcv.df = NULL,
  libsize = NULL,
  bcv.shrink = 1,
  Dropout_rate = NULL,
  mode = "GP-trendedBCV",
  de.prob = NULL,
  de.downProb = NULL,
  de.facLoc = NULL,
  de.facScale = NULL,
  path.skew = NULL,
  batch.facLoc = NULL,
  batch.facScale = NULL,
  path.nSteps = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>data matrix required to fit the mean-BCV trend for simulation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>params</code></td>
<td>
<p>SplatParams object containing parameters for the simulation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>"single", "groups" or "paths"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base_allcellmeans_SC</code></td>
<td>
<p>base mean vector provided to help setting DE analysis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pre.bcv.df</code></td>
<td>
<p>BCV.df enables us to change the variation of BCV values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>libsize</code></td>
<td>
<p>library size can be provided directly</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bcv.shrink</code></td>
<td>
<p>factor to control the BCV levels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Dropout_rate</code></td>
<td>
<p>factor to control the dropout rate directly</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>"GP-commonBCV", "BP-commonBCV", "BP", "BGP-commonBCV" and "BGP-trendedBCV"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.prob</code></td>
<td>
<p>the proportion of DE genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.downProb</code></td>
<td>
<p>the proportion of down-regulated DE genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.facLoc</code></td>
<td>
<p>DE location factor</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de.facScale</code></td>
<td>
<p>DE scale factor</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path.skew</code></td>
<td>
<p>Controls how likely cells are from the start or end point of the path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch.facLoc</code></td>
<td>
<p>DE location factor in batch</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch.facScale</code></td>
<td>
<p>DE scale factor in batch</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path.nSteps</code></td>
<td>
<p>number of steps between the start point and end point for each path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other parameters</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>SingleCellExperiment file
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(params_acinar)
data(acinar.data)
sim_trend = SCRIPsimu(data=acinar.data, params=params_acinar, mode="GP-trendedBCV")

</code></pre>


</div>