<div class="container">

<table style="width: 100%;"><tr>
<td>colEpistatic</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Cordell's Test for Epistatic Interactions
</h2>

<h3>Description</h3>

<p>Performs the likelihood ratio test for epistatic interactions proposed by Cordell (2002) for case-control data, where
<code>colEpistatic</code> assumes that each column represents a SNP, and <code>rowEpistatic</code> assumes that each row represents a SNP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">colEpistatic(mat.snp, cl, genes = NULL, warnError = TRUE)
rowEpistatic(mat.snp, cl, genes = NULL, warnError = TRUE)

## S3 method for class 'colEpi'
print(x, top = 5, digits = 4, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat.snp</code></td>
<td>
<p>a matrix containing genotype data, where the genotypes of each SNPs need to be coded by the number of minor alleles,
i.e. 0, 1, and 2. Missing values are allowed. For <code>colEpistatic</code>, each column of <code>mat.snp</code> is assumed to represent a SNP, and
each row a subject. For <code>rowEpistatic</code>, each row of <code>mat.snp</code> is assumed to represent a subject, and each column a SNP. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cl</code></td>
<td>
<p>a numeric vector of ones and zeros specifying which of the subjects in <code>mat.snp</code> are cases (coded by 1) and which are
controls (coded by 0).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>a character vector containing the names of the genes (or, e.g., LD-blocks or pathways) to which the SNPs belong. If specified, 
only the two-way interactions between SNPs from different genetic sets (e.g., genes, LD-blocks, or pathways) are tested. 
If <code>NULL</code>, all two-way interactions between all possible pairs of SNPs are tested.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warnError</code></td>
<td>
<p>logical indicating whether the statistics for the gTDT for pairs of SNPs should be returned as <code>NA</code> if the fitting 
of one of the logistic regression models with <code>glm</code> returns warning. If <code>warnError = FALSE</code>, 
these warnings will be ignored, and the gTDT will be performed also for these SNP pairs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>colEpi</code>, i.e. the output of either <code>colEpistatic</code> or <code>rowEpistatic</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top</code></td>
<td>
<p>number of interactions that should be printed. If <code>top</code> is set to <code>0</code> or the number of interactions is smaller than 
or equal to <code>top</code>, then the statistics for all interactions are printed in the order of their computation. Otherwise, 
the <code>top</code> interactions with the smallest p-values are printed (sorted by their p-values).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digits</code></td>
<td>
<p>number of digits that should be printed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>ignored.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class <code>colEpi</code> consisting of
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ll.main</code></td>
<td>
<p>a numeric vector containing the values of the maximized loglikelihoods of the logistic regression models 
considering only main effects,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ll.full</code></td>
<td>
<p>a numeric vector containing the values the maximized loglikelihoods of the logistic regression models additionally 
containing interaction terms,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat</code></td>
<td>
<p>a vector comprising the values of the test statistic,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval</code></td>
<td>
<p>a vector comprising the corresponding p-values,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>if <code>genes</code> has been specified, a vector composed of the gene pairs to which the SNP pairs belong,</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vec.error</code></td>
<td>
<p>if <code>warnError = TRUE</code>, a list in which an entry corresponding to a SNP pair for which the fitting of (at least)
one of the logistic regression models led to a warning shows this warning message of <code>glm</code>, and all other entries
are <code>NULL</code>.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Holger Schwender, <a href="mailto:holger.schwender@udo.edu">holger.schwender@udo.edu</a>
</p>


<h3>References</h3>

<p>Cordell, H. J. (2002). Epistasis: What it Means, what it Doesn't mean, and Statistical Methods to Detect it in Humans.
Human Molecular Genetics, 11, 2463-2468.
</p>


</div>