<div class="container">

<table style="width: 100%;"><tr>
<td>recodeAffySNP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Recoding of Affymetrix SNP Values</h2>

<h3>Description</h3>

<p>Recodes the values used on Affymetrix SNP chips to code the genotypes to
other values â€“ required, e.g., by other functions of this package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">recodeAffySNP(mat, refAA = FALSE, geno = 1:3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat</code></td>
<td>
<p>a matrix or data frame consisting of the character strings <code>"AA"</code>, <code>"AB"</code>,
<code>"BB"</code>. Missing values can be coded by either <code>"NN"</code> or <code>NA</code>. Each
row is assumed to correspond to a variable, and each column to a microarray.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refAA</code></td>
<td>
<p>codes <code>"AA"</code> always for the homozygous reference genotype? If
<code>TRUE</code>, <code>"AA"</code> is always replaced by <code>geno[1]</code>, and <code>"BB"</code> by
<code>geno[3]</code>. If <code>FALSE</code>, it is evaluated rowwise whether <code>"AA"</code> or
<code>"BB"</code> occurs more often, and the more frequently occuring value is set to <code>geno[1]</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>a numeric or character vector of length 3 giving the three values that
should be used to recode the genotypes. By default, <code>geno = 1:3</code> which is the
coding, e.g., required by <code>rowChisqStats</code> or <code>pamCat</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix of the same size as <code>mat</code> containing the recoded genotypes. (Missing values are
coded by <code>NA</code>.)
</p>


<h3>See Also</h3>

<p><code>recodeSNPs</code>, <code>snp2bin</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Generate a sample data set consisting of 10 rows and 12 columns,
# and randomly replace 5 of the values by "NN".

mat &lt;- matrix("", 10, 12)
mat[1:5,] &lt;- sample(c("AA", "AB", "BB"), 60, TRUE,
   prob = c(0.49, 0.42, 0.09))
mat[6:10,] &lt;- sample(c("AA", "AB", "BB"), 60, TRUE,
   prob = c(0.09, 0.42, 0.49))
mat[sample(120, 5)] &lt;- "NN"
mat

# Recode the SNPs.

recodeAffySNP(mat)

# Recode the SNPs assuming that "A" is always the major allele.

recodeAffySNP(mat, refAA = TRUE)


## End(Not run)</code></pre>


</div>