<div class="container">

<table style="width: 100%;"><tr>
<td>snp2bin</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transformation of SNPs to Binary Variables</h2>

<h3>Description</h3>

<p>Transforms SNPs to binary variables.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp2bin(mat, domrec = TRUE, refAA = FALSE, snp.in.col = TRUE, 
   monomorph = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat</code></td>
<td>
<p>a matrix or data frame in which the genotypes of all SNPs are coded
either by <code>0</code>, <code>1</code> and <code>2</code>,
or by <code>1</code>, <code>2</code> and <code>3</code>, or by <code>"AA"</code>, <code>"AB"</code> and <code>"BB"</code>.
Missing values are allowed. In the latter coding not only <code>NA</code>, but also <code>"NN"</code>
is allowed for specifying missing values. Using the former two codings it is assumed
that the smallest value codes the homozygous reference genotype, the second value the
heterozygous genotype, and the largest value the homozygous variant genotype. For the third
coding, see <code>refAA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>domrec</code></td>
<td>
<p>should each SNP be coded by two dummy variables from which one codes for
a recessive, and the other for a dominant effect? If <code>TRUE</code>, then the first
binary variable is set to 1 if the SNP is of the heterozygous or the homozygous variant
genotype, and the second dummy variable is set to 1 if the SNP is of the homozygous
variant genotype. If <code>FALSE</code>, three dummy variables are used and each of the three
genotypes of a SNP is coded by one of these binary variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refAA</code></td>
<td>
<p>codes <code>"AA"</code> always for the homozygous reference gentoype? Only considered
if the SNPs are coded by <code>"AA"</code>, <code>"AB"</code> and <code>"BB"</code>.
If <code>FALSE</code>, it is evaluated SNPwise whether <code>"AA"</code> or
<code>"BB"</code> occurs more often, and the more frequently occuring value is assumed to be
the homozygous reference genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.in.col</code></td>
<td>
<p>does each column of <code>mat</code> correspond to a SNP (and each row to an observation)?
If <code>FALSE</code>, it is assumed that each row represents a SNP, and each column an observation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>monomorph</code></td>
<td>
<p>a non-negative number. If a dummy variable contains <code>monomorph</code> or less
values that differ from the more frequent value of this variable, then the variable is
removed from the data set.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix containing the binary dummy variables.
</p>


<h3>Author(s)</h3>

<p>Holger Schwender, <a href="mailto:holger.schwender@udo.edu">holger.schwender@udo.edu</a></p>


<h3>See Also</h3>

<p><code>recodeSNPs</code>, <code>recodeAffySNP</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Generate an example data set consisting of 10 rows (observations)
# and 5 columns (SNPs).

mat &lt;- matrix(sample(3, 50, TRUE), 10)
colnames(mat) &lt;- paste("SNP", 1:5, sep = "")

# Transform each SNP into two dummy variables, one that codes for
# a recessive effect and one that codes for a dominant effect.

snp2bin(mat)

# Transform each SNP into three dummy variables, where each of
# these variables codes for one of the three genotypes.

snp2bin(mat, domrec = FALSE)

## End(Not run)</code></pre>


</div>