<div class="container">

<table style="width: 100%;"><tr>
<td>simulateSNPcatResponse</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulation of SNP Data with Categorical Response</h2>

<h3>Description</h3>

<p>Simulates SNP data. Interactions of some of the simulated SNPs
are then used to specify a categorical response
by level-wise or multinomial logistic regression.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulateSNPcatResponse(n.obs = 1000, n.snp = 50, list.ia = NULL,
   list.snp = NULL, withRef = FALSE, beta0 = -0.5, beta = 1.5, 
   maf = 0.25, sample.y = TRUE, rand = NA)
  
## S3 method for class 'simSNPcatResponse'
print(x, justify = c("left", "right"), spaces = 2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.obs</code></td>
<td>
<p>number of observations that should be generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.snp</code></td>
<td>
<p>number of SNPs that should be generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>list.ia</code></td>
<td>
<p>a list consisting of <code class="reqn">n_{cat}</code> objects, where
<code class="reqn">n_{cat}</code> is the number of levels the response should have. If one
interaction of SNPs should be explanatory for a specific level of the response,
then the corresponding object in <code>list.ia</code> must be a numeric vector specifying
the genotypes of the interacting SNPs by the integers -3, -2, -1, 1, 2, or 3, where
1 codes for the homozygous reference genotype, 2 for the heterozygous genotype, and
3 for the homozygous variant genotype, and a minus before these numbers means that
the corresponding SNP should be not of this genotype. If more than one interaction
should be explanatory for a specific category, then the corresponding object of
<code>list.ia</code> must be a list containing one numeric vector composed of the integers
-3, -2, -1, 1, 2, and 3 for each of the interactions.
</p>
<p>If, e.g., one of the
vectors is given by <code>c(1, -1, -3)</code> and the corresponding vector in
<code>list.snp</code> is <code>c(5, 7, 8)</code>, then the corresponding interaction
explanatory for a level of the response is given by
</p>
<p><code>(SNP5 == 1) &amp; (SNP7 != 1) &amp; (SNP8 != 3)</code>.
</p>
<p>For more details, see Details. Must be specified if <code>list.snp</code> is specified.
If both <code>list.ia</code> and <code>list.snp</code> are <code>NULL</code>, then the interactions
shown in the Details section are used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>list.snp</code></td>
<td>
<p>a list consisting of numeric vectors (if one interaction should be explanatory
for a level of the response) or lists of numeric vectors (if there should be more than one
explanatory interaction) specifying the SNPs that compose
the interactions. <code>list.snp</code> must have the same structure as <code>list.ia</code>, and each
entry of <code>list.snp</code> must be an integer between 1 and <code>n.snp</code>.
If <code>list.ia</code> is specified but
not <code>list.snp</code>, then the first <code class="reqn">n</code> SNPs are used to generate the interactions,
where <code class="reqn">n</code> is the total number of values in <code>list.ia</code>. For the case that both
<code>list.ia</code> and <code>list.snp</code> are not specified, see Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>withRef</code></td>
<td>
<p>should there be an additional reference group (i.e.\ a control group) denoted by a 
zero? If <code>TRUE</code>, a multinomial logistic regression is used to specify the class labels.
If <code>FALSE</code>, level-wise logistic regressions are employed to generate the class labels.
For details, see Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta0</code></td>
<td>
<p>a numeric value or vector of <code>length(list.ia)</code> specifying the intercept of
the logistic regression models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>either a non-negative numeric value or a list of non-negative numeric values specifying
the parameters in the logistic regression models. If a numeric value, all parameters (except for
the intercept) in all logistic regression models will be equal to this value. If a list, then
this list must have the same length as <code>list.ia</code>, and each object must consist of as many
numeric values as interactions are specified by the corresponding object in <code>list.ia</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>either an integer, or a vector of length 2 or <code>n.snp</code> specifying the
minor allele frequency. If an integer, all the SNPs will have the same
minor allele frequency. If a vector of length <code>n.snp</code>, each SNP will have the minor
allele frequency specified in the corresponding entry of <code>maf</code>. If length 2, then
<code>maf</code> is interpreted as the range of the minor allele frequencies, and for each SNP,
a minor allele frequency will be randomly drawn from a uniform distribution with
the range given by <code>maf</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.y</code></td>
<td>
<p>should the values of the response be randomly drawn using the probabilities
determined by the logistic regression models? If <code>FALSE</code>, then for each of the <code>n.obs</code>
observations, the value of the response is given by the level exhibiting the largest probability
at this observation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rand</code></td>
<td>
<p>a numeric value for setting the random number generator in a reproducible state.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>the output of <code>simulateSNPcatResponse</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>justify</code></td>
<td>
<p>a character string specifying whether the column of the summarizing table that
names the explanatory interactions should be <code>"left"</code>- or <code>"right"</code>-adjusted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spaces</code></td>
<td>
<p>integer specifying the distance from the left end of the column mentioned in <code>justify</code>
to the position at which the column name is presented.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>ignored.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>simulateSNPcatResponse</code> first simulates a matrix consisting of <code>n.obs</code> observations and <code>n.snp</code>
SNPs, where the minor allele frequencies of these SNPs are given by <code>maf</code>.
</p>
<p>Note that all SNPs are currently simulated independently of each other such that they are unlinked.
Moreover, an observation is currently not allowed to have genotypes/interactions that are explanatory
for more than one of the levels of the response. If, e.g., the response has three categories, then
an observation can either exhibit one (or more) of the genotypes explaining the first level, or
one (or more) of the genotypes explanatory for the second level, or one (or more) of the genotypes
explaining the third level, or none of these genotypes.
</p>
<p>Afterwards, the response is generated by employing the specifications of <code>list.ia</code>,
<code>list.snp</code>, <code>beta0</code> and <code>beta</code>.
</p>
<p>By default, i.e.\ if both <code>list.ia</code> and <code>list.snp</code> are <code>NULL</code>, <code>list.ia</code> is set
to
</p>
<p><code>list(c(-1, 1), c(1, 1, 1), list(c(-1, 1), c(1, 1, 1)))</code>,
</p>
<p>and <code>list.snp</code> is set to
</p>
<p><code>list(c(6, 7), c(3, 9, 10), list(c(2, 5), c(1, 4, 8)))</code>
</p>
<p>such that the interaction
</p>
<p><code>(SNP6 != 1) &amp; (SNP7 == 1)</code>
</p>
<p>is assumed to be explanatory for the first level of the three-categorical response, the interaction
</p>
<p><code>(SNP3 == 1) &amp; (SNP9 == 1) &amp; (SNP10 == 1)</code>
</p>
<p>is assumed to be explanatory for the second level, and the interactions
</p>
<p><code>(SNP2 != 1) &amp; (SNP5 == 1)</code>\ \ \ and
</p>
<p><code>(SNP1 == 1) &amp; (SNP4 == 1) &amp; (SNP8 == 1)</code>,
</p>
<p>are assumed to be explanatory for the third level.
</p>
<p>If <code>withRef = FALSE</code>, then for each of the levels, 
the probability of having this level given that an observation exhibits one, two, ...
of the interactions intended to be explanatory for that level is determined using the corresponding
logistic regression model. Afterwards, the value of the response for an observation showing one, two, ...
of the interactions explanatory for a particular level is randomly drawn using the above probability <code class="reqn">p</code>
for the particular level and <code class="reqn">(1-p)/(n_{cat}-1)</code> as probabilities for the other
<code class="reqn">(n_{cat}-1)</code> levels. If an observation exhibits none of the explanatory interactions,
its response value is randomly drawn using the probabilities <code class="reqn">\exp{beta0}/(1+\exp{beta0})</code>.
</p>
<p>If <code>withRef = TRUE</code>, a multinomial logistic regression is used to specify the class labels. In this
case the probabilities <code class="reqn">p_j</code>, <code class="reqn">j = 1, ..., n.cat</code>, are given by 
<code class="reqn">p_j = \exp(q_j) * p_0</code>, where <code class="reqn">q_j</code> are the probabilities on the
logit-scale (i.e.\ the probabilities on the scale of the linear predictors) and 
<code class="reqn">p_0^{-1} = 1 + p_1 + ... + p_{n.cat}</code> is the reciprocal
of the probability for the control/reference group.  
</p>


<h3>Value</h3>

<p>An object of class <code>simSNPcatResponse</code> consisting of
</p>
<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a matrix with <code>n.obs</code> rows and <code>n.snp</code> columns containing the simulated SNP values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a vector of length <code>n.obs</code> composed of the values of the response.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>models</code></td>
<td>
<p>a character vector naming the level-wise logistic regression models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>a vector of length <code>n.snp</code> composed of the minor allele frequencies.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tab.explain</code></td>
<td>
<p>a data frame summarizing the results of the simulation.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Holger Schwender, <a href="mailto:holger.schwender@udo.edu">holger.schwender@udo.edu</a></p>


<h3>See Also</h3>

<p><code>simulateSNPs</code>, <code>simulateSNPglm</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# The simulated data set described in Details.

sim1 &lt;- simulateSNPcatResponse()
sim1

# Specifying the values of the response by the levels with
# the largest probability.

sim2 &lt;- simulateSNPcatResponse(sample.y = FALSE)
sim2

# If ((SNP4 != 2) &amp; (SNP3 == 1)), (SNP5 ==3), and
# ((SNP12 !=1) &amp; (SNP9 == 3)) should be the three interactions
# (or variables) that are explanatory for the three levels
# of the response, list.ia and list.snp are specified as follows.

list.ia &lt;- list(c(-2, 1), 3, c(-1,3))
list.snp &lt;- list(c(4, 3), 5, c(12,9))

# The categorical response and a data set consisting of 
# 800 observations and 25 SNPs, where the minor allele
# frequency of each SNP is randomly drawn from a
# uniform distribution with minimum 0.1 and maximum 0.4,
# is then generated by

sim3 &lt;- simulateSNPcatResponse(n.obs = 800, n.snp = 25,
  list.ia = list.ia, list.snp = list.snp, maf = c(0.1, 0.4))
sim3


## End(Not run)</code></pre>


</div>