<div class="container">

<table style="width: 100%;"><tr>
<td>pamCat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prediction Analysis of Categorical Data</h2>

<h3>Description</h3>

<p>Performs a Prediction Analysis of Categorical Data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pamCat(data, cl, theta = NULL, n.theta = 10, newdata = NULL, newcl = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a numeric matrix composed of the integers between 1 and <code class="reqn">n_{cat}</code>,
where <code class="reqn">n_{cat}</code> is the number of levels each of the variables represented
by the rows of <code>data</code> must take. No missing values allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cl</code></td>
<td>
<p>a numeric vector of length <code>ncol(data)</code> comprising the class labels of
the observations represented by the columns of <code>data</code>. <code>cl</code> must consist
of the integers between 1 and <code class="reqn">n_{cl}</code>, where <code class="reqn">n_{cl}</code> is the number
of classes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>a numeric vector consisting of the strictly positive values of the shrinkage parameter used
in the Prediction Analysis. If <code>NULL</code>, a vector consisting of <code>n.theta</code> values for
the shrinkage parameter are determined automatically.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.theta</code></td>
<td>
<p>an integer specifying the number of values for the shrinkage parameter of the
Prediction Analysis. Ignored if <code>theta</code> is specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>newdata</code></td>
<td>
<p>a numeric matrix composed of the integers between 1 and <code class="reqn">n_{cat}</code>.
Must have the same number of rows as <code>data</code>, and each row of <code>newdata</code> must contain
the same variable as the corresponding row of <code>data</code>. <code>newdata</code> is employed to
compute the misclassification rates of the Prediction Analysis for the given values of the
shrinkage parameter. If <code>NULL</code>, <code>data</code> is used to determine the misclassification rates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>newcl</code></td>
<td>
<p>a numeric vector of length <code>ncol(newdata)</code> that consists of integers between
1 and <code class="reqn">n_{cl}</code>, and specifies the class labels of the observations in <code>newdata</code>.
Must be specified, if <code>newdata</code> is specified.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class <code>pamCat</code> composed of
</p>
<table>
<tr style="vertical-align: top;">
<td><code>mat.chisq</code></td>
<td>
<p>a matrix with <code class="reqn">m</code> rows and <code class="reqn">n_{cl}</code> columns consisting of the
classwise values of Pearson's <code class="reqn">\chi^2</code> statistic for each of the <code class="reqn">m</code> variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mat.obs</code></td>
<td>
<p>a matrix with <code class="reqn">m</code> rows and <code class="reqn">n_{cat} * n_{cl}</code> columns
in which each row shows a contingency table between the corresponding variable and <code>cl</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mat.exp</code></td>
<td>
<p>a matrix of the same size as <code>mat.obs</code> containing the numbers of observations
expected under the null hypothesis of an association between the respective variable and <code>cl</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mat.theta</code></td>
<td>
<p>a data frame consisting of the numbers of variables used in the classification
of the observations in <code>newdata</code> and the corresponding misclassification rates for a set of values of
the shrinkage parameter <code class="reqn">\theta</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tab.cl</code></td>
<td>
<p>a table summarizing the values of the response, i.e.\ the class labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.cat</code></td>
<td>
<p><code class="reqn">n_{cat}</code>.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Holger Schwender, <a href="mailto:holger.schwender@udo.edu">holger.schwender@udo.edu</a></p>


<h3>References</h3>

<p>Schwender, H.\ (2007). Statistical Analysis of Genotype and Gene Expression Data.
<em>Dissertation</em>, Department of Statistics, University of Dortmund.
</p>


<h3>See Also</h3>

<p><code>predict.pamCat</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Generate a data set consisting of 2000 rows (variables) and 50 columns.
# Assume that the first 25 observations belong to class 1, and the other
# 50 observations to class 2.

mat &lt;- matrix(sample(3, 100000, TRUE), 2000)
rownames(mat) &lt;- paste("SNP", 1:2000, sep = "")
cl &lt;- rep(1:2, e = 25)

# Apply PAM for categorical data to this matrix, and compute the
# misclassification rate on the training set, i.e. on mat.

pam.out &lt;- pamCat(mat, cl)
pam.out

# Now generate a new data set consisting of 20 observations, 
# and predict the classes of these observations using the
# value of theta that has led to the smallest misclassification
# rate in pam.out.

mat2 &lt;- matrix(sample(3, 40000, TRUE), 2000)
rownames(mat2) &lt;- paste("SNP", 1:2000, sep = "")
predict(pam.out, mat2)

# Let's assume that the predicted classes are the real classes
# of the observations. Then, mat2 can also be used in pamCat
# to compute the misclassification rate. 

cl2 &lt;- predict(pam.out, mat2)
pamCat(mat, cl, newdata = mat2, newcl = cl2)


## End(Not run)</code></pre>


</div>