<div class="container">

<table style="width: 100%;"><tr>
<td>recodeSNPs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Recoding of SNP Values</h2>

<h3>Description</h3>

<p>Recodes the values used to specify the genotypes of the SNPs to
other values. Such a recoding might be required to use other
functions contained in this package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">recodeSNPs(mat, first.ref = FALSE, geno = 1:3, snp.in.col = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat</code></td>
<td>
<p>a matrix or data frame consisting of character strings of length 2 that
specify the genotypes of the SNPs. Each of these character strings
must be a combination of the letters A, T, C, and G. Missing values can
be specified by <code>"NN"</code> or <code>NA</code>. Depending on
<code>snp.in.col</code> it is assumed that each row of <code>mat</code> represents
a SNP and each column a variable (<code>snp.in.col = FALSE</code>), or vice versa.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>first.ref</code></td>
<td>
<p>does the first letter in the string coding the heterozygous
genotype always stands for the more frequent allele? E.g., codes <code>"CC"</code>
for the homozygous reference genotype if the genotypes
of a SNP are coded by <code>"CC"</code>, <code>"CG"</code> and <code>"GG"</code>? If <code>TRUE</code>,
the value made up only of this first letter is set to <code>geno[1]</code>, and the
value made up only of the second letter is set to <code>geno[3]</code>. If <code>FALSE</code>,
it is evaluated rowwise which of the homozygous genotypes has the higher frequency
and the more often occuring value is set to <code>geno[1]</code>, and the other to <code>geno[3]</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>a numeric or character vector of length 3 giving the three values that
should be used to recode the genotypes. By default, <code>geno = 1:3</code> which is the
coding, e.g., required by <code>rowChisqStats</code> or <code>pamCat</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp.in.col</code></td>
<td>
<p>does each column of <code>mat</code> correspond to a SNP (and each row to an array)?
If <code>FALSE</code>, it is assumed that each row represents a SNP, and each column an array.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix of the same size as <code>mat</code> containing the recoded genotypes. (Missing values are
coded by <code>NA</code>).
</p>


<h3>See Also</h3>

<p><code>recodeAffySNP</code>, <code>snp2bin</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Generate an example data set consisting of 5 rows and 12 columns,
# where it is assumed that each row corresponds to a SNP.

mat &lt;- matrix("", 10, 12)
mat[c(1, 4, 6),] &lt;- sample(c("AA", "AT", "TT"), 18, TRUE)
mat[c(2, 3, 10),] &lt;- sample(c("CC", "CG", "GG"), 18, TRUE)
mat[c(5, 8),] &lt;- sample(c("GG", "GT", "TT"), 12, TRUE)
mat[c(7, 9),] &lt;- sample(c("AA", "AC", "CC"), 12, TRUE)
mat

# Recode the SNPs

recodeSNPs(mat)

# Recode the SNPs by assuming that the first letter in
# the heterogyzous genotype refers to the major allele.

recodeSNPs(mat, first.ref = TRUE)


## End(Not run)</code></pre>


</div>