<div class="container">

<table style="width: 100%;"><tr>
<td>scRNAtools_DEGsA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identification of differentially expressed genes</h2>

<h3>Description</h3>

<p>Users can identify differentially expressed genes between two type of cells based on fold change value.</p>


<h3>Usage</h3>

<pre><code class="language-R">scRNAtools_DEGsA(example, types_all, type1, type2, num)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>example</code></td>
<td>

<p>scRNA sequencing data with header.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>types_all</code></td>
<td>

<p>Cell types in the example data.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type1</code></td>
<td>

<p>Cell type one.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type2</code></td>
<td>

<p>Cell type two.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num</code></td>
<td>

<p>Threshold value of expressed genes in appointed cell types. For example, we set 0.8 in example section.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The output data is the fold change value of differentially expressed genes.
</p>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(example)
data(types)
type1&lt;-"No malignant"
type2&lt;-"Malignant"
num&lt;-0.8;###type1 Vs type2
pdf(file=file.path(tempdir(), "DEGs.pdf"))
scRNAtools_DEGsA(example,types_all,type1,type2,num)
dev.off()
</code></pre>


</div>