<div class="container">

<table style="width: 100%;"><tr>
<td>scRNAtools_cor_map</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Correlation analysis</h2>

<h3>Description</h3>

<p>Correlation analysis of interested gene set or differentially expressed gene set.</p>


<h3>Usage</h3>

<pre><code class="language-R">scRNAtools_cor_map(exam1, types_all, type, methods)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>exam1</code></td>
<td>

<p>scRNA sequencing data of several genes and cells.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>types_all</code></td>
<td>

<p>Cell names of each type.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>

<p>Cell type.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>methods</code></td>
<td>

<p>correlation methods including "pearson", "kendall" and "spearman".
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Return the correlation index of each two genes.
</p>


<h3>Author(s)</h3>

<p>Qian Yang
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(exam1)
data(types_all)
type&lt;-"Malignant";
methods&lt;-"pearson";##methods = c("pearson", "kendall", "spearman").
pdf(file=file.path(tempdir(), "correlation_color.pdf"))
scRNAtools_cor_map(exam1,types_all,type,methods)
dev.off()
</code></pre>


</div>