<div class="container">

<table style="width: 100%;"><tr>
<td>scTenifoldKnk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>scTenifoldKNK</h2>

<h3>Description</h3>

<p>Predict gene perturbations
</p>


<h3>Usage</h3>

<pre><code class="language-R">scTenifoldKnk(
  countMatrix,
  gKO = NULL,
  qc_mtThreshold = 0.1,
  qc_minLSize = 1000,
  nc_lambda = 0,
  nc_nNet = 10,
  nc_nCells = 500,
  nc_nComp = 3,
  nc_scaleScores = TRUE,
  nc_symmetric = FALSE,
  nc_q = 0.9,
  td_K = 3,
  td_maxIter = 1000,
  td_maxError = 1e-05,
  td_nDecimal = 3,
  ma_nDim = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>countMatrix</code></td>
<td>
<p>countMatrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gKO</code></td>
<td>
<p>gKO</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qc_mtThreshold</code></td>
<td>
<p>A decimal value between 0 and 1. Defines the maximum ratio of mitochondrial reads (mithocondrial reads / library size) present in a cell to be included in the analysis. It's computed using the symbol genes starting with 'MT-' non-case sensitive.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qc_minLSize</code></td>
<td>
<p>An integer value. Defines the minimum library size required for a cell to be included in the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc_lambda</code></td>
<td>
<p>A continuous value between 0 and 1. Defines the multiplicative value (1-lambda) to be applied over the weaker edge connecting two genes to maximize the adjacency matrix directionality.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc_nNet</code></td>
<td>
<p>An integer value. The number of networks based on principal components regression to generate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc_nCells</code></td>
<td>
<p>An integer value. The number of cells to subsample each time to generate a network.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc_nComp</code></td>
<td>
<p>An integer value. The number of principal components in PCA to generate the networks. Should be greater than 2 and lower than the total number of genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc_scaleScores</code></td>
<td>
<p>A boolean value (TRUE/FALSE), if TRUE, the weights will be normalized such that the maximum absolute value is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc_symmetric</code></td>
<td>
<p>A boolean value (TRUE/FALSE), if TRUE, the weights matrix returned will be symmetric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nc_q</code></td>
<td>
<p>A decimal value between 0 and 1. Defines the cut-off threshold of top q% relationships to be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>td_K</code></td>
<td>
<p>An integer value. Defines the number of rank-one tensors used to approximate the data using CANDECOMP/PARAFAC (CP) Tensor Decomposition.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>td_maxIter</code></td>
<td>
<p>An integer value. Defines the maximum number of iterations if error stay above <code>td_maxError</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>td_maxError</code></td>
<td>
<p>A decimal value between 0 and 1. Defines the relative Frobenius norm error tolerance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>td_nDecimal</code></td>
<td>
<p>An integer value indicating the number of decimal places to be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ma_nDim</code></td>
<td>
<p>An integer value. Defines the number of dimensions of the low-dimensional feature space to be returned from the non-linear manifold alignment.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Daniel Osorio &lt;dcosorioh@tamu.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Loading single-cell data
scRNAseq &lt;- system.file("single-cell/example.csv",package="scTenifoldKnk")
scRNAseq &lt;- read.csv(scRNAseq, row.names = 1)

# Running scTenifoldKnk
scTenifoldKnk(countMatrix = scRNAseq, gKO = 'G100', qc_minLSize = 0)
</code></pre>


</div>