<div class="container">

<table style="width: 100%;"><tr>
<td>makeNetworks</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Computes gene regulatory networks for subsamples of cells based on principal component regression.</h2>

<h3>Description</h3>

<p>This function computes <code>nNet</code> gene regulatory networks for a randomly selected subsample of <code>nCells</code> cells based on principal component regression (PCR), a technique based on principal component analysis. In PCR, the outcome variable is regressed over a <code>nComp</code> number of for principal components computed from a set of covariates to estimate the unknown regression coefficients in the model. <code>pcNet</code> function computes the PCR coefficients for each gene one at a time using all the others as covariates, to construct an all by all gene regulatory network.
</p>


<h3>Usage</h3>

<pre><code class="language-R">makeNetworks(
  X,
  nNet = 10,
  nCells = 500,
  nComp = 3,
  scaleScores = TRUE,
  symmetric = FALSE,
  q = 0.95,
  nCores = parallel::detectCores()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A filtered and normlized gene expression matrix with cells as columns and genes as rows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nNet</code></td>
<td>
<p>An integer value. The number of networks based on principal components regression to generate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCells</code></td>
<td>
<p>An integer value. The number of cells to subsample each time to generate a network.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nComp</code></td>
<td>
<p>An integer value. The number of principal components in PCA to generate the networks. Should be greater than 2 and lower than the total number of genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scaleScores</code></td>
<td>
<p>A boolean value (<code>TRUE/FALSE</code>), if <code>TRUE</code>, the weights will be normalized such that the maximum absolute value is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>symmetric</code></td>
<td>
<p>A boolean value (<code>TRUE/FALSE</code>), if <code>TRUE</code>, the weights matrix returned will be symmetric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>
<p>A decimal value between 0 and 1. Represent the cut-off threshold of top q% relationships to be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCores</code></td>
<td>
<p>An integer value. Defines the number of cores to be used.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Principal component regression may be broadly divided into three major steps: </p>

<ol>
<li>
<p> Perform PCA on the observed covariates data matrix to obtain <code>nComp</code> number of the principal components.
</p>
</li>
<li>
<p> Regress the observed vector of outcomes on the selected principal components as covariates, using ordinary least squares regression to get a vector of estimated regression coefficients
</p>
</li>
<li>
<p> Transform this vector back to the scale of the actual covariates, using the eigenvectors corresponding to the selected principal components to get the final PCR estimator for estimating the regression coefficients characterizing the original model.
</p>
</li>
</ol>
<h3>Value</h3>

<p>A list with <code>nNet</code> gene regulatory networks in dgCMatrix format. Each one computed from a randomly selected subsample of <code>nCells</code> cells.
</p>


<h3>References</h3>


<ul>
<li>
<p> Gill, Ryan, Somnath Datta, and Susmita Datta. "dna: An R package for differential network analysis." Bioinformation 10.4 (2014): 233.
</p>
</li>
<li>
<p> Jolliffe, Ian T. "A note on the use of principal components in regression." Journal of the Royal Statistical Society: Series C (Applied Statistics) 31.3 (1982): 300-303.
</p>
</li>
<li>
<p> Massy, William F. "Principal components regression in exploratory statistical research." Journal of the American Statistical Association 60.309 (1965): 234-256.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">library(scTenifoldNet)

# Simulating of a dataset following a negative binomial distribution with high sparcity (~67%)
nCells = 2000
nGenes = 100
set.seed(1)
X &lt;- rnbinom(n = nGenes * nCells, size = 20, prob = 0.98)
X &lt;- round(X)
X &lt;- matrix(X, ncol = nCells)
rownames(X) &lt;- c(paste0('ng', 1:90), paste0('mt-', 1:10))

# Performing Single cell quality control
qcOutput &lt;- scQC(
  X = X,
  minLibSize = 30,
  removeOutlierCells = TRUE,
  minPCT = 0.05,
  maxMTratio = 0.1
)

# Computing 3 single-cell gene regulatory networks each one from a subsample of 500 cells
mnOutput &lt;- makeNetworks(X = X,
                         nNet = 3, 
                         nCells = 500, 
                         nComp = 3, 
                         scaleScores = TRUE, 
                         symmetric = FALSE, 
                         q = 0.95
                         )

# Verifying the class
class(mnOutput)

# Verifying the number of networks
length(mnOutput)

# Veryfying the dimention of the networks
lapply(mnOutput,dim)

# Single-cell gene regulatory networks
mnOutput[[1]][1:10,1:10]
mnOutput[[2]][1:10,1:10]
mnOutput[[3]][1:10,1:10]
</code></pre>


</div>