<div class="container">

<table style="width: 100%;"><tr>
<td>cpmNormalization</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Performs counts per million (CPM) data normalization</h2>

<h3>Description</h3>

<p>This function normalizes the count data present in a given matrix using counts per million normalization (CPM). Each gene count for each cell is divided by the total counts for that cell and multiplied by 1e6. No log-transformation is applied.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cpmNormalization(X)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Raw counts matrix with cells as columns and genes (symbols) as rows</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A dgCMatrix object with the count per million (CPM) normalized values.
</p>


<h3>References</h3>

<p>Vallejos, Catalina A., et al. "Normalizing single-cell RNA sequencing data: challenges and opportunities." Nature methods 14.6 (2017): 565.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(scTenifoldNet)

# Simulating of a dataset following a negative binomial distribution with high sparcity (~67%)
nCells = 2000
nGenes = 100
set.seed(1)
X &lt;- rnbinom(n = nGenes * nCells, size = 20, prob = 0.98)
X &lt;- round(X)
X &lt;- matrix(X, ncol = nCells)
rownames(X) &lt;- c(paste0('ng', 1:90), paste0('mt-', 1:10))

# Performing Single cell quality control
qcOutput &lt;- scQC(
  X = X,
  minLibSize = 30,
  removeOutlierCells = TRUE,
  minPCT = 0.05,
  maxMTratio = 0.1
)
# Performing Counts per million Normalization (CPM)
normalizationOutput &lt;- cpmNormalization(qcOutput)

# Visualizing the differences
oldPar &lt;- par(no.readonly = TRUE)

par(
  mfrow = c(1, 2),
  mar = c(3, 3, 1, 1),
  mgp = c(1.5, 0.5, 0)
)
plot(
  Matrix::colSums(qcOutput),
  ylab = 'Library Size',
  xlab = 'Cell',
  main = 'Before CPM Normalization'
)
plot(
  Matrix::colSums(normalizationOutput),
  ylab = 'Library Size',
  xlab = 'Cell',
  main = 'After CPM Normalization'
)

par(oldPar)
</code></pre>


</div>