<div class="container">

<table style="width: 100%;"><tr>
<td>feat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Feature Plot</h2>

<h3>Description</h3>

<p>Highlight gene expression data in a 2D-embedding (UMAP, tSNE, etc.).
</p>


<h3>Usage</h3>

<pre><code class="language-R">feat(embedding, expression, legend_name = "Expression")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>embedding</code></td>
<td>
<p>A matrix/data.frame/tibble/... with exactly two columns.
If colnames are missing, the axis will be named "Dim1" and "Dim2".
Other classes than matrix/data.frame/tibble are possible, as long as
<code>data.frame(embedding)</code>) produces a numeric data.frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expression</code></td>
<td>
<p>Numeric vector with expression values of the gene of
interest. Order has to correspond to the row order in <code>embedding</code>.
Typically, <code>expression</code> is normalized gene expression and we recommend
<code>k/s/mean(1/s)</code>, where <code>k</code> are UMI counts for the gene of interest
and <code>s</code> are totalUMI of the cell (aka library size).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_name</code></td>
<td>
<p>Text displayed above the legend. Most commonly the name
of the displayed gene.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function discourages customization on purpose, because it
bundles geoms, themes and settings that I found important for
visualizing gene expression in scRNAseq data:
</p>

<ul>
<li>
<p> coord_fixed, to avoid distortion of embeddings
</p>
</li>
<li>
<p> geom_point with size=.4, to ameliorate overplotting
</p>
</li>
<li>
<p> No background grid, because distances and axis units
in embeddings do not
carry meaning for most dimensionality reduction techniques.
</p>
</li>
<li>
<p> Intensity-coded color scales (viridis) displayed with
log2-transformation. Makes visualization independent of colorblindness
and appropriate for gene expression data (which is usually Log Normal
distributed).
</p>
</li>
<li>
<p> Color scale breaks are displayed as 'closed interval', i.e.
<code>max(expression)</code> and <code>min(expression)</code> are the most extreme
breaks. Rounding makes them human-readable. This functionality is provided
by closed_breaks_log2 and closed_labels.
</p>
</li>
</ul>
<p>If you insist on customizing, think of this function as a great starting point, you can simply
copy-paste the code after typing <code>feat</code> into your
console.
</p>


<h3>Value</h3>

<p>A <code>ggplot2</code> object storing a colored scatter plot.
</p>


<h3>See Also</h3>

<p><code>ggplot</code>,
<code>closed_labels</code>,
<code>closed_breaks_log2</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"> # expression goes from 0 to 22:
 set.seed(100)
 feat(matrix(rnorm(2000, c(.1, 3)), ncol=2), rpois(1000, c(.1, 11)))
 # expression goes from 2 to 52:
 set.seed(100)
 feat(matrix(rnorm(2000, c(.1, 3)), ncol=2), rpois(1000, c(10, 31)))
</code></pre>


</div>