<div class="container">

<table style="width: 100%;"><tr>
<td>sda.ranking</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Shrinkage Discriminant Analysis 1: Predictor Ranking</h2>

<h3>Description</h3>

<p><code>sda.ranking</code> determines a ranking of predictors by computing  CAT scores
(correlation-adjusted t-scores)
between the group centroids and the pooled mean.
</p>
<p><code>plot.sda.ranking</code> provides a graphical visualization of the top ranking features.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sda.ranking(Xtrain, L, lambda, lambda.var, lambda.freqs,
  ranking.score=c("entropy", "avg", "max"), 
  diagonal=FALSE, fdr=TRUE, plot.fdr=FALSE, verbose=TRUE)
## S3 method for class 'sda.ranking'
plot(x, top=40, arrow.col="blue", zeroaxis.col="red",
   ylab="Features", main, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Xtrain</code></td>
<td>
<p>A matrix  containing the training data set. Note that 
the rows correspond to observations and the columns
to variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>L</code></td>
<td>
<p>A factor with the class labels of the training samples. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>Shrinkage intensity for the correlation matrix. If not specified it is 
estimated from the data. <code>lambda=0</code> implies no shrinkage
and <code>lambda=1</code> complete shrinkage. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda.var</code></td>
<td>
<p>Shrinkage intensity for the variances. If not specified it is 
estimated from the data. <code>lambda.var=0</code> implies no shrinkage
and <code>lambda.var=1</code> complete shrinkage. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda.freqs</code></td>
<td>
<p>Shrinkage intensity for the frequencies. If not specified it is 
estimated from the data. <code>lambda.freqs=0</code> implies no shrinkage (i.e. empirical frequencies)
and <code>lambda.freqs=1</code> complete shrinkage (i.e. uniform frequencies). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diagonal</code></td>
<td>
<p>Chooses between LDA (default, <code>diagonal=FALSE</code>) and DDA (<code>diagonal=TRUE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ranking.score</code></td>
<td>
<p>how to compute the summary score for each variable from the CAT scores of all classes - see Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>compute FDR values and HC scores for each feature.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.fdr</code></td>
<td>
<p>Show plot with estimated FDR values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print out some info while computing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An "sda.ranking" object â€“ this is produced by the sda.ranking() function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top</code></td>
<td>
<p>The number of top-ranking features shown in the plot (default: 40).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrow.col</code></td>
<td>
<p>Color of the arrows in the plot (default is <code>"blue"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zeroaxis.col</code></td>
<td>
<p>Color for the center zero axis (default is <code>"red"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>Label written next to feature list (default is <code>"Features"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>Main title (if missing, <code>"The", top, "Top Ranking Features"</code> is used).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other options passed on to generic plot().</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For each predictor variable and centroid a shrinkage CAT scores of the mean versus
the pooled mean is computed.   If there are only two classes the CAT score vs. the pooled
mean reduces to the CAT score between the two group means.
Moreover, in the diagonal case (LDA) the (shrinkage) CAT score reduces to the (shrinkage) t-score.
</p>
<p>The overall ranking of a feature is determine by computing a summary score from the CAT scores.
This is controlled by the option <code>ranking.score</code>.  The default setting
(<code>ranking.score="entropy"</code>) uses mutual information
between the response and the respective predictors (<code>ranking.score</code>) for ranking.  This is equivalent to
a weighted sum of squared CAT scores across the classes.  Another possibility is to employ
the average of the squared CAT scores for ranking (as suggested in Ahdesm\"aki and Strimmer 2010) 
by setting <code>ranking.score="avg"</code>.  A third option is to use the maximum of the squared CAT scores across groups (similarly as in the PAM algorithm) via setting <code>ranking.score="max"</code>.
Note that in the case of two classes all three options are equivalent and
lead to identical scores.  Thus, the choice of <code>ranking.score</code> is important only
in the multi-class setting.  In the two-class case the features are simply ranked according to the
(shrinkage) squared CAT-scores (or t-scores, if there is no correlation among predictors).
</p>
<p>The current default approach is to use ranking by mutual information (i.e. relative entropy
between full model vs. model without predictor) and to use shrinkage estimators of frequencies.
In order to reproduce exactly the ranking computed by previous versions (1.1.0 to 1.3.0) of the <code>sda</code> package set the options  <code>ranking.score="avg"</code> and <code>lambda.freqs=0</code>.
</p>
<p>Calling <code>sda.ranking</code> is step 1 in a classification analysis with the
sda package.  Steps 2 and 3 are 
<code>sda</code> and <code>predict.sda</code>
</p>
<p>See Zuber and Strimmer (2009) for CAT scores in general, and 
Ahdesm\"aki and Strimmer (2010) for details on multi-class CAT scores.
For shrinkage t scores see Opgen-Rhein and Strimmer (2007). 
</p>


<h3>Value</h3>

<p><code>sda.ranking</code> returns a matrix with the following columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>idx</code></td>
<td>
<p>original feature number</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>score</code></td>
<td>
<p>sum of the squared CAT scores across groups - this determines the overall ranking of a feature</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cat</code></td>
<td>
<p>for each group and feature the cat score of the centroid versus the pooled mean</p>
</td>
</tr>
</table>
<p>If <code>fdr=TRUE</code> then additionally local false discovery rate (FDR) values 
as well as higher criticism (HC) scores are computed for each feature
(using <code>fdrtool</code>).
</p>


<h3>Author(s)</h3>

<p>Miika Ahdesm\"aki, Verena Zuber, Sebastian Gibb, and Korbinian Strimmer (<a href="https://strimmerlab.github.io">https://strimmerlab.github.io</a>).
</p>


<h3>References</h3>

<p>Ahdesm\"aki, A., and K. Strimmer. 2010.  Feature selection in omics prediction problems 
using cat scores and false non-discovery rate control. Ann. Appl. Stat. 4: 503-519.
&lt;DOI:10.1214/09-AOAS277&gt;
</p>
<p>Opgen-Rhein, R., and K. Strimmer. 2007. Accurate ranking of differentially expressed 
genes by a distribution-free shrinkage approach.  
Statist. Appl. Genet. Mol. Biol. 6:9.
&lt;DOI:10.2202/1544-6115.1252&gt;
</p>
<p>Zuber, V., and K. Strimmer. 2009.  Gene ranking and biomarker discovery under correlation.
Bioinformatics 25: 2700-2707. &lt;DOI:10.1093/bioinformatics/btp460&gt;
</p>


<h3>See Also</h3>

<p><code>catscore</code>, <code>sda</code>, <code>predict.sda</code>.</p>


<h3>Examples</h3>

<pre><code class="language-R"># load sda library
library("sda")

################# 
# training data #
#################

# prostate cancer set
data(singh2002)

# training data
Xtrain = singh2002$x
Ytrain = singh2002$y

######################################### 
# feature ranking (diagonal covariance) #
#########################################

# ranking using t-scores (DDA)
ranking.DDA = sda.ranking(Xtrain, Ytrain, diagonal=TRUE)
ranking.DDA[1:10,]

# plot t-scores for the top 40 genes
plot(ranking.DDA, top=40) 

# number of features with local FDR &lt; 0.8 
# (i.e. features useful for prediction)
sum(ranking.DDA[,"lfdr"] &lt; 0.8)

# number of features with local FDR &lt; 0.2 
# (i.e. significant non-null features)
sum(ranking.DDA[,"lfdr"] &lt; 0.2)

# optimal feature set according to HC score
plot(ranking.DDA[,"HC"], type="l")
which.max( ranking.DDA[1:1000,"HC"] ) 


##################################### 
# feature ranking (full covariance) #
#####################################

# ranking using CAT-scores (LDA)
ranking.LDA = sda.ranking(Xtrain, Ytrain, diagonal=FALSE)
ranking.LDA[1:10,]

# plot t-scores for the top 40 genes
plot(ranking.LDA, top=40) 

# number of features with local FDR &lt; 0.8 
# (i.e. features useful for prediction)
sum(ranking.LDA[,"lfdr"] &lt; 0.8)

# number of features with local FDR &lt; 0.2 
# (i.e. significant non-null features)
sum(ranking.LDA[,"lfdr"] &lt; 0.2)

# optimal feature set according to HC score
plot(ranking.LDA[,"HC"], type="l")
which.max( ranking.LDA[1:1000,"HC"] ) 

</code></pre>


</div>