<div class="container">

<table style="width: 100%;"><tr>
<td>SeedMatchR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find seed matches in genomic features</h2>

<h3>Description</h3>

<p>Find seed matches in a <code>DNAStringSet</code> object of
sequences. This function will use <code>get.seed</code> extract the seed sequence
from the guide sequence. The seed is then searched across all rows of the
<code>DNAStringSet</code> object using <code>vpatterncount</code>.
</p>
<p>This function returns the input DESeq2 results <code>data.frame</code> with an
additional column that contains the counts for the input <code>seed.name</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SeedMatchR(
  res,
  gtf,
  seqs,
  sequence,
  seed.name = "mer7m8",
  col.name = NULL,
  mismatches = 0,
  indels = FALSE,
  tx.id.col = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>res</code></td>
<td>
<p>A DESeq2 results <code>data.frame</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gtf</code></td>
<td>
<p>GTF file used to map features to genes. The object must have
columns transcript_id and gene_id</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>The <code>DNAStringSet</code> object with sequence information
for features. The names of the sequences should be the transcript names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>The <code>DNAString</code> guide sequence oriented 5' &gt; 3'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed.name</code></td>
<td>
<p>The name of specific seed to extract. Options are:  mer8,
mer7A1, mer7m8, mer6</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.name</code></td>
<td>
<p>The string to use for the column name. Defaults to seed name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mismatches</code></td>
<td>
<p>The number of mismatches to allow in search</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indels</code></td>
<td>
<p>Whether to allow indels in search</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tx.id.col</code></td>
<td>
<p>Use the transcript_id column instead of gene_id</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A modified DESeq2 results dataframe that has column named after the
seed of choice representing the number of match counts.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
library(dplyr)

seq = "UUAUAGAGCAAGAACACUGUUUU"

anno.db = load_species_anno_db("human")

features = get_feature_seqs(anno.db$tx.db, anno.db$dna)

# Load test data
res &lt;- Schlegel_2022_Ttr_D1_30mkg

# Filter DESeq2 results for SeedMatchR
res = filter_deseq(res, fdr.cutoff=1, fc.cutoff=0, rm.na.log2fc = TRUE)

res = SeedMatchR(res, anno.db$gtf, features$seqs, seq, "mer7m8")

</code></pre>


</div>