<div class="container">

<table style="width: 100%;"><tr>
<td>get_feature_seqs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get transcripts features and feature sequences</h2>

<h3>Description</h3>

<p>This function is used to get the genomic features of interest
and the DNA sequences associated with them. This function takes advantage of
the <code>GenomicFeatures</code> package functions <code>threeUTRsByTranscript</code>,
<code>fiveUTRsByTranscript</code>, <code>exonsBy</code>, <code>intronsByTranscript</code>, and <code>cdsBy</code>. These
functions are used to generate the features given an input <code>tx.db</code> object. A
2bit <code>dna</code> input is also required for extracting features sequences.
</p>
<p>The output of the this function is:
</p>

<ul>
<li> <p><code style="white-space: pre;">⁠$db⁠</code>: the feature <code>GRanges</code> object
</p>
</li>
<li> <p><code style="white-space: pre;">⁠$seqs⁠</code>: <code>DNAStringSet</code> of sequences associated to those features
</p>
</li>
</ul>
<h3>Usage</h3>

<pre><code class="language-R">get_feature_seqs(tx.db, dna, feature.type = "3UTR")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tx.db</code></td>
<td>
<p>A tx.db object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dna</code></td>
<td>
<p>A 2bit dna sequence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feature.type</code></td>
<td>
<p>The type of feature to return. Options: 3UTR, 5UTR,
exons, introns, cds</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list containing the feature db object and the feature sequences
</p>


<h3>Examples</h3>

<pre><code class="language-R">
anno.db = load_species_anno_db("human")

features = get_feature_seqs(anno.db$tx.db, anno.db$dna)

</code></pre>


</div>