<div class="container">

<table style="width: 100%;"><tr>
<td>trimgalore</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run Trim Galore!</h2>

<h3>Description</h3>

<p>This function calls
<a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">trim_galore</a>
using <code>system2()</code>, and is only designed to handle standard adapter/quality
trimming. To run in parallel, register a parallel backend, e.g., using
<code>doParallel::registerDoParallel()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">trimgalore(
  filepaths,
  outputDir = "trimgalore_output",
  cmd = "trim_galore",
  args = NULL,
  pigzCmd = "pigz"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filepaths</code></td>
<td>
<p>Paths to fastq files. For single-end reads, each element
should be a single filepath. For paired-end reads, each element should be
two filepaths separated by ";".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputDir</code></td>
<td>
<p>Directory in which to store output. Will be created if it
doesn't exist.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cmd</code></td>
<td>
<p>Name or path of the command-line interface.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>args</code></td>
<td>
<p>Additional arguments to pass to the command-line interface.
Output files will always be compressed. Arguments "–gzip", "–cores",
"-j", and "–basename" are not allowed. Arguments "-o" and "–paired"
should not be specified here.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pigzCmd</code></td>
<td>
<p>String for pigz command, which will gzip the output files.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector of exit codes, invisibly.
</p>


<h3>See Also</h3>

<p><code>seeker()</code>
</p>


</div>