<div class="container">

<table style="width: 100%;"><tr>
<td>EXP_raw</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulated Gene Expression
</h2>

<h3>Description</h3>

<p>Gene expression profiles have been generated for 30 patients and 500 genes. Background correlation is set to 0.08 and the correlation for H1 regions to 0.5. The location of the H1 regions is as suggested in the work of Lai et al.(2005), i.e region 1 [101, 105], region 2 [201, 210], region 3 [301, 320] and region 4 [401, 440].  </p>


<h3>Usage</h3>

<pre><code class="language-R">data("EXP_raw")</code></pre>


<h3>Format</h3>

<p>Data frame containing the gene expression signal for 500 genes (rows) on 30 patients (columns).
</p>


<h3>References</h3>

<p>Lai, W. R., Johnson, M. D., Kucherlapati, R., &amp; Park, P. J. (2005). Comparative analysis of algorithms for identifying amplifications and deletions in array CGH data. Bioinformatics, 21(19), 3763-3770.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#data(EXP_raw)
#G = cor(t(EXP_raw))## calculating the gene x gene correlation matrix
#image(G)## plotting the correlation matrix
</code></pre>


</div>