<div class="container">

<table style="width: 100%;"><tr>
<td>segmented_signal</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Mean Segmentation
</h2>

<h3>Description</h3>

<p>Mean segmentation on the genomic signal is performed using the Fpsn function of the jointseg package. </p>


<h3>Usage</h3>

<pre><code class="language-R">segmented_signal(SNP.Chr, Kmax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SNP.Chr</code></td>
<td>

<p>SNP/CGH profile matrix for a given chromosome (NA's not allowed). Columns correspond to patients and rows to probes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Kmax</code></td>
<td>

<p>Maximum number of segments.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Smoothed genomic signal matrix. Rows correspond to probes and columns to patients.
</p>


<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Morgane Pierre-Jean, Guillem Rigaill and Pierre Neuvial. Performance evaluation of DNA copy number segmentation methods. Briefings in Bioinformatics (2015) 16 (4): 600-615.
</p>


<h3>See Also</h3>

<p><code>CNV_correction</code>, <code>Fpsn</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#data(SNP)
#mu.SNP = segmented_signal(SNP ,100)
</code></pre>


</div>