<div class="container">

<table style="width: 100%;"><tr>
<td>CNV_correction</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Corrects Gene Expression for CNV
</h2>

<h3>Description</h3>

<p>Correcting gene expression signal for CNV. 

</p>


<h3>Usage</h3>

<pre><code class="language-R">CNV_correction(s.Position.EXP, e.Position.EXP, Position.SNP, mu.SNP, EXP)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>s.Position.EXP</code></td>
<td>

<p>vector with gene start position
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>e.Position.EXP</code></td>
<td>
<p>vector with gene end position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Position.SNP</code></td>
<td>
<p>vector with SNP/CGH positions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu.SNP</code></td>
<td>
<p>Smoothed genomic signal matrix not containing NA values. Rows correspond to probes, while columns to patients. The ordering of the patients must be the same as in the EXP matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP</code></td>
<td>
<p>Gene expression matrix must not contain NA's and genes with same expression value (i.e. null gene). Rows correspond to probes, while columns to patients.  Again, ordering of patients must be the same between EXP and mu.SNP matrices.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Overlapping genes may correspond to the same SNP/CGH probes.
</p>



<h3>Value</h3>

<p>CNV corrected signal matrix.
</p>


<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Delatola E. I., Lebarbier E., Mary-Huard T., Radvanyi F., Robin S., Wong J.(2017). SegCorr: a statistical procedure for the detection of genomic regions of correlated expression. BMC Bioinformatics, 18:333.
</p>


<h3>See Also</h3>

<p><code>segmented_signal</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#data.sets = c('SNP','EXP_raw')
## Each gene corresponds to one SNP probe ##
#Position_EXP = matrix(1:1000,nrow=500,byrow=TRUE)
#Position_SNP = seq(2,1000,by=2)
#data(list=data.sets)
#mu.SNP = segmented_signal(SNP ,100) ## smoothed SNP signal
#EXP.CNV = CNV_correction(Position_EXP[,1], Position_EXP[,2], Position_SNP,
#mu.SNP, EXP_raw)## corrected signal

</code></pre>


</div>