<div class="container">

<table style="width: 100%;"><tr>
<td>SegCorr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Performs CNV Correction and Correlation Matrix Segmentation
</h2>

<h3>Description</h3>

<p>Gene expression is corrected for CNV events  must not contain NA's and genes with same expression value (i.e. null gene expression). Segmentation is used to detect changes in the correlation pattern. Regions with high
correlation are identified using an exact test.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SegCorr(CHR, EXP, genes,S, CNV, SNPSMOOTH, Position.EXP, SNP.CHR, SNP, Position.SNP, Kmax)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>CHR</code></td>
<td>

<p>Chromosome allocation vector for the genes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP</code></td>
<td>

<p>Gene expression matrix (raw/corrected for CNV). Columns correspond to patients and rows to genes. The expression matrix must not contain either NA's or genes with same expression value (i.e. null gene expression)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>

<p>Gene ID(name) vector.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>S</code></td>
<td>

<p>Threshold for model selection. Default S=0.7.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CNV</code></td>
<td>

<p>Logical variable indicating whether to perform CNV correction. When CNV=T, the correction is performed. Default value CNV=F.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNPSMOOTH</code></td>
<td>

<p>(Optional Argument when CNV=T) Logical variable indicating whether to perform SNPSMOOTH. When SNPSMOOTH=T, the smoothing is performed. Default value SNPSMOOTH=F.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Position.EXP</code></td>
<td>
<p>(Optional Argument when CNV=T) Expression position matrix.
First column is the start position and the second is the end position.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNP.CHR</code></td>
<td>
<p>(Optional Argument when CNV=T)
Chromosome allocation vector for genomic probes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNP</code></td>
<td>
<p>(Optional Argument when CNV=T)
SNP profile matrix not containing NA's. Columns correspond to patients and rows to probes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Position.SNP</code></td>
<td>
<p>(Optional Argument when CNV=T)
vector with SNP positions
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Kmax</code></td>
<td>
<p>(Optional Argument when CNV=T and SNPSMOOTH=T)
Maximum number of segments. (mean profile segmentation)
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Overlapping genes may correspond to the same genomic probes.

</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Results</code></td>
<td>
<p>Matrix containing information about the genomic regions. Each region corresponds to a row of the matrix, the one  with the smallest p-value is on the top of the list.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$CHR</code></td>
<td>
<p>Chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$Start/End</code></td>
<td>
<p>the region boundaries  with repsect to the physical location of the gene in the chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$Rho</code></td>
<td>
<p><code class="reqn">\rho</code> correlation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$length</code></td>
<td>
<p>number of genes in the region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$first/last gene</code></td>
<td>
<p>name of the first/last gene in the region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$p-value</code></td>
<td>
<p>p-value as obtained from the test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$genes</code></td>
<td>
<p>names of the genes belonging to the region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$p-valueadj</code></td>
<td>
<p>p-value of the region corrected for multiple testing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chromosome.Inf</code></td>
<td>
<p>Matrix containing the estimated background correlation (rho0.hat) per chromsome, the number of segments and the log-loglikehood.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP.corrected</code></td>
<td>
<p>If the CNV option is chosen, the corrected signal is given.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Delatola E. I., Lebarbier E., Mary-Huard T., Radvanyi F., Robin S., Wong J.(2017). SegCorr: a statistical procedure for the detection of genomic regions of correlated expression. BMC Bioinformatics, 18:333.
</p>


<h3>See Also</h3>

<p><code>CNV_correction</code>,<code>segmentation</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#data('EXP_raw')
#CHR = rep(1,dim(EXP_raw)[1])

#results = SegCorr(CHR = CHR, EXP = EXP_raw, CNV = FALSE,S=0.7)

################drawing the heatmap for one region ###########################
#tau = results$Region.List[1,2]: results$Region.List[1,3]
#heatmap(as.matrix(EXP_raw[tau,]))

</code></pre>


</div>