<div class="container">

<table style="width: 100%;"><tr>
<td>segmentation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Correlation Matrix Segmentation
</h2>

<h3>Description</h3>

<p>For a given chromosome, gene correlation matrix segmentation is performed. Regions with high
correlation are identified using an exact test. The expression matrix must not contain NA's and genes with same expression value (i.e. null gene expression).
</p>


<h3>Usage</h3>

<pre><code class="language-R">segmentation(CHR, EXP, genes, S)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>CHR</code></td>
<td>

<p>chromosome name
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EXP</code></td>
<td>

<p>Gene expression matrix (raw/corrected for CNV). Columns correspond to patients and rows to genes. The expression matrix must not contain either NA's or genes with same expression value (i.e. null gene expression).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>

<p>Gene ID(name) vector.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>S</code></td>
<td>

<p>Threshold for model selection. Default S=0.7.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Results</code></td>
<td>
<p>Matrix containing information about the genomic regions. Each region corresponds to a row of the matrix, the one  with the smallest p-value is on the top of the list.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$CHR</code></td>
<td>
<p>Chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$Start/End</code></td>
<td>
<p>region boundaries  with respect to the physical location of the gene in the chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$Rho</code></td>
<td>
<p><code class="reqn">\rho</code> correlation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$length</code></td>
<td>
<p>number of genes in the region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$first/last gene</code></td>
<td>
<p>name of the first/last gene in the region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$p-value</code></td>
<td>
<p>p-value as obtained from the test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results$genes</code></td>
<td>
<p>names of genes belonging to the region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho0</code></td>
<td>
<p>estimate of the background correlation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>likelihood</code></td>
<td>
<p>log-likelihood</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>number of segments</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>E. I. Delatola, E. Lebarbier, T. Mary-Huard, F. Radvanyi, S. Robin, J. Wong.
</p>


<h3>References</h3>

<p>Delatola E. I., Lebarbier E., Mary-Huard T., Radvanyi F., Robin S., Wong J.(2017). SegCorr: a statistical procedure for the detection of genomic regions of correlated expression. BMC Bioinformatics, 18:333.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#data(EXP_raw)
#G = cor(t(EXP_raw))## calculating the gene x gene correlation matrix
#image(G)## plotting the correlation matrix
#results = segmentation(EXP = EXP_raw)
</code></pre>


</div>