<div class="container">

<table style="width: 100%;"><tr>
<td>haploProb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Probability of observing the target haplotype given the parental genotype.</h2>

<h3>Description</h3>

<p>These functions derive the symbolic formula for the
probability of observing the target haplotype given the parental
genotype.</p>


<h3>Usage</h3>

<pre><code class="language-R">haploProb.2M(gam1,gam2,target) ## two-marker genotypes
haploProb.3M(gam1,gam2,target) ## three-marker genotypes
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gam1,gam2</code></td>
<td>
<p>Three (three-marker genotypes) or two (two-marker
genotypes) character string with the configuration of gamete one and
two of the parental genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target</code></td>
<td>
<p>Three (three-marker genotypes) or two (two-marker
genotypes) character string with the configuration of the target
haplotype.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The idea behind the algorithm is to conceptually "recode" the
alleles of the parental genotype into "target" and "non-target",
where "target" is relative to the target haplotype. Then the rules are
determined that would rearrange the gametes of the parental genotype into a
"target-target-target" haplotype. These rearrangement rules are then
translated into the symbolic formula.</p>


<h3>Value</h3>

<p>A character string with the symbolic formula. For three-marker
genotypes, <code>x</code> is the recombination frequency between markers 1
and 2 and <code>y</code> that between markers 2 and 3. For two-marker
genotypes, <code>z</code> is the recombination frequency between markers 1 and
2.</p>


<h3>Author(s)</h3>

<p>Frank Technow
</p>


<h3>Examples</h3>

<pre><code class="language-R">haploProb.2M("AA","BB","AB")
haploProb.3M("AAA","BBB","ABA")
</code></pre>


</div>