<div class="container">

<table style="width: 100%;"><tr>
<td>select_counts</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Subsample the rows and columns of a count matrix.</h2>

<h3>Description</h3>

<p>It is a good idea to subsample (each iteration) the genes and samples from
a real RNA-seq dataset prior to applying <code>thin_diff</code>
(and related functions) so that your conclusions are not dependent on the
specific structure of your dataset. This function is designed to efficiently
do this for you.
</p>


<h3>Usage</h3>

<pre><code class="language-R">select_counts(
  mat,
  nsamp = ncol(mat),
  ngene = nrow(mat),
  gselect = c("random", "max", "mean_max", "custom"),
  gvec = NULL,
  filter_first = FALSE,
  nskip = 0L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat</code></td>
<td>
<p>A numeric matrix of RNA-seq counts. The rows index the genes
and the columns index the samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsamp</code></td>
<td>
<p>The number of samples (columns) to select from <code>mat</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ngene</code></td>
<td>
<p>The number of genes (rows) to select from <code>mat</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gselect</code></td>
<td>
<p>How should we select the subset of genes? Options include:
</p>

<dl>
<dt><code>random</code></dt>
<dd>
<p>Randomly select the genes, with each gene having
an equal probability of being included in the subsampled matrix.</p>
</dd>
<dt><code>max</code></dt>
<dd>
<p>Choose the <code>ngene</code> most median-expressed genes.
Ties are broken by mean-expression.</p>
</dd>
<dt><code>mean_max</code></dt>
<dd>
<p>Choose the <code>ngene</code> most mean-expressed
genes.</p>
</dd>
<dt><code>custom</code></dt>
<dd>
<p>A user-specified list of genes. If
<code>gselect = "custom"</code> then <code>gvec</code> needs to be
non-<code>NULL</code>.</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gvec</code></td>
<td>
<p>A logical vector of length <code>nrow(mat)</code>. A <code>TRUE</code>
in position <code class="reqn">i</code> indicates inclusion into the smaller dataset.
Hence, <code>sum(gvec)</code> should equal <code>ngene</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_first</code></td>
<td>
<p>Should we first filter genes by the method of
Chen et al. (2016) (<code>TRUE</code>) or not (<code>FALSE</code>)? If
<code>TRUE</code> then the <code>edgeR</code> package should be installed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nskip</code></td>
<td>
<p>The number of median-maximally expressed genes to skip.
Not used if <code>gselect = "custom"</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The samples (columns) are chosen randomly, with each sample having
an equal probability of being in the sub-matrix. The genes are selected
according to one of four schemes (see the description of the <code>gselect</code>
argument).
</p>
<p>If you have edgeR installed, then some functionality is provided for
filtering out the lowest expressed genes prior to applying subsampling
(see the <code>filter_first</code> argument).
This filtering scheme is described in Chen et al. (2016).
If you want more control over this filtering, you should use
the <code>filterByExpr</code> function from edgeR directly. You
can install edgeR by following instructions at
<a href="https://doi.org/10.18129/B9.bioc.edgeR">doi:10.18129/B9.bioc.edgeR</a>.
</p>


<h3>Value</h3>

<p>A numeric matrix, which is a <code>ngene</code> by <code>nsamp</code> sub-matrix
of <code>mat</code>. If <code>rownames(mat)</code> is <code>NULL</code>, then the
row names of the returned matrix are the indices in <code>mat</code> of the
selected genes. If <code>colnames(mat)</code> is <code>NULL</code>, then the
column names of the returned matrix are the indices in <code>mat</code> of
the selected samples.
</p>


<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul><li>
<p>Chen, Yunshun, Aaron TL Lun, and Gordon K. Smyth. "From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline." <em>F1000Research</em> 5 (2016). <a href="https://doi.org/10.12688/f1000research.8987.2">doi:10.12688/f1000research.8987.2</a>.
</p>
</li></ul>
<h3>Examples</h3>

<pre><code class="language-R">## Simulate data from given matrix of counts
## In practice, you would obtain mat from a real dataset, not simulate it.
set.seed(1)
n   &lt;- 100
p   &lt;- 1000
mat &lt;- matrix(stats::rpois(n * p, lambda = 50), nrow = p)

## Subsample the matrix, then feed it into a thinning function
submat &lt;- select_counts(mat = mat, nsamp = 10, ngene = 100)
thout  &lt;- thin_2group(mat = submat, prop_null = 0.5)

## The rownames and colnames (if NULL in mat) tell you which genes/samples
## were selected.
rownames(submat)
colnames(submat)

</code></pre>


</div>