<div class="container">

<table style="width: 100%;"><tr>
<td>seqgendiff-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>seqgendiff: RNA-Seq Generation/Modification for Simulation</h2>

<h3>Description</h3>

<p>This package is designed to take real RNA-seq data and alter it by
adding a known amount of signal. You can then use this modified dataset
in simulation studies for differential expression analysis, factor
analysis, confounder adjustment, or library size adjustment. The
advantage of this way of simulating data is that you can see how
your method behaves when the simulated data exhibit common
(and annoying) features of real data. For example, in the real world
data are not normally (or negative binomially) distributed and
unobserved confounding is a major issue. This package will simulate
data that exhibit these characteristics. The methods used in this
package are described in detail in Gerard (2020).
</p>


<h3>seqgendiff Functions</h3>


<dl>
<dt><code>select_counts()</code></dt>
<dd>
<p>Subsample the columns and rows
of a real RNA-seq count matrix. You would then feed this sub-matrix
into one of the thinning functions below.</p>
</dd>
<dt><code>thin_diff()</code></dt>
<dd>
<p>The function most users should
be using for general-purpose binomial thinning. For the special
applications of the two-group model or library/gene thinning, see
the functions listed below.</p>
</dd>
<dt><code>thin_2group()</code></dt>
<dd>
<p>The specific application of
thinning in the two-group model.</p>
</dd>
<dt><code>thin_lib()</code></dt>
<dd>
<p>The specific application of
library size thinning.</p>
</dd>
<dt><code>thin_gene()</code></dt>
<dd>
<p>The specific application of
total gene expression thinning.</p>
</dd>
<dt><code>thin_all()</code></dt>
<dd>
<p>The specific application of thinning
all counts.</p>
</dd>
<dt><code>effective_cor()</code></dt>
<dd>
<p>Returns an estimate of the actual
correlation between surrogate variables and a user-specified
design matrix.</p>
</dd>
<dt><code>ThinDataToSummarizedExperiment()</code></dt>
<dd>
<p>Converts a
ThinData object to a SummarizedExperiment object.</p>
</dd>
<dt><code>ThinDataToDESeqDataSet()</code></dt>
<dd>
<p>Converts a
ThinData object to a DESeqDataSet object.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>David Gerard
</p>


<h3>References</h3>


<ul><li>
<p>Gerard, D (2020). "Data-based RNA-seq simulations by binomial thinning." <em>BMC Bioinformatics</em>. 21(1), 206. <a href="https://doi.org/10.1186/s12859-020-3450-9">doi:10.1186/s12859-020-3450-9</a>.
</p>
</li></ul>
<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/dcgerard/seqgendiff">https://github.com/dcgerard/seqgendiff</a>
</p>
</li>
<li>
<p> Report bugs at <a href="https://github.com/dcgerard/seqgendiff/issues">https://github.com/dcgerard/seqgendiff/issues</a>
</p>
</li>
</ul>
</div>