<div class="container">

<table style="width: 100%;"><tr>
<td>gbk2g2.euk</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Conversion of a GenBank format file into a glimmer-like
one. Eukaryotic version.</h2>

<h3>Description</h3>

<p>This function reads a file in GenBank format and converts the features
corresponding to CDS (Coding Sequences) into a format similar to
glimmer program output. This function is specifically made for
eukaryotic sequences, i.e. with introns. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">gbk2g2.euk(gbkfile = system.file("sequences/ame1.gbk", package ="seqinr"),
g2.coord = "g2.coord")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gbkfile</code></td>
<td>
<p> The name of the GenBank file </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g2.coord</code></td>
<td>
<p> The name of the output file </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function returns the coordinates of the exons annotated in the
GenBank format file. 
</p>


<h3>Value</h3>

<p>A data frame with three columns will be written to the <code>g2.coord</code>
file. The first column corresponds to the name of the gene, given in the
GenBank file through the <code>/gene</code> feature. The second and third
column contain the start and the stop position of the exon.
</p>


<h3>Author(s)</h3>

<p>J.R. Lobry, A. Nec≈üulea</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

 <p><code>oriloc</code>, <code>gbk2g2</code> </p>


<h3>Examples</h3>

<pre><code class="language-R"> ## Not run:  gbk2g2.euk() 
</code></pre>


</div>