<div class="container">

<table style="width: 100%;"><tr>
<td>uco</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Codon usage indices </h2>

<h3>Description</h3>

<p><code>uco</code> calculates some codon usage indices: the codon counts <code>eff</code>, the relative frequencies <code>freq</code> or the Relative Synonymous Codon Usage <code>rscu</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">uco(seq, frame = 0, index = c("eff", "freq", "rscu"), as.data.frame = FALSE,
NA.rscu = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p> a coding sequence as a vector of chars </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frame</code></td>
<td>
<p> an integer (0, 1, 2) giving the frame of the coding sequence </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index</code></td>
<td>
<p> codon usage index choice, partial matching is allowed.
<code>eff</code> for codon counts,
<code>freq</code> for codon relative frequencies,
and <code>rscu</code> the RSCU index.<br></p>
<p>"eff", "freq", and "rscu" correspond to "R0", "R1", and "R3", respectively, in Suzuki et al. (2005) "2.2 Normalization of codon usage data".<br></p>
<p>"eff" and "rscu" correspond to "AF" and "RSCU", respectively, in Suzuki et al. (2008) "2.2. Definitions of codon usage data".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as.data.frame</code></td>
<td>
<p> logical. If <code>TRUE</code>: all indices are returned into a data frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NA.rscu</code></td>
<td>
<p> when an amino-acid is missing, RSCU are no more defined and repported
as missing values (<code>NA</code>). You can force them to another value (typically 0 or
1) with this argument.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Codons with ambiguous bases are ignored.<br></p>
<p>RSCU is a simple measure of non-uniform usage of synonymous codons in a coding sequence
(Sharp <em>et al.</em> 1986).
RSCU values are the number of times a particular codon is observed, relative to the number
of times that the codon would be observed for a uniform synonymous codon usage (i.e. all the
codons for a given amino-acid have the same probability).
In the absence of any codon usage bias, the RSCU values would be 1.00 (this is the case
for sequence <code>cds</code> in the exemple thereafter). A codon that is used
less frequently than expected will have an RSCU value of less than 1.00 and vice versa for a codon
that is used more frequently than expected.<br></p>
<p>Do not use correspondence analysis on RSCU tables as this is a source of artifacts
(Perrière and Thioulouse 2002, Suzuki <em>et al.</em> 2008). Within-aminoacid correspondence analysis is a
simple way to study synonymous codon usage (Charif <em>et al.</em> 2005). For an introduction
to correspondence analysis and within-aminoacid correspondence analysis see the
chapter titled <em>Multivariate analyses</em> in the seqinR manual that ships with the
seqinR package in the <b>doc</b> folder. You can also use internal correspondence
analysis if you want to analyze simultaneously a row-block structure such as the
within and between species variability (Lobry and Chessel 2003).<br></p>
<p>If <code>as.data.frame</code> is FALSE, <code>uco</code> returns one of these:
</p>

<dl>
<dt> eff </dt>
<dd>
<p> a table of codon counts </p>
</dd>
<dt> freq </dt>
<dd>
<p> a table of codon relative frequencies </p>
</dd>
<dt> rscu </dt>
<dd>
<p> a numeric vector of relative synonymous codon usage values</p>
</dd>
</dl>
<p>If <code>as.data.frame</code> is TRUE, <code>uco</code> returns a data frame with five columns:
</p>

<dl>
<dt> aa </dt>
<dd>
<p> a vector containing the name of amino-acid </p>
</dd>
<dt> codon </dt>
<dd>
<p> a vector containing the corresponding codon </p>
</dd>
<dt> eff </dt>
<dd>
<p> a numeric vector of codon counts </p>
</dd>
<dt> freq </dt>
<dd>
<p> a numeric vector of codon relative frequencies </p>
</dd>
<dt> rscu </dt>
<dd>
<p> a numeric vector of RSCU index </p>
</dd>
</dl>
<h3>Value</h3>

<p>If <code>as.data.frame</code> is FALSE, the default, a table for <code>eff</code> and <code>freq</code> and
a numeric vector for <code>rscu</code>. If <code>as.data.frame</code> is TRUE,
a data frame with all indices is returned.
</p>


<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry, G. Perrière</p>


<h3>References</h3>

<p><code>citation("seqinr")</code> <br></p>
<p>Sharp, P.M., Tuohy, T.M.F., Mosurski, K.R. (1986) Codon usage in yeast: cluster
analysis clearly differentiates highly and lowly expressed genes.
<em>Nucl. Acids. Res.</em>, <b>14</b>:5125-5143.<br></p>
<p>Perrière, G., Thioulouse, J. (2002) Use and misuse of correspondence analysis in
codon usage studies. <em>Nucl. Acids. Res.</em>, <b>30</b>:4548-4555.<br></p>
<p>Lobry, J.R., Chessel, D. (2003) Internal correspondence analysis of codon and
amino-acid usage in thermophilic bacteria.
<em>Journal of Applied Genetics</em>, <b>44</b>:235-261. <a href="http://jag.igr.poznan.pl/2003-Volume-44/2/pdf/2003_Volume_44_2-235-261.pdf">http://jag.igr.poznan.pl/2003-Volume-44/2/pdf/2003_Volume_44_2-235-261.pdf</a>.<br></p>
<p>Charif, D., Thioulouse, J., Lobry, J.R., Perrière, G. (2005) Online
Synonymous Codon Usage Analyses with the ade4 and seqinR packages.
<em>Bioinformatics</em>, <b>21</b>:545-547. <a href="https://pbil.univ-lyon1.fr/members/lobry/repro/bioinfo04/">https://pbil.univ-lyon1.fr/members/lobry/repro/bioinfo04/</a>.<br></p>
<p>Suzuki, H., Saito, R. Tomita, R. (2005)
A problem in multivariate analysis of codon usage data and a possible solution.
<em>FEBS Lett.</em>, <b>579</b>:6499-504. <a href="https://www.thermofisher.com/de/de/home/brands/applied-biosystems.html">https://www.thermofisher.com/de/de/home/brands/applied-biosystems.html</a>.<br></p>
<p>Suzuki, H., Brown, C.J., Forney, L.J., Top, E. (2008)
Comparison of Correspondence Analysis Methods for Synonymous Codon Usage in Bacteria.
<em>DNA Research</em>, <b>15</b>:357-365. <a href="https://academic.oup.com/dnaresearch/article/15/6/357/513030">https://academic.oup.com/dnaresearch/article/15/6/357/513030</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Show all possible codons:
words()

## Make a coding sequence from this:
(cds &lt;- s2c(paste(words(), collapse = "")))

## Get codon counts:
uco(cds, index = "eff")

## Get codon relative frequencies:
uco(cds, index = "freq")

## Get RSCU values:
uco(cds, index = "rscu")

## Show what happens with ambiguous bases:
uco(s2c("aaannnttt"))

## Use a real coding sequence:
rcds &lt;- read.fasta(file = system.file("sequences/malM.fasta", package = "seqinr"))[[1]]
uco( rcds, index = "freq")
uco( rcds, index = "eff")
uco( rcds, index = "rscu")
uco( rcds, as.data.frame = TRUE)

## Show what happens with RSCU when an amino-acid is missing:
ecolicgpe5 &lt;- read.fasta(file = system.file("sequences/ecolicgpe5.fasta",package="seqinr"))[[1]]
uco(ecolicgpe5, index = "rscu")

## Force NA to zero:
uco(ecolicgpe5, index = "rscu", NA.rscu = 0)
</code></pre>


</div>