<div class="container">

<table style="width: 100%;"><tr>
<td>rearranged.oriloc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Detection of replication-associated effects on base composition asymmetry in
prokaryotic chromosomes.</h2>

<h3>Description</h3>

<p>Detection of replication-associated effects on base composition asymmetry in
prokaryotic chromosomes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rearranged.oriloc(seq.fasta = system.file("sequences/ct.fasta.gz", package = "seqinr"),
  g2.coord = system.file("sequences/ct.predict", package = "seqinr"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq.fasta</code></td>
<td>
<p>The path of the file containing a FASTA-format
sequence. Default value: 
the FASTA sequence of the Chlamydia trachomatis chromosome. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g2.coord</code></td>
<td>
<p>The path of the file containing the coordinates of the
protein coding genes found on this chromosome. This file can be
obtained using the function <code>gbk2g2</code>. The format of the file is
similar to the output of the Glimmer2 program. The first column
contains the index or the name of the gene, the second one contains
the start position and the third column contains the end position. For
reverse transcribed genes, the start position is greater than the end position.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The purpose of this method is to decouple replication-related
and coding sequence-related  effects on base composition asymmetry. In
order to do so, the analyzed chromosome is artificially rearranged to
obtain a perfect gene orientation bias - all forward transcribed genes
on the first half of the chromosome, and all reverse transcribed genes
on the other half.
This rearrangement conserves the relative order of genes within each of
the two groups  - both forward-encoded and reverse-encoded genes are
placed on the rearranged chromosome in increasing order of their
coordinates on the real chromosome.
If the replication mechanism has a significant effect on base
composition asymmetry, this should be seen as a change of slope in the
nucleotide skews computed on the rearranged chromosome; the change of
slope should take place at the origin or the terminus of replication.
Use <code>extract.breakpoints</code> to detect the position of the changes in
slope on the rearranged nucleotide skews.
</p>


<h3>Value</h3>

<p>A data.frame with six columns: <code>meancoord.rearr</code> contains the
gene index on the rearranged chromosome; <code>gcskew.rearr</code> contains
the normalized GC-skew ((G-C)/(G+C)) computed on the third codon positions of
protein coding genes, still on the rearranged chromosome;  <code>atskew.rearr</code> contains
the normalized AT-skew ((A-T)/(A+T)) computed on the third codon positions of
protein coding genes; <code>strand.rearr</code> contains the transcription
strand of the gene (either "forward" or "reverse"); <code>order</code>
contains the permutation that was used to obtain a perfect gene
orientation bias; <code>meancoord.real</code> contains the mid-coordinate of
the genes on the real chromosome (before the rearrangement).
</p>


<h3>Author(s)</h3>

<p>A. Necşulea</p>


<h3>References</h3>

<p>Necşulea, A. and Lobry, J.R. (2007) A New Method for Assessing the 
Effect of Replication on DNA Base Composition Asymmetry.
<em>Molecular Biology and Evolution</em>, <b>24</b>:2169-2179.
</p>


<h3>See Also</h3>

 <p><code>oriloc</code>, <code>draw.rearranged.oriloc</code>,
<code>extract.breakpoints</code>  </p>


<h3>Examples</h3>

<pre><code class="language-R">
### Example for Chlamydia trachomatis ####

### Rearrange the chromosome and compute the nucleotide skews ###

## Not run: r.ori &lt;- rearranged.oriloc(seq.fasta = 
   system.file("sequences/ct.fasta.gz", package = "seqinr"),
    g2.coord =  system.file("sequences/ct.predict", package = "seqinr"))
## End(Not run)

### Extract the breakpoints for the rearranged nucleotide skews ###



## Not run: breaks &lt;- extract.breakpoints(r.ori, type = c("gcfw", "gcrev"), 
 nbreaks =c(2, 2), gridsize = 50, it.max = 100)
## End(Not run)



### Draw the rearranged nucleotide skews and place the position of the breakpoints ### 
### on the graphics ###

## Not run: draw.rearranged.oriloc(r.ori, breaks.gcfw = breaks$gcfw$breaks,
 breaks.gcrev = breaks$gcrev$breaks)
## End(Not run)


</code></pre>


</div>