<div class="container">

<table style="width: 100%;"><tr>
<td>EXP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Vectors of coefficients to compute linear forms.</h2>

<h3>Description</h3>

<p>This dataset is used to compute linear forms on codon frequencies:
if <code>codfreq</code> is a vector of codon frequencies then
<code>drop(freq %*% EXP$CG3)</code> will return for instance the G+C content
in third codon positions. Base order is the lexical order: a,
c, g, t (or u).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(EXP)</code></pre>


<h3>Format</h3>

<p>List of 24 vectors of coefficients
</p>

<dl>
<dt>A</dt>
<dd>
<p>num [1:4] 1 0 0 0</p>
</dd>
<dt>A3</dt>
<dd>
<p>num [1:64] 1 0 0 0 1 0 0 0 1 0 ...</p>
</dd>
<dt>AGZ</dt>
<dd>
<p>num [1:64] 0 0 0 0 0 0 0 0 1 0 ...</p>
</dd>
<dt>ARG</dt>
<dd>
<p>num [1:64] 0 0 0 0 0 0 0 0 1 0 ...</p>
</dd>
<dt>AU3</dt>
<dd>
<p>num [1:64] 1 0 0 1 1 0 0 1 1 0 ...</p>
</dd>
<dt>BC</dt>
<dd>
<p>num [1:64] 0 1 0 0 0 0 0 0 0 0 ...</p>
</dd>
<dt>C</dt>
<dd>
<p>num [1:4] 0 1 0 0</p>
</dd>
<dt>C3</dt>
<dd>
<p>num [1:64] 0 1 0 0 0 1 0 0 0 1 ...</p>
</dd>
<dt>CAI</dt>
<dd>
<p>num [1:64]  0.00  0.00 -1.37 -2.98 -2.58 ...</p>
</dd>
<dt>CG</dt>
<dd>
<p>num [1:4] 0 1 1 0</p>
</dd>
<dt>CG1</dt>
<dd>
<p>num [1:64] 0 0 0 0 0 0 0 0 0 0 ...</p>
</dd>
<dt>CG12</dt>
<dd>
<p>num [1:64] 0 0 0 0 0.5 0.5 0.5 0.5 0.5 0.5 ...</p>
</dd>
<dt>CG2</dt>
<dd>
<p>num [1:64] 0 0 0 0 1 1 1 1 1 1 ...</p>
</dd>
<dt>CG3</dt>
<dd>
<p>num [1:64] 0 1 1 0 0 1 1 0 0 1 ...</p>
</dd>
<dt>CGN</dt>
<dd>
<p>num [1:64] 0 0 0 0 0 0 0 0 0 0 ...</p>
</dd>
<dt>F1</dt>
<dd>
<p>num [1:64]  1.026  0.239  1.026  0.239 -0.097 ...</p>
</dd>
<dt>G</dt>
<dd>
<p>num [1:4] 0 0 1 0</p>
</dd>
<dt>G3</dt>
<dd>
<p>num [1:64] 0 0 1 0 0 0 1 0 0 0 ...</p>
</dd>
<dt>KD</dt>
<dd>
<p>num [1:64] -3.9 -3.5 -3.9 -3.5 -0.7 -0.7 -0.7 -0.7 -4.5 -0.8 ...</p>
</dd>
<dt>Q</dt>
<dd>
<p>num [1:64] 0 0 0 0 1 1 1 1 0 0 ...</p>
</dd>
<dt>QA3</dt>
<dd>
<p>num [1:64] 0 0 0 0 1 0 0 0 0 0 ...</p>
</dd>
<dt>QC3</dt>
<dd>
<p>num [1:64] 0 0 0 0 0 1 0 0 0 0 ...</p>
</dd>
<dt>U</dt>
<dd>
<p>num [1:4] 0 0 0 1</p>
</dd>
<dt>U3</dt>
<dd>
<p>num [1:64] 0 0 0 1 0 0 0 1 0 0 ...</p>
</dd>
</dl>
<h3>Details</h3>

<p>It's better to work directly at the amino-acid level
when computing linear forms on amino-acid frequencies so as to have
a single coefficient vector. For instance <code>EXP$KD</code> to compute the Kyte
and Doolittle hydrophaty index from codon frequencies is valid only
for the standard genetic code.<br><br>
An alternative for <code>drop(freq %*% EXP$CG3)</code> is <code>
sum( freq * EXP$CG3 )</code>, but this is less efficient in terms of CPU
time. The advantage of the latter, however, is that thanks to
recycling rules you can use either <code>sum( freq * EXP$A )</code>
or <code>sum( freq * EXP$A3 )</code>. To do the same with the %*%
operator you have to explicit the recycling rule as in <code>
drop( freq %*% rep(EXP$A, 16))</code>.
</p>


<h3>Source</h3>

<p>ANALSEQ EXPFILEs for command EXP.<br><a href="http://pbil.univ-lyon1.fr/software/doclogi/docanals/manuel.html">http://pbil.univ-lyon1.fr/software/doclogi/docanals/manuel.html</a>
</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>

<dl>
<dt>A</dt>
<dd>
<p>content in A nucleotide</p>
</dd>
<dt>A3</dt>
<dd>
<p>content in A nucleotide in third position of codon</p>
</dd>	
<dt>AGZ</dt>
<dd>
<p>Arg content (aga and agg codons)</p>
</dd>
<dt>ARG</dt>
<dd>
<p>Arg content</p>
</dd>
<dt>AU3</dt>
<dd>
<p>content in A and U nucleotides in third position of codon</p>
</dd>
<dt>BC</dt>
<dd>
<p>Good choice (Bon choix). Gouy M., Gautier C. (1982)
codon usage in bacteria : Correlation with gene expressivity. <em>Nucleic Acids Research</em>,<b>10(22)</b>:7055-7074.</p>
</dd>
<dt>C</dt>
<dd>
<p>content in C nucleotides</p>
</dd>
<dt>C3</dt>
<dd>
<p>content in A nucleotides in third position of codon</p>
</dd>
<dt>CAI</dt>
<dd>
<p>Codon adaptation index for E. coli. Sharp, P.M., Li, W.-H. (1987) The codon adaptation index -
a measure of directionam synonymous codon usage bias, and its potential
applications. <em>Nucleic Acids Research</em>,<b>15</b>:1281-1295.</p>
</dd>
<dt>CG</dt>
<dd>
<p>content in G + C nucleotides</p>
</dd>
<dt>CG1</dt>
<dd>
<p>content in G + C nucleotides in first position of codon</p>
</dd>
<dt>CG12</dt>
<dd>
<p>content in G + C nucleotides in first and second position of codon</p>
</dd>
<dt>CG2</dt>
<dd>
<p>content in G + C nucleotides in second position of codon</p>
</dd>
<dt>CG3</dt>
<dd>
<p>content in G + C nucleotides in third position of codon</p>
</dd>
<dt>CGN</dt>
<dd>
<p>content in CGA + CGU + CGA + CGG</p>
</dd>
<dt>F1</dt>
<dd>
<p>From Table 2 in Lobry, J.R., Gautier, C. (1994) Hydrophobicity,
expressivity and aromaticity are the major trends of amino-acid usage in
999 <em>Escherichia coli</em> chromosome-encode genes. <em>Nucleic Acids
Research</em>,<b>22</b>:3174-3180.</p>
</dd>
<dt>G3</dt>
<dd>
<p>content in G nucleotides in third position of codon</p>
</dd>
<dt>KD</dt>
<dd>
<p>Kyte, J., Doolittle, R.F. (1982) A simple method for displaying
the hydropathic character of a protein. <em>J. Mol. Biol.</em>,<b>157</b>
:105-132.</p>
</dd>
<dt>Q</dt>
<dd>
<p>content in quartet</p>
</dd>
<dt>QA3</dt>
<dd>
<p>content in quartet with the A nucleotide in third position</p>
</dd>
<dt>QC3</dt>
<dd>
<p>content in quartet with the A nucleotide in third position</p>
</dd>
<dt>U</dt>
<dd>
<p>content in U nucleotide</p>
</dd>
<dt>U3</dt>
<dd>
<p>content in U nucleotides in third position of codon</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R">data(EXP)
</code></pre>


</div>