<div class="container">

<table style="width: 100%;"><tr>
<td>translate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Translate nucleic acid sequences into proteins </h2>

<h3>Description</h3>

<p>This function translates nucleic acid sequences into the corresponding
peptide sequence. It can translate in any of the 3 forward or three
reverse sense frames. In the case of reverse sense, the reverse-complement
of the sequence is taken. It can translate using the standard (universal)
genetic code and also with non-standard codes. Ambiguous bases can also
be handled.
</p>


<h3>Usage</h3>

<pre><code class="language-R">translate(seq, frame = 0, sens = "F", numcode = 1, NAstring = "X", ambiguous = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p> the sequence to translate as a vector of single characters in lower case letters. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frame</code></td>
<td>
<p> Frame(s) (0,1,2) to translate. By default the frame <code>0</code> is used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sens</code></td>
<td>
<p> Sense to translate: <code>F</code> for forward sense and <code>R</code> for reverse sense. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numcode</code></td>
<td>
<p> The ncbi genetic code number for translation. By default the standard genetic code is used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NAstring</code></td>
<td>
<p> How to translate amino-acids when there are ambiguous bases in codons. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ambiguous</code></td>
<td>
<p> If TRUE, ambiguous bases are taken into account so that for instance
GGN is translated to Gly in the standard genetic code. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The following genetic codes are described here. The number preceding each code
corresponds to <code>numcode</code>.
</p>

<dl>
<dt>1</dt>
<dd>
<p> standard </p>
</dd>
<dt>2</dt>
<dd>
<p> vertebrate.mitochondrial </p>
</dd>
<dt>3</dt>
<dd>
<p> yeast.mitochondrial </p>
</dd>
<dt>4</dt>
<dd>
<p> protozoan.mitochondrial+mycoplasma </p>
</dd>
<dt>5</dt>
<dd>
<p> invertebrate.mitochondrial </p>
</dd>
<dt>6</dt>
<dd>
<p> ciliate+dasycladaceal </p>
</dd>
<dt>9</dt>
<dd>
<p> echinoderm+flatworm.mitochondrial </p>
</dd>
<dt>10</dt>
<dd>
<p> euplotid </p>
</dd>
<dt>11</dt>
<dd>
<p> bacterial+plantplastid </p>
</dd>
<dt>12</dt>
<dd>
<p> alternativeyeast </p>
</dd>
<dt>13</dt>
<dd>
<p> ascidian.mitochondrial </p>
</dd>
<dt>14</dt>
<dd>
<p> alternativeflatworm.mitochondrial </p>
</dd>
<dt>15</dt>
<dd>
<p> blepharism </p>
</dd>
<dt>16</dt>
<dd>
<p> chlorophycean.mitochondrial </p>
</dd>
<dt>21</dt>
<dd>
<p> trematode.mitochondrial </p>
</dd>
<dt>22</dt>
<dd>
<p> scenedesmus.mitochondrial </p>
</dd>
<dt>23</dt>
<dd>
<p> thraustochytrium.mitochondria </p>
</dd>
<dt>24</dt>
<dd>
<p>Pterobranchia.mitochondrial</p>
</dd>
<dt>25</dt>
<dd>
<p>CandidateDivision.SR1+Gracilibacteria</p>
</dd>
<dt>26</dt>
<dd>
<p>Pachysolen.tannophilus</p>
</dd>
</dl>
<h3>Value</h3>

<p><code>translate</code> returns a vector of single characters containing the peptide sequence in
the standard one-letter IUPAC code. Termination (STOP) codons are translated by
the character '*'.
</p>


<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry</p>


<h3>References</h3>

<p>The genetic codes have been taken from the ncbi taxonomy database:
<a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a>.
Last update October 05, 2000.<br>
The IUPAC one-letter code for aminoacids is described at:
<a href="https://www.bioinformatics.org/sms/iupac.html">https://www.bioinformatics.org/sms/iupac.html</a>
</p>
<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

<p>Use <code>tolower</code> to change upper case letters into lower case letters.
For coding sequences obtained from an ACNUC server with <code>query</code> it's
better to use the function <code>getTrans</code> so that the relevant genetic
code and the relevant frame are automatically used.
The genetic codes are given in the object <code>SEQINR.UTIL</code>, a more
human readable form is given by the function <code>tablecode</code>.
Use <code>aaa</code> to get the three-letter code for amino-acids.</p>


<h3>Examples</h3>

<pre><code class="language-R">##
## Toy CDS example invented by Leonor Palmeira:
##
toycds &lt;- s2c("tctgagcaaataaatcgg")
translate(seq = toycds) # should be c("S", "E", "Q", "I", "N", "R")
##
## Toy CDS example with ambiguous bases:
##
toycds2 &lt;- s2c("tcngarcarathaaycgn")
translate(toycds2) # should be c("X", "X", "X", "X", "X", "X")
translate(toycds2, ambiguous = TRUE) # should be c("S", "E", "Q", "I", "N", "R")
translate(toycds2, ambiguous = TRUE, numcode = 2) # should be c("S", "E", "Q", "X", "N", "R")
##
## Real CDS example:
##
realcds &lt;- read.fasta(file = system.file("sequences/malM.fasta", package ="seqinr"))[[1]]
translate(seq = realcds)
# Biologically correct, only one stop codon at the end
translate(seq = realcds, frame = 3, sens = "R", numcode = 6)
# Biologically meaningless, note the in-frame stop codons

# Read from an alignment as suggested by Dr. H. Suzuki
fasta.res    &lt;- read.alignment(file = system.file("sequences/Anouk.fasta", package = "seqinr"),
 format = "fasta")

AA1 &lt;- seqinr::getTrans(s2c(fasta.res$seq[[1]]))
AA2 &lt;- seqinr::translate(s2c(fasta.res$seq[[1]]))
identical(AA1, AA2)

AA1 &lt;- lapply(fasta.res$seq, function(x) seqinr::getTrans(s2c(x)))
AA2 &lt;- lapply(fasta.res$seq, function(x) seqinr::translate(s2c(x)))
identical(AA1, AA2)

## Not run: 
## Need internet connection.
## Translation of the following EMBL entry:
##
## FT   CDS             join(complement(153944..154157),complement(153727..153866),
## FT                   complement(152185..153037),138523..138735,138795..138955)
## FT                   /codon_start=1
## FT                   /db_xref="FLYBASE:FBgn0002781"
## FT                   /db_xref="GOA:Q86B86"
## FT                   /db_xref="TrEMBL:Q86B86"
## FT                   /note="mod(mdg4) gene product from transcript CG32491-RZ;
## FT                   trans splicing"
## FT                   /gene="mod(mdg4)"
## FT                   /product="CG32491-PZ"
## FT                   /locus_tag="CG32491"
## FT                   /protein_id="AAO41581.1"
## FT                   /translation="MADDEQFSLCWNNFNTNLSAGFHESLCRGDLVDVSLAAEGQIVKA
## FT                   HRLVLSVCSPFFRKMFTQMPSNTHAIVFLNNVSHSALKDLIQFMYCGEVNVKQDALPAF
## FT                   ISTAESLQIKGLTDNDPAPQPPQESSPPPAAPHVQQQQIPAQRVQRQQPRASARYKIET
## FT                   VDDGLGDEKQSTTQIVIQTTAAPQATIVQQQQPQQAAQQIQSQQLQTGTTTTATLVSTN
## FT                   KRSAQRSSLTPASSSAGVKRSKTSTSANVMDPLDSTTETGATTTAQLVPQQITVQTSVV
## FT                   SAAEAKLHQQSPQQVRQEEAEYIDLPMELPTKSEPDYSEDHGDAAGDAEGTYVEDDTYG
## FT                   DMRYDDSYFTENEDAGNQTAANTSGGGVTATTSKAVVKQQSQNYSESSFVDTSGDQGNT
## FT                   EAQVTQHVRNCGPQMFLISRKGGTLLTINNFVYRSNLKFFGKSNNILYWECVQNRSVKC
## FT                   RSRLKTIGDDLYVTNDVHNHMGDNKRIEAAKAAGMLIHKKLSSLTAADKIQGSWKMDTE
## FT                   GNPDHLPKM"
choosebank("emblTP")
trans &lt;- query("trans", "N=AE003734.PE35")
trans1 &lt;- getTrans(trans$req[[1]])
## Complex transsplicing operations, the correct frame and the correct
## genetic code are automatically used for translation into protein.
seq &lt;- getSequence(trans$req[[1]])
identical(translate(seq),trans1)
#default frame and genetic code are correct
trans &lt;- query("trans", "N=AB004237")
trans1 &lt;- getTrans(trans$req[[1]])
## Complex transsplicing operations, the correct frame and the correct
## genetic code are automatically used for translation into protein.
seq &lt;- getSequence(trans$req[[1]])
identical(translate(seq),trans1)
#default  genetic code is not correct
identical(translate(seq,numcode=2),trans1)
#genetic code is 2

## End(Not run)
</code></pre>


</div>