<div class="container">

<table style="width: 100%;"><tr>
<td>aacost</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Aerobic cost of amino-acids in Escherichia coli and G+C classes</h2>

<h3>Description</h3>

<p>The metabolic cost of amino-acid biosynthesis in <em>E. coli</em> under aerobic conditions
from table 1 in Akashi and Gojobori (2002). The G+C classes are from Lobry (1997).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(aacost)</code></pre>


<h3>Format</h3>

<p>A data frame with 20 rows for the amino-acids and the following 7 columns:
</p>

<dl>
<dt>aaa</dt>
<dd>
<p>amino-acid (three-letters code).</p>
</dd>
<dt>a</dt>
<dd>
<p>amino-acid (one-letter code).</p>
</dd>
<dt>prec</dt>
<dd>
<p>precursor metabolites (see details).</p>
</dd>
<dt>p</dt>
<dd>
<p>number of high-energy phosphate bonds contained in ATP and GTP molecules.</p>
</dd>
<dt>h</dt>
<dd>
<p>number of available hydrogen atoms carried in NADH, NADPH,
and FADH2 molcules.</p>
</dd>
<dt>tot</dt>
<dd>
<p>total metabolic cost assuming 2 high-energy phosphate bonds
per hydrogen atom.</p>
</dd>
<dt>gc</dt>
<dd>
<p>an ordered factor (l&lt;m&lt;h) for the G+C class of the amino-acid (see details)</p>
</dd>
</dl>
<h3>Details</h3>

<p>Precursor metabolites are: penP, ribose 5-phosphate; PRPP,
5-phosphoribosyl pyrophosphate; eryP, erythrose 4-phosphate; 3pg,
3-phosphoglycerate; pep, phospho<em>enol</em>pyruvate; pyr, pyruvate;
acCoA, acetyl-CoA; akg, alpha-ketoglutarate; oaa, oxaloacetate. Negative
signs on precursor metabolites indicate chemicals <em>gained</em> through
biosynthetic pathways. Costs of precursors reflect averages for growth on
glucose, acetate, and malate (see Table 6 in the supporting information
from Akashi and Gojobori 2002).<br></p>
<p>The levels l&lt;m&lt;h for the <code>gc</code> ordered factor stand for Low G+C, Middle G+C,
High G+C amino-acid, respectively. The frequencies of Low G+C amino-acids
monotonously decrease with G+C content. The frequencies of High G+C amino-
acids monotonously increase with G+C content. The frequencies of Middle G+C
amino-acids first increase and then decrease with G+C content. These G+C classes 
are from Lobry (1997).<br></p>
<p><code>example(aacost)</code> reproduces figure 2 from Lobry (2004).
</p>
<p><img src="../help/figures/aka.pdf" width="400" alt="aka.pdf"></p>


<h3>Source</h3>

<p>Akashi, H, Gojobori, T. (2002) Metabolic efficiency and amino acid composition
in the proteomes of <em>Escherichia coli</em> and <em>Bacillus subtilis</em>.
<em>Proceedings of the National Academy of Sciences of the United States of America</em>, 
<b>99</b>:3695-3700.<br></p>
<p>Lobry, J.R. (1997) Influence of genomic G+C content on average amino-acid composition 
of proteins from 59 bacterial species. <em>Gene</em>, <b>205</b>:309-316.<br></p>
<p>Lobry, J.R. (2004) Life history traits and genome structure: aerobiosis and G+C content in bacteria. 
<em>Lecture Notes in Computer Sciences</em>, <b>3039</b>:679-686.
</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(aacost)
levels(aacost$gc) &lt;- c("low G+C", "mid G+C", "high G+C")
stripchart(aacost$tot~aacost$gc, pch = 19, ylim = c(0.5,3.5),
     xlim = c(0, max(aacost$tot)),
     xlab = "Metabolic cost (high-energy phosphate bonds equivalent)",
     main = "Metabolic cost of the 20 amino-acids\nas function of their G+C class" )
boxplot(aacost$tot~aacost$gc, horizontal = TRUE, add = TRUE)   
</code></pre>


</div>