<div class="container">

<table style="width: 100%;"><tr>
<td>read.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> read FASTA formatted files </h2>

<h3>Description</h3>

<p>Read nucleic or amino-acid sequences from a file in FASTA format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.fasta(file = system.file("sequences/ct.fasta.gz", package = "seqinr"),
  seqtype = c("DNA", "AA"), as.string = FALSE, forceDNAtolower = TRUE,
  set.attributes = TRUE, legacy.mode = TRUE, seqonly = FALSE, strip.desc = FALSE,
  whole.header = FALSE,
  bfa = FALSE, sizeof.longlong = .Machine$sizeof.longlong,
  endian = .Platform$endian, apply.mask = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> The name of the file which the sequences in fasta format are to be
read from. If it does not contain an absolute or relative path, the file name is relative
to the current working directory, <code>getwd</code>. The default here is to
read the <code>ct.fasta.gz</code> file which is present in the <code>sequences</code> folder
of the seqinR package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqtype</code></td>
<td>
<p> the nature of the sequence: <code>DNA</code> or <code>AA</code>, defaulting
to <code>DNA</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as.string</code></td>
<td>
<p> if TRUE sequences are returned as a string instead of a
vector of single characters</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forceDNAtolower</code></td>
<td>
<p> whether sequences with <code>seqtype == "DNA"</code> should be
returned as lower case letters </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>set.attributes</code></td>
<td>
<p> whether sequence attributes should be set</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legacy.mode</code></td>
<td>
<p>if TRUE lines starting with a semicolon ';' are ignored</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqonly</code></td>
<td>
<p>if TRUE, only sequences as returned without attempt to modify
them or to get their names and annotations (execution time is divided approximately
by a factor 3)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strip.desc</code></td>
<td>
<p>if TRUE the '&gt;' at the beginning of the description lines is removed
in the annotations of the sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>whole.header</code></td>
<td>
<p>if TRUE the whole header line, except the first '&gt;' character,
is kept for sequence name. If FALSE, the default, the name is truncated at the first
space (" ") character.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bfa</code></td>
<td>
<p>logical. If TRUE the fasta file is in MAQ binary format (see details).
Only for DNA sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sizeof.longlong</code></td>
<td>
<p>the number of bytes in a C <code>long long</code> type.
Only relevant for <code>bfa = TRUE</code>. See <code>.Machine</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>endian</code></td>
<td>
<p>character string, <code>"big"</code> or <code>"little"</code>, giving the
endianness of the processor in use. Only relevant for <code>bfa = TRUE</code>.
See <code>.Platform</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>apply.mask</code></td>
<td>
<p>logical defaulting to <code>TRUE</code>. Only relevant for
<code>bfa = TRUE</code>. When this flag is <code>TRUE</code> the mask in the MAQ
binary format is used to replace non acgt characters in the sequence
by the n character. For pure acgt sequences (without gaps or ambiguous
bases) turning this to <code>FALSE</code> will save time.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>FASTA is a widely used format in biology, some FASTA files are distributed
with the seqinr package, see the examples section below.
Sequence in FASTA format begins with a single-line
description (distinguished by a greater-than '&gt;' symbol), followed
by sequence data on the next lines. Lines starting by a semicolon ';'
are ignored, as in the original FASTA program (Pearson and Lipman 1988).
The sequence name is just after the '&gt;' up to the next space ' ' character,
trailling infos are ignored for the name but saved in the annotations.
</p>
<p>There is no standard file extension name for a FASTA file. Commonly
found values are .fasta, .fas, .fa and .seq for generic FASTA files.
More specific file extension names are also used for fasta sequence
alignement (.fsa), fasta nucleic acid (.fna), fasta functional
nucleotide (.ffn), fasta amino acid (.faa), multiple protein
fasta (.mpfa), fasta RNA non-coding (.frn).
</p>
<p>The MAQ fasta binary format was introduced in seqinR 1.1-7 and has not
been extensively tested. This format is used in the MAQ (Mapping and
Assembly with Qualities) software (<a href="https://maq.sourceforge.net/">https://maq.sourceforge.net/</a>).
In this format the four nucleotides are coded with two bits and the
sequence is stored as a vector of C <code>unsigned long long</code>. There
is in addition a mask to locate non-acgt characters.
</p>


<h3>Value</h3>

<p>By default <code>read.fasta</code> return a list of vector of chars. Each element
is a sequence object of the class <code>SeqFastadna</code> or <code>SeqFastaAA</code>.
</p>


<h3>Note</h3>

<p>The old argument <code>File</code> that was deprecated since seqinR &gt;= 1.1-3 is
no more valid since seqinR &gt;= 2.0-6. Just use <code>file</code> instead.
</p>


<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry</p>


<h3>References</h3>

<p>Pearson, W.R. and Lipman, D.J. (1988) Improved tools for biological
sequence comparison. <em>Proceedings of the National Academy
of Sciences of the United States of America</em>, <b>85</b>:2444-2448
</p>
<p>According to MAQ's FAQ page <a href="https://maq.sourceforge.net/faq.shtml">https://maq.sourceforge.net/faq.shtml</a>
last consulted 2016-06-07 the MAQ manuscript has not been published.
</p>
<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

<p><code>write.fasta</code> to write sequences in a FASTA file,
<code>gb2fasta</code> to convert a GenBank file into a FASTA file,
<code>read.alignment</code> to read aligned sequences,
<code>reverse.align</code> to get an alignment at the nucleic level from the
one at the amino-acid level </p>


<h3>Examples</h3>

<pre><code class="language-R">#
# Simple sanity check with a small FASTA file:
#
  smallFastaFile &lt;- system.file("sequences/smallAA.fasta", package = "seqinr")
  mySmallProtein &lt;- read.fasta(file = smallFastaFile, as.string = TRUE, seqtype = "AA")[[1]]
  stopifnot(mySmallProtein == "SEQINRSEQINRSEQINRSEQINR*")
#
# Simple sanity check with the gzipped version of the same small FASTA file:
#
  smallFastaFile &lt;- system.file("sequences/smallAA.fasta.gz", package = "seqinr")
  mySmallProtein &lt;- read.fasta(file = smallFastaFile, as.string = TRUE, seqtype = "AA")[[1]]
  stopifnot(mySmallProtein == "SEQINRSEQINRSEQINRSEQINR*")
#
# Example of a DNA file in FASTA format:
#
  dnafile &lt;- system.file("sequences/malM.fasta", package = "seqinr")
#
# Read with defaults arguments, looks like:
#
# $XYLEECOM.MALM
# [1] "a" "t" "g" "a" "a" "a" "a" "t" "g" "a" "a" "t" "a" "a" "a" "a" "g" "t"
# ...
  read.fasta(file = dnafile)
#
# The same but do not turn the sequence into a vector of single characters, looks like:
#
# $XYLEECOM.MALM
# [1] "atgaaaatgaataaaagtctcatcgtcctctgtttatcagcagggttactggcaagcgc
# ...
  read.fasta(file = dnafile, as.string = TRUE)
#
# The same but do not force lower case letters, looks like:
#
# $XYLEECOM.MALM
# [1] "ATGAAAATGAATAAAAGTCTCATCGTCCTCTGTTTATCAGCAGGGTTACTGGCAAGC
# ...
  read.fasta(file = dnafile, as.string = TRUE, forceDNAtolower = FALSE)
#
# Example of a protein file in FASTA format:
#
  aafile &lt;- system.file("sequences/seqAA.fasta", package = "seqinr")
#
# Read the protein sequence file, looks like:
#
# $A06852
# [1] "M" "P" "R" "L" "F" "S" "Y" "L" "L" "G" "V" "W" "L" "L" "L" "S" "Q" "L"
# ...
  read.fasta(aafile, seqtype = "AA")
#
# The same, but as string and without attributes, looks like:
#
# $A06852
# [1] "MPRLFSYLLGVWLLLSQLPREIPGQSTNDFIKACGRELVRLWVEICGSVSWGRTALSLEEP
# QLETGPPAETMPSSITKDAEILKMMLEFVPNLPQELKATLSERQPSLRELQQSASKDSNLNFEEFK
# KIILNRQNEAEDKSLLELKNLGLDKHSRKKRLFRMTLSEKCCQVGCIRKDIARLC*"
#
  read.fasta(aafile, seqtype = "AA", as.string = TRUE, set.attributes = FALSE)
#
# Example with a FASTA file that contains comment lines starting with
# a semicolon character ';'
#
  legacyfile &lt;- system.file("sequences/legacy.fasta", package = "seqinr")
  legacyseq &lt;- read.fasta(file = legacyfile, as.string = TRUE)
  stopifnot( nchar(legacyseq) == 921 )
#
# Example of a MAQ binary fasta file produced with maq fasta2bfa ct.fasta ct.bfa
# on a platform where .Platform$endian == "little" and .Machine$sizeof.longlong == 8
#
  fastafile &lt;- system.file("sequences/ct.fasta.gz", package = "seqinr")
  bfafile &lt;- system.file("sequences/ct.bfa", package = "seqinr")

  original &lt;- read.fasta(fastafile, as.string = TRUE, set.att = FALSE)
  bfavers &lt;- read.fasta(bfafile, as.string = TRUE, set.att = FALSE, bfa = TRUE,
    endian = "little", sizeof.longlong = 8)
  if(!identical(original, bfavers)){
     warning(paste("trouble reading bfa file on a platform with endian =",
     .Platform$endian, "and sizeof.longlong =", .Machine$sizeof.longlong))
  }
</code></pre>


</div>