<div class="container">

<table style="width: 100%;"><tr>
<td>count</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Composition of dimer/trimer/etc oligomers</h2>

<h3>Description</h3>

<p>Counts the number of times dimer/trimer/etc oligomers occur in a
sequence. Note that the oligomers are overlapping by default.
</p>


<h3>Usage</h3>

<pre><code class="language-R">count(seq, wordsize, start = 0, by = 1,
 freq = FALSE, alphabet = s2c("acgt"), frame = start)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>a vector of single characters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wordsize</code></td>
<td>
<p>an integer giving the size of word (n-mer) to count.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>an integer (0, 1, 2,...) giving the starting
position to consider in the sequence. The default value 0 means that
we start at the first nucleotide in the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by</code></td>
<td>
<p>an integer defaulting to 1 for the window step.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>if TRUE, word relative frequencies (summing to 1) are returned instead of counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphabet</code></td>
<td>
<p>a vector of single characters used to build the oligomer set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frame</code></td>
<td>
<p>synonymous for start</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>count</code> counts the occurence of all words by moving a window of
length <code>word</code>. The window step is controlled by the argument <code>by</code>. 
<code>start</code> controls the starting position in the sequence for the count.
</p>


<h3>Value</h3>

<p>This function returns a <code>table</code> whose <code>dimnames</code> are all the possible
oligomers. All oligomers are returned, even if absent from
the sequence.		
</p>


<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry with suggestions from Gabriel Valiente, Stefanie Hartmann and Christian Gautier</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

 <p><code>table</code> for the class of the returned objet. See <code>rho</code> and
<code>zscore</code> for dinucleotide statistics.</p>


<h3>Examples</h3>

<pre><code class="language-R">a &lt;- s2c("acgggtacggtcccatcgaa")
##
## To count dinucleotide occurrences in sequence a:
##
count(a, word = 2)
##
## To count trinucleotide occurrences in sequence a, with start = 2:
##
count(a, word = 3, start = 2)
##
## To count dinucleotide relative frequencies in sequence a:
##
count(a, word = 2, freq = TRUE)
##
## To count dinucleotides in codon positions III-I in a coding sequence:
##
alldinuclIIIpI &lt;- s2c("NNaaNatNttNtgNgtNtcNctNtaNagNggNgcNcgNgaNacNccNcaNN")
resIIIpI &lt;- count(alldinuclIIIpI, word = 2, start = 2, by = 3)
stopifnot(all( resIIIpI == 1))
##
## Simple sanity check:
##
#alldinucl &lt;- "aattgtctaggcgacca"
#stopifnot(all(count(s2c(alldinucl), 2) == 1))
#alldiaa &lt;- "aaxxzxbxvxyxwxtxsxpxfxmxkxlxixhxgxexqxcxdxnxrxazzbzvzyzwztzszpzfzmzkzlzizhzgzezqzczdznz
#rzabbvbybwbtbsbpbfbmbkblbibhbgbebqbcbdbnbrbavvyvwvtvsvpvfvmvkvlvivhvgvevqvcvdvnvrvayywytysypyfymyky
#lyiyhygyeyqycydynyryawwtwswpwfwmwkwlwiwhwgwewqwcwdwnwrwattstptftmtktltithtgtetqtctdtntrtasspsfsmsks
#lsishsgsesqscsdsnsrsappfpmpkplpiphpgpepqpcpdpnprpaffmfkflfifhfgfefqfcfdfnfrfammkmlmimhmgmemqmcmdmnm
#rmakklkikhkgkekqkckdknkrkallilhlglelqlcldlnlrlaiihigieiqicidiniriahhghehqhchdhnhrhaggegqgcgdgngrgae
#eqecedenereaqqcqdqnqrqaccdcncrcaddndrdannrnarra"
#stopifnot(all(count(s2c(alldiaa), 2, alphabet = s2c("arndcqeghilkmfpstwyvbzx")) == 1))
##
## Example with dinucleotide count in the complete Human mitochondrion genome:
##
humanMito &lt;- read.fasta(file = system.file("sequences/humanMito.fasta", package = "seqinr"))
##
## Get the dinucleotide count:
##
dinu &lt;- count(humanMito[[1]], 2)
##
## Put the results in a 4 X 4 array:
##
dinu2 &lt;- dinu
dim(dinu2) &lt;- c(4, 4)
nucl &lt;- s2c("ACGT")
dimnames(dinu2) &lt;- list(paste(nucl, "-3\'", sep = ""), paste("5\'-", nucl, sep = ""))
##
## Show that CpG and GpT dinucleotides are depleted:
##
mosaicplot(t(dinu2), shade = TRUE,
  main = "Dinucleotide XpY frequencies in the Human\nmitochondrion complete genome", 
  xlab = "First nucleotide: Xp", 
  ylab = "Second nucleotide: pY", las = 1, cex = 1)
mtext("Note the depletion in CpG and GpT dinucleotides", side = 1, line = 3)
</code></pre>


</div>