<div class="container">

<table style="width: 100%;"><tr>
<td>dinucleotides</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Statistical over- and under- representation of dinucleotides in a
sequence</h2>

<h3>Description</h3>

<p>These two functions compute two different types of statistics for the
measure of statistical dinculeotide over- and under-representation :
the rho statistic, and the z-score, each computed for all 16 dinucleotides.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rho(sequence, wordsize = 2, alphabet = s2c("acgt"))
zscore(sequence, simulations = NULL, modele, exact = FALSE, alphabet = s2c("acgt"), ... )
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>a vector of single characters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wordsize</code></td>
<td>
<p>an integer giving the size of word (n-mer) to consider.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simulations</code></td>
<td>
<p> If <code>NULL</code>, analytical solution is computed
when available (models <code>base</code> and <code>codon</code>). Otherwise, it
should be the number of permutations for the z-score computation </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modele</code></td>
<td>
<p> A string of characters describing the model chosen for
the random generation </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exact</code></td>
<td>
<p> Whether exact analytical calculation or an
approximation should be used </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphabet</code></td>
<td>
<p> A vector of single characters. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> Optional parameters for specific model permutations are
passed on to <code>permutation</code> function. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>rho</code> statistic, as presented in Karlin S., Cardon LR. (1994), can
be computed on each of the 16 dinucleotides. It is the frequence of
dinucleotide <em>xy</em> divided by the product of frequencies of
nucleotide <em>x</em> and nucleotide <em>y</em>. It is equal to 1.00 when
dinucleotide <em>xy</em> is formed by pure chance, and it is superior
(respectively inferior) to 1.00 when dinucleotide <em>xy</em> is over-
(respectively under-) represented. Note that if you want to reproduce
Karlin's results you have to compute the statistic from the sequence 
concatenated with its inverted complement that is with something 
like <code>rho(c(myseq, rev(comp(myseq))))</code>.
</p>
<p>The <code>zscore</code> statistic, as presented in Palmeira, L., Guéguen, L.
and Lobry JR. (2006). The statistic is the normalization of the
<code>rho</code> statistic by its expectation and variance according to a
given random sequence generation model, and follows the
standard normal distribution. This statistic can be computed
with several models (cf. <code>permutation</code> for the description
of each of the models). We provide analytical calculus for two of
them: the <code>base</code> permutations model and the  <code>codon</code>
permutations model.
</p>
<p>The <code>base</code> model allows for random sequence generation by
shuffling (with/without replacement) of all bases in the sequence.
Analytical computations are available for this model: either as an 
approximation for large sequences (cf. Palmeira, L., Guéguen, L.
and Lobry JR. (2006)), either as the exact analytical formulae
(cf. Schbath, S. (1995)).
</p>
<p>The <code>position</code> model allows for random sequence generation
by shuffling (with/without replacement) of bases within their
position in the codon (bases in position I, II or III stay in
position I, II or III in the new sequence.
</p>
<p>The <code>codon</code> model allows for random sequence generation by
shuffling (with/without replacement) of codons. Analytical
computation is available for this model (Gautier, C., Gouy, M. and
Louail, S. (1985)).
</p>
<p>The <code>syncodon</code> model allows for random sequence generation
by shuffling (with/without replacement) of synonymous codons.
</p>


<h3>Value</h3>

<p>a table containing the computed statistic for each dinucleotide
</p>


<h3>Author(s)</h3>

<p>L. Palmeira, J.R. Lobry with suggestions from A. Coghlan.</p>


<h3>References</h3>

<p>Gautier, C., Gouy, M. and Louail, S. (1985) Non-parametric statistics
for nucleic acid sequence study. <em>Biochimie</em>, <b>67</b>:449-453.
</p>
<p>Karlin S. and Cardon LR. (1994) Computational DNA sequence analysis.
<em>Annu Rev Microbiol</em>, <b>48</b>:619-654.
</p>
<p>Schbath, S. (1995) Étude asymptotique du nombre d'occurrences d'un
mot dans une chaîne de Markov et application à la recherche de mots
de fréquence exceptionnelle dans les séquences d'ADN.
<em>Thèse de l'Université René Descartes, Paris V</em>
</p>
<p>Palmeira, L., Guéguen, L. and Lobry, J.R. (2006) UV-targeted dinucleotides
are not depleted in light-exposed Prokaryotic genomes.
<em>Molecular Biology and Evolution</em>,
<b>23</b>:2214-2219.
<a href="https://academic.oup.com/mbe/article/23/11/2214/1335460">https://academic.oup.com/mbe/article/23/11/2214/1335460</a>
</p>
<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

 <p><code>permutation</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
sequence &lt;- sample(x = s2c("acgt"), size = 6000, replace = TRUE)
rho(sequence)
zscore(sequence, modele = "base")
zscore(sequence, modele = "base", exact = TRUE)
zscore(sequence, modele = "codon")
zscore(sequence, simulations = 1000, modele = "syncodon")

## End(Not run)
</code></pre>


</div>