<div class="container">

<table style="width: 100%;"><tr>
<td>write.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Write sequence(s) into a file in fasta format </h2>

<h3>Description</h3>

<p>Writes one or more sequences into a file in FASTA format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">write.fasta(sequences, names, file.out, open = "w", nbchar = 60, as.string = FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sequences</code></td>
<td>
<p> A DNA or protein sequence (in the form of a
vector of single characters by default) or a list of such sequences. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as.string</code></td>
<td>
<p> FALSE. When set to TRUE sequences are in the
form of strings instead of vectors of single characters. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>names</code></td>
<td>
<p> The name(s) of the sequences. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbchar</code></td>
<td>
<p> The number of characters per line (default: 60) </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file.out</code></td>
<td>
<p> The name of the output file. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>open</code></td>
<td>
<p> Mode to open the output file, use "w" to write into
a new file, use "a" to append at the end of an already existing file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>none.
</p>


<h3>Author(s)</h3>

<p>A. Nec≈üulea</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Read 3 sequences from a FASTA file:
ortho &lt;- read.fasta(file = system.file("sequences/ortho.fasta", package =
"seqinr"))

## Select only third codon positions:
ortho3 &lt;- lapply(ortho, function(x) x[seq(from = 3, to = length(x), by = 3)])

## Write the 3 modified sequences to a file:
fname &lt;- tempfile(pattern = "ortho3", tmpdir = tempdir(), fileext = "fasta")
#write.fasta(sequences = ortho3, names = names(ortho3), nbchar = 80, file.out = "ortho3.fasta")
write.fasta(sequences = ortho3, names = names(ortho3), nbchar = 80, file.out = fname)

## Read them again from the same file and check that sequences are preserved:
ortho3bis &lt;- read.fasta(fname, set.attributes = FALSE)
stopifnot(identical(ortho3bis, ortho3))
</code></pre>


</div>