<div class="container">

<table style="width: 100%;"><tr>
<td>draw.oriloc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Graphical representation for nucleotide skews in
prokaryotic chromosomes.</h2>

<h3>Description</h3>

<p>Graphical representation for nucleotide skews in
prokaryotic chromosomes.</p>


<h3>Usage</h3>

<pre><code class="language-R">draw.oriloc(ori, main = "Title",
  xlab = "Map position in Kb",
  ylab = "Cumulated combined skew in Kb", las = 1, las.right = 3,
  ta.mtext = "Cumul. T-A skew", ta.col = "pink", ta.lwd = 1,
  cg.mtext = "Cumul. C-G skew", cg.col = "lightblue", cg.lwd = 1,
  cds.mtext = "Cumul. CDS skew", cds.col = "lightgreen", cds.lwd = 1,
  sk.col = "black", sk.lwd = 2,
  add.grid = TRUE, ...)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ori</code></td>
<td>
<p>A data frame obtained with the <code>oriloc</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>The main title of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>The x-axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>The y-axis title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>las</code></td>
<td>
<p>The style of axis labels for the bottom and left axes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>las.right</code></td>
<td>
<p>The style of axis labels for the right axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ta.mtext</code></td>
<td>
<p>The marginal legend for the TA skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ta.col</code></td>
<td>
<p>The color for the TA skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ta.lwd</code></td>
<td>
<p>The line width for the TA skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cg.mtext</code></td>
<td>
<p>The marginal legend for the CG skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cg.col</code></td>
<td>
<p>The color for the CG skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cg.lwd</code></td>
<td>
<p>The line width for the CG skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cds.mtext</code></td>
<td>
<p>The marginal legend for the CDS skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cds.col</code></td>
<td>
<p>The color for the CDS skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cds.lwd</code></td>
<td>
<p>The line width for the CDS skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sk.col</code></td>
<td>
<p>The color for the cumulated combined skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sk.lwd</code></td>
<td>
<p>The line width for the cumulated combined skew.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add.grid</code></td>
<td>
<p>Logical, if <code>TRUE</code> a vertical grid is added to the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments are passed to the function <code>plot</code>.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>J.R. Lobry</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

 <p><code>oriloc</code>, <code>rearranged.oriloc</code>,
<code>extract.breakpoints</code>  </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run:  # need internet connection
#
# Example with Chlamydia trachomatis complete genome
#
  ori &lt;- oriloc()
  draw.oriloc(ori)
#
# The same, using more options from function draw.oriloc()
#
draw.oriloc(ori, 
  main = expression(italic(Chlamydia~~trachomatis)~~complete~~genome),
  ta.mtext = "TA skew", ta.col = "red",
  cg.mtext = "CG skew", cg.col = "blue",
  cds.mtext = "CDS skew", cds.col = "seagreen",
  add.grid = FALSE)

## End(Not run)
</code></pre>


</div>