<div class="container">

<table style="width: 100%;"><tr>
<td>AAstat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> To Get Some Protein Statistics </h2>

<h3>Description</h3>

<p>Returns simple protein sequence information including the number of residues,
the percentage physico-chemical classes and the theoretical isoelectric point.
The functions ignore ambiguous amino acids (e.g. "B", "Z", "X", "J").
</p>


<h3>Usage</h3>

<pre><code class="language-R">AAstat(seq, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p> a protein sequence as a vector of upper-case chars </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p> if <code>TRUE</code>, plots the presence of residues splited by
physico-chemical classes along the sequence. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with the three following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Compo</code></td>
<td>
<p>A factor giving the amino acid counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Prop</code></td>
<td>
<p>A list giving the percentage of each physico-chemical classes
(Tiny, Small, Aliphatic, Aromatic, Non-polar, Polar, Charged, Positive, Negative).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pi</code></td>
<td>
<p>The theoretical isoelectric point</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>D. Charif, J.R. Lobry</p>


<h3>References</h3>

<p><code>citation("seqinr")</code>
</p>


<h3>See Also</h3>

 <p><code>computePI</code>, <code>SEQINR.UTIL</code>, <code>SeqFastaAA</code></p>


<h3>Examples</h3>

<pre><code class="language-R">  seqAA &lt;- read.fasta(file = system.file("sequences/seqAA.fasta", package = "seqinr"),
   seqtype = "AA")
  AAstat(seqAA[[1]])
</code></pre>


</div>