<div class="container">

<table style="width: 100%;"><tr>
<td>seqkat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SeqKat</h2>

<h3>Description</h3>

<p>Kataegis detection from SNV BED files
</p>


<h3>Usage</h3>

<pre><code class="language-R">seqkat(sigcutoff = 5, mutdistance = 3.2, segnum = 4, ref.dir = NULL,
  bed.file = "./", output.dir = "./", chromosome = "all",
  chromosome.length.file = NULL, trinucleotide.count.file = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sigcutoff</code></td>
<td>
<p>The minimum hypermutation score used to classify the windows in the sliding binomial test as significant windows. The score is calculated per window as follows: -log10(binomial test p-value). Recommended value: 5</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutdistance</code></td>
<td>
<p>The maximum intermutational distance allowed for SNVs to be grouped in the same kataegic event. Recommended value: 3.2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segnum</code></td>
<td>
<p>Minimum mutation count. The minimum number of mutations required within a cluster to be identified as kataegic. Recommended value: 4</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome. Each chromosome should have its own .fa file and chromosomes X and Y are named as chr23 and chr24. The fasta files should contain no header</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bed.file</code></td>
<td>
<p>Path to the SNV BED file. The BED file should contain the following information: Chromosome, Position, Reference allele, Alternate allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.dir</code></td>
<td>
<p>Path to a directory where output will be created.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p>The chromosome to be analysed. This can be (1, 2, ..., 23, 24) or "all" to run sequentially on all chromosomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome.length.file</code></td>
<td>
<p>A tab separated file containing the lengths of all chromosomes in the reference genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trinucleotide.count.file</code></td>
<td>
<p>A tab seprarated file containing a count of all trinucleotides present in the reference genome. This can be generated with the get.trinucleotide.counts() function in this package.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The default paramters in SeqKat have been optimized using Alexanrov's "Signatures of mutational processes in human cancer" dataset. SeqKat accepts a BED file and outputs the results in TXT format. A file per chromosome is generated if a kataegic event is detected, otherwise no file is generated. SeqKat reports two scores per kataegic event, a hypermutation score and an APOBEC mediated kataegic score.
</p>


<h3>Author(s)</h3>

<p>Fouad Yousif
</p>
<p>Fan Fan
</p>
<p>Christopher Lalansingh
</p>


<h3>Examples</h3>

<pre><code class="language-R">example.bed.file &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/PD4120a-chr4-1-2000000_test_snvs.bed"
	);
example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);
example.chromosome.length.file &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/length_hg19_chr_test.txt"
	);
seqkat(
	5,
	3.2,
	2,
	bed.file = example.bed.file,
	output.dir = tempdir(),
	chromosome = "4",
	ref.dir = example.ref.dir,
	chromosome.length.file = example.chromosome.length.file
	);
</code></pre>


</div>