<div class="container">

<table style="width: 100%;"><tr>
<td>test.kataegis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Test Kataegis</h2>

<h3>Description</h3>

<p>Performs exact binomial test to test the deviation of the 32 tri-nucleotides counts from expected
</p>


<h3>Usage</h3>

<pre><code class="language-R">test.kataegis(chromosome.num, somatic, units, exprobntcx, output.name, ref.dir,
  chromosome.length.file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>chromosome.num</code></td>
<td>
<p>Chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>somatic</code></td>
<td>
<p>Data frame of somatic variants</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>units</code></td>
<td>
<p>Base window size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exprobntcx</code></td>
<td>
<p>Expected probability for each trinucleotide and total number of tcx</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.name</code></td>
<td>
<p>Name of the generated output directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.dir</code></td>
<td>
<p>Path to a directory containing the reference genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome.length.file</code></td>
<td>
<p>A tab separated file containing the lengths of all chromosomes in the reference genome.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class="language-R">load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/somatic.rda"
		)
	);

load(
	paste0(
		path.package("SeqKat"),
		"/extdata/test/exprobntcx.rda"
		)
	);

example.chromosome.length.file &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/length_hg19_chr_test.txt"
	);

example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);

test.kataegis(
	4,
	somatic,
	2,
	exprobntcx,
	tempdir(),
	example.ref.dir,
	example.chromosome.length.file
	);
</code></pre>


</div>