<div class="container">

<table style="width: 100%;"><tr>
<td>get.nucleotide.chunk.counts</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Nucleotide Chunk Counts</h2>

<h3>Description</h3>

<p>Obtain counts for all possible trinucleotides within a specified genomic region
</p>


<h3>Usage</h3>

<pre><code class="language-R">get.nucleotide.chunk.counts(key, chr, upstream = 1, downstream = 1,
  start = 1, end = -1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>key</code></td>
<td>
<p>List of specify trinucleotides to count</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>Chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upstream</code></td>
<td>
<p>Length upstream to read</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>downstream</code></td>
<td>
<p>Length downstream to read</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>
<p>Starting position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>
<p>Ending position</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Fouad Yousif
</p>


<h3>Examples</h3>

<pre><code class="language-R">example.ref.dir &lt;- paste0(
	path.package("SeqKat"),
	"/extdata/test/ref/"
	);

bases.raw &lt;- c('A','C','G','T','N');
tri.types.raw &lt;- c(
	outer(
		c(outer(bases.raw, bases.raw, function(x, y) paste0(x,y))),
		bases.raw, function(x, y) paste0(x,y))
	);
tri.types.raw &lt;- sort(tri.types.raw);
get.nucleotide.chunk.counts(
	tri.types.raw,
	file.path(example.ref.dir, 'chr4.fa'),
	upstream = 1,
	downstream = 1,
	start = 1,
	end = -1
	);
</code></pre>


</div>