<div class="container">

<table style="width: 100%;"><tr>
<td>gen_rnaseq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate RNA-seq data from an underlying network</h2>

<h3>Description</h3>

<p>The expression data are generated based on the gene-gene associations of an
underlying network. An association structure is imposed by first generating
data from a multivariate Gaussian distribution. Those data are then used to
sample from the empirical distribution of gene expression profiles in the
reference dataset using the inverse transform method.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gen_rnaseq(n, network, reference = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The number of samples to generate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>network</code></td>
<td>
<p>A 'network' object or list of 'network' objects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>A data.frame containing reference gene expression data. Rows
should correspond to samples and columns to genes. If <code>NULL</code>, then the
<code>reference</code> dataset is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Boolean indicator for message output.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing the simulated expression data and the reference
dataset. If a list of networks were provided, then the results for
each network are returned as a list.
</p>


<h3>References</h3>

<p>Grimes T, Datta S (2021).
“SeqNet: An R Package for Generating Gene-Gene Networks and Simulating RNA-Seq Data.”
<em>Journal of Statistical Software</em>, <b>98</b>(12), 1–49.
doi: <a href="https://doi.org/10.18637/jss.v098.i12">10.18637/jss.v098.i12</a>, <a href="https://doi.org/10.18637/jss.v098.i12">https://doi.org/10.18637/jss.v098.i12</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">nw &lt;- random_network(10) # Create a random network with 10 nodes.
nw &lt;- gen_partial_correlations(nw) # Add weights to connections in the network.
# If no reference is provided, the internal RNA-seq reference dataset is used.
x &lt;- gen_rnaseq(20, nw)$x # Simulate 20 observations from the network.
</code></pre>


</div>