<div class="container">

<table style="width: 100%;"><tr>
<td>plot_gene_pair</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Scatter plot of two gene expressions</h2>

<h3>Description</h3>

<p>Plots the expression of two genes for visual assessment of association.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_gene_pair(
  x_list,
  geneA,
  geneB,
  method = "loess",
  se_alpha = 0.1,
  do_facet_wrap = FALSE,
  scales = "fixed"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x_list</code></td>
<td>
<p>A named list containing one or more n by p gene expression
profiles, one for each group or subpopulation under consideration.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneA</code></td>
<td>
<p>The name of the first gene to plot. Must be either a character
string matching a column name in each matrix of <code>x_list</code> or an integer
to index the columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneB</code></td>
<td>
<p>The name of the second gene to plot. Must be either a character
string matching a column name in each matrix of <code>x_list</code> or an integer
to index the columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>Charater string either "lm" or "loess" used for plotting.
For no line, set <code>method = NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>se_alpha</code></td>
<td>
<p>Sets transparancy of confidence interval around association
trend line. Set to 0 to remove the confidence interval.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do_facet_wrap</code></td>
<td>
<p>If <code>TRUE</code>, the groups are plotted in seperate graphs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scales</code></td>
<td>
<p>Only used if <code>do_facet_wrap</code> is <code>TRUE</code>. See
<code>facet_wrap</code> for details.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns the generated plot.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(reference)
rnaseq &lt;- reference$rnaseq
genes &lt;- colnames(rnaseq)
plot_gene_pair(rnaseq, genes[1], genes[2])
# Suppose we had multiple data frames.
control &lt;- rnaseq[1:100, 1:10]
treatment1 &lt;- rnaseq[101:200, 1:10]
treatment2 &lt;- rnaseq[201:250, 1:10]
plot_gene_pair(list(ctrl = control, trt1 = treatment1, trt2 = treatment2),
               genes[1], genes[2], method = NA)
plot_gene_pair(list(ctrl = control, trt = treatment1),
               genes[1], genes[2], do_facet_wrap = TRUE, method = "lm")

</code></pre>


</div>