<div class="container">

<table style="width: 100%;"><tr>
<td>EstEr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimate genotyping error rate (REMOVED; will be re-implemented)</h2>

<h3>Description</h3>

<p>Estimate the genotyping error rates in SNP data, based on a
pedigree and/or duplicates. Estimates probabilities (observed given
actual) hom|other hom, het|hom, and hom|het. THESE ARE APPROXIMATE VALUES!
</p>


<h3>Usage</h3>

<pre><code class="language-R">EstEr(
  GenoM,
  Pedigree,
  Duplicates = NULL,
  Er_start = c(0.05, 0.05, 0.05),
  perSNP = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>GenoM</code></td>
<td>
<p>Genotype matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pedigree</code></td>
<td>
<p>data.frame with columns id - dam - sire</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Duplicates</code></td>
<td>
<p>matrix or data.frame with 2 columns, id1 &amp; id2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Er_start</code></td>
<td>
<p>vector of length 3 with starting values for <code>optim</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perSNP</code></td>
<td>
<p>logical, estimate error rate per SNP. WARNING not very
precise, use only as an approximate indicator! Try on simulated data first,
e.g. with <code>SimGeno</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The result should be interpreted as approximate, ballpark estimates!
The estimated error rates from a pedigree will not be as accurate as from
duplicate samples. Errors in individuals without parents or offspring will
not be counted, and errors in individuals with only few offspring may not be
noted either. Deviation of genotype frequencies among founders from
Hardy-Weinberg equilibrium may wrongly be attributed to genotyping errors.
Last but not least, any pedigree errors will result in higher estimated
genotyping errors.
</p>


<h3>Value</h3>

<p>vector of length 3 with estimated genotyping error rates: the
probabilities that
</p>

<ul>
<li>
<p> hom|hom: an actual homozygote is observed as the other homozygote
</p>
</li>
<li>
<p> het|hom: an actual homozygote is observed as heterozygote
</p>
</li>
<li>
<p> hom|het: an actual heterozygote is observed as homozygote
</p>
</li>
</ul>
<p>These are three independent parameters, that define the genotyping error
matrix (see <code>ErrToM</code>) as follows:
</p>

<table>
<tr>
<td style="text-align: left;">
            </td>
<td style="text-align: center;"> <strong>0</strong>       </td>
<td style="text-align: center;"> <strong>1</strong>   </td>
<td style="text-align: center;"> <strong>2</strong>  </td>
</tr>
<tr>
<td style="text-align: left;">
 <strong>0</strong> </td>
<td style="text-align: center;"> <code class="reqn">1-E_1-E_2</code> </td>
<td style="text-align: center;"> <code class="reqn">E_2</code>    </td>
<td style="text-align: center;"> <code class="reqn">E_1</code> </td>
</tr>
<tr>
<td style="text-align: left;">
 <strong>1</strong> </td>
<td style="text-align: center;"> <code class="reqn">E_3</code>        </td>
<td style="text-align: center;"> <code class="reqn">1-2E_3</code> </td>
<td style="text-align: center;"> <code class="reqn">E_3</code>   </td>
</tr>
<tr>
<td style="text-align: left;">
 <strong>2</strong> </td>
<td style="text-align: center;"> <code class="reqn">E_1</code>       </td>
<td style="text-align: center;"> <code class="reqn">E_2</code>    </td>
<td style="text-align: center;"> <code class="reqn">1-E_1-E_2</code> </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Note that for <code>optim</code> a lower bound of 1e-6 and upper bound of 0.499
are used; if these values are returned this should be interpreted as
'inestimably small' and 'inestimably large', respectively. PLEASE DO NOT USE
THESE VALUES AS INPUT IN SUBSEQUENT ANALYSIS BUT SUBSITUTE BY A SENSIBLE
VALUE!!
</p>


<h3>Examples</h3>

<pre><code class="language-R">GenoX &lt;- SimGeno(Ped_griffin, nSnp=400, SnpError=c(0.01,0.07, 0.1),
                ParMis=0.1, CallRate=0.9)
# EstEr(GenoM=GenoX, Pedigree=Ped_griffin)

</code></pre>


</div>