<div class="container">

<table style="width: 100%;"><tr>
<td>GenoConvert</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert Genotype Data</h2>

<h3>Description</h3>

<p>Convert genotype data in various formats to sequoia's
1-column-per-marker format, PLINK's ped format, or Colony's
2-columns-per-marker format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GenoConvert(
  InData = NULL,
  InFile = NULL,
  InFormat = "raw",
  OutFile = NA,
  OutFormat = "seq",
  Missing = c("-9", "NA", "??", "?", "NULL", "-1", c("0")[InFormat %in% c("col",
    "ped")]),
  sep = c(" ", "\t", ",", ";"),
  header = NA,
  IDcol = NA,
  FIDcol = NA,
  FIDsep = "__",
  dropcol = NA,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>InData</code></td>
<td>
<p>dataframe, matrix or
<code>genlight</code> object with genotypes to be
converted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>InFile</code></td>
<td>
<p>character string with name of genotype file to be converted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>InFormat</code></td>
<td>
<p>One of 'seq' (sequoia), 'ped' (PLINK .ped file), 'col'
(COLONY), 'raw' (PLINK –recodeA), 'vcf' (requires library <code>{vcfR}</code>),
'single' (1 column per SNP), or 'double' (2 columns per SNP); see Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OutFile</code></td>
<td>
<p>character string with name of converted file. If NA, return
matrix with genotypes in console (default); if NULL, write to
'GenoForSequoia.txt' in current working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OutFormat</code></td>
<td>
<p>as <code>InFormat</code>; only 'seq', 'col', and 'ped' are
implemented. For 'ped' also a sham .map file is created, so that the file
can be read by PLINK. Only for 'ped' are extensions .ped &amp; .map added to
the specified OutFile filename.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Missing</code></td>
<td>
<p>vector with symbols interpreted as missing data. '0' is
missing data for <code>InFormat</code>s 'col' and 'ped' only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>vector with field separator strings that will be tried on
<code>InFile</code>. Ignored if package <span class="pkg">data.table</span> is present or if
<code>InFormat='vcf'</code>. The <code>OutFile</code> separator uses the
<code>write.table</code> default, i.e. one blank space.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header</code></td>
<td>
<p>a logical value indicating whether the file contains a header
as its first line. If NA (default), set to TRUE for 'raw', and FALSE
otherwise.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>IDcol</code></td>
<td>
<p>number giving the column with individual IDs; 0 indicates the
rownames (for InData only). If NA (default), set to 2 for InFormat 'raw'
and 'ped', and otherwise to 1 for InFile and 0 (rownames) for InData,
except when InData has a column labeled 'ID'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FIDcol</code></td>
<td>
<p>column with the family IDs, if any are wished to
be used. This is column 1 for InFormat 'raw' and 'seq', but those are by
default not used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FIDsep</code></td>
<td>
<p>string used to paste FID and IID together into a composite-ID
(value passed to <code>paste</code>'s <code>collapse</code>). This joining can
be reversed using <code>PedStripFID</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropcol</code></td>
<td>
<p>columns to exclude from the output data, on top of IDcol and
FIDcol (which become rownames). When NA, defaults to columns 3-6 for
InFormat 'raw' and 'seq'. Can also be used to drop some SNPs, see example
below on how to do this for the 2-columns-per-SNP input formats.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>suppress messages and warnings.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The first two arguments are interchangeable, and can be given
unnamed. The first argument is assumed to be a file name if it is of class
'character' and length 1, and to be the genetic data if it is a matrix or
dataframe.
</p>
<p>If package <span class="pkg">data.table</span> is detected, <code>fread</code>
is used to read in the data from file. Otherwise, a combination of
<code>readLines</code> and <code>strsplit</code> is used.
</p>


<h3>Value</h3>

<p>A genotype matrix in the specified output format; the default sequoia
format ('seq') has 1 column per SNP coded in 0/1/2 format (major homozygote
/heterozygote /minor homozygote) with -9 for missing values, sample IDs in
row names and SNP names in column names. If 'OutFile' is specified, the
matrix is written to this file and nothing is returned inside R.
</p>


<h3>Input formats</h3>

<p>The following formats can be specified by <code>InFormat</code>:
</p>

<dl>
<dt>seq</dt>
<dd>
<p>(sequoia) genotypes are coded as 0, 1, 2, missing as <code class="reqn">-9</code> (in input
any negative number or NA are OK),
in 1 column per marker. Column 1 contains IDs, there is no header row.</p>
</dd>
<dt>ped</dt>
<dd>
<p>(PLINK) genotypes are coded as A, C, T, G, missing as 0, in 2
columns per marker. The first 6 columns are descriptive (1:FID, 2:IID, 3 to
6 ignored). If an associated .map file exists, SNP names will be read from
there.</p>
</dd>
<dt>raw</dt>
<dd>
<p>(PLINK) genotypes are coded as 0, 1, 2, missing as NA, in 1
column per marker. The first 6 columns are descriptive (1:FID, 2:IID, 3 to
6 ignored), and there is a header row. This is produced by PLINK's option
–recodeA</p>
</dd>
<dt>col</dt>
<dd>
<p>(Colony) genotypes are coded as numeric values, missing as 0, in
2 columns per marker. Column 1 contains IDs.</p>
</dd>
<dt>vcf</dt>
<dd>
<p>(VCF) genotypes are coded as '0/0','0/1','1/1', variable number
of header rows followed by 1 row per SNP, with various columns of
metadata followed by 1 column per individual. Requires package
<span class="pkg">vcfR</span>.</p>
</dd>
<dt>single</dt>
<dd>
<p>1 column per marker, otherwise unspecified</p>
</dd>
<dt>double</dt>
<dd>
<p>2 columns per marker, otherwise unspecified</p>
</dd>
</dl>
<p>For each <code>InFormat</code>, its default values for <code>Missing, header,
 IDcol, FIDcol</code>, and <code>dropcol</code> can be overruled by specifying the
corresponding input parameters.
</p>


<h3>Error messages</h3>

<p>Occasionally when reading in a file <code>GenoConvert</code> may give an error
that 'rows have unequal length'. <code>GenoConvert</code> makes use of
<code>readLines</code> and <code>strsplit</code>, which is much faster
than <code>read.table</code> for large datafiles, but also more sensitive
to unusual line endings, unusual end-of-file characters, or invisible
characters (spaces or tabs) after the end of some lines. In these cases,
try to read the data from file using read.table or read.csv, and then use
<code>GenoConvert</code> on this dataframe or matrix, see example.
</p>


<h3>Author(s)</h3>

<p>Jisca Huisman, <a href="mailto:jisca.huisman@gmail.com">jisca.huisman@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code>CheckGeno, SnpStats, LHConvert</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Requires PLINK installed &amp; in system PATH:

# tinker with window size, window overlap and VIF to get a set of
# 400 - 800 markers (100-200 enough for just parentage):
system("cmd", input = "plink --file mydata --indep 50 5 2")
system("cmd", input = "plink --file mydata --extract plink.prune.in
  --recodeA --out PlinkOUT")

GenoM &lt;- GenoConvert(InFile = "PlinkOUT.raw", InFormat='raw')
# which is the same as
GenoM &lt;- GenoConvert(PlinkOUT.raw, InFormat='single',
                    IDcol=2, dropcol=c(1,3:6), header=TRUE)
# (but it will complain that InFormat='single' is not consistent with .raw
# file extension)

# save time on file conversion next time:
write.table(GenoM, file="Geno_sequoia.txt", quote=FALSE, col.names=FALSE)
GenoM &lt;- as.matrix(read.table("Geno_sequoia.txt", row.names=1, header=FALSE))

# drop some SNPs, e.g. after a warning of &gt;2 alleles:
dropSNP &lt;- c(5,68,101,128)
GenoM &lt;- GenoConvert(ColonyFile, InFormat = "col",
                     dropcol = 1 + c(2*dropSNP-1, 2*dropSNP) )

# circumvent a 'rows have unequal length' error:
GenoTmp &lt;- as.matrix(read.table("mydata.txt", header=TRUE, row.names=1))
GenoM &lt;- GenoConvert(InData=GenoTmp, InFormat="single", IDcol=0)

## End(Not run)

</code></pre>


</div>