<div class="container">

<table style="width: 100%;"><tr>
<td>OHperSNP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimate Genotyping Error Rate</h2>

<h3>Description</h3>

<p>Estimate genotyping error rate from Mendelian errors per SNP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">OHperSNP(GenoM, Par, Dups = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>GenoM</code></td>
<td>
<p>genotype matrix, in sequoia's format: 1 column per SNP, 1 row
per individual, genotypes coded as 0/1/2/-9, and rownames giving individual
IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Par</code></td>
<td>
<p>pedigree dataframe, only genotyped parents are used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Dups</code></td>
<td>
<p>pairs of duplicates</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe with columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>n.dam, n.sire, n.pair</code></td>
<td>
<p>Number of dams, sires, parent-pairs successfully
genotyped for the SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OHdam, OHsire</code></td>
<td>
<p>Count of number of opposing homozygous cases</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MEpair</code></td>
<td>
<p>Count of Mendelian errors, includes opposing homozygous cases</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.dups, n.diff</code></td>
<td>
<p>Number of duplicate pairs successfully genotyped for
the SNP; number of differences</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>SnpStats</code>.
</p>


</div>