<div class="container">

<table style="width: 100%;"><tr>
<td>CalcPairLL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Likelihoods for Alternative Relationships</h2>

<h3>Description</h3>

<p>For each specified pair of individuals, calculate the
log10-likelihoods of being PO, FS, HS, GP, FA, HA, U (see Details).
Individuals must be genotyped or have at least one genotyped offspring.
</p>
<p><strong>NOTE</strong> values <code class="reqn">&gt;0</code> are various <code>NA</code> types, see 'Likelihood
special codes' in 'Value' section below.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CalcPairLL(
  Pairs = NULL,
  GenoM = NULL,
  Pedigree = NULL,
  LifeHistData = NULL,
  AgePrior = TRUE,
  SeqList = NULL,
  Complex = "full",
  Herm = "no",
  Err = 1e-04,
  ErrFlavour = "version2.9",
  Tassign = 0.5,
  Tfilter = -2,
  quiet = FALSE,
  Plot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Pairs</code></td>
<td>
<p>dataframe with columns <code>ID1</code> and <code>ID2</code>, and
optionally
</p>

<dl>
<dt>Sex1</dt>
<dd>
<p>Sex of ID1, 1=female, 2=male, 3=unknown, or <code>NA</code> to take
from <code>LifeHistData</code>. The sex of individuals occurring as parent in
<code>Pedigree</code> cannot be altered.</p>
</dd>
<dt>Sex2</dt>
<dd>
<p>Sex of ID2</p>
</dd>
<dt>AgeDif</dt>
<dd>
<p>Age difference in whole time units, BirthYear1 - BirthYear2
(i.e. positive if ID2 is born before ID1). If <code>NA</code>, calculated from
<code>LifeHistData</code>. Use '999' to explicitly specify 'unknown'.</p>
</dd>
<dt>focal</dt>
<dd>
<p>relationship character abbreviation; PO, FS, HS, GP or U. See
Details for its effect and explanation of abbreviations. Default: U</p>
</dd>
<dt>patmat</dt>
<dd>
<p>1=maternal relatives, 2=paternal relatives. Only relevant for
HS &amp; GP, for which it defaults to Sex1, or 1 if Sex1=3, but is currently
only predictably implemented for pairs of two genotyped individuals.
Always equal to Sex2 for PO pairs when Sex2 is known.</p>
</dd>
<dt>dropPar1</dt>
<dd>
<p>Drop the parents of <code>ID1</code> before calculating the pair
likelihood, rather than conditioning on them; choose from 'none', 'dam',
'sire', or 'both'. See example. If e.g. the pair shares a common mother,
'none' and 'sire' will condition on this shared mother and not calculate the
likelihood that they are maternal siblings, while dropPar1='dam' or 'both'
will calculate that likelihood, and the other likelihoods as if the mother
of ID1 were unknown.</p>
</dd>
<dt>dropPar2</dt>
<dd>
<p>as <code>dropPar1</code>, for <code>ID2</code></p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GenoM</code></td>
<td>
<p>numeric matrix with genotype data: One row per individual,
one column per SNP, coded as 0, 1, 2, missing values as a negative number
or NA. You can reformat data with <code>GenoConvert</code>, or use other
packages to get it into a genlight object and then use <code>as.matrix</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pedigree</code></td>
<td>
<p>dataframe with columns id-dam-sire; likelihoods will be
calculated conditional on the pedigree. May include non-genotyped
individuals, which will be treated as dummy individuals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LifeHistData</code></td>
<td>
<p>data.frame with up to 6 columns:
</p>

<dl>
<dt>ID</dt>
<dd>
<p>max. 30 characters long</p>
</dd>
<dt>Sex</dt>
<dd>
<p>1 = female, 2 = male, 3 = unknown, 4 = hermaphrodite,
other numbers or NA = unknown</p>
</dd>
<dt>BirthYear </dt>
<dd>
<p>birth or hatching year, integer, with missing values as NA
or any negative number.</p>
</dd>
<dt>BY.min</dt>
<dd>
<p>minimum birth year, only used if BirthYear is missing</p>
</dd>
<dt>BY.max</dt>
<dd>
<p>maximum birth year, only used if BirthYear is missing</p>
</dd>
<dt>Year.last</dt>
<dd>
<p>Last year in which individual could have had offspring. Can
e.g. in mammals be the year before death for females, and year after death
for males. </p>
</dd> </dl>
<p>"Birth year" may be in any arbitrary discrete time unit relevant to the
species (day, month, decade), as long as parents are never born in the same
time unit as their offspring, and only integers are used. Individuals do not
need to be in the same order as in ‘GenoM’, nor do all genotyped individuals
need to be included.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>AgePrior</code></td>
<td>
<p>logical (<code>TRUE/FALSE</code>) whether to estimate the ageprior
from <code>Pedigree</code> and <code>LifeHistData</code>, or a matrix as generated by
<code>MakeAgePrior</code> and included in the <code>sequoia</code>
output. The <code>AgePrior</code> affects which relationships are considered
possible: only those where <code class="reqn">P(A|R) / P(A) &gt; 0</code>.  When <code>TRUE</code>,
<code>MakeAgePrior</code> is called using its default values. When
<code>FALSE</code>, all relationships are considered possible for all age
differences, except that parent-offspring pairs cannot have age difference
zero, and grand-parental pairs have an age difference of at least two.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SeqList</code></td>
<td>
<p>list with output from <code>sequoia</code>. If input
parameter <code>Pedigree=NULL</code>, <code>SeqList$Pedigree</code> will be used if
present, and <code>SeqList$PedigreePar</code> otherwise. If <code>SeqList$Specs</code>
is present, input parameters with the same name as its items are ignored.
The list elements 'LifeHist', 'AgePriors', and 'ErrM' are also used if
present, and override the corresponding input parameters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Complex</code></td>
<td>
<p>Breeding system complexity. Either "full" (default), "simp"
(simplified, no explicit consideration of inbred relationships), "mono"
(monogamous).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Herm</code></td>
<td>
<p>Hermaphrodites, either "no", "A" (distinguish between dam and
sire role, default if at least 1 individual with sex=4), or "B" (no
distinction between dam and sire role). Both of the latter deal with
selfing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Err</code></td>
<td>
<p>estimated genotyping error rate, as a single number, or a length 3
vector with P(hom|hom), P(het|hom), P(hom|het), or a 3x3 matrix. See
details below. The error rate is presumed constant across SNPs, and
missingness is presumed random with respect to actual genotype. Using
<code>Err</code> &gt;5% is not recommended, and <code>Err</code> &gt;10% strongly
discouraged.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ErrFlavour</code></td>
<td>
<p>function that takes <code>Err</code> (single number) as input,
and returns a length 3 vector or 3x3 matrix, or choose from inbuilt options
'version2.9', 'version2.0', 'version1.3', or 'version1.1', referring to the
sequoia version in which they were the default. Ignored if <code>Err</code> is a
vector or matrix. See <code>ErrToM</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Tassign</code></td>
<td>
<p>minimum LLR required for acceptance of proposed relationship,
relative to next most likely relationship. Higher values result in more
conservative assignments. Must be zero or positive.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Tfilter</code></td>
<td>
<p>threshold log10-likelihood ratio (LLR) between a proposed
relationship versus unrelated, to select candidate relatives. Typically a
negative value, related to the fact that unconditional likelihoods are
calculated during the filtering steps. More negative values may decrease
non-assignment, but will increase computational time.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>logical, suppress messages</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Plot</code></td>
<td>
<p>logical, display scatter plots by <code>PlotPairLL</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The same pair may be included multiple times, e.g. with different
sex, age difference, or focal relationship, to explore their effect on the
likelihoods. Likelihoods are only calculated for relationships that are
possible given the age difference, e.g. PO (parent-offspring) is not
calculated for pairs with an age difference of 0.
</p>
<p>Non-genotyped individuals can be included if they have at least one
genotyped offspring and can be turned into a dummy (see
<code>getAssignCat</code>); to establish this a pedigree must be provided.
</p>
<p><strong>Warning 1</strong>: There is no check whether the input pedigree is genetically
sensible, it is simply conditioned upon. Checking whether a pedigree is
compatible with the SNP data can be done with <code>CalcOHLLR</code>.
</p>
<p><strong>Warning 2</strong>: Conditioning on a <code>Pedigree</code> can make computation
orders of magnitude slower.
</p>


<h3>Value</h3>

<p>The <code>Pairs</code> dataframe including all optional columns listed
above, plus the additional columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>xx</code></td>
<td>
<p>Log10-Likelihood of this pair having relationship xx, with xx
being the relationship abbreviations listed below.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TopRel</code></td>
<td>
<p>Abbreviation of most likely relationship</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LLR</code></td>
<td>
<p>Log10-Likelihood ratio between most-likely and second most likely
relationships. Other LLRs, e.g. between most-likely and unrelated, can
easily be computed.</p>
</td>
</tr>
</table>
<p><strong>Relationship abbreviations:</strong>
</p>
<table>
<tr style="vertical-align: top;">
<td><code>PO</code></td>
<td>
<p>Parent - offspring</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FS</code></td>
<td>
<p>Full siblings</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>HS</code></td>
<td>
<p>Half siblings</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GP</code></td>
<td>
<p>Grandparent</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FA</code></td>
<td>
<p>Full avuncular</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>HA</code></td>
<td>
<p>Half avuncular and other 3rd degree relationships</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>U</code></td>
<td>
<p>Unrelated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>2nd</code></td>
<td>
<p>Unclear which type of 2nd degree relatives
(HS, GP, or FA)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>??</code></td>
<td>
<p>Unclear which type of 1st, 2nd or 3rd degree
relatives</p>
</td>
</tr>
</table>
<p><strong>Likelihood special codes:</strong>
</p>
<table>
<tr style="vertical-align: top;">
<td><code>222</code></td>
<td>
<p>Maybe (via) other parent (e.g. focal="GP", but as likely to be
maternal as paternal grandparent, and therefore not assignable)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>333</code></td>
<td>
<p>Excluded from comparison (shouldn't occur)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>444</code></td>
<td>
<p>Not implemented (e.g. would create an odd double/triple
relationship in combination with the provided pedigree)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>777</code></td>
<td>
<p>Impossible (e.g. cannot be both full sibling and grandparent)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>888</code></td>
<td>
<p>Already assigned in the provided pedigree (see <code>dropPar</code>
arguments)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>999</code></td>
<td>
<p>NA</p>
</td>
</tr>
</table>
<h3>Why does it say 777 (impossible)?</h3>

<p>This function uses the same machinery as <code>sequoia</code>, which will to save
time not calculate the likelihood when it is quickly obvious that the pair
cannot be related in the specified manner.
</p>
<p>For PO (putative parent-offspring pairs) this is the case when:
</p>

<ul>
<li>
<p> the sex of the candidate parent, via <code>Pairs$Sex2</code> or
<code>LifeHistData</code>, does not match <code>Pairs$patmat</code>, which defaults
to 1 (maternal relatives, i.e. dam)
</p>
</li>
<li>
<p> a dam is already assigned via <code>Pedigree</code> and <code>Pairs$dropPar1
    ='none'</code>, and <code>Pairs$patmat = 1</code>
</p>
</li>
<li> <p><code>Pairs$focal</code> is not 'U' (the default), and the OH count between the
two individuals exceeds MaxMismatchOH. This value can be found in
<code>SeqList$Specs</code>), and is calculated by <code>CalcMaxMismatch</code>
</p>
</li>
<li>
<p> the age difference, either calculated from <code>LifeHistData</code> or
specified via <code>Pairs$AgeDif</code>, is impossible for a parent-offspring
pair according to the age prior. The latter can be specified via
<code>AgePrior</code>, or is taken from <code>SeqList</code>, or is calculated when
both <code>Pedigree</code> and <code>LifeHistData</code> are provided.
</p>
</li>
</ul>
<p>For FS (putative full siblings) this happens when e.g. ID1 has a dam
assigned which is not dropped (<code>Pairs$dropPar1='none'</code> or
<code>'sire'</code>), and the OH count between ID1's dam and ID2 exceeds
MaxMismatchOH. The easiest way to 'fix' this is by increasing the presumed
genotyping error rate.
</p>


<h3>Double relationships &amp; focal relationship</h3>

<p>Especially when Complex='full', not only the seven relationship
alternatives listed above are considered, but a whole range of possible
double and even triple relationships. For example, mother A and offspring B
(PO) may also be paternal half-siblings (HS, A and A's mother mated with
same male), grandmother and grand-offspring (GP, B's father is A's son), or
paternal aunt (B's father is a full or half sib of A).
</p>
<p>The likelihood reported as 'LL_PO' is the most-likely one of the possible
alternatives, among those that are not impossible due to age differences or
due to the pedigree (as reconstructed up to that point). Whether e.g. the
likelihood to be both PO &amp; HS is counted as PO or as HS, depends on the
situation and is determined by the variable 'focal': During parentage
assignment, it is counted as PO but not HS, while during sibship
clustering, it is counted as HS but not PO – not omitting from the
alternative relationship would result in a deadlock.
</p>


<h3>See Also</h3>

<p><code>PlotPairLL</code> to plot alternative relationship pairs from
the output; <code>CalcOHLLR</code> to calculate LLR for parents &amp;
parent-pairs in a pedigree; <code>GetRelM</code> to find all pairwise
relatives according to the pedigree; <code>GetMaybeRel</code> to get
likely relative pairs not in the pedigree.
</p>


<h3>Examples</h3>

<pre><code class="language-R">CalcPairLL(Pairs = data.frame(ID1='i116_2006_M', ID2='i119_2006_M'),
           GenoM = Geno_griffin, Err = 1e-04, Plot=FALSE)

## likelihoods underlying parent LLR in pedigree:
# Example: dams for bottom 3 individuals
tail(SeqOUT_griffin$PedigreePar, n=3)
# set up dataframe with these pairs. LLRdam &amp; LLRsire ignore any co-parent
Pairs_d &lt;- data.frame(ID1 = SeqOUT_griffin$PedigreePar$id[140:142],
                      ID2 = SeqOUT_griffin$PedigreePar$dam[140:142],
                      focal = "PO",
                      dropPar1 = 'both')

# Calculate LL's, conditional on the rest of the pedigree + age differences
CalcPairLL(Pairs_d, GenoM = Geno_griffin, Err = 1e-04,
           LifeHistData = LH_griffin, Pedigree = SeqOUT_griffin$PedigreePar)

# LLR changes when ignoring age and/or pedigree, as different relationships
# become (im)possible
CalcPairLL(Pairs_d, GenoM = Geno_griffin, Err = 1e-04)

# LLRpair is calculated conditional on co-parent, and min. of dam &amp; sire LLR
Pairs_d$dropPar1 &lt;- 'dam'
Pairs_s &lt;- data.frame(ID1 = SeqOUT_griffin$PedigreePar$id[141:142],
                      ID2 = SeqOUT_griffin$PedigreePar$sire[141:142],
                      focal = "PO",
                      dropPar1 = 'sire')
CalcPairLL(rbind(Pairs_d, Pairs_s), GenoM = Geno_griffin, Err = 1e-04,
           LifeHistData = LH_griffin, Pedigree = SeqOUT_griffin$PedigreePar)


## likelihoods underlying LLR in getMaybeRel output:
MaybeRel_griffin$MaybePar[1:5, ]
FivePairs &lt;- MaybeRel_griffin$MaybePar[1:5, c("ID1", "ID2", "Sex1", "Sex2")]
PairLL &lt;- CalcPairLL(Pairs = rbind( cbind(FivePairs, focal = "PO"),
                                    cbind(FivePairs, focal = "HS"),
                                    cbind(FivePairs, focal = "GP")),
                     GenoM = Geno_griffin, Plot=FALSE)
PairLL[PairLL$ID1=="i121_2007_M", ]
# LL(FS)==222 : HSHA, HSGP, FAHA more likely than FS
# LL(GP) higher when focal=HS: GP via 'other' parent also considered
# LL(FA) higher when focal=PO: FAHA, or FS of 'other' parent

</code></pre>


</div>