<div class="container">

<table style="width: 100%;"><tr>
<td>CalcMaxMismatch</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Maximum Number of Mismatches</h2>

<h3>Description</h3>

<p>Calculate the maximum expected number of mismatches for
duplicate samples, parent-offspring pairs, and parent-parent-offspring
trios.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CalcMaxMismatch(Err, MAF, ErrFlavour = "version2.9", qntl = 1 - 1e-05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Err</code></td>
<td>
<p>estimated genotyping error rate, as a single number or 3x3 matrix
(averaged value(s) across SNPs), or a vector with the same length as MAF,
or a nSnp x 3 x 3 array. If a matrix, this should be the probability of
observed genotype (columns) conditional on actual genotype (rows). Each row
must therefore sum to 1. If an array, each 3x3 slice should abide this rule.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>vector with minor allele frequency at each SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ErrFlavour</code></td>
<td>
<p>function that takes <code>Err</code> as input, and returns a 3x3
matrix of observed (columns) conditional on actual (rows) genotypes, or
choose from inbuilt ones as used in sequoia 'version2.0', 'version1.3', or
'version1.1'. Ignored if <code>Err</code> is a matrix. See <code>ErrToM</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>qntl</code></td>
<td>
<p>quantile of binomial distribution to be used as the maximum, of
individual-level probability. For a desired dataset-level probability
quantile <code class="reqn">Q</code>, use <code>qntl</code><code class="reqn">= Q^{(1/N)}</code>, where <code class="reqn">N</code> is
the number of individuals.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The thresholds for maximum number of mismatches calculated here aim
to minimise false negatives, i.e. to minimise the chance that any true
duplicates or true parent-offspring pairs are already excluded during the
filtering steps where these <code>MaxMismatch</code> values are used.
Consequently, there is a high probability of false positives, i.e. it is
likely that some sample pairs with fewer mismatches than the
<code>MaxMismatch</code> threshold, are in fact not duplicate samples or
parent-offspring pairs. Use of these <code>MaxMismatch</code> thresholds is
therefore only the first step of pedigree reconstruction by
<code>sequoia</code>.
</p>


<h3>Value</h3>

<p>A vector with three integers:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>DUP</code></td>
<td>
<p>Maximum number of differences between 2 samples from the
same individual</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OH</code></td>
<td>
<p>Maximum number of Opposing Homozygous SNPs between a true
parent-offspring pair</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ME</code></td>
<td>
<p>Maximum number of Mendelian Errors among a true parent-parent-
offspring trio</p>
</td>
</tr>
</table>
<p>.
</p>


<h3>See Also</h3>

<p><code>SnpStats</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">CalcMaxMismatch(Err = 0.05, MAF = runif(n=100, min=0.3, max=0.5))
## Not run: 
CalcMaxMismatch(Err = 0.02, MAF = SnpStats(MyGenoMatrix, Plot=FALSE)[,"AF"])

## End(Not run)

</code></pre>


</div>