<div class="container">

<table style="width: 100%;"><tr>
<td>writeSeq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Write Sequoia Output to File</h2>

<h3>Description</h3>

<p>The various list elements returned by <code>sequoia</code> are each
written to text files in the specified folder, or to separate sheets in a
single excel file (requires library <span class="pkg">openxlsx</span>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">writeSeq(
  SeqList,
  GenoM = NULL,
  MaybeRel = NULL,
  PedComp = NULL,
  OutFormat = "txt",
  folder = "Sequoia-OUT",
  file = "Sequoia-OUT.xlsx",
  ForVersion = 2,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SeqList</code></td>
<td>
<p>list returned by <code>sequoia</code>, to be written out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GenoM</code></td>
<td>
<p>matrix with genetic data (optional). Ignored if
OutFormat='xls', as the resulting file could become too large for excel.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MaybeRel</code></td>
<td>
<p>list with results from <code>GetMaybeRel</code> (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PedComp</code></td>
<td>
<p>list with results from <code>PedCompare</code> (optional).
<code>SeqList$DummyIDs</code> is combined with <code>PedComp$DummyMatch</code> if both
are provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OutFormat</code></td>
<td>
<p>'xls' or 'txt'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>folder</code></td>
<td>
<p>the directory where the text files will be written; will be
created if it does not already exists. Relative to the current working
directory, or NULL for current working directory. Ignored if
<code>OutFormat='xls'</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>the name of the excel file to write to, ignored if
<code>OutFormat='txt'</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ForVersion</code></td>
<td>
<p>choose '1' for back-compatibility with stand-alone sequoia
versions 1.x</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>suppress messages.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The text files can be used as input for the stand-alone Fortran
version of sequoia, e.g. when the genotype data is too large for R. See
<code>vignette('sequoia')</code> for further details.
</p>


<h3>See Also</h3>

<p><code>writeColumns</code> to write to a text file, using white
space padding to keep columns aligned.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
writeSeq(SeqList, OutFormat="xls", file="MyFile.xlsx")

# add additional sheet to the excel file:
library(openxlsx)
wb &lt;- loadWorkbook("MyFile.xlsx")
addWorksheet(wb, sheetName = "ExtraData")
writeData(wb, sheet = "ExtraData", MyData, rowNames=FALSE)
saveWorkbook(wb, "MyFile.xlsx", overwrite=TRUE, returnValue=TRUE)

# or: (package requires java &amp; is trickier to install)
xlsx::write.xlsx(MyData, file = "MyFile.xlsx", sheetName="ExtraData",
      col.names=TRUE, row.names=FALSE, append=TRUE, showNA=FALSE)

## End(Not run)

</code></pre>


</div>