<div class="container">

<table style="width: 100%;"><tr>
<td>SnpStats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SNP Summary Statistics</h2>

<h3>Description</h3>

<p>Estimate allele frequency (AF), missingness and Mendelian
errors per SNP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SnpStats(
  GenoM,
  Pedigree = NULL,
  Duplicates = NULL,
  Plot = TRUE,
  quiet = TRUE,
  ErrFlavour
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>GenoM</code></td>
<td>
<p>genotype matrix, in sequoia's format: 1 column per SNP, 1 row
per individual, genotypes coded as 0/1/2/-9, and row names giving individual
IDs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pedigree</code></td>
<td>
<p>dataframe with 3 columns: ID - parent1 - parent2.
Additional columns and non-genotyped individuals are ignored. Used to count
Mendelian errors per SNP and (poorly) estimate the error rate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Duplicates</code></td>
<td>
<p>dataframe with pairs of duplicated samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Plot</code></td>
<td>
<p>show histograms of the results?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>suppress messages</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ErrFlavour</code></td>
<td>
<p>DEPRECATED AND IGNORED. Was used to estimate <code>Err.hat</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Calculation of these summary statistics can be done in PLINK, and
SNPs with low minor allele frequency or high missingness should be filtered
out prior to pedigree reconstruction. This function is provided as an aid
to inspect the relationship between AF, missingness and genotyping error to
find a suitable combination of SNP filtering thresholds to use.
</p>
<p>For pedigree reconstruction, SNPs with zero or one copies of the alternate
allele in the dataset (MAF <code class="reqn">\le 1/2N</code>) are considered fixed, and
excluded.
</p>


<h3>Value</h3>

<p>A matrix with a number of rows equal to the number of SNPs
(=number of columns of GenoM), and when no Pedigree is provided 2 columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>AF</code></td>
<td>
<p>Allele frequency of the 'second allele' (the one for which the
homozygote is coded 2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Mis</code></td>
<td>
<p>Proportion of missing calls</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>HWE.p</code></td>
<td>
<p>p-value from chi-square test for Hardy-Weinberg equilibrium</p>
</td>
</tr>
</table>
<p>When a Pedigree is provided, there are 8 additional columns:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>n.dam, n.sire, n.pair</code></td>
<td>
<p>Number of dams, sires, parent-pairs successfully
genotyped for the SNP</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OHdam, OHsire</code></td>
<td>
<p>Count of number of opposing homozygous cases</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MEpair</code></td>
<td>
<p>Count of Mendelian errors, includes opposing homozygous cases
when only one parent is genotyped</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.dups, n.diff</code></td>
<td>
<p>Number of duplicate pairs successfully genotyped for
the SNP; number of differences. The latter does not count cases where one
duplicate is not successfully genotyped at the SNP</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>GenoConvert</code> to convert from various data formats;
<code>CheckGeno</code> to check the data is in valid format for sequoia
and exclude monomorphic SNPs etc., <code>CalcOHLLR</code> to calculate OH
&amp; ME per individual.
</p>


<h3>Examples</h3>

<pre><code class="language-R">Genotypes &lt;- SimGeno(Ped_HSg5, nSnp=100, CallRate = runif(100, 0.5, 0.8),
                     SnpError = 0.05)
SnpStats(Genotypes)   # only plots; data is returned invisibly
SNPstats &lt;- SnpStats(Genotypes, Pedigree=Ped_HSg5)

</code></pre>


</div>