<div class="container">

<table style="width: 100%;"><tr>
<td>DuplicateCheck</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Check Data for Duplicates.</h2>

<h3>Description</h3>

<p>Check the genotype and life history data for duplicate IDs (not
permitted) and duplicated genotypes (not advised), and count how many
individuals in the genotype data are not included in the life history data
(permitted). The order of IDs in the genotype and life history data is not
required to be identical.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DuplicateCheck(GenoM = NULL, FortPARAM.dup, quiet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>GenoM</code></td>
<td>
<p>matrix with genotype data, size nInd x nSnp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FortPARAM.dup</code></td>
<td>
<p>list with Fortran-ready parameter values, as generated by
<code>MkFortParams</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>suppress messages.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with one or more of the following elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>DupGenoID</code></td>
<td>
<p>Dataframe, row numbers of duplicated IDs in genotype data.
Please do remove or relabel these to avoid downstream confusion.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DupGenotype</code></td>
<td>
<p>Dataframe, duplicated genotypes (with or without identical
IDs). The specified number of maximum mismatches is allowed, and this
dataframe may include pairs of closely related individuals. Mismatch =
number of SNPs at which genotypes differ, LLR = likelihood ratio between
'self' and most likely non-self.</p>
</td>
</tr>
</table>
<h3>See Also</h3>

<p><code>CheckLH</code>, which performs the check for duplicated IDs
in the life history data, as well as for IDs (in genotype data) for which
no life history data is provided.
</p>


</div>