<div class="container">

<table style="width: 100%;"><tr>
<td>Geno_HSg5</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Example genotype file: 'HSg5'</h2>

<h3>Description</h3>

<p>Simulated genotype data for all* individuals in Pedigree
<code>Ped_HSg5</code> (*: with 40
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(Geno_HSg5)
</code></pre>


<h3>Format</h3>

<p>A genotype matrix with 920 rows (ids) and 200 columns (SNPs). Each
SNP is coded as 0/1/2 copies of the reference allele, with -9 for missing
values. Ids are stored as rownames.
</p>


<h3>Author(s)</h3>

<p>Jisca Huisman, <a href="mailto:jisca.huisman@gmail.com">jisca.huisman@gmail.com</a>
</p>


<h3>See Also</h3>

<p><code>LH_HSg5, SimGeno, SeqOUT_HSg5</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# this output was created as follows:
Geno_HSg5 &lt;- SimGeno(Ped = Ped_HSg5, nSnp = 200, ParMis=0.4,
                     CallRate = 0.9, SnpError = 0.005)

## End(Not run)
</code></pre>


</div>