<div class="container">

<table style="width: 100%;"><tr>
<td>DEenrichRPlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DE and EnrichR pathway visualization barplot</h2>

<h3>Description</h3>

<p>DE and EnrichR pathway visualization barplot
</p>


<h3>Usage</h3>

<pre><code class="language-R">DEenrichRPlot(
  object,
  ident.1 = NULL,
  ident.2 = NULL,
  balanced = TRUE,
  logfc.threshold = 0.25,
  assay = NULL,
  max.genes,
  test.use = "wilcox",
  p.val.cutoff = 0.05,
  cols = NULL,
  enrich.database = NULL,
  num.pathway = 10,
  return.gene.list = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>Name of object class Seurat.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ident.1</code></td>
<td>
<p>Cell class identity 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ident.2</code></td>
<td>
<p>Cell class identity 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>balanced</code></td>
<td>
<p>Option to display pathway enrichments for both negative and
positive DE genes.If false, only positive DE gene will be displayed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logfc.threshold</code></td>
<td>
<p>Limit testing to genes which show, on average, at least
X-fold difference (log-scale) between the two groups of cells. Default is 0.25. 
Increasing logfc.threshold speeds up the function, but can miss weaker signals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Assay to use in differential expression testing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.genes</code></td>
<td>
<p>Maximum number of genes to use as input to enrichR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.use</code></td>
<td>
<p>Denotes which test to use. Available options are:
</p>

<ul>
<li>
<p>"wilcox" : Identifies differentially expressed genes between two
groups of cells using a Wilcoxon Rank Sum test (default); will use a fast
implementation by Presto if installed
</p>
</li>
<li>
<p>"wilcox_limma" : Identifies differentially expressed genes between two
groups of cells using the limma implementation of the Wilcoxon Rank Sum test;
set this option to reproduce results from Seurat v4
</p>
</li>
<li>
<p>"bimod" : Likelihood-ratio test for single cell gene expression,
(McDavid et al., Bioinformatics, 2013)
</p>
</li>
<li>
<p>"roc" : Identifies 'markers' of gene expression using ROC analysis.
For each gene, evaluates (using AUC) a classifier built on that gene alone,
to classify between two groups of cells. An AUC value of 1 means that
expression values for this gene alone can perfectly classify the two
groupings (i.e. Each of the cells in cells.1 exhibit a higher level than
each of the cells in cells.2). An AUC value of 0 also means there is perfect
classification, but in the other direction. A value of 0.5 implies that
the gene has no predictive power to classify the two groups. Returns a
'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially
expressed genes.
</p>
</li>
<li>
<p>"t" : Identify differentially expressed genes between two groups of
cells using the Student's t-test.
</p>
</li>
<li>
<p>"negbinom" : Identifies differentially expressed genes between two
groups of cells using a negative binomial generalized linear model.
Use only for UMI-based datasets
</p>
</li>
<li>
<p>"poisson" : Identifies differentially expressed genes between two
groups of cells using a poisson generalized linear model.
Use only for UMI-based datasets
</p>
</li>
<li>
<p>"LR" : Uses a logistic regression framework to determine differentially
expressed genes. Constructs a logistic regression model predicting group
membership based on each feature individually and compares this to a null
model with a likelihood ratio test.
</p>
</li>
<li>
<p>"MAST" : Identifies differentially expressed genes between two groups
of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST
package to run the DE testing.
</p>
</li>
<li>
<p>"DESeq2" : Identifies differentially expressed genes between two groups
of cells based on a model using DESeq2 which uses a negative binomial
distribution (Love et al, Genome Biology, 2014).This test does not support
pre-filtering of genes based on average difference (or percent detection rate)
between cell groups. However, genes may be pre-filtered based on their
minimum detection rate (min.pct) across both cell groups. To use this method,
please install DESeq2, using the instructions at
https://bioconductor.org/packages/release/bioc/html/DESeq2.html
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.val.cutoff</code></td>
<td>
<p>Cutoff to select DE genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cols</code></td>
<td>
<p>A list of colors to use for barplots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enrich.database</code></td>
<td>
<p>Database to use from enrichR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.pathway</code></td>
<td>
<p>Number of pathways to display in barplot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.gene.list</code></td>
<td>
<p>Return list of DE genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to other methods and to specific DE methods</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns one (only enriched) or two (both enriched and depleted)
barplots with the top enriched/depleted GO terms from EnrichR.
</p>


</div>