<div class="container">

<table style="width: 100%;"><tr>
<td>seurat-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The Seurat Class</h2>

<h3>Description</h3>

<p>The Seurat object is the center of each single cell analysis. It stores all
information associated with the dataset, including data, annotations,
analyses, etc. All that is needed to construct a Seurat object is an
expression matrix (rows are genes, columns are cells), which should
be log-scale
</p>


<h3>Details</h3>

<p>Each Seurat object has a number of slots which store information. Key slots
to access are listed below.
</p>


<h3>Slots</h3>


<dl>
<dt><code>raw.data</code></dt>
<dd>
<p>The raw project data</p>
</dd>
<dt><code>data</code></dt>
<dd>
<p>The normalized expression matrix (log-scale)</p>
</dd>
<dt><code>scale.data</code></dt>
<dd>
<p>scaled (default is z-scoring each gene) expression matrix;
used for dimensional reduction and heatmap visualization</p>
</dd>
<dt><code>var.genes</code></dt>
<dd>
<p>Vector of genes exhibiting high variance across single cells</p>
</dd>
<dt><code>is.expr</code></dt>
<dd>
<p>Expression threshold to determine if a gene is expressed
(0 by default)</p>
</dd>
<dt><code>ident</code></dt>
<dd>
<p>THe 'identity class' for each cell</p>
</dd>
<dt><code>meta.data</code></dt>
<dd>
<p>Contains meta-information about each cell, starting with
number of genes detected (nFeature) and the original identity class
(orig.ident); more information is added using <code>AddMetaData</code></p>
</dd>
<dt><code>project.name</code></dt>
<dd>
<p>Name of the project (for record keeping)</p>
</dd>
<dt><code>dr</code></dt>
<dd>
<p>List of stored dimensional reductions; named by technique</p>
</dd>
<dt><code>assay</code></dt>
<dd>
<p>List of additional assays for multimodal analysis; named by
technique</p>
</dd>
<dt><code>hvg.info</code></dt>
<dd>
<p>The output of the mean/variability analysis for all genes</p>
</dd>
<dt><code>imputed</code></dt>
<dd>
<p>Matrix of imputed gene scores</p>
</dd>
<dt><code>cell.names</code></dt>
<dd>
<p>Names of all single cells
(column names of the expression matrix)</p>
</dd>
<dt><code>cluster.tree</code></dt>
<dd>
<p>List where the first element is a phylo object containing
the phylogenetic tree relating different identity classes</p>
</dd>
<dt><code>snn</code></dt>
<dd>
<p>Spare matrix object representation of the SNN graph</p>
</dd>
<dt><code>calc.params</code></dt>
<dd>
<p>Named list to store all calculation-related
parameter choices</p>
</dd>
<dt><code>kmeans</code></dt>
<dd>
<p>Stores output of gene-based clustering from <code>DoKMeans</code></p>
</dd>
<dt><code>spatial</code></dt>
<dd>
<p>Stores internal data and calculations for spatial mapping of
single cells</p>
</dd>
<dt><code>misc</code></dt>
<dd>
<p>Miscellaneous spot to store any data alongside the object
(for example, gene lists)</p>
</dd>
<dt><code>version</code></dt>
<dd>
<p>Version of package used in object creation</p>
</dd>
</dl>
</div>