<div class="container">

<table style="width: 100%;"><tr>
<td>pbmc_small</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A small example version of the PBMC dataset</h2>

<h3>Description</h3>

<p>A subsetted version of 10X Genomics' 3k PBMC dataset
</p>


<h3>Usage</h3>

<pre><code class="language-R">pbmc_small
</code></pre>


<h3>Format</h3>

<p>A Seurat object with the following slots filled
</p>

<dl>
<dt>assays</dt>
<dd>
<p>Currently only contains one assay ("RNA" - scRNA-seq expression data)
</p>
<ul>
<li>
<p>counts - Raw expression data
</p>
</li>
<li>
<p>data - Normalized expression data
</p>
</li>
<li>
<p>scale.data - Scaled expression data
</p>
</li>
<li>
<p>var.features - names of the current features selected as variable
</p>
</li>
<li>
<p>meta.features - Assay level metadata such as mean and variance
</p>
</li>
</ul>
</dd>
<dt>meta.data</dt>
<dd>
<p>Cell level metadata</p>
</dd>
<dt>active.assay</dt>
<dd>
<p>Current default assay</p>
</dd>
<dt>active.ident</dt>
<dd>
<p>Current default idents</p>
</dd>
<dt>graphs</dt>
<dd>
<p>Neighbor graphs computed, currently stores the SNN</p>
</dd>
<dt>reductions</dt>
<dd>
<p>Dimensional reductions: currently PCA and tSNE</p>
</dd>
<dt>version</dt>
<dd>
<p>Seurat version used to create the object</p>
</dd>
<dt>commands</dt>
<dd>
<p>Command history</p>
</dd>
</dl>
<h3>Source</h3>

<p><a href="https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/pbmc3k">https://support.10xgenomics.com/single-cell-gene-expression/datasets/1.1.0/pbmc3k</a>
</p>


</div>