<div class="container">

<table style="width: 100%;"><tr>
<td>FeaturePlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Visualize 'features' on a dimensional reduction plot</h2>

<h3>Description</h3>

<p>Colors single cells on a dimensional reduction plot according to a 'feature'
(i.e. gene expression, PC scores, number of genes detected, etc.)
</p>


<h3>Usage</h3>

<pre><code class="language-R">FeaturePlot(
  object,
  features,
  dims = c(1, 2),
  cells = NULL,
  cols = if (blend) {
     c("lightgrey", "#ff0000", "#00ff00")
 } else {
    
    c("lightgrey", "blue")
 },
  pt.size = NULL,
  alpha = 1,
  order = FALSE,
  min.cutoff = NA,
  max.cutoff = NA,
  reduction = NULL,
  split.by = NULL,
  keep.scale = "feature",
  shape.by = NULL,
  slot = "data",
  blend = FALSE,
  blend.threshold = 0.5,
  label = FALSE,
  label.size = 4,
  label.color = "black",
  repel = FALSE,
  ncol = NULL,
  coord.fixed = FALSE,
  by.col = TRUE,
  sort.cell = deprecated(),
  interactive = FALSE,
  combine = TRUE,
  raster = NULL,
  raster.dpi = c(512, 512)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>Seurat object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>Vector of features to plot. Features can come from:
</p>

<ul>
<li>
<p> An <code>Assay</code> feature (e.g. a gene name - "MS4A1")
</p>
</li>
<li>
<p> A column name from meta.data (e.g. mitochondrial percentage -
"percent.mito")
</p>
</li>
<li>
<p> A column name from a <code>DimReduc</code> object corresponding to the
cell embedding values (e.g. the PC 1 scores - "PC_1")
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dims</code></td>
<td>
<p>Dimensions to plot, must be a two-length numeric vector specifying x- and y-dimensions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cells</code></td>
<td>
<p>Vector of cells to plot (default is all cells)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cols</code></td>
<td>
<p>The two colors to form the gradient over. Provide as string vector with
the first color corresponding to low values, the second to high. Also accepts a Brewer
color scale or vector of colors. Note: this will bin the data into number of colors provided.
When blend is <code>TRUE</code>, takes anywhere from 1-3 colors:
</p>

<dl>
<dt>1 color:</dt>
<dd>
<p>Treated as color for double-negatives, will use default colors 2 and 3 for per-feature expression</p>
</dd>
<dt>2 colors:</dt>
<dd>
<p>Treated as colors for per-feature expression, will use default color 1 for double-negatives</p>
</dd>
<dt>3+ colors:</dt>
<dd>
<p>First color used for double-negatives, colors 2 and 3 used for per-feature expression, all others ignored</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pt.size</code></td>
<td>
<p>Adjust point size for plotting</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Alpha value for plotting (default is 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>order</code></td>
<td>
<p>Boolean determining whether to plot cells in order of expression. Can be useful if
cells expressing given feature are getting buried.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cutoff, max.cutoff</code></td>
<td>
<p>Vector of minimum and maximum cutoff values for each feature,
may specify quantile in the form of 'q##' where '##' is the quantile (eg, 'q1', 'q10')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reduction</code></td>
<td>
<p>Which dimensionality reduction to use. If not specified, first searches for umap, then tsne, then pca</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split.by</code></td>
<td>
<p>A factor in object metadata to split the plot by, pass 'ident'
to split by cell identity'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.scale</code></td>
<td>
<p>How to handle the color scale across multiple plots. Options are:
</p>

<ul>
<li> <p>“feature” (default; by row/feature scaling): The plots for
each individual feature are scaled to the maximum expression of the
feature across the conditions provided to <code>split.by</code>
</p>
</li>
<li> <p>“all” (universal scaling): The plots for all features and
conditions are scaled to the maximum expression value for the feature
with the highest overall expression
</p>
</li>
<li> <p><code>all</code> (no scaling): Each individual plot is scaled to the
maximum expression value of the feature in the condition provided to
<code>split.by</code>. Be aware setting <code>NULL</code> will result in color
scales that are not comparable between plots
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shape.by</code></td>
<td>
<p>If NULL, all points are circles (default). You can specify any
cell attribute (that can be pulled with FetchData) allowing for both
different colors and different shapes on cells.  Only applicable if <code>raster = FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slot</code></td>
<td>
<p>Which slot to pull expression data from?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blend</code></td>
<td>
<p>Scale and blend expression values to visualize coexpression of two features</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blend.threshold</code></td>
<td>
<p>The color cutoff from weak signal to strong signal; ranges from 0 to 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>Whether to label the clusters</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.size</code></td>
<td>
<p>Sets size of labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label.color</code></td>
<td>
<p>Sets the color of the label text</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>repel</code></td>
<td>
<p>Repel labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncol</code></td>
<td>
<p>Number of columns to combine multiple feature plots to, ignored if <code>split.by</code> is not <code>NULL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coord.fixed</code></td>
<td>
<p>Plot cartesian coordinates with fixed aspect ratio</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by.col</code></td>
<td>
<p>If splitting by a factor, plot the splits per column with the features as rows; ignored if <code>blend = TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort.cell</code></td>
<td>
<p>Redundant with <code>order</code>. This argument is being
deprecated. Please use <code>order</code> instead.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interactive</code></td>
<td>
<p>Launch an interactive <code>FeaturePlot</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>combine</code></td>
<td>
<p>Combine plots into a single <code>patchworked</code>
ggplot object. If <code>FALSE</code>, return a list of ggplot objects</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>raster</code></td>
<td>
<p>Convert points to raster format, default is <code>NULL</code> which
automatically rasterizes if plotting more than 100,000 cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>raster.dpi</code></td>
<td>
<p>Pixel resolution for rasterized plots, passed to geom_scattermore().
Default is c(512, 512).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>patchworked</code> ggplot object if
<code>combine = TRUE</code>; otherwise, a list of ggplot objects
</p>


<h3>Note</h3>

<p>For the old <code>do.hover</code> and <code>do.identify</code> functionality, please see
<code>HoverLocator</code> and <code>CellSelector</code>, respectively.
</p>


<h3>See Also</h3>

<p><code>DimPlot</code> <code>HoverLocator</code>
<code>CellSelector</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("pbmc_small")
FeaturePlot(object = pbmc_small, features = 'PC_1')

</code></pre>


</div>