<div class="container">

<table style="width: 100%;"><tr>
<td>BuildClusterTree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Phylogenetic Analysis of Identity Classes</h2>

<h3>Description</h3>

<p>Constructs a phylogenetic tree relating the 'aggregate' cell from each
identity class. Tree is estimated based on a distance matrix constructed in
either gene expression space or PCA space.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BuildClusterTree(
  object,
  assay = NULL,
  features = NULL,
  dims = NULL,
  reduction = "pca",
  graph = NULL,
  slot = "data",
  reorder = FALSE,
  reorder.numeric = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>Seurat object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Assay to use for the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>Genes to use for the analysis. Default is the set of
variable genes (<code>VariableFeatures(object = object)</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dims</code></td>
<td>
<p>If set, tree is calculated in dimension reduction space;
overrides <code>features</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reduction</code></td>
<td>
<p>Name of dimension reduction to use. Only used if <code>dims</code>
is not NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>graph</code></td>
<td>
<p>If graph is passed, build tree based on graph connectivity between
clusters; overrides <code>dims</code> and <code>features</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slot</code></td>
<td>
<p>slot/layer to use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reorder</code></td>
<td>
<p>Re-order identity classes (factor ordering), according to
position on the tree. This groups similar classes together which can be
helpful, for example, when drawing violin plots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reorder.numeric</code></td>
<td>
<p>Re-order identity classes according to position on
the tree, assigning a numeric value ('1' is the leftmost node)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Show progress updates</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Note that the tree is calculated for an 'aggregate' cell, so gene expression
or PC scores are summed across all cells in an identity class before the
tree is constructed.
</p>


<h3>Value</h3>

<p>A Seurat object where the cluster tree can be accessed with <code>Tool</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
if (requireNamespace("ape", quietly = TRUE)) {
  data("pbmc_small")
  pbmc_small
  pbmc_small &lt;- BuildClusterTree(object = pbmc_small)
  Tool(object = pbmc_small, slot = 'BuildClusterTree')
}

## End(Not run)

</code></pre>


</div>