<div class="container">

<table style="width: 100%;"><tr>
<td>Read10X_h5</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read 10X hdf5 file</h2>

<h3>Description</h3>

<p>Read count matrix from 10X CellRanger hdf5 file.
This can be used to read both scATAC-seq and scRNA-seq matrices.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Read10X_h5(filename, use.names = TRUE, unique.features = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Path to h5 file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.names</code></td>
<td>
<p>Label row names with feature names rather than ID numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unique.features</code></td>
<td>
<p>Make feature names unique (default TRUE)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a sparse matrix with rows and columns labeled. If multiple
genomes are present, returns a list of sparse matrices (one per genome).
</p>


</div>