<div class="container">

<table style="width: 100%;"><tr>
<td>MixscapeHeatmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Differential expression heatmap for mixscape</h2>

<h3>Description</h3>

<p>Draws a heatmap of single cell feature expression with cells ordered by their
mixscape ko probabilities.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MixscapeHeatmap(
  object,
  ident.1 = NULL,
  ident.2 = NULL,
  balanced = TRUE,
  logfc.threshold = 0.25,
  assay = "RNA",
  max.genes = 100,
  test.use = "wilcox",
  max.cells.group = NULL,
  order.by.prob = TRUE,
  group.by = NULL,
  mixscape.class = "mixscape_class",
  prtb.type = "KO",
  fc.name = "avg_log2FC",
  pval.cutoff = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ident.1</code></td>
<td>
<p>Identity class to define markers for; pass an object of class
<code>phylo</code> or 'clustertree' to find markers for a node in a cluster tree;
passing 'clustertree' requires <code>BuildClusterTree</code> to have been run</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ident.2</code></td>
<td>
<p>A second identity class for comparison; if <code>NULL</code>,
use all other cells for comparison; if an object of class <code>phylo</code> or
'clustertree' is passed to <code>ident.1</code>, must pass a node to find markers for</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>balanced</code></td>
<td>
<p>Plot an equal number of genes with both groups of cells.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logfc.threshold</code></td>
<td>
<p>Limit testing to genes which show, on average, at least
X-fold difference (log-scale) between the two groups of cells. Default is 0.25. 
Increasing logfc.threshold speeds up the function, but can miss weaker signals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Assay to use in differential expression testing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.genes</code></td>
<td>
<p>Total number of DE genes to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.use</code></td>
<td>
<p>Denotes which test to use. Available options are:
</p>

<ul>
<li>
<p>"wilcox" : Identifies differentially expressed genes between two
groups of cells using a Wilcoxon Rank Sum test (default); will use a fast
implementation by Presto if installed
</p>
</li>
<li>
<p>"wilcox_limma" : Identifies differentially expressed genes between two
groups of cells using the limma implementation of the Wilcoxon Rank Sum test;
set this option to reproduce results from Seurat v4
</p>
</li>
<li>
<p>"bimod" : Likelihood-ratio test for single cell gene expression,
(McDavid et al., Bioinformatics, 2013)
</p>
</li>
<li>
<p>"roc" : Identifies 'markers' of gene expression using ROC analysis.
For each gene, evaluates (using AUC) a classifier built on that gene alone,
to classify between two groups of cells. An AUC value of 1 means that
expression values for this gene alone can perfectly classify the two
groupings (i.e. Each of the cells in cells.1 exhibit a higher level than
each of the cells in cells.2). An AUC value of 0 also means there is perfect
classification, but in the other direction. A value of 0.5 implies that
the gene has no predictive power to classify the two groups. Returns a
'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially
expressed genes.
</p>
</li>
<li>
<p>"t" : Identify differentially expressed genes between two groups of
cells using the Student's t-test.
</p>
</li>
<li>
<p>"negbinom" : Identifies differentially expressed genes between two
groups of cells using a negative binomial generalized linear model.
Use only for UMI-based datasets
</p>
</li>
<li>
<p>"poisson" : Identifies differentially expressed genes between two
groups of cells using a poisson generalized linear model.
Use only for UMI-based datasets
</p>
</li>
<li>
<p>"LR" : Uses a logistic regression framework to determine differentially
expressed genes. Constructs a logistic regression model predicting group
membership based on each feature individually and compares this to a null
model with a likelihood ratio test.
</p>
</li>
<li>
<p>"MAST" : Identifies differentially expressed genes between two groups
of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST
package to run the DE testing.
</p>
</li>
<li>
<p>"DESeq2" : Identifies differentially expressed genes between two groups
of cells based on a model using DESeq2 which uses a negative binomial
distribution (Love et al, Genome Biology, 2014).This test does not support
pre-filtering of genes based on average difference (or percent detection rate)
between cell groups. However, genes may be pre-filtered based on their
minimum detection rate (min.pct) across both cell groups. To use this method,
please install DESeq2, using the instructions at
https://bioconductor.org/packages/release/bioc/html/DESeq2.html
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cells.group</code></td>
<td>
<p>Number of cells per identity to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>order.by.prob</code></td>
<td>
<p>Order cells on heatmap based on their mixscape knockout
probability from highest to lowest score.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group.by</code></td>
<td>
<p>(Deprecated) Option to split densities based on mixscape
classification. Please use mixscape.class instead</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mixscape.class</code></td>
<td>
<p>metadata column with mixscape classifications.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prtb.type</code></td>
<td>
<p>specify type of CRISPR perturbation expected for labeling
mixscape classifications. Default is KO.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc.name</code></td>
<td>
<p>Name of the fold change, average difference, or custom
function column in the output data.frame. Default is avg_log2FC</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval.cutoff</code></td>
<td>
<p>P-value cut-off for selection of significantly DE genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to other methods and to specific DE methods</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot object.
</p>


</div>