<div class="container">

<table style="width: 100%;"><tr>
<td>Read10X</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Load in data from 10X</h2>

<h3>Description</h3>

<p>Enables easy loading of sparse data matrices provided by 10X genomics.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Read10X(
  data.dir,
  gene.column = 2,
  cell.column = 1,
  unique.features = TRUE,
  strip.suffix = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data.dir</code></td>
<td>
<p>Directory containing the matrix.mtx, genes.tsv (or features.tsv), and barcodes.tsv
files provided by 10X. A vector or named vector can be given in order to load
several data directories. If a named vector is given, the cell barcode names
will be prefixed with the name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.column</code></td>
<td>
<p>Specify which column of genes.tsv or features.tsv to use for gene names; default is 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell.column</code></td>
<td>
<p>Specify which column of barcodes.tsv to use for cell names; default is 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unique.features</code></td>
<td>
<p>Make feature names unique (default TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strip.suffix</code></td>
<td>
<p>Remove trailing "-1" if present in all cell barcodes.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If features.csv indicates the data has multiple data types, a list
containing a sparse matrix of the data from each type will be returned.
Otherwise a sparse matrix containing the expression data will be returned.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# For output from CellRanger &lt; 3.0
data_dir &lt;- 'path/to/data/directory'
list.files(data_dir) # Should show barcodes.tsv, genes.tsv, and matrix.mtx
expression_matrix &lt;- Read10X(data.dir = data_dir)
seurat_object = CreateSeuratObject(counts = expression_matrix)

# For output from CellRanger &gt;= 3.0 with multiple data types
data_dir &lt;- 'path/to/data/directory'
list.files(data_dir) # Should show barcodes.tsv.gz, features.tsv.gz, and matrix.mtx.gz
data &lt;- Read10X(data.dir = data_dir)
seurat_object = CreateSeuratObject(counts = data$`Gene Expression`)
seurat_object[['Protein']] = CreateAssayObject(counts = data$`Antibody Capture`)

## End(Not run)

</code></pre>


</div>