<div class="container">

<table style="width: 100%;"><tr>
<td>LoadXenium</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read and Load 10x Genomics Xenium in-situ data</h2>

<h3>Description</h3>

<p>Read and Load 10x Genomics Xenium in-situ data
</p>


<h3>Usage</h3>

<pre><code class="language-R">LoadXenium(data.dir, fov = "fov", assay = "Xenium")

ReadXenium(
  data.dir,
  outs = c("matrix", "microns"),
  type = "centroids",
  mols.qv.threshold = 20
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data.dir</code></td>
<td>
<p>Directory containing all Xenium output files with
default filenames</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fov</code></td>
<td>
<p>FOV name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Assay name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outs</code></td>
<td>
<p>Types of molecular outputs to read; choose one or more of:
</p>

<ul>
<li> <p>“matrix”: the counts matrix
</p>
</li>
<li> <p>“microns”: molecule coordinates
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Type of cell spatial coordinate matrices to read; choose one
or more of:
</p>

<ul>
<li> <p>“centroids”: cell centroids in pixel coordinate space
</p>
</li>
<li> <p>“segmentations”: cell segmentations in pixel coordinate space
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mols.qv.threshold</code></td>
<td>
<p>Remove transcript molecules with
a QV less than this threshold. QV &gt;= 20 is the standard threshold
used to construct the cell x gene count matrix.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>LoadXenium</code>: A <code>Seurat</code> object
</p>
<p><code>ReadXenium</code>: A list with some combination of the
following values:
</p>

<ul>
<li> <p>“<code>matrix</code>”: a
sparse matrix with expression data; cells
are columns and features are rows
</p>
</li>
<li> <p>“<code>centroids</code>”: a data frame with cell centroid
coordinates in three columns: “x”, “y”, and “cell”
</p>
</li>
<li> <p>“<code>pixels</code>”: a data frame with molecule pixel coordinates
in three columns: “x”, “y”, and “gene”
</p>
</li>
</ul>
</div>