<div class="container">

<table style="width: 100%;"><tr>
<td>RidgePlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Single cell ridge plot</h2>

<h3>Description</h3>

<p>Draws a ridge plot of single cell data (gene expression, metrics, PC
scores, etc.)
</p>


<h3>Usage</h3>

<pre><code class="language-R">RidgePlot(
  object,
  features,
  cols = NULL,
  idents = NULL,
  sort = FALSE,
  assay = NULL,
  group.by = NULL,
  y.max = NULL,
  same.y.lims = FALSE,
  log = FALSE,
  ncol = NULL,
  slot = deprecated(),
  layer = "data",
  stack = FALSE,
  combine = TRUE,
  fill.by = "feature"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>Seurat object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>Features to plot (gene expression, metrics, PC scores,
anything that can be retreived by FetchData)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cols</code></td>
<td>
<p>Colors to use for plotting</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idents</code></td>
<td>
<p>Which classes to include in the plot (default is all)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort</code></td>
<td>
<p>Sort identity classes (on the x-axis) by the average
expression of the attribute being potted, can also pass 'increasing' or 'decreasing' to change sort direction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Name of assay to use, defaults to the active assay</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group.by</code></td>
<td>
<p>Group (color) cells in different ways (for example, orig.ident)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.max</code></td>
<td>
<p>Maximum y axis value</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>same.y.lims</code></td>
<td>
<p>Set all the y-axis limits to the same values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log</code></td>
<td>
<p>plot the feature axis on log scale</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncol</code></td>
<td>
<p>Number of columns if multiple plots are displayed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slot</code></td>
<td>
<p>Slot to pull expression data from (e.g. "counts" or "data")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>layer</code></td>
<td>
<p>Layer to pull expression data from (e.g. "counts" or "data")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stack</code></td>
<td>
<p>Horizontally stack plots for each feature</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>combine</code></td>
<td>
<p>Combine plots into a single <code>patchworked</code>
ggplot object. If <code>FALSE</code>, return a list of ggplot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fill.by</code></td>
<td>
<p>Color violins/ridges based on either 'feature' or 'ident'</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>patchworked</code> ggplot object if
<code>combine = TRUE</code>; otherwise, a list of ggplot objects
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("pbmc_small")
RidgePlot(object = pbmc_small, features = 'PC_1')

</code></pre>


</div>