<div class="container">

<table style="width: 100%;"><tr>
<td>FindMarkers</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene expression markers of identity classes</h2>

<h3>Description</h3>

<p>Finds markers (differentially expressed genes) for identity classes
</p>


<h3>Usage</h3>

<pre><code class="language-R">FindMarkers(object, ...)

## Default S3 method:
FindMarkers(
  object,
  slot = "data",
  cells.1 = NULL,
  cells.2 = NULL,
  features = NULL,
  logfc.threshold = 0.1,
  test.use = "wilcox",
  min.pct = 0.01,
  min.diff.pct = -Inf,
  verbose = TRUE,
  only.pos = FALSE,
  max.cells.per.ident = Inf,
  random.seed = 1,
  latent.vars = NULL,
  min.cells.feature = 3,
  min.cells.group = 3,
  fc.results = NULL,
  densify = FALSE,
  ...
)

## S3 method for class 'Assay'
FindMarkers(
  object,
  slot = "data",
  cells.1 = NULL,
  cells.2 = NULL,
  features = NULL,
  test.use = "wilcox",
  fc.slot = "data",
  pseudocount.use = 1,
  norm.method = NULL,
  mean.fxn = NULL,
  fc.name = NULL,
  base = 2,
  ...
)

## S3 method for class 'SCTAssay'
FindMarkers(
  object,
  cells.1 = NULL,
  cells.2 = NULL,
  features = NULL,
  test.use = "wilcox",
  pseudocount.use = 1,
  slot = "data",
  fc.slot = "data",
  mean.fxn = NULL,
  fc.name = NULL,
  base = 2,
  recorrect_umi = TRUE,
  ...
)

## S3 method for class 'DimReduc'
FindMarkers(
  object,
  cells.1 = NULL,
  cells.2 = NULL,
  features = NULL,
  logfc.threshold = 0.1,
  test.use = "wilcox",
  min.pct = 0.01,
  min.diff.pct = -Inf,
  verbose = TRUE,
  only.pos = FALSE,
  max.cells.per.ident = Inf,
  random.seed = 1,
  latent.vars = NULL,
  min.cells.feature = 3,
  min.cells.group = 3,
  densify = FALSE,
  mean.fxn = rowMeans,
  fc.name = NULL,
  ...
)

## S3 method for class 'Seurat'
FindMarkers(
  object,
  ident.1 = NULL,
  ident.2 = NULL,
  latent.vars = NULL,
  group.by = NULL,
  subset.ident = NULL,
  assay = NULL,
  reduction = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to other methods and to specific DE methods</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slot</code></td>
<td>
<p>Slot to pull data from; note that if <code>test.use</code> is
"negbinom", "poisson", or "DESeq2", <code>slot</code> will be set to "counts"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cells.1</code></td>
<td>
<p>Vector of cell names belonging to group 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cells.2</code></td>
<td>
<p>Vector of cell names belonging to group 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>Genes to test. Default is to use all genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logfc.threshold</code></td>
<td>
<p>Limit testing to genes which show, on average, at least
X-fold difference (log-scale) between the two groups of cells. Default is 0.1
Increasing logfc.threshold speeds up the function, but can miss weaker signals.
If the <code>slot</code> parameter is "scale.data" no filtering is performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.use</code></td>
<td>
<p>Denotes which test to use. Available options are:
</p>

<ul>
<li>
<p>"wilcox" : Identifies differentially expressed genes between two
groups of cells using a Wilcoxon Rank Sum test (default); will use a fast
implementation by Presto if installed
</p>
</li>
<li>
<p>"wilcox_limma" : Identifies differentially expressed genes between two
groups of cells using the limma implementation of the Wilcoxon Rank Sum test;
set this option to reproduce results from Seurat v4
</p>
</li>
<li>
<p>"bimod" : Likelihood-ratio test for single cell gene expression,
(McDavid et al., Bioinformatics, 2013)
</p>
</li>
<li>
<p>"roc" : Identifies 'markers' of gene expression using ROC analysis.
For each gene, evaluates (using AUC) a classifier built on that gene alone,
to classify between two groups of cells. An AUC value of 1 means that
expression values for this gene alone can perfectly classify the two
groupings (i.e. Each of the cells in cells.1 exhibit a higher level than
each of the cells in cells.2). An AUC value of 0 also means there is perfect
classification, but in the other direction. A value of 0.5 implies that
the gene has no predictive power to classify the two groups. Returns a
'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially
expressed genes.
</p>
</li>
<li>
<p>"t" : Identify differentially expressed genes between two groups of
cells using the Student's t-test.
</p>
</li>
<li>
<p>"negbinom" : Identifies differentially expressed genes between two
groups of cells using a negative binomial generalized linear model.
Use only for UMI-based datasets
</p>
</li>
<li>
<p>"poisson" : Identifies differentially expressed genes between two
groups of cells using a poisson generalized linear model.
Use only for UMI-based datasets
</p>
</li>
<li>
<p>"LR" : Uses a logistic regression framework to determine differentially
expressed genes. Constructs a logistic regression model predicting group
membership based on each feature individually and compares this to a null
model with a likelihood ratio test.
</p>
</li>
<li>
<p>"MAST" : Identifies differentially expressed genes between two groups
of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST
package to run the DE testing.
</p>
</li>
<li>
<p>"DESeq2" : Identifies differentially expressed genes between two groups
of cells based on a model using DESeq2 which uses a negative binomial
distribution (Love et al, Genome Biology, 2014).This test does not support
pre-filtering of genes based on average difference (or percent detection rate)
between cell groups. However, genes may be pre-filtered based on their
minimum detection rate (min.pct) across both cell groups. To use this method,
please install DESeq2, using the instructions at
https://bioconductor.org/packages/release/bioc/html/DESeq2.html
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.pct</code></td>
<td>
<p>only test genes that are detected in a minimum fraction of
min.pct cells in either of the two populations. Meant to speed up the function
by not testing genes that are very infrequently expressed. Default is 0.01</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.diff.pct</code></td>
<td>
<p>only test genes that show a minimum difference in the
fraction of detection between the two groups. Set to -Inf by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Print a progress bar once expression testing begins</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only.pos</code></td>
<td>
<p>Only return positive markers (FALSE by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.cells.per.ident</code></td>
<td>
<p>Down sample each identity class to a max number.
Default is no downsampling. Not activated by default (set to Inf)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random.seed</code></td>
<td>
<p>Random seed for downsampling</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>latent.vars</code></td>
<td>
<p>Variables to test, used only when <code>test.use</code> is one of
'LR', 'negbinom', 'poisson', or 'MAST'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cells.feature</code></td>
<td>
<p>Minimum number of cells expressing the feature in at least one
of the two groups, currently only used for poisson and negative binomial tests</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cells.group</code></td>
<td>
<p>Minimum number of cells in one of the groups</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc.results</code></td>
<td>
<p>data.frame from FoldChange</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>densify</code></td>
<td>
<p>Convert the sparse matrix to a dense form before running the 
DE test. This can provide speedups but might require higher memory; default is FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc.slot</code></td>
<td>
<p>Slot used to calculate fold-change - will also affect the 
default for <code>mean.fxn</code>, see below for more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pseudocount.use</code></td>
<td>
<p>Pseudocount to add to averaged expression values when
calculating logFC. 1 by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm.method</code></td>
<td>
<p>Normalization method for fold change calculation when
<code>slot</code> is “<code>data</code>”</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mean.fxn</code></td>
<td>
<p>Function to use for fold change or average difference calculation.
The default depends on the the value of <code>fc.slot</code>:
</p>

<ul>
<li>
<p>"counts" : difference in the log of the mean counts, with pseudocount.
</p>
</li>
<li>
<p>"data" : difference in the log of the average exponentiated data, with pseudocount.
This adjusts for differences in sequencing depth between cells, and assumes that "data"
has been log-normalized.
</p>
</li>
<li>
<p>"scale.data" : difference in the means of scale.data.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc.name</code></td>
<td>
<p>Name of the fold change, average difference, or custom function column
in the output data.frame. If NULL, the fold change column will be named
according to the logarithm base (eg, "avg_log2FC"), or if using the scale.data
slot "avg_diff".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base</code></td>
<td>
<p>The base with respect to which logarithms are computed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>recorrect_umi</code></td>
<td>
<p>Recalculate corrected UMI counts using minimum of the 
median UMIs when performing DE using multiple SCT objects; default is TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ident.1</code></td>
<td>
<p>Identity class to define markers for; pass an object of class
<code>phylo</code> or 'clustertree' to find markers for a node in a cluster tree;
passing 'clustertree' requires <code>BuildClusterTree</code> to have been run</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ident.2</code></td>
<td>
<p>A second identity class for comparison; if <code>NULL</code>,
use all other cells for comparison; if an object of class <code>phylo</code> or
'clustertree' is passed to <code>ident.1</code>, must pass a node to find markers for</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group.by</code></td>
<td>
<p>Regroup cells into a different identity class prior to 
performing differential expression (see example)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset.ident</code></td>
<td>
<p>Subset a particular identity class prior to regrouping. 
Only relevant if group.by is set (see example)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Assay to use in differential expression testing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reduction</code></td>
<td>
<p>Reduction to use in differential expression testing - will 
test for DE on cell embeddings</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>p-value adjustment is performed using bonferroni correction based on
the total number of genes in the dataset. Other correction methods are not
recommended, as Seurat pre-filters genes using the arguments above, reducing
the number of tests performed. Lastly, as Aaron Lun has pointed out, p-values
should be interpreted cautiously, as the genes used for clustering are the
same genes tested for differential expression.
</p>


<h3>Value</h3>

<p>data.frame with a ranked list of putative markers as rows, and associated
statistics as columns (p-values, ROC score, etc., depending on the test used (<code>test.use</code>)). The following columns are always present:
</p>

<ul>
<li> <p><code>avg_logFC</code>: log fold-chage of the average expression between the two groups. Positive values indicate that the gene is more highly expressed in the first group
</p>
</li>
<li> <p><code>pct.1</code>: The percentage of cells where the gene is detected in the first group
</p>
</li>
<li> <p><code>pct.2</code>: The percentage of cells where the gene is detected in the second group
</p>
</li>
<li> <p><code>p_val_adj</code>: Adjusted p-value, based on bonferroni correction using all genes in the dataset
</p>
</li>
</ul>
<h3>References</h3>

<p>McDavid A, Finak G, Chattopadyay PK, et al. Data exploration,
quality control and testing in single-cell qPCR-based gene expression experiments.
Bioinformatics. 2013;29(4):461-467. doi:10.1093/bioinformatics/bts714
</p>
<p>Trapnell C, et al. The dynamics and regulators of cell fate
decisions are revealed by pseudotemporal ordering of single cells. Nature
Biotechnology volume 32, pages 381-386 (2014)
</p>
<p>Andrew McDavid, Greg Finak and Masanao Yajima (2017). MAST: Model-based
Analysis of Single Cell Transcriptomics. R package version 1.2.1.
https://github.com/RGLab/MAST/
</p>
<p>Love MI, Huber W and Anders S (2014). "Moderated estimation of
fold change and dispersion for RNA-seq data with DESeq2." Genome Biology.
https://bioconductor.org/packages/release/bioc/html/DESeq2.html
</p>


<h3>See Also</h3>

<p><code>FoldChange</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data("pbmc_small")
# Find markers for cluster 2
markers &lt;- FindMarkers(object = pbmc_small, ident.1 = 2)
head(x = markers)

# Take all cells in cluster 2, and find markers that separate cells in the 'g1' group (metadata
# variable 'group')
markers &lt;- FindMarkers(pbmc_small, ident.1 = "g1", group.by = 'groups', subset.ident = "2")
head(x = markers)

# Pass 'clustertree' or an object of class phylo to ident.1 and
# a node to ident.2 as a replacement for FindMarkersNode
if (requireNamespace("ape", quietly = TRUE)) {
  pbmc_small &lt;- BuildClusterTree(object = pbmc_small)
  markers &lt;- FindMarkers(object = pbmc_small, ident.1 = 'clustertree', ident.2 = 5)
  head(x = markers)
}

## End(Not run)

</code></pre>


</div>