<div class="container">

<table style="width: 100%;"><tr>
<td>sharp-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>sharp: Stability-enHanced Approaches using Resampling Procedures</h2>

<h3>Description</h3>

<p>In stability selection and consensus clustering, resampling techniques are
used to enhance the reliability of the results. In this package,
hyper-parameters are calibrated by maximising model stability, which is
measured under the null hypothesis that all selection (or co-membership)
probabilities are identical. Functions are readily implemented for the use of
LASSO regression, sparse PCA, sparse (group) PLS or graphical LASSO in
stability selection, and hierarchical clustering, partitioning around
medoids, K means or Gaussian mixture models in consensus clustering.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;"> Package: </td>
<td style="text-align: left;"> sharp</td>
</tr>
<tr>
<td style="text-align: left;"> Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;"> Version: </td>
<td style="text-align: left;">
1.4.6 </td>
</tr>
<tr>
<td style="text-align: left;"> Date: </td>
<td style="text-align: left;"> 2024-02-03 </td>
</tr>
<tr>
<td style="text-align: left;"> License: </td>
<td style="text-align: left;"> GPL (&gt;= 3)</td>
</tr>
<tr>
<td style="text-align: left;"> Maintainer:
</td>
<td style="text-align: left;"> Barbara Bodinier <a href="mailto:barbara.bodinier@gmail.com">barbara.bodinier@gmail.com</a>
</td>
</tr>
</table>
<h3>References</h3>

<p>Bodinier B, Vuckovic D, Rodrigues S, Filippi S, Chiquet J, Chadeau-Hyam M (2023).
“Automated calibration of consensus weighted distance-based clustering approaches using sharp.”
<em>Bioinformatics</em>, btad635.
ISSN 1367-4811, <a href="https://doi.org/10.1093/bioinformatics/btad635">doi:10.1093/bioinformatics/btad635</a>, https://academic.oup.com/bioinformatics/advance-article-pdf/doi/10.1093/bioinformatics/btad635/52191190/btad635.pdf.
</p>
<p>Bodinier B, Filippi S, Nøst TH, Chiquet J, Chadeau-Hyam M (2023).
“Automated calibration for stability selection in penalised regression and graphical models.”
<em>Journal of the Royal Statistical Society Series C: Applied Statistics</em>, qlad058.
ISSN 0035-9254, <a href="https://doi.org/10.1093/jrsssc/qlad058">doi:10.1093/jrsssc/qlad058</a>, https://academic.oup.com/jrsssc/advance-article-pdf/doi/10.1093/jrsssc/qlad058/50878777/qlad058.pdf.
</p>
<p>Meinshausen N, Bühlmann P (2010).
“Stability selection.”
<em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em>, <b>72</b>(4), 417-473.
<a href="https://doi.org/10.1111/j.1467-9868.2010.00740.x">doi:10.1111/j.1467-9868.2010.00740.x</a>.
</p>
<p>Monti S, Tamayo P, Mesirov J, Golub T (2003).
“Consensus Clustering: A Resampling-Based Method for Class Discovery and Visualization of Gene Expression Microarray Data.”
<em>Machine Learning</em>, <b>52</b>(1), 91–118.
<a href="https://doi.org/10.1023/A%3A1023949509487">doi:10.1023/A:1023949509487</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
oldpar &lt;- par(no.readonly = TRUE)
par(mar = c(5, 5, 5, 5))

## Regression models
# Data simulation
set.seed(1)
simul &lt;- SimulateRegression(n = 100, pk = 50)

# Stability selection
stab &lt;- VariableSelection(xdata = simul$xdata, ydata = simul$ydata)
CalibrationPlot(stab)
summary(stab)
SelectedVariables(stab)


## Graphical models
# Data simulation
set.seed(1)
simul &lt;- SimulateGraphical(n = 100, pk = 20, topology = "scale-free")

# Stability selection
stab &lt;- GraphicalModel(xdata = simul$data)
CalibrationPlot(stab)
summary(stab)
plot(stab)


## PCA models
if (requireNamespace("elasticnet", quietly = TRUE)) {
  # Data simulation
  set.seed(1)
  simul &lt;- SimulateComponents(pk = c(5, 3, 4))
  plot(simul)

  # Stability selection
  stab &lt;- BiSelection(
    xdata = simul$data,
    ncomp = 3,
    implementation = SparsePCA
  )
  CalibrationPlot(stab)
  summary(stab)
  SelectedVariables(stab)
}


## PLS models
if (requireNamespace("sgPLS", quietly = TRUE)) {
  # Data simulation
  set.seed(1)
  simul &lt;- SimulateRegression(n = 50, pk = c(10, 20, 30), family = "gaussian")

  # Stability selection
  stab &lt;- BiSelection(
    xdata = simul$xdata, ydata = simul$ydata,
    family = "gaussian", ncomp = 3,
    implementation = SparsePLS
  )
  CalibrationPlot(stab)
  summary(stab)
  plot(stab)
}

par(oldpar)

</code></pre>


</div>