<div class="container">

<table style="width: 100%;"><tr>
<td>sharpr2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>sharpr2</h2>

<h3>Description</h3>

<p>For a HiDRA dataset on a given chromosome, this function calls tiled regions (the regions covered by at least one fragment), and calculates regulatory scores for each tiled region. The regulatory scores are based on standardized log(RNA/PLASMID).
</p>


<h3>Usage</h3>

<pre><code class="language-R">sharpr2(data, l_min = 150, l_max = 600, f_rna = 10, f_dna = 0,
  s_a = 300, verbose = FALSE, auto = TRUE, sig = TRUE, len = FALSE, 
  alpha = 0.05, win = 5, mse = FALSE, max_t = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>A data.frame containing an ATAC-STARR dataset for one chromosome. The data.frame must contain four columns: 'start', 'end', 'PLASMID', 'RNA'. 'PLASMID' and 'RNA' are the values for DNA and RNA, which should be non-negative real numbers (average value over multiple replicates) or integers (counts).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l_min</code></td>
<td>
<p>The fragments with a length smaller than l_min will not be processed. The default is 150.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l_max</code></td>
<td>
<p>The fragments with a length larger than l_max will not be processed. The default is 600.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f_rna</code></td>
<td>
<p>The fragments with an RNA count smaller than f_rna will not be processed. The default is 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f_dna</code></td>
<td>
<p>The fragments with an DNA count smaller than f_rna will not be processed. The default is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s_a</code></td>
<td>
<p>A variance hyperparameter in the prior for the latent regulatory scores. The default is 1000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>An indicator of whether to show processing information. The default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>auto</code></td>
<td>
<p>An indicator of whether to automatically estimate the ridge coefficient <code class="reqn">\lambda</code> from the data for each tiled region using a data-driven way described in the reference. The default is TRUE. If <em>auto</em> is TRUE, <em>s_a</em> is ignored and a ridge coefficient is estimated for each tiled region separately. If <em>auto</em> is FALSE, a global user-defined ridge coefficient (1/s_a) is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig</code></td>
<td>
<p>An indicator of whether to identify significant motif regions for the estimated scores. Only valid if auto=TRUE. The default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>len</code></td>
<td>
<p>An indicator of whether to model log(RNA/PLASMID) of each fragment as the average or the sum of the latent regulatory scores. The default is FALSE, which is the sum.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>A regional FWER to call high resolution driver elements (the significant regulatory region). The default is 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>win</code></td>
<td>
<p>A window size for removing sporadic identified significant regions. If a significant consecutive region is small than <em>win</em>, it will be treated as false signals. The default is 5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mse</code></td>
<td>
<p>An indicator of whether mean square errors are included in the output results. The default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_t</code></td>
<td>
<p>A value between 0 and 1, indicating the proportion of non-zero eigenvectors used to calculate <code class="reqn">\lambda</code> when auto=TRUE. The default is 1.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The default value of s_a is set to be 300, which is equivalent to a ridge coefficient of 0.0033. This default ridge coefficient value is selected by the median of the estimated <code class="reqn">\lambda</code> from the first library. 
</p>


<h3>Value</h3>

<p>score: the regulatory scores for each tiled region. This list contains four components: est_a (the regulatory scores at each locus), sd_e (the sqare root of the mean square error), var_nb (the variance of the esitmate at each locus), <code class="reqn">\lambda</code> (the ridge coefficient).
</p>
<p>region: the start and end positions for each tiled region.
</p>
<p>n_reg: total number of tiled regions.
</p>
<p>n_read: the number of reads in each tiled region.
</p>
<p>sig_reg: identified high resolution driver elements based on the cutoff.
</p>
<p>motif: predicted 20bp motifs 
</p>
<p>cutoff: the cutoff used to call high resolution driver elements for the tiled region.
</p>


<h3>References</h3>

<p>Xinchen Wang, Liang He, Sarah Goggin, Alham Saadat, Li Wang, Melina Claussnitzer, Manolis Kellis. 
High-resolution genome-wide functional dissection of transcriptional regulatory regions in human.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(hidra_ex)
re &lt;- sharpr2(hidra_ex[1:2000,], l_min = 150, l_max = 600, f_dna = 5, f_rna = 0, sig=FALSE)
</code></pre>


</div>