<div class="container">

<table style="width: 100%;"><tr>
<td>createTargetingMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculates a targeting rate matrix</h2>

<h3>Description</h3>

<p><code>createTargetingMatrix</code> calculates the targeting model matrix as the
combined probability of mutability and substitution.
</p>


<h3>Usage</h3>

<pre><code class="language-R">createTargetingMatrix(substitutionModel, mutabilityModel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>substitutionModel</code></td>
<td>
<p>matrix of 5-mers substitution rates built by 
createSubstitutionMatrix or 
extendSubstitutionMatrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutabilityModel</code></td>
<td>
<p>vector of 5-mers mutability rates built by 
createMutabilityMatrix or 
extendMutabilityMatrix.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Targeting rates are calculated by multiplying the normalized mutability rate by the 
normalized substitution rates for each individual 5-mer.
</p>


<h3>Value</h3>

<p>A <code>TargetingMatrix</code> with the same dimensions as the input <code>substitutionModel</code> 
containing normalized targeting probabilities for each 5-mer motif with 
row names defining the center nucleotide and column names defining the 
5-mer nucleotide sequence. 
</p>
<p>If the input <code>mutabilityModel</code> is of class <code>MutabilityModel</code>, then the output 
<code>TargetingMatrix</code> will carry over the input <code>numMutS</code> and <code>numMutR</code> slots.
</p>


<h3>References</h3>


<ol><li>
<p>  Yaari G, et al. Models of somatic hypermutation targeting and substitution based
on synonymous mutations from high-throughput immunoglobulin sequencing data.
Front Immunol. 2013 4(November):358.
</p>
</li></ol>
<h3>See Also</h3>

<p>createSubstitutionMatrix, extendSubstitutionMatrix, 
createMutabilityMatrix, extendMutabilityMatrix, 
TargetingMatrix, createTargetingModel
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Subset example data to 50 sequences, of one isotype and sample as a demo
data(ExampleDb, package="alakazam")
db &lt;- subset(ExampleDb, c_call == "IGHA" &amp; sample_id == "-1h")[1:50,]

# Create 4x1024 models using only silent mutations
sub_model &lt;- createSubstitutionMatrix(db, model="s", sequenceColumn="sequence_alignment",
                                      germlineColumn="germline_alignment_d_mask",
                                      vCallColumn="v_call")
mut_model &lt;- createMutabilityMatrix(db, sub_model, model="s",
                                    sequenceColumn="sequence_alignment",
                                    germlineColumn="germline_alignment_d_mask",
                                    vCallColumn="v_call")

# Extend substitution and mutability to including Ns (5x3125 model)
sub_model &lt;- extendSubstitutionMatrix(sub_model)
mut_model &lt;- extendMutabilityMatrix(mut_model)

# Create targeting model from substitution and mutability
tar_model &lt;- createTargetingMatrix(sub_model, mut_model)


</code></pre>


</div>