<div class="container">

<table style="width: 100%;"><tr>
<td>distToNearest</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Distance to nearest neighbor</h2>

<h3>Description</h3>

<p>Get non-zero distance of every heavy chain (<code>IGH</code>) sequence (as defined by 
<code>sequenceColumn</code>) to its nearest sequence in a partition of heavy chains sharing the same 
V gene, J gene, and junction length (V-J-length), or in a partition of single cells with heavy/long chains
sharing the same heavy/long chain V-J-length combination, or of single cells with heavy/long and light/short chains 
sharing the same heavy/long chain V-J-length and light/short chain V-J-length combinations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">distToNearest(
  db,
  sequenceColumn = "junction",
  vCallColumn = "v_call",
  jCallColumn = "j_call",
  model = c("ham", "aa", "hh_s1f", "hh_s5f", "mk_rs1nf", "mk_rs5nf", "m1n_compat",
    "hs1f_compat"),
  normalize = c("len", "none"),
  symmetry = c("avg", "min"),
  first = TRUE,
  VJthenLen = TRUE,
  nproc = 1,
  fields = NULL,
  cross = NULL,
  mst = FALSE,
  subsample = NULL,
  progress = FALSE,
  cellIdColumn = NULL,
  locusColumn = "locus",
  locusValues = c("IGH"),
  onlyHeavy = TRUE,
  keepVJLgroup = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>data.frame containing sequence data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequenceColumn</code></td>
<td>
<p>name of the column containing the junction for grouping and for calculating
nearest neighbor distances. Note that while both heavy/long and light/short chain junctions
may be used for V-J-length grouping, only the heavy/long chain (IGH, TRB, TRD) junction is 
used to calculate distances.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vCallColumn</code></td>
<td>
<p>name of the column containing the V-segment allele calls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>jCallColumn</code></td>
<td>
<p>name of the column containing the J-segment allele calls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>underlying SHM model, which must be one of 
<code>c("ham", "aa", "hh_s1f", "hh_s5f", "mk_rs1nf", "hs1f_compat", "m1n_compat")</code>.
See Details for further information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize</code></td>
<td>
<p>method of normalization. The default is <code>"len"</code>, which 
divides the distance by the length of the sequence group. If 
<code>"none"</code> then no normalization if performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>symmetry</code></td>
<td>
<p>if model is hs5f, distance between seq1 and seq2 is either the
average (avg) of seq1-&gt;seq2 and seq2-&gt;seq1 or the minimum (min).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>first</code></td>
<td>
<p>if <code>TRUE</code> only the first call of the gene assignments 
is used. if <code>FALSE</code> the union of ambiguous gene 
assignments is used to group all sequences with any 
overlapping gene calls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>VJthenLen</code></td>
<td>
<p>logical value specifying whether to perform partitioning as a 2-stage
process. If <code>TRUE</code>, partitions are made first based on V and J
gene, and then further split based on junction lengths corresponding 
to <code>sequenceColumn</code>. If <code>FALSE</code>, perform partition as a 1-stage 
process during which V gene, J gene, and junction length are used 
to create partitions simultaneously. Defaults to <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>number of cores to distribute the function over.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fields</code></td>
<td>
<p>additional fields to use for grouping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cross</code></td>
<td>
<p>character vector of column names to use for grouping to calculate 
distances across groups. Meaning the columns that define self versus others.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mst</code></td>
<td>
<p>if <code>TRUE</code>, return comma-separated branch lengths from minimum 
spanning tree.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subsample</code></td>
<td>
<p>number of sequences to subsample for speeding up pairwise-distance-matrix calculation. 
Subsampling is performed without replacement in each V-J-length group of heavy chain sequences. 
If <code>subsample</code> is larger than the unique number of heavy chain sequences in each 
VJL group, then the subsampling process is ignored for that group. For each heavy chain
sequence in <code>db</code>, the reported <code>dist_nearest</code> is the distance to the closest
heavy chain sequence in the subsampled set for the V-J-length group. If <code>NULL</code> no 
subsampling is performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress</code></td>
<td>
<p>if <code>TRUE</code> print a progress bar.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cellIdColumn</code></td>
<td>
<p>name of the character column containing cell identifiers or barcodes. 
If specified, grouping will be performed in single-cell mode
with the behavior governed by the <code>locusColumn</code> and 
<code>onlyHeavy</code> arguments. If set to <code>NULL</code> then the 
bulk sequencing data is assumed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locusColumn</code></td>
<td>
<p>name of the column containing locus information. 
Valid loci values
are "IGH", "IGI", "IGK", "IGL", "TRA", "TRB", 
"TRD", and "TRG".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locusValues</code></td>
<td>
<p>Loci values to focus the analysis on.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>onlyHeavy</code></td>
<td>
<p>use only the IGH (BCR) or TRB/TRD (TCR) sequences 
for grouping. Only applicable to single-cell data.
Ignored if <code>cellIdColumn=NULL</code>. 
See groupGenes for further details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keepVJLgroup</code></td>
<td>
<p>logical value specifying whether to keep in the output the the column 
column indicating grouping based on V-J-length combinations. Only applicable for
1-stage partitioning (i.e. <code>VJthenLen=FALSE</code>). Also see 
groupGenes.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To invoke single-cell mode the <code>cellIdColumn</code> argument must be specified and <code>locusColumn</code> 
must be correct. Otherwise, <code>distToNearest</code> will be run with bulk sequencing assumptions, 
using all input sequences regardless of the values in the <code>locusColumn</code> column.
</p>
<p>Under single-cell mode, only heavy/long chain (IGH, TRB, TRD) sequences will be used for calculating 
nearest neighbor distances. Under non-single-cell mode, all input sequences will be used for 
calculating nearest neighbor distances, regardless of the values in the <code>locusColumn</code> field (if present).
</p>
<p>Values in the <code>locusColumn</code> must be one of <code>c("IGH", "IGI", "IGK", "IGL")</code> for BCR 
or <code>c("TRA", "TRB", "TRD", "TRG")</code> for TCR sequences. Otherwise, the function returns an 
error message and stops.
</p>
<p>For single-cell mode, the input format is the same as that for groupGenes. 
Namely, each row represents a sequence/chain. Sequences/chains from the same cell are linked
by a cell ID in the <code>cellIdColumn</code> field. In this mode, there is a choice of whether 
grouping should be done by (a) using IGH (BCR) or TRB/TRD (TCR) sequences only or
(b) using IGH plus IGK/IGL (BCR) or TRB/TRD plus TRA/TRG (TCR). 
This is governed by the <code>onlyHeavy</code> argument.
</p>
<p>Note, <code>distToNearest</code> required that each cell (each unique value in <code>cellIdColumn</code>)
correspond to only a single <code>IGH</code> (BCR) or <code>TRB/TRD</code> (TCR) sequence.
</p>
<p>The distance to nearest neighbor can be used to estimate a threshold for assigning 
Ig sequences to clonal groups. A histogram of the resulting vector is often bimodal, with the 
ideal threshold being a value that separates the two modes.
</p>
<p>The following distance measures are accepted by the <code>model</code> parameter.
</p>

<ul>
<li> <p><code>"ham"</code>:          Single nucleotide Hamming distance matrix from getDNAMatrix 
with gaps assigned zero distance.
</p>
</li>
<li> <p><code>"aa"</code>:           Single amino acid Hamming distance matrix from getAAMatrix.
</p>
</li>
<li> <p><code>"hh_s1f"</code>:       Human single nucleotide distance matrix derived from HH_S1F with 
calcTargetingDistance.
</p>
</li>
<li> <p><code>"hh_s5f"</code>:       Human 5-mer nucleotide context distance matix derived from HH_S5F with 
calcTargetingDistance.
</p>
</li>
<li> <p><code>"mk_rs1nf"</code>:     Mouse single nucleotide distance matrix derived from MK_RS1NF with 
calcTargetingDistance.
</p>
</li>
<li> <p><code>"mk_rs5nf"</code>:     Mouse 5-mer nucleotide context distance matrix derived from MK_RS1NF with 
calcTargetingDistance.
</p>
</li>
<li> <p><code>"hs1f_compat"</code>:  Backwards compatible human single nucleotide distance matrix used in 
SHazaM v0.1.4 and Change-O v0.3.3.
</p>
</li>
<li> <p><code>"m1n_compat"</code>:   Backwards compatibley mouse single nucleotide distance matrix used in 
SHazaM v0.1.4 and Change-O v0.3.3.
</p>
</li>
</ul>
<p>Note on <code>NA</code>s: if, for a given combination of V gene, J gene, and junction length,
there is only 1  heavy chain sequence (as defined by <code>sequenceColumn</code>), <code>NA</code> is 
returned instead of a distance (since it has no heavy/long chain neighbor). If for a given combination 
there are multiple heavy/long chain sequences but only 1 unique one, (in which case every heavy/long cahin 
sequence in this group is the de facto nearest neighbor to each other, thus giving rise to distances 
of 0), <code>NA</code>s are returned instead of zero-distances.
</p>
<p>Note on <code>subsample</code>: Subsampling is performed independently in each V-J-length group for 
heavy/long chain sequences. If <code>subsample</code> is larger than number of heavy/long chain sequences 
in the group, it is ignored. In other words, subsampling is performed only on groups in which the 
number of heavy/long chain sequences is equal to or greater than <code>subsample</code>. <code>dist_nearest</code> 
has values calculated using all heavy chain sequences in the group for groups with fewer than 
<code>subsample</code> heavy/long chain sequences, and values calculated using a subset of heavy/long chain 
sequences for the larger groups. To select a value of <code>subsample</code>, it can be useful to explore 
the group sizes in <code>db</code> (and the number of heavy/long chain sequences in those groups).
</p>


<h3>Value</h3>

<p>Returns a modified <code>db</code> data.frame with nearest neighbor distances between heavy chain
sequences in the <code>dist_nearest</code> column if <code>cross=NULL</code>. If <code>cross</code> was 
specified, distances will be added as the <code>cross_dist_nearest</code> column. 
</p>
<p>Note that distances between light/short (IGK, IGL, TRA, TRG) chain sequences are not calculated, 
even if light/short chains were used for V-J-length grouping via <code>onlyHeavy=FALSE</code>. 
Light/short chain sequences, if any, will have <code>NA</code> in the <code>dist_nearest</code> output column.
</p>
<p>Note that the output <code>vCallColumn</code> and <code>jCallColumn</code> columns will be converted to 
type <code>character</code> if they were type <code>factor</code> in the input <code>db</code>.
</p>


<h3>References</h3>


<ol>
<li>
<p>  Smith DS, et al. Di- and trinucleotide target preferences of somatic 
mutagenesis in normal and autoreactive B cells. 
J Immunol. 1996 156:2642-52. 
</p>
</li>
<li>
<p>  Glanville J, Kuo TC, von Budingen H-C, et al. 
Naive antibody gene-segment frequencies are heritable and unaltered by 
chronic lymphocyte ablation. 
Proc Natl Acad Sci USA. 2011 108(50):20066-71.
</p>
</li>
<li>
<p>  Yaari G, et al. Models of somatic hypermutation targeting and substitution based 
on synonymous mutations from high-throughput immunoglobulin sequencing data. 
Front Immunol. 2013 4:358.
</p>
</li>
</ol>
<h3>See Also</h3>

<p>See calcTargetingDistance for generating nucleotide distance matrices 
from a TargetingModel object. See HH_S5F, HH_S1F, 
MK_RS1NF, getDNAMatrix, and getAAMatrix
for individual model details. getLocus to get locus
values based on allele calls.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Subset example data to one sample as a demo
data(ExampleDb, package="alakazam")
db &lt;- subset(ExampleDb, sample_id == "-1h")

# Use genotyped V assignments, Hamming distance, and normalize by junction length
# First partition based on V and J assignments, then by junction length
# Take into consideration ambiguous V and J annotations
dist &lt;- distToNearest(db, sequenceColumn="junction", 
                      vCallColumn="v_call_genotyped", jCallColumn="j_call",
                      model="ham", first=FALSE, VJthenLen=TRUE, normalize="len")
                           
# Plot histogram of non-NA distances
p1 &lt;- ggplot(data=subset(dist, !is.na(dist_nearest))) + 
      theme_bw() + 
      ggtitle("Distance to nearest: Hamming") + 
      xlab("distance") +
      geom_histogram(aes(x=dist_nearest), binwidth=0.025, 
                     fill="steelblue", color="white")
plot(p1)

</code></pre>


</div>