<div class="container">

<table style="width: 100%;"><tr>
<td>Gap.code</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gap coding</h2>

<h3>Description</h3>

<p>Gap coding of DNA nucleotide alignments</p>


<h3>Usage</h3>

<pre><code class="language-R">Gap.code(seqs)</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>Character vector of aligned (and preferably flank trimmed) DNA sequences.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>FastGap-like gap code nucleotide alignments ('ATGCN-' are allowed).
</p>
<p>Encodes gap presence as 'A' and absence as 'C'.
</p>
<p>Likely too straightforward, and only weakly optimized (really slow).
</p>


<h3>Value</h3>

<p>Outputs character matrix where each column is a gapcoded position.</p>


<h3>Author(s)</h3>

<p>Alexey Shipunov</p>


<h3>References</h3>

<p>Borchsenius F. 2009. FastGap 1.2. Department of Biosciences, Aarhus
University, Denmark. See "http://www.aubot.dk/FastGap_home.htm".
</p>


<h3>Examples</h3>

<pre><code class="language-R">write(file=file.path(tempdir(), "tmp.fasta"),  c(
 "&gt;1\nGAAC------ATGC",
 "&gt;2\nGAAC------TTGC",
 "&gt;3\nGAAC---CCTTTGC",
 "&gt;4\nGAA---------GC"))
write(file=file.path(tempdir(), "tmp_expected.fasta"), c(
 "&gt;1\nGAAC------ATGCCA-",
 "&gt;2\nGAAC------TTGCCA-",
 "&gt;3\nGAAC---CCTTTGCCCA",
 "&gt;4\nGAA---------GCA--"))
tmp &lt;- Read.fasta(file=file.path(tempdir(), "tmp.fasta"))
expected &lt;- Read.fasta(file=file.path(tempdir(), "tmp_expected.fasta"))
seqs &lt;- tmp$sequence
gc &lt;- Gap.code(seqs)
tmp$sequence &lt;- apply(cbind(seqs, gc), 1, paste, collapse="")
identical(tmp, expected) # TRUE, isn't it?
</code></pre>


</div>