<div class="container">

<table style="width: 100%;"><tr>
<td>fitNL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fit Negative Binomial Mixture Model</h2>

<h3>Description</h3>

<p>The function fits a two-component Negative Binomial mixture model. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">fitNL(y, d=NULL, model='E')
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>A vector representing the transformed data that follows the normal mixture distribution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>A vector of the same length as y representing the normalization constant to be applied to the data. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Character specifying E or V model. E model fits the mixture model with equal variance while V model doesn't put any constraint. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calls the mclust package to fit the 2-component normal mixture. 
</p>


<h3>Value</h3>

<p>A vector consisting parameter estimates of mu1, mu2, phi1, phi2, pi1, logLik and BIC. 
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Wang, J.,Wen, S., Symmans,W., Pusztai, L., and Coombes, K. (2009). The bimodality
index: a criterion for discovering and ranking bimodal signatures from cancer gene
expression profiling data. Cancer informatics, 7, 199.
</p>
<p>Fraley, C. and Raftery, A. (2002). Model-based clustering, discriminant analysis, and
density estimation. Journal of the american statistical association,
97(458), 611:631.
</p>
<p>Tong, P., Chen, Y., Su, X. and Coombes, K. R. (2012). Systematic
Identification of Bimodally Expressed Genes Using RNAseq
Data. Bioinformatics, 2013 Mar 1;29(5):605-13. 
</p>


<h3>See Also</h3>

<p>SIBER
fitLN
fitNB
fitGP
</p>


<h3>Examples</h3>

<pre><code class="language-R"># artificial microarray data from normal distribution
set.seed(1000)
dat &lt;- rnorm(100, mean=5, sd=1)
fitNL(y=dat)
</code></pre>


</div>