<div class="container">

<table style="width: 100%;"><tr>
<td>fitNB</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fit Negative Binomial Mixture Model</h2>

<h3>Description</h3>

<p>The function fits a two-component Negative Binomial mixture model. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">fitNB(y, d=NULL, inits=NULL, model='V', zeroPercentThr=0.2)
</code></pre>


<h3>Arguments</h3>

  
<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>A vector representing the RNAseq raw count.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>A vector of the same length as y representing the normalization constant to be applied to the data. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inits</code></td>
<td>
<p>Initial value to fit the mixture model. A vector with elements mu1, mu2, phi1, phi2 and pi1. For 0-inflated model,
only mu2, phi2, pi1 are used while the other elements can be arbitrary. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>Character specifying E or V model. E model fits the mixture model with equal dispersion phi while V model doesn't put any constraint. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zeroPercentThr</code></td>
<td>
<p>A scalar specifying the minimum percent of zero counts needed when fitting a zero-inflated Negative Binomial model. This
parameter is used to deal with zero-inflation in RNAseq count data. When the percent of zero exceeds this threshold, 
rather than fitting a 2-component negative binomial mixture, a mixture of point mass at 0 and negative binomial is fitted. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function directly maximize the log likelihood function through optimization. 
With this function, three models can be fitted: (1) negative binomial mixture with equal dispersion (E model); 
(2) negative binomial mixture with unequal dispersion (V model); (3) 0-inflated negative binomial model.
The 0-inflated negative binomial has the following density function:
</p>
<p><code class="reqn">P(Y=y)=\pi D(y) + (1-\pi)NB(\mu, \phi)</code> where D is the point mass at 0 while <code class="reqn">NB(\mu, \phi)</code> is the density
of negative binomial distribution with mean <code class="reqn">\mu</code> and dispersion <code class="reqn">\phi</code>. The variance is <code class="reqn">\mu+\phi \mu^2</code>.
</p>
<p>The rule to fit 0-inflated model is that the observed percentage of count exceeds the user specified threshold. This
rule overrides the model argument when observed percentae of zero count exceeds the threshold.
</p>


<h3>Value</h3>

<p>A vector consisting parameter estimates of mu1, mu2, phi1, phi2, pi1, logLik and BIC. 
For 0-inflated model, mu1=phi1=0.
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Tong, P., Chen, Y., Su, X. and Coombes, K. R. (2012). Systematic
Identification of Bimodally Expressed Genes Using RNAseq
Data. Bioinformatics, 2013 Mar 1;29(5):605-13. 
</p>


<h3>See Also</h3>

<p>SIBER
fitLN
fitGP
fitNL
</p>


<h3>Examples</h3>

<pre><code class="language-R"># artificial RNAseq data from negative binomial distribution
set.seed(1000)
dat &lt;- rnbinom(100, mu=1000, size=1/0.2)
fitNB(y=dat)
</code></pre>


</div>