<div class="container">

<table style="width: 100%;"><tr>
<td>mutationSummary</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Summary of observed mutations
</h2>

<h3>Description</h3>

<p>This function computes the number of substitutions and indels observed in a given alignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mutationSummary(align, addExtremes = FALSE, output = "brief")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>align</code></td>
<td>

<p>the name of the "DNAbin" alignment to be analysed. See "?read.dna" in the ape  package for details about reading alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addExtremes</code></td>
<td>

<p>a logical; if TRUE, additional nucleotide sites are included in both extremes of the alignment. This will allow estimating distances for alignments showing gaps in terminal positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>a string; defines the kind of output. Two values are accepted:
- "brief" (default) produces an output showing the number of mutations (sites and events).
- "detailed" produces an output showing the number of mutations (sites and events), the position of each mutation, and the state of these sites per sequence (A, T, C, G or - for substitutions and 1 or 0 for indels).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Sites </code></td>
<td>
<p>A matrix containing: the number of sites per sequence (Length); the number of constant and variable sites; the number of singletons and informative sites.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Events </code></td>
<td>
<p>A matrix containing: the number of substitution (singletons, informative, and total) and indel (singletons, informative, and total) events</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Constants.Alignment</code></td>
<td>
<p>A matrix showing constant sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Variables.Alignment</code></td>
<td>
<p>A matrix showing variable sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Singletons.Alignment</code></td>
<td>
<p>A matrix showing singleton sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Inforatives.Alignment</code></td>
<td>
<p>A matrix showing informative sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Substitutions</code></td>
<td>
<p>A matrix showing substitution sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Subst.Single</code></td>
<td>
<p>A matrix showing singleton substitution sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Subst.Info</code></td>
<td>
<p>A matrix showing informative substitution sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gaps</code></td>
<td>
<p>A matrix showing gap sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gaps.Single</code></td>
<td>
<p>A matrix showing singleton gap sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gaps.Info</code></td>
<td>
<p>A matrix showing informative gap sites in the alignment (Only shown if output=="detailed").</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>A. J. Mu√±oz-Pajares</p>


<h3>Examples</h3>

<pre><code class="language-R"># cat("&gt;Population1_sequence1",
# "A-AGGGTC-CT---G",
# "&gt;Population1_sequence2",
# "TAA---TCGCT---G",
# "&gt;Population1_sequence3",
# "TAAGGGTCGCT---G",
# "&gt;Population1_sequence4",
# "TAA---TCGCT---G",
# "&gt;Population2_sequence1",
# "TTACGGTCG---TTG",
# "&gt;Population2_sequence2",
# "TAA---TCG---TTG",
# "&gt;Population2_sequence3",
# "TAA---TCGCTATTG",
# "&gt;Population2_sequence4",
# "TTACGGTCG---TTG",
# "&gt;Population3_sequence1",
# "TTA---TCG---TAG",
# "&gt;Population3_sequence2",
# "TTA---TCG---TAG",
# "&gt;Population3_sequence3",
# "TTA---TCG---TAG",
# "&gt;Population3_sequence4",
# "TTA---TCG---TAG",
#      file = "ex3.fas", sep = "\n")
# 
# # Reading the alignment directly from file and saving no output file:
# library(ape)
# mutationSummary (align=read.dna("ex3.fas",format="fasta"))
# mutationSummary (align=read.dna("ex3.fas",format="fasta"),output="detailed")
# 
# #A more complex alignment
# data(ex_alignment1) # this will read a fasta file with the name 'alignExample'
# mutationSummary(align=alignExample,addExtremes=TRUE)
# 
</code></pre>


</div>