<div class="container">

<table style="width: 100%;"><tr>
<td>nt.gap.comb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>substitution and indel distance combinations</h2>

<h3>Description</h3>

<p>This function obtains a lineal combination from two original matrices. The weight of each matrix in the combination must be defined. If it is a range of values, several matrices are computed.</p>


<h3>Usage</h3>

<pre><code class="language-R">nt.gap.comb(DISTnuc = NA, DISTgap = NA, alpha = seq(0, 1, 0.1),
method = "Corrected", saveFile = TRUE, align = NA, silent = FALSE)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DISTnuc</code></td>
<td>
<p>a matrix containing substitution genetic distances. See "dist.dna" in "ape" package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DISTgap</code></td>
<td>
<p>a matrix containing indel genetic distances.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>a numeric between 0 and 1, is the weight given to the indel genetic distance matrix in the combination. By definition, the weight of the substitution genetic matrix is the complementary value (i.e., 1-alpha). The value "info" will use the proportion of informative substitutions per informative indel event as weight. It is also possible to define multiple weights to estimate different combinations (See examples to obtain 11 corrected combined matrices using a range of alpha values).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a string defining whether each distance matrix must be divided by its maximum value before the combination ("Corrected") or not ("Uncorrected"). Consequently, if the "Corrected" method is chosen, both matrices will range between 0 and 1 before to be combined.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveFile</code></td>
<td>
<p>a logical; if TRUE (default), each output matrix is saved in a different text file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align</code></td>
<td>
<p>if alpha="info" must contain the name of the alignment to be analysed. See "read.dna" in ape package for details about reading alignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent</code></td>
<td>
<p>a logical; if FALSE (default), it prints the number of unique sequences found and the name of the output file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If "alpha" is a single value, this function generates a data frame containing the estimated combination of substitution and indel distance matrices. If "alpha" is a vector of values, this function generates a list of data frames.
</p>


<h3>Author(s)</h3>

<p>A. J. Mu√±oz-Pajares
</p>


<h3>See Also</h3>

<p><code>MCIC</code>,<code>BARRIEL</code>,<code>SIC</code>,<code>FIFTH</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># cat("&gt;Population1_sequence1",
# "TTATAAAATCTA----TAGC",
# "&gt;Population1_sequence2",
# "TAAT----TCTA----TAAC",
# "&gt;Population1_sequence3",
# "TTATAAAAATTA----TAGC",
# "&gt;Population1_sequence4",
# "TAAT----TCTA----TAAC",
# "&gt;Population2_sequence1",
# "TTAT----TCGAGGGGTAGC",
# "&gt;Population2_sequence2",
# "TAAT----TCTA----TAAC",
# "&gt;Population2_sequence3",
# "TTATAAAA--------TAGC",
# "&gt;Population2_sequence4",
# "TTAT----TCGAGGGGTAGC",
# "&gt;Population3_sequence1",
# "TTAT----TCGA----TAGC",
# "&gt;Population3_sequence2",
# "TTAT----TCGA----TAGC",
# "&gt;Population3_sequence3",
# "TTAT----TCGA----TAGC",
# "&gt;Population3_sequence4",
# "TTAT----TCGA----TAGC",
#      file = "ex2.fas", sep = "\n")
# 
#  # Estimating indel distances after reading the alignment from file:
# distGap&lt;-MCIC(input="ex2.fas",saveFile=FALSE)
#  # Estimating substitution distances after reading the alignment from file:
# library(ape)
# align&lt;-read.dna(file="ex2.fas",format="fasta")
# dist.nt&lt;-dist.dna(align,model="raw",pairwise.deletion=TRUE)
# DISTnt&lt;-as.matrix(dist.nt)
#  # Obtaining 11 corrected combined matrices using a range of alpha values:
# nt.gap.comb(DISTgap=distGap, alpha=seq(0,1,0.1), method="Corrected", 
# saveFile=FALSE, DISTnuc=DISTnt)
#  # Obtaining the arithmetic mean of both matrices using both the corrected
#  # and the uncorrected methods.
# nt.gap.comb(DISTgap=distGap, alpha=0.5, method="Uncorrected", saveFile=FALSE,
#  DISTnuc=DISTnt)
#  # Obtaining a range of combinations...
# Range01&lt;-nt.gap.comb(DISTgap=distGap, alpha=seq(0,1,0.1), method="Uncorrected",
#  saveFile=FALSE, DISTnuc=DISTnt)
#  # ...and displaying the arithmetic mean (alpha=0.5 is the element number 6
#  # in the resulting data frame):
# Range01[[6]]
</code></pre>


</div>