<div class="container">

<table style="width: 100%;"><tr>
<td>sidier-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SIDIER: Substitution and Indel Distances to Infer Evolutionary Relationships</h2>

<h3>Description</h3>

<p>Package for evolutionary reconstruction and bar code analyses based on substitutions and insertion-deletion (indels) in a distance-based framework.</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> sidier</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Substitution and Indel Distances to Infer Evolutionary
Relationships</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 4.1.0</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2021-06-25</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> A. Jesus Muñoz Pajares</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> A.J. Muñoz-Pajares &lt;ajesusmp@ugr.es&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
Depends: </td>
<td style="text-align: left;"> R (&gt;= 3.5.0)</td>
</tr>
<tr>
<td style="text-align: left;">
Imports: </td>
<td style="text-align: left;"> ape, network, igraph, gridBase, grid, ggmap, ggplot2</td>
</tr>
<tr>
<td style="text-align: left;">
Encoding: </td>
<td style="text-align: left;"> UTF-8</td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> Evolutionary reconstruction based on substitutions and insertion-deletion (indels) analyses in a distance-based framework.</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
BARRIEL                 Indel distances following Barriel method
Example_Spatial.plot_Alignment
                        example alignment #1 (fasta format)
FIFTH                   Indel distances following the fifth state
                        rationale
FilterHaplo             Filter haplotypes by occurrence
FindHaplo               Find equal haplotypes
GetHaplo                Get sequences of unique haplotypes
HapPerPop               Returns the number of haplotypes per
                        population.
MCIC                    Modified Complex Indel Coding as distance
                        matrix
NINA.thr                No Isolated Nodes Allowed network
SIC                     Indel distances following the Simple Index
                        Coding method
alignExample            example alignment #1 ('DNAbin' class)
assign.whole.taxo       Get taxonomy described in sequence names
barcode.gap             Barcode gap identification
barcode.quality         Estimates of barcode quality
barcode.summary         Summary of the inter- and intraspecific
                        distances
colour.scheme           internal function for node colour scheme
compare.dist            Threshold to discriminate species comparing
                        intra- and interspecific distance distributions
distance.comb           Distance matrices combination
double.plot             Haplotype and population networks including
                        mutations and haplotype frequencies.
ex_BLAST                example BLAST output
ex_Coords               example coordinates
ex_alignment1           example alignment #1
filter.whole.taxo       Get consensus taxonomy
genbank.sp.names        Species names from genbank accessions
get.majority.taxo       Get majority taxonomy for a sequence
inter.intra.plot        Histogram of the intra- and interspecific
                        distances
mergeNodes              Merges nodes showing distance values equal to
                        zero
mutation.network        Haplotype network depiction including mutations
mutationSummary         Summary of observed mutations
nt.gap.comb             substitution and indel distance combinations
perc.thr                Percolation threshold network
pie.network             Population network depiction including
                        haplotype frequencies
pop.dist                Distances among populations
rule                    Threshold to discriminate species.
sidier-package          SIDIER: Substitution and Indel Distances to
                        Infer Evolutionary Relationships
simplify.network        Network showing modules as nodes
simuEvolution           Simulate sequences evolution
single.network          Plot a network given a threshold
single.network.module   Get modules and network given a threshold
spatial.plot            spatial plot of populations
zero.thr                Zero distance networks
</pre>
<p>Functions can be classified according to the following groups:
</p>
<p>- <em>Barcode analysis</em>:
<code>barcode.gap</code>; 
<code>barcode.quality</code>; 
<code>barcode.summary</code>; 
<code>compare.dist</code>; 
<code>genbank.sp.names</code>; 
<code>inter.intra.plot</code>; 
<code>rule</code>. 
</p>
<p>- <em>Example files</em>:
<code>alignExample</code>; 
<code>Example_Spatial.plot_Alignment</code>; 
<code>ex_alignment1</code>; 
<code>ex_Coords</code>.
</p>
<p>- <em>Indel coding methods</em>:
<code>BARRIEL</code>; 
<code>FIFTH</code>; 
<code>MCIC</code>; 
<code>SIC</code>.
</p>
<p>- <em>Matrix/network manipulation</em>:
<code>distance.comb</code>; 
<code>mergeNodes</code>; 
<code>nt.gap.comb</code>; 
<code>simplify.network</code>.
</p>
<p>- <em>Network-from-distance methods</em>:
<code>NINA.thr</code>; 
<code>perc.thr</code>; 
<code>zero.thr</code>; 
<code>single.network</code>; 
<code>single.network.module</code>.
</p>
<p>- <em>Network visualization</em>:
<code>mutation.network</code>; 
<code>pie.network</code>; 
<code>double.plot</code>; 
<code>colour.scheme</code>; 
<code>spatial.plot</code>.
</p>
<p>- <em>Sequence/haplotype analysis</em>:
<code>FilterHaplo</code>; 
<code>FindHaplo</code>; 
<code>GetHaplo</code>; 
<code>HapPerPop</code>; 
<code>mutationSummary</code>; 
<code>pop.dist</code>; 
<code>simuEvolution</code>.
</p>


<h3>Author(s)</h3>

<p>A. Jesus Muñoz Pajares
</p>
<p>Maintainer: A.J. Muñoz-Pajares &lt;ajesusmp@ugr.es&gt;
</p>


<h3>References</h3>

<p>Muñoz-Pajares, A. J. (2013). SIDIER: substitution and indel distances to infer evolutionary relationships. Methods in Ecology and Evolution 4, 1195-1200. doi: 10.1111/2041-210X.12118
</p>


</div>