<div class="container">

<table style="width: 100%;"><tr>
<td>transform_seg_table</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transform Copy Number Table</h2>

<h3>Description</h3>

<p>Transform Copy Number Table
</p>


<h3>Usage</h3>

<pre><code class="language-R">transform_seg_table(
  data,
  genome_build = c("hg19", "hg38", "T2T", "mm10", "mm9", "ce11"),
  ref_type = c("cytoband", "gene"),
  values_fill = NA,
  values_fn = function(x, ...) {
     round(mean(x, ...))
 },
  resolution_factor = 1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a <code>CopyNumber</code> object or a data.frame containing
at least 'chromosome', 'start', 'end', 'segVal', 'sample' these columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>genome build version, used when <code>data</code> is a <code>data.frame</code>, should be 'hg19' or 'hg38'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_type</code></td>
<td>
<p>annotation data type used for constructing matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>values_fill</code></td>
<td>
<p>Optionally, a (scalar) value that specifies what each
<code>value</code> should be filled in with when missing.
</p>
<p>This can be a named list if you want to apply different fill values to
different value columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>values_fn</code></td>
<td>
<p>Optionally, a function applied to the value in each cell
in the output. You will typically use this when the combination of
<code>id_cols</code> and <code>names_from</code> columns does not uniquely identify an
observation.
</p>
<p>This can be a named list if you want to apply different aggregations
to different <code>values_from</code> columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resolution_factor</code></td>
<td>
<p>an integer to control the resolution.
When it is <code>1</code> (default), compute frequency in each cytoband.
When it is <code>2</code>, use compute frequency in each half cytoband.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a <code>data.table</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
load(system.file("extdata", "toy_copynumber.RData",
  package = "sigminer", mustWork = TRUE
))
# Compute the mean segVal in each cytoband
x &lt;- transform_seg_table(cn, resolution_factor = 1)
x
# Compute the mean segVal in each half-cytoband
x2 &lt;- transform_seg_table(cn, resolution_factor = 2)
x2

</code></pre>


</div>