<div class="container">

<table style="width: 100%;"><tr>
<td>get_pLOH_score</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get proportions of pLOH score from Allele Specific Copy Number Profile</h2>

<h3>Description</h3>

<p>pLOH score represents the genome that displayed LOH.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_pLOH_score(data, rm_chrs = c("chrX", "chrY"), genome_build = "hg19")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a CopyNumber object or a <code>data.frame</code> containing at least
'chromosome', 'start', 'end', 'segVal', "minor_cn", 'sample' these columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm_chrs</code></td>
<td>
<p>chromosomes to be removed in calculation. Default is sex
chromosomes (recommended).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>genome build version, should be 'hg19', 'hg38', 'mm9' or 'mm10'.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>data.frame</code>
</p>


<h3>References</h3>

<p>Steele, Christopher D., et al. "Signatures of copy number alterations in human cancer." bioRxiv (2021).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Load toy dataset of absolute copynumber profile
load(system.file("extdata", "toy_segTab.RData",
  package = "sigminer", mustWork = TRUE
))

set.seed(1234)
segTabs$minor_cn &lt;- sample(c(0, 1), size = nrow(segTabs), replace = TRUE)
cn &lt;- read_copynumber(segTabs,
  seg_cols = c("chromosome", "start", "end", "segVal"),
  genome_measure = "wg", complement = TRUE, add_loh = TRUE
)

df &lt;- get_pLOH_score(cn)
df

df2 &lt;- get_pLOH_score(cn@data)
df2

</code></pre>


</div>