<div class="container">

<table style="width: 100%;"><tr>
<td>sig_tally</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Tally a Genomic Alteration Object</h2>

<h3>Description</h3>

<p>Tally a variation object like MAF, CopyNumber and return a matrix for NMF de-composition and more.
This is a generic function,
so it can be further extended to other mutation cases.
<strong>Please read details about how to set sex for identifying copy number signatures</strong>.
Please read <a href="https://osf.io/s93d5/">https://osf.io/s93d5/</a> for the generation of SBS, DBS and ID (INDEL)
components.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sig_tally(object, ...)

## S3 method for class 'CopyNumber'
sig_tally(
  object,
  method = "Wang",
  ignore_chrs = NULL,
  indices = NULL,
  add_loh = FALSE,
  feature_setting = sigminer::CN.features,
  cores = 1,
  keep_only_matrix = FALSE,
  ...
)

## S3 method for class 'RS'
sig_tally(object, keep_only_matrix = FALSE, ...)

## S3 method for class 'MAF'
sig_tally(
  object,
  mode = c("SBS", "DBS", "ID", "ALL"),
  ref_genome = "BSgenome.Hsapiens.UCSC.hg19",
  genome_build = NULL,
  add_trans_bias = FALSE,
  ignore_chrs = NULL,
  use_syn = TRUE,
  keep_only_matrix = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>a CopyNumber object or MAF object or SV object (from read_sv_as_rs).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>custom setting for operating object. Detail see S3 method for
corresponding class (e.g. <code>CopyNumber</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>method for feature classification, can be one of
"Wang" ("W"), "S" (for method described in Steele et al. 2019),
"X" (for method described in Tao et al. 2023).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_chrs</code></td>
<td>
<p>Chromsomes to ignore from analysis. e.g. chrX and chrY.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indices</code></td>
<td>
<p>integer vector indicating segments to keep.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add_loh</code></td>
<td>
<p>flag to add LOH classifications.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>feature_setting</code></td>
<td>
<p>a <code>data.frame</code> used for classification.
<strong>Only used when method is "Wang" ("W")</strong>.
Default is CN.features. Users can also set custom input with "feature",
"min" and "max" columns available. Valid features can be printed by
<code>unique(CN.features$feature)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>number of computer cores to run this task.
You can use <code>future::availableCores()</code> function to check how
many cores you can use.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_only_matrix</code></td>
<td>
<p>if <code>TRUE</code>, keep only matrix for signature extraction.
For a <code>MAF</code> object, this will just return the most useful matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>type of mutation matrix to extract, can be one of 'SBS', 'DBS' and 'ID'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref_genome</code></td>
<td>
<p>'BSgenome.Hsapiens.UCSC.hg19', 'BSgenome.Hsapiens.UCSC.hg38',
'BSgenome.Mmusculus.UCSC.mm10',  'BSgenome.Mmusculus.UCSC.mm9', etc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>genome build 'hg19', 'hg38', 'mm9' or "mm10", if not set, guess it by <code>ref_genome</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>add_trans_bias</code></td>
<td>
<p>if <code>TRUE</code>, consider transcriptional bias categories.
'T:' for Transcribed (the variant is on the transcribed strand);
'U:' for Un-transcribed (the variant is on the untranscribed strand);
'B:' for Bi-directional (the variant is on both strand and is transcribed either way);
'N:' for Non-transcribed (the variant is in a non-coding region and is untranslated);
'Q:' for Questionable.
<strong>NOTE</strong>: the result counts of 'B' and 'N' labels are a little different from
SigProfilerMatrixGenerator, the reason is unknown (may be caused by annotation file).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_syn</code></td>
<td>
<p>Logical. If <code>TRUE</code>, include synonymous variants in analysis.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For identifying copy number signatures, we have to derive copy number
features firstly. Due to the difference of copy number values in sex chromosomes
between male and female, we have to do an extra step <strong>if we don't want to
ignore them</strong>.
</p>
<p>I create two options to control this, the default values are shown as
the following, you can use the same way to set (per R session).
</p>
<p><code>options(sigminer.sex = "female", sigminer.copynumber.max = NA_integer_)</code>
</p>

<ul>
<li>
<p> If your cohort are all females, you can totally ignore this.
</p>
</li>
<li>
<p> If your cohort are all males, set <code>sigminer.sex</code> to 'male' and
<code>sigminer.copynumber.max</code> to a proper value (the best is consistent
with read_copynumber).
</p>
</li>
<li>
<p> If your cohort contains both males and females, set <code>sigminer.sex</code>
as a <code>data.frame</code> with two columns "sample" and "sex". And
set <code>sigminer.copynumber.max</code> to a proper value (the best is consistent
with read_copynumber).
</p>
</li>
</ul>
<h3>Value</h3>

<p>a <code>list</code> contains a <code>matrix</code> used for NMF de-composition.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>sig_tally(CopyNumber)</code>: Returns copy number features, components and component-by-sample matrix
</p>
</li>
<li> <p><code>sig_tally(RS)</code>: Returns genome rearrangement sample-by-component matrix
</p>
</li>
<li> <p><code>sig_tally(MAF)</code>: Returns SBS mutation sample-by-component matrix and APOBEC enrichment
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Shixiang Wang
</p>


<h3>References</h3>

<p>Wang, Shixiang, et al. "Copy number signature analyses in prostate cancer reveal
distinct etiologies and clinical outcomes." medRxiv (2020).
</p>
<p>Steele, Christopher D., et al. "Undifferentiated sarcomas develop through
distinct evolutionary pathways." Cancer Cell 35.3 (2019): 441-456.
</p>
<p>Mayakonda, Anand, et al. "Maftools: efficient and comprehensive analysis of somatic variants in cancer." Genome research 28.11 (2018): 1747-1756.
</p>
<p>Roberts SA, Lawrence MS, Klimczak LJ, et al. An APOBEC Cytidine Deaminase Mutagenesis Pattern is Widespread in Human Cancers. Nature genetics. 2013;45(9):970-976. doi:10.1038/ng.2702.
</p>
<p>Bergstrom EN, Huang MN, Mahto U, Barnes M, Stratton MR, Rozen SG, Alexandrov LB: SigProfilerMatrixGenerator: a tool for visualizing and exploring patterns of small mutational events. BMC Genomics 2019, 20:685 https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6041-2
</p>


<h3>See Also</h3>

<p>sig_estimate for estimating signature number for sig_extract,
sig_auto_extract for extracting signatures using automatic relevance determination technique.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Load copy number object
load(system.file("extdata", "toy_copynumber.RData",
  package = "sigminer", mustWork = TRUE
))

# Use method designed by Wang, Shixiang et al.
cn_tally_W &lt;- sig_tally(cn, method = "W")

# Use method designed by Steele et al.
# See example in read_copynumber

# Prepare SBS signature analysis
laml.maf &lt;- system.file("extdata", "tcga_laml.maf.gz", package = "maftools")
laml &lt;- read_maf(maf = laml.maf)
if (require("BSgenome.Hsapiens.UCSC.hg19")) {
  mt_tally &lt;- sig_tally(
    laml,
    ref_genome = "BSgenome.Hsapiens.UCSC.hg19",
    use_syn = TRUE
  )
  mt_tally$nmf_matrix[1:5, 1:5]

  ## Use strand bias categories
  mt_tally &lt;- sig_tally(
    laml,
    ref_genome = "BSgenome.Hsapiens.UCSC.hg19",
    use_syn = TRUE, add_trans_bias = TRUE
  )
  ## Test it by enrichment analysis
  enrich_component_strand_bias(mt_tally$nmf_matrix)
  enrich_component_strand_bias(mt_tally$all_matrices$SBS_24)
} else {
  message("Please install package 'BSgenome.Hsapiens.UCSC.hg19' firstly!")
}

</code></pre>


</div>