<div class="container">

<table style="width: 100%;"><tr>
<td>get_genome_annotation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Genome Annotation</h2>

<h3>Description</h3>

<p>Get Genome Annotation
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_genome_annotation(
  data_type = c("chr_size", "centro_loc", "cytobands", "transcript", "gene"),
  chrs = paste0("chr", c(1:22, "X", "Y")),
  genome_build = c("hg19", "hg38", "T2T", "mm10", "mm9", "ce11")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data_type</code></td>
<td>
<p>'chr_size' for chromosome size,
'centro_loc' for location of centromeres,
'cytobands' for location of chromosome cytobands
and 'transcript' for location of transcripts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrs</code></td>
<td>
<p>chromosomes start with 'chr'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>one of 'hg19', 'hg38'</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a <code>data.frame</code> containing annotation data
</p>


<h3>Examples</h3>

<pre><code class="language-R">df1 &lt;- get_genome_annotation()
df1

df2 &lt;- get_genome_annotation(genome_build = "hg38")
df2

df3 &lt;- get_genome_annotation(data_type = "centro_loc")
df3

df4 &lt;- get_genome_annotation(data_type = "centro_loc", genome_build = "hg38")
df4

df5 &lt;- get_genome_annotation(data_type = "cytobands")
df5

df6 &lt;- get_genome_annotation(data_type = "cytobands", genome_build = "hg38")
df6
</code></pre>


</div>