<div class="container">

<table style="width: 100%;"><tr>
<td>get_sig_db</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Curated Reference Signature Database</h2>

<h3>Description</h3>

<p>Reference mutational signatures and their aetiologies,
mainly obtained from COSMIC database
(SigProfiler results) and cleaned before saving into
<strong>sigminer</strong> package. You can obtain:
</p>

<ul>
<li>
<p> COSMIC legacy SBS signatures.
</p>
</li>
<li>
<p> COSMIC v3 SBS signatures.
</p>
</li>
<li>
<p> COSMIC v3 DBS signatures.
</p>
</li>
<li>
<p> COSMIC v3 ID (indel) signatures.
</p>
</li>
<li>
<p> SBS and RS (rearrangement) signatures from Nik lab 2020 Nature Cancer paper.
</p>
</li>
<li>
<p> RS signatures from BRCA560 and USARC cohorts.
</p>
</li>
<li>
<p> Copy number signatures from USARC cohort and TCGA.
</p>
</li>
<li>
<p> Copy number signatures from Liu lab 2023. It supports both PCAWG and TCGA cohort.
</p>
</li>
</ul>
<h3>Usage</h3>

<pre><code class="language-R">get_sig_db(sig_db = "legacy")
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>sig_db</code></td>
<td>
<p>default 'legacy', it can be 'legacy' (for <a href="https://cancer.sanger.ac.uk/cosmic/signatures_v2.tt/">COSMIC v2 'SBS'</a>),
'SBS', 'DBS', 'ID' and 'TSB' (for <a href="https://cancer.sanger.ac.uk/cosmic/signatures/">COSMIV v3.1 signatures</a>)
for small scale mutations.
For more specific details, it can also be 'SBS_hg19', 'SBS_hg38',
'SBS_mm9', 'SBS_mm10', 'DBS_hg19', 'DBS_hg38', 'DBS_mm9', 'DBS_mm10' to use
COSMIC v3 reference signatures from Alexandrov, Ludmil B., et al. (2020) (reference #1).
In addition, it can be one of "SBS_Nik_lab_Organ", "RS_Nik_lab_Organ",
"SBS_Nik_lab", "RS_Nik_lab" to refer reference signatures from
Degasperi, Andrea, et al. (2020) (reference #2);
"RS_BRCA560", "RS_USARC" to reference signatures from BRCA560 and USARC cohorts;
"CNS_USARC" (40 categories), "CNS_TCGA" (48 categories) to reference copy number signatures from USARC cohort and TCGA;
"CNS_TCGA176" (176 categories) and "CNS_PCAWG176" (176 categories) to reference copy number signatures from PCAWG and TCGA separately.
<strong>UPDATE</strong>, the latest version of reference version can be automatically
downloaded and loaded from <a href="https://cancer.sanger.ac.uk/signatures/downloads/">https://cancer.sanger.ac.uk/signatures/downloads/</a>
when a option with <code>latest_</code> prefix is specified (e.g. "latest_SBS_GRCh37").
<strong>Note</strong>: the signature profile for different genome builds are basically same.
And specific database (e.g. 'SBS_mm10') contains less signatures than all COSMIC
signatures (because some signatures are not detected from Alexandrov, Ludmil B., et al. (2020)).
For all available options, check the parameter setting.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>a <code>list</code>.
</p>


<h3>References</h3>


<ul>
<li>
<p> Steele, Christopher D., et al. "Signatures of copy number alterations in human cancer." Nature 606.7916 (2022): 984-991.
</p>
</li>
<li>
<p> Alexandrov, Ludmil B., et al. "The repertoire of mutational signatures in human cancer." Nature 578.7793 (2020): 94-101.
</p>
</li>
<li>
<p> Steele, Christopher D., et al. "Undifferentiated sarcomas develop through distinct evolutionary pathways." Cancer Cell 35.3 (2019): 441-456.
</p>
</li>
<li>
<p> Ziyu Tao, et al. "The repertoire of copy number alteration signatures in human cancer." Briefings in Bioinformatics (2023): bbad053.
</p>
</li>
</ul>
<h3>See Also</h3>

<p>get_sig_similarity, sig_fit and show_cosmic_sig_profile.
</p>


<h3>Examples</h3>

<pre><code class="language-R">s1 &lt;- get_sig_db()
s2 &lt;- get_sig_db("SBS")
s3 &lt;- get_sig_db("DBS")
s4 &lt;- get_sig_db("DBS_mm10")
s5 &lt;- get_sig_db("SBS_Nik_lab")
s6 &lt;- get_sig_db("ID")
s7 &lt;- get_sig_db("RS_BRCA560")
s8 &lt;- get_sig_db("RS_USARC")
s9 &lt;- get_sig_db("RS_Nik_lab")
s10 &lt;- get_sig_db("CNS_USARC")
s11 &lt;- get_sig_db("CNS_TCGA")
s12 &lt;- get_sig_db("CNS_TCGA176")
s13 &lt;- get_sig_db("CNS_PCAWG176")
s1
s2
s3
s4
s5
s6
s7
s8
s9
s10
s11
s12
s13
</code></pre>


</div>