<div class="container">

<table style="width: 100%;"><tr>
<td>CopyNumber-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Class CopyNumber</h2>

<h3>Description</h3>

<p>S4 class for storing summarized absolute copy number profile.
</p>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt>
<dd>
<p>data.table of absolute copy number calling.</p>
</dd>
<dt><code>summary.per.sample</code></dt>
<dd>
<p>data.table of copy number variation summary per sample.</p>
</dd>
<dt><code>genome_build</code></dt>
<dd>
<p>genome build version, should be one of 'hg19' or 'hg38'.</p>
</dd>
<dt><code>genome_measure</code></dt>
<dd>
<p>Set 'called' will use autosomo called segments size to compute total size
for CNA burden calculation, this option is useful for WES and target sequencing.
Set 'wg' will autosome size from genome build, this option is useful for WGS, SNP etc..</p>
</dd>
<dt><code>annotation</code></dt>
<dd>
<p>data.table of annotation for copy number segments.</p>
</dd>
<dt><code>dropoff.segs</code></dt>
<dd>
<p>data.table of copy number segments dropped from raw input.</p>
</dd>
</dl>
</div>