<div class="container">

<table style="width: 100%;"><tr>
<td>show_cn_group_profile</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Show Summary Copy Number Profile for Sample Groups</h2>

<h3>Description</h3>

<p>Show Summary Copy Number Profile for Sample Groups
</p>


<h3>Usage</h3>

<pre><code class="language-R">show_cn_group_profile(
  data,
  groups = NULL,
  fill_area = TRUE,
  cols = NULL,
  chrs = paste0("chr", c(1:22, "X")),
  genome_build = c("hg19", "hg38", "T2T", "mm10", "mm9", "ce11"),
  cutoff = 2L,
  resolution_factor = 1L,
  force_y_limit = TRUE,
  highlight_genes = NULL,
  repel = FALSE,
  nrow = NULL,
  ncol = NULL,
  return_plotlist = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a <code>CopyNumber</code> object or a data.frame containing
at least 'chromosome', 'start', 'end', 'segVal', 'sample' these columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groups</code></td>
<td>
<p>a named list or a column name for specifying groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fill_area</code></td>
<td>
<p>default is <code>TRUE</code>, fill area with colors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cols</code></td>
<td>
<p>length-2 colors for AMP and DEL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrs</code></td>
<td>
<p>chromosomes start with 'chr'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_build</code></td>
<td>
<p>genome build version, used when <code>data</code> is a <code>data.frame</code>, should be 'hg19' or 'hg38'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p>copy number value cutoff for splitting data into AMP and DEL.
The values equal to cutoff are discarded. Default is <code>2</code>, you can also set
a length-2 vector, e.g. <code>c(2, 2)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resolution_factor</code></td>
<td>
<p>an integer to control the resolution.
When it is <code>1</code> (default), compute frequency in each cytoband.
When it is <code>2</code>, use compute frequency in each half cytoband.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>force_y_limit</code></td>
<td>
<p>default is <code>TRUE</code>, force multiple plots</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight_genes</code></td>
<td>
<p>gene list to highlight.
have same y ranges. You can also set a length-2 numeric value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>repel</code></td>
<td>
<p>if <code>TRUE</code> (default is <code>FALSE</code>), repel highlight genes to
avoid overlap.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nrow</code></td>
<td>
<p>number of rows in the plot grid when multiple samples are selected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncol</code></td>
<td>
<p>number of columns in the plot grid when multiple samples are selected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_plotlist</code></td>
<td>
<p>default is <code>FALSE</code>, if <code>TRUE</code>, return a plot list instead of a combined plot.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a (list of) <code>ggplot</code> object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">load(system.file("extdata", "toy_copynumber.RData",
  package = "sigminer", mustWork = TRUE
))

p1 &lt;- show_cn_group_profile(cn)
p1

ss &lt;- unique(cn@data$sample)
p2 &lt;- show_cn_group_profile(cn, groups = list(a = ss[1:5], b = ss[6:10]))
p2
p3 &lt;- show_cn_group_profile(cn,
  groups = list(g1 = ss[1:5], g2 = ss[6:10]),
  force_y_limit = c(-1, 1), nrow = 2
)
p3

## Set custom cutoff for custom data
data &lt;- cn@data
data$segVal &lt;- data$segVal - 2L
p4 &lt;- show_cn_group_profile(data,
  groups = list(g1 = ss[1:5], g2 = ss[6:10]),
  force_y_limit = c(-1, 1), nrow = 2,
  cutoff = c(0, 0)
)
p4

## Add highlight gene
p5 &lt;- show_cn_group_profile(cn, highlight_genes = c("TP53", "EGFR"))
p5

</code></pre>


</div>