<div class="container">

<table style="width: 100%;"><tr>
<td>run_signalHsmm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Predict presence of signal peptide in protein</h2>

<h3>Description</h3>

<p>Using the hidden semi-Markov model predict presence of signal peptide in 
eukaryotic proteins.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_signalHsmm(test_data)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>test_data</code></td>
<td>
<p>single protein sequence (<code>character</code> vector) or list of 
sequences. It may be an object of class <code>SeqFastaAA</code>.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>Function <code>signalHsmm</code> returns respectively probability of presence of 
signal peptide, start of signal peptide and the probable cleavage site localization.
If input consists of more than one sequence, result is a data.frame where each column
contains above values for different proteins.
</p>


<h3>Value</h3>

<p>An object of class <code>hsmm_pred_list</code>.
</p>


<h3>Note</h3>

<p>Currently start of signal peptide is naively set as 1 amino acid. The prediction 
of a cleavage site is still an experimental feature, use on your own risk.
</p>


<h3>See Also</h3>

<p><code>hsmm_pred_list</code> <code>hsmm_pred</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#run signalHsmm on one sequence
x1 &lt;- run_signalHsmm(benchmark_dat[[1]])

#run signalHsmm on one sequence, but input is a character vector
x2 &lt;- run_signalHsmm(c("M", "A", "G", "K", "E", "V", "I", "F", "I", "M", "A", "L", 
"F", "I", "A", "V", "E", "S", "S", "P", "I", "F", "S", "F", "D", 
"D", "L", "V", "C", "P", "S", "V", "T", "S", "L", "R", "V", "N", 
"V", "E", "K", "N", "E", "C", "S", "T", "K", "K", "D", "C", "G", 
"R", "N", "L", "C", "C", "E", "N", "Q", "N", "K", "I", "N", "V", 
"C", "V", "G", "G", "I", "M", "P", "L", "P", "K", "P", "N", "L", 
"D", "V", "N", "N", "I", "G", "G", "A", "V", "S", "E", "S", "V", 
"K", "Q", "K", "R", "E", "T", "A", "E", "S", "L"))

#run signalHsmm on list of sequences
x3 &lt;- run_signalHsmm(benchmark_dat[1:3])
#see summary of results
summary(x3)
#print results as data frame
pred2df(x3)
#summary one result
summary(x3[[1]])
plot(x3[[1]])
</code></pre>


</div>