<div class="container">

<table style="width: 100%;"><tr>
<td>FeatureMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Feature Matrix</h2>

<h3>Description</h3>

<p>Construct a feature x cell matrix from a genomic fragments file
</p>


<h3>Usage</h3>

<pre><code class="language-R">FeatureMatrix(
  fragments,
  features,
  cells = NULL,
  process_n = 2000,
  sep = c("-", "-"),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fragments</code></td>
<td>
<p>A list of <code>Fragment</code> objects. Note that if
setting the <code>cells</code> parameter, the requested cells should be present in
the supplied <code>Fragment</code> objects. However, if the cells information in
the fragment object is not set (<code>Cells(fragments)</code> is <code>NULL</code>), then
the fragment object will still be searched.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>A GRanges object containing a set of genomic intervals.
These will form the rows of the matrix, with each entry recording the number
of unique reads falling in the genomic region for each cell. If a genomic 
region provided is on a chromosome that is not present in the fragment file,
it will not be included in the returned matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cells</code></td>
<td>
<p>Vector of cells to include. If NULL, include all cells found
in the fragments file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>process_n</code></td>
<td>
<p>Number of regions to load into memory at a time, per thread.
Processing more regions at once can be faster but uses more memory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>Vector of separators to use for genomic string. First element is
used to separate chromosome and coordinates, second separator is used to
separate start and end coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Display messages</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a sparse matrix
</p>


<h3>Examples</h3>

<pre><code class="language-R">fpath &lt;- system.file("extdata", "fragments.tsv.gz", package="Signac")
fragments &lt;- CreateFragmentObject(fpath)
FeatureMatrix(
  fragments = fragments,
  features = granges(atac_small)
)
</code></pre>


</div>