<div class="container">

<table style="width: 100%;"><tr>
<td>IdentifyVariants</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify mitochondrial variants</h2>

<h3>Description</h3>

<p>Identify mitochondrial variants present in single cells.
</p>


<h3>Usage</h3>

<pre><code class="language-R">IdentifyVariants(object, ...)

## Default S3 method:
IdentifyVariants(
  object,
  refallele,
  stabilize_variance = TRUE,
  low_coverage_threshold = 10,
  verbose = TRUE,
  ...
)

## S3 method for class 'Assay'
IdentifyVariants(object, refallele, ...)

## S3 method for class 'StdAssay'
IdentifyVariants(object, refallele, ...)

## S3 method for class 'Seurat'
IdentifyVariants(object, refallele, assay = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A Seurat object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refallele</code></td>
<td>
<p>A dataframe containing reference alleles for the
mitochondrial genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stabilize_variance</code></td>
<td>
<p>Stabilize variance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>low_coverage_threshold</code></td>
<td>
<p>Low coverage threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Display messages</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Name of assay to use. If NULL, use the default assay.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a dataframe
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
data.dir &lt;- "path/to/data/directory"
mgatk &lt;- ReadMGATK(dir = data.dir)
variant.df &lt;- IdentifyVariants(
  object = mgatk$counts,
  refallele = mgatk$refallele
)

## End(Not run)
</code></pre>


</div>