<div class="container">

<table style="width: 100%;"><tr>
<td>SignacFast</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fast classification of cellular phenotypes</h2>

<h3>Description</h3>

<p><code>SignacFast</code> uses pre-computed neural network models to classify cellular phenotypes in single cell data:
these models were pre-trained with the HPCA training data. Any features that are
present in the training data and absent in the single cell data are set to zero. 
This is a factor of ~5-10 speed improvement over <code>Signac</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SignacFast(
  E,
  Models = "default",
  spring.dir = NULL,
  num.cores = 1,
  threshold = 0,
  smooth = TRUE,
  impute = TRUE,
  verbose = TRUE,
  do.normalize = TRUE,
  return.probability = FALSE,
  graph.used = "nn"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>E</code></td>
<td>
<p>a gene (rows) by cell (column) matrix, sparse matrix or a Seurat object. Rows are HUGO symbols.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Models</code></td>
<td>
<p>if 'default', as returned by <code>GetModels_HPCA</code>. An ensemble of 1,800 neural network models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spring.dir</code></td>
<td>
<p>If using SPRING, directory to categorical_coloring_data.json. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.cores</code></td>
<td>
<p>number of cores to use for parallel computation. Default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>Probability threshold for assigning cells to "Unclassified." Default is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>smooth</code></td>
<td>
<p>if TRUE, smooths the cell type classifications. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute</code></td>
<td>
<p>if TRUE, gene expression values are imputed prior to cell type classification (see <code>KSoftImpute</code>). Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>if TRUE, code will report outputs. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.normalize</code></td>
<td>
<p>if TRUE, cells are normalized to the mean library size. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.probability</code></td>
<td>
<p>if TRUE, returns the probability associated with each cell type label. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>graph.used</code></td>
<td>
<p>If using Seurat object by default, Signac uses the nearest neighbor graph in the graphs field of the Seurat object. Other options are "wnn" to use weighted nearest neighbors, as well as "snn" to use shared nearest neighbors.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of character vectors: cell type annotations (L1, L2, ...) at each level of the hierarchy
as well as 'clusters' for the Louvain clustering results.
</p>


<h3>See Also</h3>

<p><code>Signac</code> for another classification function.
</p>
<p><code>Signac</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# download single cell data for classification
file.dir = "https://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v3/"
file = "pbmc_1k_v3_filtered_feature_bc_matrix.h5"
download.file(paste0(file.dir, file), "Ex.h5")

# load data, process with Seurat
library(Seurat)
E = Read10X_h5(filename = "Ex.h5")
pbmc &lt;- CreateSeuratObject(counts = E, project = "pbmc")
pbmc &lt;- SCTransform(pbmc)
pbmc &lt;- RunPCA(pbmc, verbose = FALSE)
pbmc &lt;- RunUMAP(pbmc, dims = 1:30, verbose = FALSE)
pbmc &lt;- FindNeighbors(pbmc, dims = 1:30, verbose = FALSE)

# classify cells
labels = SignacFast(E = pbmc)
celltypes = GenerateLabels(labels, E = pbmc)

# add labels to Seurat object, visualize
lbls &lt;- factor(celltypes$CellStates)
levels(lbls) &lt;- sort(unique(lbls))
pbmc &lt;- AddMetaData(pbmc, metadata=celltypes$CellStates, col.name = "celltypes")
pbmc &lt;- SetIdent(pbmc, value='celltypes')
DimPlot(pbmc, label = T)

# save results
saveRDS(pbmc, "pbmcs.rds")
saveRDS(celltypes, "celltypes.rds")

## End(Not run)
</code></pre>


</div>