<div class="container">

<table style="width: 100%;"><tr>
<td>ExpressionPlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot gene expression</h2>

<h3>Description</h3>

<p>Display gene expression values for different groups of cells and different
genes. Genes will be arranged on the x-axis and different groups stacked on
the y-axis, with expression value distribution for each group shown as a
violin plot. This is designed to work alongside a genomic coverage track,
and the plot will be able to be aligned with coverage tracks for the same
groups of cells.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ExpressionPlot(
  object,
  features,
  assay = NULL,
  group.by = NULL,
  idents = NULL,
  slot = "data"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A Seurat object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>features</code></td>
<td>
<p>A list of features to plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Name of the assay storing expression information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group.by</code></td>
<td>
<p>A grouping variable to group cells by. If NULL, use the
current cell identities</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idents</code></td>
<td>
<p>A list of identities to include in the plot. If NULL, include
all identities</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>slot</code></td>
<td>
<p>Which slot to pull expression data from</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">
ExpressionPlot(atac_small, features = "TSPAN6", assay = "RNA")

</code></pre>


</div>