<div class="container">

<table style="width: 100%;"><tr>
<td>KSoftImpute</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>KNN-based imputation</h2>

<h3>Description</h3>

<p><code>KSoftImpute</code> is an ultra-fast method for imputing missing gene expression values in single cell data.
<code>KSoftImpute</code> uses k-nearest neighbors to impute the expression of each gene by the weighted average of itself
and it's first-degree neighbors. Weights for imputation are determined by the number of detected genes. This method
works for large data sets (&gt;100,000 cells) in under a minute.
</p>


<h3>Usage</h3>

<pre><code class="language-R">KSoftImpute(E, dM = NULL, genes.to.use = NULL, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>E</code></td>
<td>
<p>A gene-by-sample count matrix (sparse matrix or matrix) with genes identified by their HUGO symbols.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dM</code></td>
<td>
<p>see ?CID.GetDistMat</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes.to.use</code></td>
<td>
<p>a character vector of genes to impute. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If TRUE, code reports outputs. Default is FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An expression matrix (sparse matrix) with imputed values.
</p>


<h3>See Also</h3>

<p><code>Signac</code> and <code>SignacFast</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# download single cell data for classification
file.dir = "https://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v3/"
file = "pbmc_1k_v3_filtered_feature_bc_matrix.h5"
download.file(paste0(file.dir, file), "Ex.h5")

# load data, process with Seurat
library(Seurat)
E = Read10X_h5(filename = "Ex.h5")
pbmc &lt;- CreateSeuratObject(counts = E, project = "pbmc")

# run Seurat pipeline
pbmc &lt;- SCTransform(pbmc, verbose = FALSE)
pbmc &lt;- RunPCA(pbmc, verbose = FALSE)
pbmc &lt;- RunUMAP(pbmc, dims = 1:30, verbose = FALSE)
pbmc &lt;- FindNeighbors(pbmc, dims = 1:30, verbose = FALSE)

# get edges from default assay from Seurat object
default.assay &lt;- Seurat::DefaultAssay(pbmc)
edges = pbmc@graphs[[which(grepl(paste0(default.assay, "_nn"), names(pbmc@graphs)))]]

# get distance matrix
dM = CID.GetDistMat(edges)

# run imputation
Z = KSoftImpute(E = E, dM = dM, verbose = TRUE)

## End(Not run)
</code></pre>


</div>