<div class="container">

<table style="width: 100%;"><tr>
<td>CombineTracks</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Combine genome region plots</h2>

<h3>Description</h3>

<p>This can be used to combine coverage plots, peak region plots, gene
annotation plots, and linked element plots. The different tracks are stacked
on top of each other and the x-axis combined.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CombineTracks(plotlist, expression.plot = NULL, heights = NULL, widths = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>plotlist</code></td>
<td>
<p>A list of plots to combine. Must be from the same genomic
region.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expression.plot</code></td>
<td>
<p>Plot containing gene expression information. If
supplied, this will be placed to the left of the coverage tracks and aligned
with each track</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>heights</code></td>
<td>
<p>Relative heights for each plot. If NULL, the first plot will
be 8x the height of the other tracks.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>widths</code></td>
<td>
<p>Relative widths for each plot. Only required if adding a gene
expression panel. If NULL, main plots will be 8x the width of the gene
expression panel</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a patchworked ggplot2 object
</p>


<h3>Examples</h3>

<pre><code class="language-R">
p1 &lt;- PeakPlot(atac_small, region = "chr1-29554-39554")
p2 &lt;- AnnotationPlot(atac_small, region = "chr1-29554-39554")
CombineTracks(plotlist = list(p1, p2), heights = c(1, 1))

</code></pre>


</div>