<div class="container">

<table style="width: 100%;"><tr>
<td>GenerateLabels</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generates cellular phenotype labels</h2>

<h3>Description</h3>

<p><code>GenerateLabels</code> returns a list of cell type and cell state labels, as well as novel cellular phenotypes and unclassified cells.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GenerateLabels(
  cr,
  E = NULL,
  smooth = TRUE,
  new_populations = NULL,
  new_categories = NULL,
  min.cells = 10,
  spring.dir = NULL,
  graph.used = "nn"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>cr</code></td>
<td>
<p>list returned by <code>Signac</code> or by <code>SignacFast</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>E</code></td>
<td>
<p>a sparse gene (rows) by cell (column) matrix, or a Seurat object. Rows are HUGO symbols.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>smooth</code></td>
<td>
<p>if TRUE, smooths the cell type classifications. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new_populations</code></td>
<td>
<p>Character vector specifying any new cell types that were learned by Signac. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new_categories</code></td>
<td>
<p>If new_populations are set to a cell type, new_category is a corresponding character vector indicating the population that the new population belongs to. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cells</code></td>
<td>
<p>If desired, any cell population with equal to or less than N cells is set to "Unclassified." Default is 10 cells.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spring.dir</code></td>
<td>
<p>If using SPRING, directory to categorical_coloring_data.json. Default is NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>graph.used</code></td>
<td>
<p>If using Seurat object by default, Signac uses the nearest neighbor graph in the graphs field of the Seurat object. Other options are "wnn" to use weighted nearest neighbors, as well as "snn" to use shared nearest neighbors.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of cell type labels for cell types, cell states and novel populations.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# download single cell data for classification
file.dir = "https://cf.10xgenomics.com/samples/cell-exp/3.0.0/pbmc_1k_v3/"
file = "pbmc_1k_v3_filtered_feature_bc_matrix.h5"
download.file(paste0(file.dir, file), "Ex.h5")

# load data, process with Seurat
library(Seurat)
E = Read10X_h5(filename = "Ex.h5")
pbmc &lt;- CreateSeuratObject(counts = E, project = "pbmc")

# run Seurat pipeline
pbmc &lt;- SCTransform(pbmc, verbose = FALSE)
pbmc &lt;- RunPCA(pbmc, verbose = FALSE)
pbmc &lt;- FindNeighbors(pbmc, dims = 1:30, verbose = FALSE)

# classify cells
labels = SignacFast(E = pbmc)
celltypes = GenerateLabels(labels, E = pbmc)

## End(Not run)
</code></pre>


</div>