<div class="container">

<table style="width: 100%;"><tr>
<td>AnnotationPlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot gene annotations</h2>

<h3>Description</h3>

<p>Display gene annotations in a given region of the genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">AnnotationPlot(
  object,
  region,
  assay = NULL,
  mode = "gene",
  sep = c("-", "-"),
  extend.upstream = 0,
  extend.downstream = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A <code>Seurat</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>A genomic region to plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Name of assay to use. If NULL, use the default assay.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>Display mode. Choose either "gene" or "transcript" to determine
whether genes or transcripts are plotted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>Separators to use for strings encoding genomic coordinates. First
element is used to separate the chromosome from the coordinates, second
element is used to separate the start from end coordinate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extend.upstream</code></td>
<td>
<p>Number of bases to extend the region upstream.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extend.downstream</code></td>
<td>
<p>Number of bases to extend the region downstream.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a <code>ggplot</code> object
</p>


<h3>Examples</h3>

<pre><code class="language-R">
AnnotationPlot(object = atac_small, region = c("chr1-29554-39554"))

</code></pre>


</div>