<div class="container">

<table style="width: 100%;"><tr>
<td>AlleleFreq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute allele frequencies per cell</h2>

<h3>Description</h3>

<p>Collapses allele counts for each strand and normalize by the total number of
counts at each nucleotide position.
</p>


<h3>Usage</h3>

<pre><code class="language-R">AlleleFreq(object, ...)

## Default S3 method:
AlleleFreq(object, variants, ...)

## S3 method for class 'Assay'
AlleleFreq(object, variants, ...)

## S3 method for class 'StdAssay'
AlleleFreq(object, variants, ...)

## S3 method for class 'Seurat'
AlleleFreq(object, variants, assay = NULL, new.assay.name = "alleles", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A Seurat object, Assay, or matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Arguments passed to other methods</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variants</code></td>
<td>
<p>A character vector of informative variants to keep. For
example, <code>c("627G&gt;A","709G&gt;A","1045G&gt;A","1793G&gt;A")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>assay</code></td>
<td>
<p>Name of assay to use</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>new.assay.name</code></td>
<td>
<p>Name of new assay to store variant data in</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a <code>Seurat</code> object with a new assay
containing the allele frequencies for the informative variants.
</p>


</div>