<div class="container">

<table style="width: 100%;"><tr>
<td>ChromatinAssay-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The ChromatinAssay class</h2>

<h3>Description</h3>

<p>The ChromatinAssay object is an extended <code>Assay</code>
for the storage and analysis of single-cell chromatin data.
</p>


<h3>Slots</h3>


<dl>
<dt><code>ranges</code></dt>
<dd>
<p>A <code>GRanges</code> object describing the
genomic location of features in the object</p>
</dd>
<dt><code>motifs</code></dt>
<dd>
<p>A <code>Motif</code> object</p>
</dd>
<dt><code>fragments</code></dt>
<dd>
<p>A list of <code>Fragment</code> objects.</p>
</dd>
<dt><code>seqinfo</code></dt>
<dd>
<p>A <code>Seqinfo</code> object containing basic
information about the genome sequence used.</p>
</dd>
<dt><code>annotation</code></dt>
<dd>
<p>A  <code>GRanges</code> object containing
genomic annotations. This should be a GRanges object with the following 
columns:
</p>

<ul>
<li>
<p>tx_id: Transcript ID
</p>
</li>
<li>
<p>gene_name: Gene name
</p>
</li>
<li>
<p>gene_id: Gene ID
</p>
</li>
<li>
<p>gene_biotype: Gene biotype (e.g. "protein_coding", "lincRNA")
</p>
</li>
<li>
<p>type: Annotation type (e.g. "exon", "gap")
</p>
</li>
</ul>
</dd>
<dt><code>bias</code></dt>
<dd>
<p>A vector containing Tn5 integration bias information
(frequency of Tn5 integration at different kmers)</p>
</dd>
<dt><code>positionEnrichment</code></dt>
<dd>
<p>A named list of matrices containing positional
enrichment scores for Tn5 integration (for example, enrichment at the TSS)</p>
</dd>
<dt><code>links</code></dt>
<dd>
<p>A <code>GRanges</code> object describing linked
genomic positions, such as co-accessible sites or enhancer-gene regulatory
relationships. This should be a <code>GRanges</code> object, where the start and
end coordinates are the two linked genomic positions, and must contain a
"score" metadata column.</p>
</dd>
</dl>
</div>