<div class="container">

<table style="width: 100%;"><tr>
<td>GetTSSPositions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find transcriptional start sites</h2>

<h3>Description</h3>

<p>Get the TSS positions from a set of genomic ranges containing gene positions.
Ranges can contain exons, introns, UTRs, etc, rather than the whole
transcript. Only protein coding gene biotypes are included in output.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GetTSSPositions(ranges, biotypes = "protein_coding")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ranges</code></td>
<td>
<p>A GRanges object containing gene annotations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>biotypes</code></td>
<td>
<p>Gene biotypes to include. If NULL, use all biotypes in the
supplied gene annotation.</p>
</td>
</tr>
</table>
</div>