<div class="container">

<table style="width: 100%;"><tr>
<td>GetGRangesFromEnsDb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extract genomic ranges from EnsDb object</h2>

<h3>Description</h3>

<p>Pulls the transcript information for all chromosomes from an EnsDb object.
This wraps <code>crunch</code> and applies the extractor
function to all chromosomes present in the EnsDb object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GetGRangesFromEnsDb(
  ensdb,
  standard.chromosomes = TRUE,
  biotypes = c("protein_coding", "lincRNA", "rRNA", "processed_transcript"),
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ensdb</code></td>
<td>
<p>An EnsDb object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>standard.chromosomes</code></td>
<td>
<p>Keep only standard chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>biotypes</code></td>
<td>
<p>Biotypes to keep</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Display messages</p>
</td>
</tr>
</table>
</div>