<div class="container">

<table style="width: 100%;"><tr>
<td>signatureSurvival-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Signature Survival Analysis
</h2>

<h3>Description</h3>

<p>When multiple Cox proportional hazard models are performed on clinical data 
    (month or year and status) and a set of differential expressions of genes, 
    the results (Hazard risks, z-scores and p-values) can be used to create gene-expression signatures. 
    Weights are calculated using the survival p-values of genes and are utilized to calculate expression 
    values of the signature across the selected genes in all patients in a cohort. A Single or multiple 
    univariate or multivariate Cox proportional hazard survival analyses of the patients in one cohort 
    can be performed by using the gene-expression signature and visualized using our survival plots.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> signatureSurvival</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Signature Survival Analysis</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.0.0</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2023-07-17</td>
</tr>
<tr>
<td style="text-align: left;">
Authors@R: </td>
<td style="text-align: left;"> 
  c(person(given = "Yuan-De",
           family = "Tan",
           role = c("aut", "cre"),
           email = "tanyuande@gmail.com",
           comment = c(ORCID = "0000-0002-0364-2223")),
    person(given = "Yuguang",
           family = "Ban",
           role = "ctb"))</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> Yuan-De Tan &lt;tanyuande@gmail.com&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> When multiple Cox proportional hazard models are performed on clinical data 
    (month or year and status) and a set of differential expressions of genes, 
    the results (Hazard risks, z-scores and p-values) can be used to create gene-expression signatures. 
    Weights are calculated using the survival p-values of genes and are utilized to calculate expression 
    values of the signature across the selected genes in all patients in a cohort. A Single or multiple 
    univariate or multivariate Cox proportional hazard survival analyses of the patients in one cohort 
    can be performed by using the gene-expression signature and visualized using our survival plots.</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL(&gt;=3)</td>
</tr>
<tr>
<td style="text-align: left;">
Depends: </td>
<td style="text-align: left;"> R(&gt;= 3.5.0)</td>
</tr>
<tr>
<td style="text-align: left;">
Imports: </td>
<td style="text-align: left;"> stats, utils,graphics,grDevices,dplyr,forestplot, gplots,
gtools, survival,survminer,ggplot2</td>
</tr>
<tr>
<td style="text-align: left;">
Suggests: </td>
<td style="text-align: left;"> Rmisc</td>
</tr>
<tr>
<td style="text-align: left;">
LazyLoad: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
NeedsCompilation: </td>
<td style="text-align: left;"> no</td>
</tr>
<tr>
<td style="text-align: left;">
Encoding: </td>
<td style="text-align: left;"> UTF-8</td>
</tr>
<tr>
<td style="text-align: left;">
LazyData: </td>
<td style="text-align: left;"> true</td>
</tr>
<tr>
<td style="text-align: left;">
Packaged: </td>
<td style="text-align: left;"> 2023-07-17 13:44:39 UTC; yxt477</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> Yuan-De Tan [aut, cre] (&lt;https://orcid.org/0000-0002-0364-2223&gt;),
  Yuguang Ban [ctb]</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
GSE50081                Survival data from cohort GSE50081
MKMplot                 Multivariate Kaplan-Meier survival curve plot
MMKMplot                Multiple multivariate Kaplan-Meier survival
                        curve plots
MUKMplot                Function multiple univariate Kaplan-Meier
                        survival curve plots
MVKMresult              Multivariate survival analysis with multiple
                        specified independent variables
SKMCresult              Univariate Cox proportional hazard survival
                        analysis with a specified independent variable
TCGA_forestplt          Data for forestplot
TCGA_survivalData       TCGA data for survival analysis
TS_signature            A signature constructed with a set of tumor
                        suppressor genes
UKMplot                 Univariate Kaplan-Meier survival curve plot
musvtest                Multiple univariate suvival tests with a set of
                        genes
mvstest                 Multivariate Cox proportional hazard survival
                        analyses with multiple genes
results                 results of univariate Cox proportional hazard
                        analysis of patients with ADC in three cohorts.
signatureExp            Signature expression or signature score
signatureSurvival-package
                        Signature Survival Analysis
signature_weight        Weights of genes in a signature
survivalForest          Forestplot for result of multivariate Cox
                        proportional hazard survival analysis
weight                  Caculation of Weights for signature genes
</pre>
<p>This package is used to create up and down signatures,do univariate or 
multivariate survival analysis and make forest plot for the results of 
multivariate Cox proportional hazard survival analysis. The steps for screening
signature are as following: At step1, users should perform differential 
expression analysis of genes in one or multiple microarray datasets or RNA-seq 
count datasets or the other expression datasets and then mark these 
differentally expressed (DE) genes selected with "up" and "down" using negative 
or positive t-values. At step2, retrieve survival (or clinical data) using these 
DE genes and construct a new survival data(age, sex, stages/smoking, month, 
status, and DE genes in column and patients in row). Note that expression data 
ofthe DE genes are listed in the right side in the survival data. At step 3,
perform musvtest.R (multiple univariate survival tests) or mvstest
(multiple multivariate survival tests) with covariates age, sex and/smoking ect.
Use p-value to select genes in big difference between low and high-survival 
probalities and use HR and up and down-regulation to classify genes selected 
into up and down groups in multiple cohorts. At step 4,use weight.R to
calculate weight values of each gene in signature and use signatureExp.R to 
caculate expression values of signature in all patients and move the expression
values to the last column in survival data. At step 5, perform MUKMplot.R or 
MMKMplot.R on signature in the survival data to plot Kaplan-Meier survival 
curves.
</p>


<h3>Author(s)</h3>

<p>Yuan-De Tan [aut, cre] (&lt;https://orcid.org/0000-0002-0364-2223&gt;),
  Yuguang Ban [ctb]
</p>
<p>Maintainer: Yuan-De Tan &lt;tanyuande@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(GSE50081)
res&lt;-musvtest(sdata=GSE50081,stn=3500,gn=3506,time="month",status="status",
quant=c("no",-0.2,0.2))

</code></pre>


</div>