<div class="container">

<table style="width: 100%;"><tr>
<td>readVCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read a vcf file, with options to filter out low or high frequency markers.</h2>

<h3>Description</h3>

<p>Read a vcf file, with options to filter out low or high frequency markers.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readVCF(filename = "data.vcf", thin = NA, maxNumberOfVariants = 400,
  min_maf = 0.02, max_maf = NA, region_start = NA, region_end = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Input VCF file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thin</code></td>
<td>
<p>How much to thin markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxNumberOfVariants</code></td>
<td>
<p>Maximum number of variants to keep from region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_maf</code></td>
<td>
<p>Minimum allele frequency of markers to keep. If NA skip min_maf filtering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_maf</code></td>
<td>
<p>Maximum allele frequency of markers to keep. If NA skip max_maf filtering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_start</code></td>
<td>
<p>Extract a region from a vcf files with this starting basepair position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_end</code></td>
<td>
<p>Extract a region from a vcf files with this ending basepair position</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>VCF object to be used by startSimulation function.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
examples_dir = system.file("examples", package = "sim1000G")
vcf_file = file.path(examples_dir,
  "region-chr4-93-TMEM156.vcf.gz")

vcf = readVCF( vcf_file, maxNumberOfVariants = 500 ,
               min_maf = 0.02 ,max_maf = NA)

str(as.list(vcf))
</code></pre>


</div>