<div class="container">

<table style="width: 100%;"><tr>
<td>collapse_do_alleles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Collapse alleles for simulated DO genotypes</h2>

<h3>Description</h3>

<p>When simulating Diversity Outbreds, we need to specify parents
1-16, with 9-16 being the males from strains 1-8.  This function
collapses replaces alleles 9-16 with 1-8, to make the result
ordinary DO-type data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">collapse_do_alleles(xodata)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>xodata</code></td>
<td>
<p>The sort of detailed genotype/crossover data
generated by <code>sim_from_pedigree()</code>.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>The input object, with alleles 9-16 replaced by 1-8.
</p>


<h3>See Also</h3>

<p><code>sim_do_pedigree()</code>, <code>sim_do_pedigree_fix_n()</code>,
<code>sim_from_pedigree()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># simulate DO pedigree
tab &lt;- sim_do_pedigree(8)

# simulate genotypes for that pedigree
dat &lt;- sim_from_pedigree(tab)
# collapse to alleles 1-8
dat &lt;- collapse_do_alleles(dat)

# also works with data on multiple chromosomes
dat &lt;- sim_from_pedigree(tab, c("1"=100, "2"=75, "X"=100), xchr="X")
dat &lt;- collapse_do_alleles(dat)
</code></pre>


</div>