<div class="container">

<table style="width: 100%;"><tr>
<td>convert2geno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert continuous allele information into marker genotypes</h2>

<h3>Description</h3>

<p>Convert the continuous crossover location information produced by
sim_from_pedigree to marker genotypes
</p>


<h3>Usage</h3>

<pre><code class="language-R">convert2geno(xodat, map, founder_geno = NULL, shift_map = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xodat</code></td>
<td>
<p>The sort of detailed genotype/crossover data generated by
<code>sim_from_pedigree()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>vector of marker locations; can also be a list of such
vectors (one per chromosome), in which case xodat and founder_geno
must be lists with the same length.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>founder_geno</code></td>
<td>
<p>Optional matrix (size <code>n_founders</code> x
<code>length(map)</code>) of founder genotypes. If coded as 1/2 (or 1/3),
results are 1/2/3 genotypes. If coded as A/T/G/C/N, results are
A/T/G/C/N/H genotypes. If coded as letters A-H for the 8 founders,
results are two-letter genotypes AA-HH with 36 possible values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shift_map</code></td>
<td>
<p>If TRUE, shift genetic map to start at 0</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>founder_geno</code> is provided or there are just two
founders, the result is a numeric matrix of genotypes, individuals
x markers, with genotypes 1/2/3 codes for 11/12/22 genotypes.
</p>
<p>If <code>founder_geno</code> is not provided and there are more than two
founders, the result is a 3-dimensional array, individuals x
markers x alleles, with the third dimensional corresponding to the
maternal and paternal allele.
</p>
<p>If the input <code>map</code> is a list (the components being
chromosomes), then <code>xodat</code> and <code>founder_geno</code> must be
lists of the same length, and the result will be a list of
matrices.
</p>


<h3>See Also</h3>

<p><code>get_geno()</code>, <code>sim_from_pedigree()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># simulate AIL pedigree
tab &lt;- sim_ail_pedigree(12, 30)
# simulate data from that pedigree
dat &lt;- sim_from_pedigree(tab)
# marker map (could also use sim.map in R/qtl)
map &lt;- seq(0, 100, by=5)
names(map) &lt;- paste0("marker", seq(along=map))
# convert data to marker genotypes
geno &lt;- convert2geno(dat, map)


# AIL with multiple chromosomes
dat &lt;- sim_from_pedigree(tab, c("1"=100, "2"=75, "X"=100), xchr="X")
# marker map
multmap &lt;- list("1"=seq(0, 100, by=5),
                "2"=seq(0, 75, by=5),
                "X"=seq(0, 100, by=5))
for(i in 1:3)
  names(multmap[[i]]) &lt;- paste0("marker", i, "_", 1:length(map[[i]]))
geno &lt;- convert2geno(dat, multmap)

# simulate DO pedigree
tab &lt;- sim_do_pedigree(8)
# simulate data from that pedigree
dat &lt;- sim_from_pedigree(tab)
# simulate founder snp alleles
fg &lt;- matrix(sample(1:2, 8*length(map), repl=TRUE), nrow=8)
# for DO, need female &amp; male founders (to deal with X chr)
fg &lt;- rbind(fg, fg)
# convert dat to SNP genotypes
geno &lt;- convert2geno(dat, map, fg)
# if fg not provided, result is a 3d array
genoarray &lt;- convert2geno(dat, map)
</code></pre>


</div>