<div class="container">

<table style="width: 100%;"><tr>
<td>sim_from_pedigree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate genotypes for pedigree</h2>

<h3>Description</h3>

<p>Simulate genotypes along one chromosome for a pedigree
</p>


<h3>Usage</h3>

<pre><code class="language-R">sim_from_pedigree(
  pedigree,
  L = 100,
  xchr = FALSE,
  m = 10,
  p = 0,
  obligate_chiasma = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pedigree</code></td>
<td>
<p>Matrix or data frame describing a pedigree, with first four
columns being individual ID, mom ID, dad ID, and sex (female as
<code>0</code>, male as <code>1</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>L</code></td>
<td>
<p>Length of chromosome in cM (or a vector of chromosome lengths)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xchr</code></td>
<td>
<p>If TRUE, simulate X chromosome. (If <code>L</code> is a
vector, this should be a vector of TRUE/FALSE values, of the same
length as <code>L</code>, or a character string with the name of the X
chromosome, in <code>L</code>.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>Crossover interference parameter, for chi-square model
(m=0 corresponds to no interference).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>proportion of crossovers coming from no-interference process</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obligate_chiasma</code></td>
<td>
<p>If TRUE, require an obligate chiasma on the
4-strand bundle at meiosis.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with each component being the data for one
individual, as produced by the <code>cross()</code> function.  Those
results are a list with two components, corresponding to the
maternal and paternal chromosomes. The chromosomes are represented
as lists with two components: an integer vector of alleles in
chromosome intervals, and a numeric vector of locations of the
right-endpoints of those intervals; these two vectors should have
the same length.
</p>
<p>If the input <code>L</code> is a vector, in order to simulate multiple
chromosomes at once, then the output will be a list with length
<code>length(L)</code>, each component being a chromosome and having the
form described above.
</p>


<h3>See Also</h3>

<p><code>check_pedigree()</code>,
<code>sim_ril_pedigree()</code>, <code>sim_ail_pedigree()</code>,
<code>sim_from_pedigree_allchr()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># simulate AIL pedigree
tab &lt;- sim_ail_pedigree(12, 30)
# simulate data from that pedigree
dat &lt;- sim_from_pedigree(tab)
# simulate multiple chromosomes
dat &lt;- sim_from_pedigree(tab, c("1"=100, "2"=75, "X"=100), xchr="X")
</code></pre>


</div>