<div class="container">

<table style="width: 100%;"><tr>
<td>convert2geno_allchr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert continuous allele information into marker genotypes for
multiple chromosomes</h2>

<h3>Description</h3>

<p>Wrap up of convert2geno to adequate multiple chromosomes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">convert2geno_allchr(
  xodat,
  map,
  id = NULL,
  founder_geno = NULL,
  return.matrix = TRUE,
  shift_map = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xodat</code></td>
<td>
<p>The sort of detailed genotype/crossover data generated
by <code>sim_from_pedigree_allchr()</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>marker locations, a list with elements for each
chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>ids for which individuals genotypes is desired</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>founder_geno</code></td>
<td>
<p>Optional list of matrices (one per chromosome)
of size <code>n_founders</code> x <code>n_markers</code>, with the founder genotypes.
If coded as 1/2 (or 1/3), results are 1/2/3
genotypes. If coded as A/T/G/C/N, results are A/T/G/C/N/H
genotypes. If coded as letters A-H (in the case of 8 founders), results are
two-letter genotypes AA-HH with 36 possible values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.matrix</code></td>
<td>
<p>If FALSE, the result is a list of length
<code>n_chrs</code>, otherwise it is converted into a matrix if size
<code>length(id)</code> x <code>n_markers</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shift_map</code></td>
<td>
<p>If TRUE, shift genetic map to start at 0</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>founder_geno</code> is provided or there are just two
founders, the result is a numeric matrix of genotypes, individuals
x markers, with genotypes 1/2/3 codes for 11/12/22 genotypes. If
there are more than two founders and <code>founder_geno</code> are
letters, the result is a character matrix, too.
</p>
<p>If <code>founder_geno</code> is not provided and there are more than two
founders, the result is a 3-dimensional array, individuals x
markers x alleles, with the third dimensional corresponding to the
maternal and paternal allele.
</p>


<h3>See Also</h3>

<p><code>convert2geno()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(qtl)
# marker map
map &lt;- sim.map(len=rep(100, 19), n.mar=10, include.x=FALSE)
# simulate AIL pedigree
tab &lt;- sim_ail_pedigree(12, 30)
# simulate data from that pedigree
dat &lt;- sim_from_pedigree_allchr(tab, map)
names(map) &lt;- paste0("marker", seq(along=map))
# convert data to marker genotypes
id &lt;- which(tab[, "gen"]==12)
geno &lt;- convert2geno_allchr(dat, map, id)

</code></pre>


</div>