<div class="container">

<table style="width: 100%;"><tr>
<td>simSBS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulate single base substitutions
</h2>

<h3>Description</h3>

<p>Given the number of genomes to be created, generate single base substitutions in those genomes from simulated mutational processes, by referring to a human reference genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simSBS(nSigs = NULL, nGenomes = NULL, refGenome = NULL,
        similarity = 0.6, noise = 0,
        presetSigs = NULL, chrs = NULL, nMutPerGenome = NULL,
        sigPrevalence = NULL, chrDistribution = NULL,
        parallel = TRUE, saveDir = './')
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nSigs</code></td>
<td>

<p><strong>Required</strong>. The number of mutational processes to be created.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nGenomes</code></td>
<td>

<p><strong>Required</strong>. The number of genomes in which to simulate single base substitutions.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refGenome</code></td>
<td>

<p><strong>Required</strong>. A <code>BSgenome</code> object of human reference genome.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>similarity</code></td>
<td>

<p>Optional. Limit the similarity between any two mutational proccesses. 0 indicates no similarity while 1 indicates the opposite. Lower similarity may require more time to simulate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noise</code></td>
<td>

<p>Optional. The value should between 0 and 1, indicating the amount of random mutations (noise) added to each simulated genome. 0 indicates no noise while 1 indicates the amount of noise is equal to the amount of mutation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>presetSigs</code></td>
<td>

<p>Optional. Use user defined mutational processes to simulate mutations in the genome. It should be a 96-by-n matrix, where 96 denotes the number of mutation motifs while n denotes the number of mutational processes. If presetSigs is given, nSigs = n.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrs</code></td>
<td>

<p>Optional. On wich chromosome(s) mutations simulated from. Default is <code>c(1:22, 'X', 'Y')</code>. This argument accepts a vector that indicates chromosomes, which should be a vector created by manual input or, for example, using R code like <code>c(1:22, 'X', 'Y', 'M')</code>, where 'X', 'Y', 'M' are case sensitive (upper case) and indicate chromosome X, Y and mitochondrial chromosome. Incompatible input could cause fatal errors cause of unidentifiable chromosome name.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nMutPerGenome</code></td>
<td>

<p>Optional. <em>NULL</em> or a numerical vector whose length equals nGenomes. Number of mutations on each genome to simulate. If not defined, Default will use the <a href="https://doi.org/10.1101/112367">distribution</a> of number of single base substitutions in all WGS projects of <a href="https://dcc.icgc.org">ICGC</a> release 23.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigPrevalence</code></td>
<td>

<p>Optional. Acceptable values are either <em>NULL</em> or <em>a numerical vector</em>. The prevalence of mutational processes in wild. The default uses known prevalances of 21 processes from <a href="http://dx.doi.org/10.1038/nature12477">Alexandrov et al.</a>'s work.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrDistribution</code></td>
<td>

<p>Optional. <em>NULL</em> or <em>a numerical vector</em> are acceptable. The percentage of mutations assigned to each chromosome in a genome. The default uses the distribution of length of chromosomes (chr1 to chr22 and chrX and chrY). If a numerical vector was given, its length should equal the length of <em>chrs</em> and values should sum up to 1.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>

<p>Optional. <code>TRUE</code> or <code>FALSE</code>. Whether enable or disable parallel computing ability.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saveDir</code></td>
<td>

<p>Optional. The directory where to save simulation output. Default is the current working directory. Other paths should also be relative to the current working directory.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If succeed, the return value is 1. Simulation results are saved in saveDir.
</p>


<h3>Examples</h3>

<pre><code class="language-R">if(require(BSgenome.Hsapiens.UCSC.hg38)){
  simSBS(nSigs = 2, nGenomes = 2,
    refGenome = BSgenome.Hsapiens.UCSC.hg38::BSgenome.Hsapiens.UCSC.hg38,
    nMutPerGenome = sample(10:50, 2),
    parallel = FALSE)
}else{
  message('Cannot proceed withoud a valid reference genome.')
}
</code></pre>


</div>