<div class="container">

<table style="width: 100%;"><tr>
<td>editCsdLocus</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Edit the csd locus</h2>

<h3>Description</h3>

<p>Edits the csd locus in an entire population of individuals to
ensure heterozygosity. The user can provide a list of csd alleles for each
individual or, alternatively, the function samples a heterozygous genotype
for each individual from all possible csd alleles. The gv slot is
recalculated to reflect the any changes due to editing, but other slots
remain the same.
</p>


<h3>Usage</h3>

<pre><code class="language-R">editCsdLocus(pop, alleles = NULL, simParamBee = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p><code>Pop-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p><code>NULL</code> or list;
If <code>NULL</code>, then the function samples a heterozygous csd genotype for
each virgin queen from all possible csd alleles.
If not <code>NULL</code>, the user provides a list of length <code>nInd</code> with each
node holding a matrix or a data.frame, each having two rows and n columns.
Each row must hold one csd haplotype (allele) that will be assigned to a
virgin queen. The n columns span the length of the csd locus as specified
in <code>SimParamBee</code>. The two csd alleles must be different to
ensure heterozygosity at the csd locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParamBee</code></td>
<td>
<p>global simulation parameters.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns an object of <code>Pop-class</code>
</p>


</div>