<div class="container">

<table style="width: 100%;"><tr>
<td>getIbdHaplo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Access IBD haplotypes of individuals in a caste</h2>

<h3>Description</h3>

<p>Level 0 function that returns IBD (identity by descent)
haplotypes of individuals in a caste.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getIbdHaplo(
  x,
  caste = NULL,
  nInd = NULL,
  chr = NULL,
  snpChip = NULL,
  dronesHaploid = TRUE,
  collapse = FALSE,
  simParamBee = NULL
)

getQueenIbdHaplo(
  x,
  chr = NULL,
  snpChip = NULL,
  collapse = FALSE,
  simParamBee = NULL
)

getFathersIbdHaplo(
  x,
  nInd = NULL,
  chr = NULL,
  snpChip = NULL,
  dronesHaploid = TRUE,
  collapse = FALSE,
  simParamBee = NULL
)

getVirginQueensIbdHaplo(
  x,
  nInd = NULL,
  chr = NULL,
  snpChip = NULL,
  collapse = FALSE,
  simParamBee = NULL
)

getWorkersIbdHaplo(
  x,
  nInd = NULL,
  chr = NULL,
  snpChip = NULL,
  collapse = FALSE,
  simParamBee = NULL
)

getDronesIbdHaplo(
  x,
  nInd = NULL,
  chr = NULL,
  snpChip = NULL,
  dronesHaploid = TRUE,
  collapse = FALSE,
  simParamBee = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p><code>Pop-class</code>, <code>Colony-class</code>, or
<code>MultiColony-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>caste</code></td>
<td>
<p>NULL or character, NULL when <code>x</code> is a <code>Pop-class</code>,
and character when <code>x</code> is a <code>Colony-class</code> or
<code>MultiColony-class</code> with the possible values of "queen", "fathers",
"workers", "drones", "virginQueens", or "all"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nInd</code></td>
<td>
<p>numeric, number of individuals to access, if <code>NULL</code> all
individuals are accessed, otherwise a random sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>numeric, chromosomes to retrieve, if <code>NULL</code>, all chromosome
are retrieved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpChip</code></td>
<td>
<p>integer, indicating which SNP array loci are to be retrieved,
if <code>NULL</code>, all sites are retrieved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dronesHaploid</code></td>
<td>
<p>logical, return haploid result for drones?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collapse</code></td>
<td>
<p>logical, if the return value should be a single matrix
with haplotypes of all the individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParamBee</code></td>
<td>
<p><code>SimParamBee</code>, global simulation parameters</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>matrix with haplotypes when <code>x</code> is <code>Colony-class</code>
and list of matrices with haplotypes when <code>x</code> is
<code>MultiColony-class</code>, named by colony id when <code>x</code> is
<code>MultiColony-class</code>
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>getQueenIbdHaplo()</code>: Access IBD haplotype data of the queen
</p>
</li>
<li> <p><code>getFathersIbdHaplo()</code>: Access IBD haplotype data of fathers
</p>
</li>
<li> <p><code>getVirginQueensIbdHaplo()</code>: Access IBD haplotype data of virgin queens
</p>
</li>
<li> <p><code>getWorkersIbdHaplo()</code>: Access IBD haplotype data of workers
</p>
</li>
<li> <p><code>getDronesIbdHaplo()</code>: Access IBD haplotype data of drones
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>getIbdHaplo</code> and <code>pullIbdHaplo</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">founderGenomes &lt;- quickHaplo(nInd = 4, nChr = 1, segSites = 50)
SP &lt;- SimParamBee$new(founderGenomes)

SP$setTrackRec(TRUE)
SP$setTrackPed(isTrackPed = TRUE)
basePop &lt;- createVirginQueens(founderGenomes)

drones &lt;- createDrones(x = basePop[1], nInd = 200)
droneGroups &lt;- pullDroneGroupsFromDCA(drones, n = 10, nDrones = nFathersPoisson)

# Create a Colony and a MultiColony class
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = droneGroups[[1]])
colony &lt;- buildUp(x = colony, nWorkers = 6, nDrones = 3)
colony &lt;- addVirginQueens(x = colony, nInd = 5)

apiary &lt;- createMultiColony(basePop[3:4], n = 2)
apiary &lt;- cross(apiary, drones = droneGroups[c(2, 3)])
apiary &lt;- buildUp(x = apiary, nWorkers = 6, nDrones = 3)
apiary &lt;- addVirginQueens(x = apiary, nInd = 5)

# Input is a population
getIbdHaplo(x = getQueen(colony))
queens &lt;- getQueen(apiary, collapse = TRUE)
getIbdHaplo(queens)

# Input is a colony
getIbdHaplo(x = colony, caste = "queen")
getQueenIbdHaplo(colony)

getIbdHaplo(colony, caste = "workers", nInd = 3)
getWorkersIbdHaplo(colony)
# Same aliases exist for all castes!

# Get haplotypes for all individuals
getIbdHaplo(colony, caste = "all")
# Get all haplotypes in a single matrix
getIbdHaplo(colony, caste = "all", collapse = TRUE)

# Input is a MultiColony
getIbdHaplo(x = apiary, caste = "queen")
getQueenIbdHaplo(apiary)
# Or collapse all the haplotypes into a single matrix
getQueenIbdHaplo(apiary, collapse = TRUE)

# Get the haplotypes of all individuals either by colony or in a single matrix
getIbdHaplo(apiary, caste = "all")
getIbdHaplo(apiary, caste = "all", collapse = TRUE)
</code></pre>


</div>