<div class="container">

<table style="width: 100%;"><tr>
<td>SimParamBee</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Honeybee simulation parameters</h2>

<h3>Description</h3>

<p>Container for global honeybee simulation parameters. Saving this
object as <code>SP</code> will allow it to be accessed by SIMplyBee functions
without repeatedly (and annoyingly!) typing out
<code>someFun(argument, simParamBee = SP)</code>. <code>SimParamBee</code> inherits
from AlphaSimR <code>SimParam</code>, so all <code>SimParam</code> slots
and functions are available in addition to <code>SimParamBee</code>-specific
slots and functions. Some <code>SimParam</code> functions could have
upgraded behaviour as documented in line with honeybee biology.
</p>


<h3>Details</h3>

<p>This documentation shows details specific to <code>SimParamBee</code>. We
suggest you also read all the options provided by the AlphaSimR
<code>SimParam</code>. Below we show minimal usage cases for each
<code>SimParamBee</code> function.
</p>
<p>See also <code>vignette(package = "SIMplyBee")</code> for descriptions of how
SIMplyBee implements the specific honeybee biology.
</p>


<h3>Super class</h3>

<p><code>AlphaSimR::SimParam</code> -&gt; <code>SimParamBee</code>
</p>


<h3>Public fields</h3>

<div class="r6-fields">

<dl>
<dt><code>nWorkers</code></dt>
<dd>
<p>numeric or function, a number of workers generated in a
colony - used in <code>createWorkers</code>, <code>addWorkers</code>,
<code>buildUp</code>.
</p>
<p>The default value is 100, that is, queen generates 100 workers - this
is for a down-scaled simulation (for efficiency) assuming that this
represents ~60,000 workers in a full/strong colony (Seeley, 2019). This
value is set in <code>SimParamBee$new()</code> to have a number to work with.
</p>
<p>You can change this setting to your needs!
</p>
<p>When <code>nWorkers</code> is a function, it should work with internals of
other functions. Therefore, the function MUST be defined like
<code>function(colony, arg = default) someCode </code>, that is, the first
argument MUST be <code>colony</code> and any following arguments MUST have a
default value.
For flexibility you can add ... argument to pass on any other argument.
See <code>nWorkersPoisson</code>, <code>nWorkersTruncPoisson</code>,
or <code>nWorkersColonyPhenotype</code> for examples.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
<dt><code>nDrones</code></dt>
<dd>
<p>numeric or function, a number of drones generated in a
colony - used in <code>createDrones</code>, <code>addDrones</code>,
<code>buildUp</code>.
</p>
<p>The default value is 100, that is, queen generates 100 drones - this is
for a down-scaled simulation (for efficiency) assuming that this
represents ~1,000 drones in a full/strong colony (Seeley, 2019). This
value is set in <code>SimParamBee$new()</code> to have a number to work with.
</p>
<p>You can change this setting to your needs!
</p>
<p>When <code>nDrones</code> is a function, it should work with internals of
other functions. Therefore, the function MUST be defined like
<code>function(x, arg = default) someCode </code>, that is, the first
argument MUST be <code>x</code> and any following arguments MUST have a
default value.
For flexibility you can add ... argument to pass on any other argument.
See <code>nDronesPoisson</code>, <code>nDronesTruncPoisson</code>, or
<code>nDronesColonyPhenotype</code> for examples.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
<dt><code>nVirginQueens</code></dt>
<dd>
<p>numeric or function, a number of virgin queens
generated when a queen dies or other situations - used in
<code>createVirginQueens</code> and <code>addVirginQueens</code>.
</p>
<p>The default value is 10, that is, when the queen dies, workers generate
10 new virgin queens (Seeley, 2019). This value is set in
<code>SimParamBee$new()</code> to have a number to work with.
</p>
<p>You can change this setting to your needs!
</p>
<p>When <code>nVirginQueens</code> is a function, it should work with internals
of other functions. Therefore, the function MUST be defined like
<code>function(colony, arg = default) someCode </code>, that is, the first
argument MUST be <code>colony</code> and any following arguments MUST have a
default value.
For flexibility you can add ... argument to pass on any other argument.
See <code>nVirginQueensPoisson</code>,
<code>nVirginQueensTruncPoisson</code>, or
<code>nVirginQueensColonyPhenotype</code> for examples.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
<dt><code>nFathers</code></dt>
<dd>
<p>numeric or function, a number of drones a queen mates
with  - used in <code>pullDroneGroupsFromDCA</code>,
<code>cross</code>.
</p>
<p>The default value is 15, that is, a virgin queen mates on average with
15 drones (Seeley, 2019). This value is set in <code>SimParamBee$new()</code>
to have a number to work with.
</p>
<p>You can change this setting to your needs!
</p>
<p>When <code>nFathers</code> is a function, it should work with internals of
other functions. Therefore, the function MUST be defined like
<code>function(arg = default) someCode </code>, that is, any arguments MUST
have a default value. We did not use the <code>colony</code> argument here,
because <code>nFathers</code> likely does not depend on the colony. Let us
know if we are wrong!
For flexibility you can add ... argument to pass on any other argument.
See <code>nFathersPoisson</code> or
<code>nFathersTruncPoisson</code> for examples.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
<dt><code>swarmP</code></dt>
<dd>
<p>numeric or a function, the swarm proportion - the proportion
of workers that leave with the old queen when the colony swarms - used
in <code>swarm</code>.
</p>
<p>The default value is 0.50, that is, about a half of workers leave colony
in a swarm (Seeley, 2019). This value is set in <code>SimParamBee$new()</code>
to have a proportion to work with.
</p>
<p>You can change this setting to your needs!
</p>
<p>When <code>swarmP</code> is a function, it should work with internals of
other functions. Therefore, the function MUST be defined like
<code>function(colony, arg = default) someCode </code>, that is, the first
argument MUST be <code>colony</code> and any following arguments MUST have a
default value.
For flexibility you can add ... argument to pass on any other argument.
See <code>swarmPUnif</code> for
examples.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
<dt><code>swarmRadius</code></dt>
<dd>
<p>numeric, radius within which to sample a location of
of the swarm - used in <code>swarm</code> - see its <code>radius</code>
argument.
</p>
<p>The default value is <code>0</code>, that is, swarm gets the same location as
the original colony.
</p>
<p>You can change this setting to your needs!</p>
</dd>
<dt><code>splitP</code></dt>
<dd>
<p>numeric or a function, the split proportion - the
proportion of workers removed in a managed split - used in
<code>split</code>.
</p>
<p>The default value is 0.30, that is, about a third of workers is put into
a split colony from a strong colony (Seeley, 2019). This value is set
in <code>SimParamBee$new()</code> to have a proportion to work with.
</p>
<p>You can change this setting to your needs!
</p>
<p>When <code>splitP</code> is a function, it should work with internals of
other functions. Therefore, the function MUST be defined like
<code>function(colony, arg = default) someCode </code>, that is, the first
argument MUST be <code>colony</code> and any following arguments MUST have a
default value.
For flexibility you can add ... argument to pass on any other argument.
See <code>splitPUnif</code> or <code>splitPColonyStrength</code> for
examples.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
<dt><code>downsizeP</code></dt>
<dd>
<p>numeric or a function, the downsize proportion - the
proportion of workers removed from the colony when downsizing, usually
in autumn - used in <code>downsize</code>.
</p>
<p>The default value is 0.85, that is, a majority of workers die before
autumn or all die but some winter workers are created (Seeley, 2019).
This value is set in <code>SimParamBee$new()</code> to have a proportion to
work with.
</p>
<p>You can change this setting to your needs!
</p>
<p>When <code>downsizeP</code> is a function, it should work with internals of
other functions. Therefore, the function MUST be defined like
<code>function(colony, arg = default) someCode </code>, that is, the first
argument MUST be <code>colony</code> and any following arguments MUST have a
default value.
For flexibility you can add ... argument to pass on any other argument.
See <code>downsizePUnif</code> for example.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
<dt><code>colonyValueFUN</code></dt>
<dd>
<p>function, to calculate colony values - used
in <code>calcColonyValue</code> - see also <code>calcColonyPheno</code>
and <code>calcColonyGv</code>.
</p>
<p>This function should work with internals of others functions -
therefore the function MUST be defined like <code>function(colony, arg
  = default) someCode </code>, that is, the first argument MUST be
<code>colony</code> and any following arguments MUST have a default value.
For flexibility you can add ... argument to pass on any other argument.
See <code>mapCasteToColonyValue</code> for an example.
</p>
<p>You can provide your own functions that satisfy your needs!</p>
</dd>
</dl>
</div>


<h3>Active bindings</h3>

<div class="r6-active-bindings">

<dl>
<dt><code>caste</code></dt>
<dd>
<p>character, caste information for every individual ever
created</p>
</dd>
<dt><code>lastColonyId</code></dt>
<dd>
<p>integer, ID of the last Colony object
created with <code>createColony</code></p>
</dd>
<dt><code>csdChr</code></dt>
<dd>
<p>integer, chromosome of the csd locus</p>
</dd>
<dt><code>csdPos</code></dt>
<dd>
<p>numeric, starting position of the csd locus on the
<code>csdChr</code> chromosome (relative at the moment, but could be in base
pairs in the future)</p>
</dd>
<dt><code>nCsdAlleles</code></dt>
<dd>
<p>integer, number of possible csd alleles</p>
</dd>
<dt><code>nCsdSites</code></dt>
<dd>
<p>integer, number of segregating sites representing the
csd locus</p>
</dd>
<dt><code>csdPosStart</code></dt>
<dd>
<p>integer, starting position of the csd locus</p>
</dd>
<dt><code>csdPosStop</code></dt>
<dd>
<p>integer, ending position of the csd locus</p>
</dd>
<dt><code>version</code></dt>
<dd>
<p>list, versions of AlphaSimR and SIMplyBee packages used to
generate this object</p>
</dd>
</dl>
</div>


<h3>Methods</h3>



<h4>Public methods</h4>


<ul>
<li> <p><a href="#method-SimParamBee-new"><code>SimParamBee$new()</code></a>
</p>
</li>
<li> <p><a href="#method-SimParamBee-addToCaste"><code>SimParamBee$addToCaste()</code></a>
</p>
</li>
<li> <p><a href="#method-SimParamBee-changeCaste"><code>SimParamBee$changeCaste()</code></a>
</p>
</li>
<li> <p><a href="#method-SimParamBee-updateLastColonyId"><code>SimParamBee$updateLastColonyId()</code></a>
</p>
</li>
<li> <p><a href="#method-SimParamBee-clone"><code>SimParamBee$clone()</code></a>
</p>
</li>
</ul>
<details><summary>Inherited methods</summary><ul>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addSnpChip"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addSnpChip"><code>AlphaSimR::SimParam$addSnpChip()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addSnpChipByName"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addSnpChipByName"><code>AlphaSimR::SimParam$addSnpChipByName()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addStructuredSnpChip"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addStructuredSnpChip"><code>AlphaSimR::SimParam$addStructuredSnpChip()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addToPed"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addToPed"><code>AlphaSimR::SimParam$addToPed()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addToRec"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addToRec"><code>AlphaSimR::SimParam$addToRec()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitA"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitA"><code>AlphaSimR::SimParam$addTraitA()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitAD"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitAD"><code>AlphaSimR::SimParam$addTraitAD()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitADE"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitADE"><code>AlphaSimR::SimParam$addTraitADE()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitADEG"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitADEG"><code>AlphaSimR::SimParam$addTraitADEG()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitADG"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitADG"><code>AlphaSimR::SimParam$addTraitADG()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitAE"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitAE"><code>AlphaSimR::SimParam$addTraitAE()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitAEG"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitAEG"><code>AlphaSimR::SimParam$addTraitAEG()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="addTraitAG"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-addTraitAG"><code>AlphaSimR::SimParam$addTraitAG()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="altAddTraitAD"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-altAddTraitAD"><code>AlphaSimR::SimParam$altAddTraitAD()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="ibdHaplo"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-ibdHaplo"><code>AlphaSimR::SimParam$ibdHaplo()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="importTrait"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-importTrait"><code>AlphaSimR::SimParam$importTrait()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="manAddTrait"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-manAddTrait"><code>AlphaSimR::SimParam$manAddTrait()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="removeTrait"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-removeTrait"><code>AlphaSimR::SimParam$removeTrait()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="rescaleTraits"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-rescaleTraits"><code>AlphaSimR::SimParam$rescaleTraits()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="resetPed"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-resetPed"><code>AlphaSimR::SimParam$resetPed()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="restrSegSites"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-restrSegSites"><code>AlphaSimR::SimParam$restrSegSites()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="setCorE"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-setCorE"><code>AlphaSimR::SimParam$setCorE()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="setFounderHap"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-setFounderHap"><code>AlphaSimR::SimParam$setFounderHap()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="setRecombRatio"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-setRecombRatio"><code>AlphaSimR::SimParam$setRecombRatio()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="setSexes"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-setSexes"><code>AlphaSimR::SimParam$setSexes()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="setTrackPed"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-setTrackPed"><code>AlphaSimR::SimParam$setTrackPed()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="setTrackRec"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-setTrackRec"><code>AlphaSimR::SimParam$setTrackRec()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="setVarE"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-setVarE"><code>AlphaSimR::SimParam$setVarE()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="switchFemaleMap"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-switchFemaleMap"><code>AlphaSimR::SimParam$switchFemaleMap()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="switchGenMap"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-switchGenMap"><code>AlphaSimR::SimParam$switchGenMap()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="switchMaleMap"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-switchMaleMap"><code>AlphaSimR::SimParam$switchMaleMap()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="switchTrait"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-switchTrait"><code>AlphaSimR::SimParam$switchTrait()</code></a></span></li>
<li><span class="pkg-link" data-pkg="AlphaSimR" data-topic="SimParam" data-id="updateLastId"><a href="../../AlphaSimR/html/SimParam.html#method-SimParam-updateLastId"><code>AlphaSimR::SimParam$updateLastId()</code></a></span></li>
</ul></details><hr>
<a id="method-SimParamBee-new"></a>



<h4>Method <code>new()</code>
</h4>

<p>Starts the process of building a new simulation by creating
a new SimParamBee object and assigning a founder population of genomes
to the this object.
</p>


<h5>Usage</h5>

<div class="r"><pre>SimParamBee$new(
  founderPop,
  nWorkers = 100,
  nDrones = 100,
  nVirginQueens = 10,
  nFathers = 15,
  swarmP = 0.5,
  swarmRadius = 0,
  splitP = 0.3,
  downsizeP = 0.85,
  csdChr = 3,
  csdPos = 0.865,
  nCsdAlleles = 128,
  colonyValueFUN = NULL
)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>founderPop</code></dt>
<dd>
<p><code>MapPop-class</code>, founder population of
genomes</p>
</dd>
<dt><code>nWorkers</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>nWorkers</code></p>
</dd>
<dt><code>nDrones</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>nDrones</code></p>
</dd>
<dt><code>nVirginQueens</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>nVirginQueens</code></p>
</dd>
<dt><code>nFathers</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>nFathers</code></p>
</dd>
<dt><code>swarmP</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>swarmP</code></p>
</dd>
<dt><code>swarmRadius</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>swarmRadius</code></p>
</dd>
<dt><code>splitP</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>splitP</code></p>
</dd>
<dt><code>downsizeP</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>downsizeP</code></p>
</dd>
<dt><code>csdChr</code></dt>
<dd>
<p>integer, chromosome that will carry the csd locus, by
default 3, but if there are less chromosomes (for a simplified
simulation), the locus is put on the last available chromosome (1 or
2); if <code>NULL</code> then csd locus is ignored in the simulation</p>
</dd>
<dt><code>csdPos</code></dt>
<dd>
<p>numeric, starting position of the csd locus on the
<code>csdChr</code> chromosome (relative at the moment, but could be in base
pairs in future)</p>
</dd>
<dt><code>nCsdAlleles</code></dt>
<dd>
<p>integer, number of possible csd alleles (this
determines how many segregating sites will be needed to represent the
csd locus from the underlying bi-allelic SNP; the minimum number of
bi-allelic SNP needed is <code>log2(nCsdAlleles)</code>); if set to <code>0</code>
then <code>csdChr=NULL</code> is triggered. By default we set <code>nCsdAlleles</code>
to 128, which is at the upper end of the reported number of csd alleles
(Lechner et al., 2014; Zareba et al., 2017; Bovo et al., 2021).</p>
</dd>
<dt><code>colonyValueFUN</code></dt>
<dd>
<p>see <code>SimParamBee</code> field <code>colonyValueFUN</code></p>
</dd>
</dl>
</div>



<h5>Examples</h5>

<div class="r example copy">
<pre>founderGenomes &lt;- quickHaplo(nInd = 10, nChr = 3, segSites = 10)
SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 2)
\dontshow{SP$nThreads = 1L}

# We need enough segregating sites
try(SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 100))
\dontshow{SP$nThreads = 1L}
founderGenomes &lt;- quickHaplo(nInd = 10, nChr = 3, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 100)
\dontshow{SP$nThreads = 1L}

# We can save the csd locus on chromosome 1 or 2, too, for quick simulations
founderGenomes &lt;- quickHaplo(nInd = 10, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 100)
\dontshow{SP$nThreads = 1L}
</pre>
</div>


<hr>
<a id="method-SimParamBee-addToCaste"></a>



<h4>Method <code>addToCaste()</code>
</h4>

<p>Store caste information (for internal use only!)
</p>


<h5>Usage</h5>

<div class="r"><pre>SimParamBee$addToCaste(id, caste)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>id</code></dt>
<dd>
<p>character, individuals whose caste will be stored</p>
</dd>
<dt><code>caste</code></dt>
<dd>
<p>character, single "Q" for queens, "W" for workers, "D" for
drones, "V" for virgin queens, and "F" for fathers</p>
</dd>
</dl>
</div>



<h5>Examples</h5>

<div class="r example copy">
<pre>founderGenomes &lt;- quickHaplo(nInd = 2, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes)
\dontshow{SP$nThreads = 1L}
SP$setTrackPed(isTrackPed = TRUE)
basePop &lt;- createVirginQueens(founderGenomes)

drones &lt;- createDrones(x = basePop[1], nInd = 10)
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = drones)
colony &lt;- addWorkers(colony, nInd = 5)
colony &lt;- addDrones(colony, nInd = 5)
colony &lt;- addVirginQueens(colony, nInd = 2)

SP$pedigree
SP$caste
</pre>
</div>


<hr>
<a id="method-SimParamBee-changeCaste"></a>



<h4>Method <code>changeCaste()</code>
</h4>

<p>Change caste information (for internal use only!)
</p>


<h5>Usage</h5>

<div class="r"><pre>SimParamBee$changeCaste(id, caste)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>id</code></dt>
<dd>
<p>character, individuals whose caste will be changed</p>
</dd>
<dt><code>caste</code></dt>
<dd>
<p>character, single "Q" for queens, "W" for workers, "D" for
drones, "V" for virgin queens, and "F" for fathers</p>
</dd>
</dl>
</div>



<h5>Examples</h5>

<div class="r example copy">
<pre>founderGenomes &lt;- quickHaplo(nInd = 2, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes)
\dontshow{SP$nThreads = 1L}
SP$setTrackPed(isTrackPed = TRUE)
basePop &lt;- createVirginQueens(founderGenomes)
SP$pedigree
SP$caste

drones &lt;- createDrones(x = basePop[1], nInd = 10)
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = drones)
SP$pedigree
SP$caste
</pre>
</div>


<hr>
<a id="method-SimParamBee-updateLastColonyId"></a>



<h4>Method <code>updateLastColonyId()</code>
</h4>

<p>A function to update the colony last
ID everytime we create a Colony-class with createColony.
For internal use only.
</p>


<h5>Usage</h5>

<div class="r"><pre>SimParamBee$updateLastColonyId()</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>lastColonyId</code></dt>
<dd>
<p>integer, last colony ID assigned</p>
</dd>
</dl>
</div>


<hr>
<a id="method-SimParamBee-clone"></a>



<h4>Method <code>clone()</code>
</h4>

<p>The objects of this class are cloneable with this method.
</p>


<h5>Usage</h5>

<div class="r"><pre>SimParamBee$clone(deep = FALSE)</pre></div>



<h5>Arguments</h5>

<div class="arguments">

<dl>
<dt><code>deep</code></dt>
<dd>
<p>Whether to make a deep clone.</p>
</dd>
</dl>
</div>




<h3>References</h3>

<p>Bovo et al. (2021) Application of Next Generation Semiconductor-Based
Sequencing for the Identification of Apis mellifera Complementary Sex
Determiner (csd) Alleles from Honey DNA. Insects, 12(10), 868.
<a href="https://doi.org/10.3390/insects12100868">doi:10.3390/insects12100868</a>
</p>
<p>Lechner et al. (2014) Nucleotide variability at its limit? Insights into
the number and evolutionary dynamics of the sex-determining specificities
of the honey bee Apis mellifera Molecular Biology and Evolution, 31,
272-287. <a href="https://doi.org/10.1093/molbev/mst207">doi:10.1093/molbev/mst207</a>
</p>
<p>Seeley (2019) The Lives of Bees: The Untold Story of the Honey
Bee in the Wild. Princeton: Princeton University Press.
<a href="https://doi.org/10.1515/9780691189383">doi:10.1515/9780691189383</a>
</p>
<p>Zareba et al. (2017) Uneven distribution of complementary sex determiner
(csd) alleles in Apis mellifera population. Scientific Reports, 7, 2317.
<a href="https://doi.org/10.1038/s41598-017-02629-9">doi:10.1038/s41598-017-02629-9</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## ------------------------------------------------
## Method `SimParamBee$new`
## ------------------------------------------------

founderGenomes &lt;- quickHaplo(nInd = 10, nChr = 3, segSites = 10)
SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 2)


# We need enough segregating sites
try(SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 100))

founderGenomes &lt;- quickHaplo(nInd = 10, nChr = 3, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 100)


# We can save the csd locus on chromosome 1 or 2, too, for quick simulations
founderGenomes &lt;- quickHaplo(nInd = 10, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 100)


## ------------------------------------------------
## Method `SimParamBee$addToCaste`
## ------------------------------------------------

founderGenomes &lt;- quickHaplo(nInd = 2, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes)

SP$setTrackPed(isTrackPed = TRUE)
basePop &lt;- createVirginQueens(founderGenomes)

drones &lt;- createDrones(x = basePop[1], nInd = 10)
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = drones)
colony &lt;- addWorkers(colony, nInd = 5)
colony &lt;- addDrones(colony, nInd = 5)
colony &lt;- addVirginQueens(colony, nInd = 2)

SP$pedigree
SP$caste

## ------------------------------------------------
## Method `SimParamBee$changeCaste`
## ------------------------------------------------

founderGenomes &lt;- quickHaplo(nInd = 2, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes)

SP$setTrackPed(isTrackPed = TRUE)
basePop &lt;- createVirginQueens(founderGenomes)
SP$pedigree
SP$caste

drones &lt;- createDrones(x = basePop[1], nInd = 10)
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = drones)
SP$pedigree
SP$caste
</code></pre>


</div>