<div class="container">

<table style="width: 100%;"><tr>
<td>getCsdAlleles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get csd alleles</h2>

<h3>Description</h3>

<p>Level 0 function that returns alleles from the csd locus. See
<code>SimParamBee</code> for more information about the csd locus.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getCsdAlleles(
  x,
  caste = NULL,
  nInd = NULL,
  allele = "all",
  dronesHaploid = TRUE,
  collapse = FALSE,
  unique = FALSE,
  simParamBee = NULL
)

getQueenCsdAlleles(
  x,
  allele = "all",
  unique = FALSE,
  collapse = FALSE,
  simParamBee = NULL
)

getFathersCsdAlleles(
  x,
  nInd = NULL,
  allele = "all",
  dronesHaploid = TRUE,
  unique = FALSE,
  collapse = FALSE,
  simParamBee = NULL
)

getVirginQueensCsdAlleles(
  x,
  nInd = NULL,
  allele = "all",
  unique = FALSE,
  collapse = FALSE,
  simParamBee = NULL
)

getWorkersCsdAlleles(
  x,
  nInd = NULL,
  allele = "all",
  unique = FALSE,
  collapse = FALSE,
  simParamBee = NULL
)

getDronesCsdAlleles(
  x,
  nInd = NULL,
  allele = "all",
  dronesHaploid = TRUE,
  unique = FALSE,
  collapse = FALSE,
  simParamBee = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p><code>Pop-class</code>, <code>Colony-class</code>, or
<code>MultiColony-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>caste</code></td>
<td>
<p>NULL or character, NULL when <code>x</code> is a <code>Pop-class</code>,
and character when <code>x</code> is a <code>Colony-class</code> or
<code>MultiColony-class</code> with the possible values of "queen", "fathers",
"workers", "drones", "virginQueens", or "all"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nInd</code></td>
<td>
<p>numeric, for how many individuals; if <code>NULL</code> all individuals
are taken; this can be useful as a test of sampling individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele</code></td>
<td>
<p>character, either "all" for both alleles or an integer for a
single allele, use a value of 1 for female allele and a value of 2 for male
allele</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dronesHaploid</code></td>
<td>
<p>logical, return haploid result for drones?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collapse</code></td>
<td>
<p>logical, if <code>TRUE</code>, the function will return a set of
csd alleles across the entire population, colony, or multicolony (not
separately for each caste when <code>x</code> is a colony or each caste of
each colony when <code>x</code> is a multicolony. This is a way to get one single
object as an output across castes or colonies. Note this has nothing to do
with the colony collapse. It's like <code>paste(..., collapse = TRUE)</code>.
Default is <code>FALSE</code>. See examples about this behaviour.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unique</code></td>
<td>
<p>logical, return only the unique set of csd alleles. This argument
interacts with <code>collapse</code>. Default is <code>FALSE</code>. See examples about
this behaviour.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParamBee</code></td>
<td>
<p><code>SimParamBee</code>, global simulation parameters</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If both collapse and unique are <code>TRUE</code>, the function returns a
unique set of csd alleles in the entire population, colony, or multicolony
</p>


<h3>Value</h3>

<p>matrix with haplotypes when <code>x</code> is <code>Pop-class</code>, list
of matrices with haplotypes when <code>x</code> is <code>Colony-class</code>
(list nodes named by caste) and list of a list of matrices with haplotypes
when <code>x</code> is <code>MultiColony-class</code>, outer list is named by
colony id when <code>x</code> is <code>MultiColony-class</code>; <code>NULL</code> when
<code>x</code> is <code>NULL</code>
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>getQueenCsdAlleles()</code>: Access csd alleles of the queen
</p>
</li>
<li> <p><code>getFathersCsdAlleles()</code>: Access csd alleles of the fathers
</p>
</li>
<li> <p><code>getVirginQueensCsdAlleles()</code>: Access csd alleles of the virgin queens
</p>
</li>
<li> <p><code>getWorkersCsdAlleles()</code>: Access csd alleles of the workers
</p>
</li>
<li> <p><code>getDronesCsdAlleles()</code>: Access csd alleles of the drones
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">founderGenomes &lt;- quickHaplo(nInd = 8, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes, nCsdAlleles = 5)

basePop &lt;- createVirginQueens(founderGenomes)

drones &lt;- createDrones(x = basePop[1], nInd = 1000)
droneGroups &lt;- pullDroneGroupsFromDCA(drones, n = 10, nDrones = nFathersPoisson)

# Create a Colony and a MultiColony class
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = droneGroups[[1]])
colony &lt;- buildUp(x = colony, nWorkers = 6, nDrones = 3)

apiary &lt;- createMultiColony(basePop[3:4], n = 2)
apiary &lt;- cross(apiary, drones = droneGroups[c(2, 3)])
apiary &lt;- buildUp(x = apiary, nWorkers = 6, nDrones = 3)

# Use getCsdAlleles on a Population
getCsdAlleles(getQueen(colony))
getCsdAlleles(getWorkers(colony))

# Use getCsdAlleles on a Colony
getCsdAlleles(colony)
getCsdAlleles(colony, caste = "queen")
getQueenCsdAlleles(colony)
getCsdAlleles(colony, caste = "workers")
getWorkersCsdAlleles(colony)
# Same aliases exist for all the castes!

getCsdAlleles(colony, unique = TRUE)
getCsdAlleles(colony, collapse = TRUE)
getCsdAlleles(colony, collapse = TRUE, unique = TRUE)

# Use getCsdAlleles on a MultiColony
getCsdAlleles(apiary)
getCsdAlleles(apiary, unique = TRUE)
getCsdAlleles(apiary, collapse = TRUE, unique = TRUE)
getCsdAlleles(apiary, nInd = 2)
</code></pre>


</div>