<div class="container">

<table style="width: 100%;"><tr>
<td>calcBeeGRMIbd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Genomic Relatedness Matrix (GRM) for honeybees from
Identical By Descent genomic data</h2>

<h3>Description</h3>

<p>Level 0 function that returns Genomic Relatedness Matrix (GRM)
for honeybees from Identical By Descent genomic data (tracked alleles
since the founders) - see references on the background theory.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calcBeeGRMIbd(x)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p><code>matrix</code> of haplotypes/genomes with allele indicators
for the founders coded as 1, 2, ... Haplotypes/genome are in rows and sites
are in columns; no missing values are allowed (this is not checked!). Row
names are essential (formated as ind_genome as returned by AlphaSimR IBD
functions) to infer the individual and their ploidy (see examples)!</p>
</td>
</tr></table>
<h3>Value</h3>

<p>a list with a matrix of gametic relatedness coefficients (genome) and
a matrix of individual relatedness coefficients (indiv)
</p>


<h3>References</h3>

<p>Grossman and Eisen (1989) Inbreeding, coancestry, and covariance between
relatives for X-chromosomal loci. The Journal of Heredity,
<a href="https://doi.org/10.1093/oxfordjournals.jhered.a110812">doi:10.1093/oxfordjournals.jhered.a110812</a>
</p>
<p>Fernando and Grossman (1989) Covariance between relatives for X-chromosomal
loci in a population in disequilibrium. Theoretical and Applied Genetics,
<a href="https://doi.org/10.1007/bf00305821">doi:10.1007/bf00305821</a>
</p>
<p>Fernando and Grossman (1990) Genetic evaluation with autosomal
and X-chromosomal inheritance. Theoretical and Applied Genetics,
<a href="https://doi.org/10.1007/bf00224018">doi:10.1007/bf00224018</a>
</p>
<p>Van Arendonk, Tier, and Kinghorn (1994) Use of multiple genetic markers in
prediction of breeding values. Genetics,
<a href="https://doi.org/10.1093/genetics/137.1.319">doi:10.1093/genetics/137.1.319</a>
</p>
<p>Hill and Weir (2011) Variation in actual relationship as a consequence of
Mendelian sampling and linkage. Genetics Research,
<a href="https://doi.org/10.1017/s0016672310000480">doi:10.1017/s0016672310000480</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">founderGenomes &lt;- quickHaplo(nInd = 3, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes)

SP$setTrackRec(TRUE)
SP$setTrackPed(isTrackPed = TRUE)

basePop &lt;- createVirginQueens(founderGenomes)
drones &lt;- createDrones(x = basePop[1], nInd = 1000)
droneGroups &lt;- pullDroneGroupsFromDCA(drones, n = 1, nDrones = nFathersPoisson)
colony &lt;- createColony(basePop[2])
colony &lt;- cross(x = colony, drones = droneGroups[[1]])
colony &lt;- buildUp(x = colony, nWorkers = 6, nDrones = 3)

haploQ &lt;- getQueenIbdHaplo(colony)
haploW &lt;- getWorkersIbdHaplo(colony)
haploD &lt;- getDronesIbdHaplo(colony)
SP$pedigree

haplo &lt;- rbind(haploQ, haploW, haploD)

GRMs &lt;- calcBeeGRMIbd(x = haplo)
# You can visualise this matrix with the image() functions from the "Matrix" package

# Inspect the diagonal of the relationship matrix between individuals
x &lt;- diag(GRMs$indiv)
hist(x)
summary(x)

 # Inspect the off-diagonal of the relationship matrix between individuals
x &lt;- GRMs$indiv[lower.tri(x = GRMs$indiv, diag = FALSE)]
hist(x)
summary(x)

ids &lt;- getCasteId(colony)
qI &lt;- ids$queen
wI &lt;- sort(ids$workers)
dI &lt;- sort(ids$drones)

qG &lt;- c(t(outer(X = qI, Y = 1:2, FUN = paste, sep = "_")))
wG &lt;- c(t(outer(X = wI, Y = 1:2, FUN = paste, sep = "_")))
dG &lt;- paste(dI, 1, sep = "_")

# Queen vs workers
GRMs$genome[wG, qG]
GRMs$indiv[wI, qI]

# Queen vs drones
GRMs$genome[dG, qG]
GRMs$indiv[dI, qI]

# Workers vs workers
GRMs$genome[wG, wG]
GRMs$indiv[wI, wI]

# Workers vs drones
GRMs$genome[dG, wG]
GRMs$indiv[dI, wI]
</code></pre>


</div>