<div class="container">

<table style="width: 100%;"><tr>
<td>getCsdGeno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get genotypes from the csd locus</h2>

<h3>Description</h3>

<p>Level 0 function that returns genotypes from the csd locus. See
<code>SimParamBee</code> for more information about the csd locus and how
we have implemented it.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getCsdGeno(
  x,
  caste = NULL,
  nInd = NULL,
  dronesHaploid = TRUE,
  collapse = FALSE,
  simParamBee = NULL
)

getQueenCsdGeno(x, collapse = FALSE, simParamBee = NULL)

getFathersCsdGeno(
  x,
  nInd = NULL,
  dronesHaploid = TRUE,
  collapse = FALSE,
  simParamBee = NULL
)

getVirginQueensCsdGeno(x, nInd = NULL, collapse = FALSE, simParamBee = NULL)

getWorkersCsdGeno(x, nInd = NULL, collapse = FALSE, simParamBee = NULL)

getDronesCsdGeno(
  x,
  nInd = NULL,
  dronesHaploid = TRUE,
  collapse = FALSE,
  simParamBee = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p><code>Pop-class</code>, <code>Colony-class</code>, or
<code>MultiColony-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>caste</code></td>
<td>
<p>NULL or character, NULL when <code>x</code> is a <code>Pop-class</code>,
and character when <code>x</code> is a <code>Colony-class</code> or
<code>MultiColony-class</code> with the possible values of "queen", "fathers",
"workers", "drones", "virginQueens", or "all"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nInd</code></td>
<td>
<p>numeric, for how many individuals; if <code>NULL</code> all individuals
are taken; this can be useful as a test of sampling individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dronesHaploid</code></td>
<td>
<p>logical, return haploid result for drones?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collapse</code></td>
<td>
<p>logical, if the return value should be a single matrix
with haplotypes of all the individuals</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParamBee</code></td>
<td>
<p><code>SimParamBee</code>, global simulation parameters</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The returned genotypes are spanning multiple bi-allelic SNP of
a non-recombining csd locus / haplotype. In most cases you will want to use
<code>getCsdAlleles</code>.
</p>


<h3>Value</h3>

<p>matrix with genotypes when <code>x</code> is <code>Pop-class</code>, list
of matrices with genotypes when <code>x</code> is <code>Colony-class</code>
(list nodes named by caste) and list of a list of matrices with genotypes
when <code>x</code> is <code>MultiColony-class</code>, outer list is named by
colony id when <code>x</code> is <code>MultiColony-class</code>; <code>NULL</code> when
<code>x</code> is <code>NULL</code>
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>getQueenCsdGeno()</code>: Access csd genotypes of the queen
</p>
</li>
<li> <p><code>getFathersCsdGeno()</code>: Access csd genotypes of the fathers
</p>
</li>
<li> <p><code>getVirginQueensCsdGeno()</code>: Access csd genotypes of the virgin queens
</p>
</li>
<li> <p><code>getWorkersCsdGeno()</code>: Access csd genotypes of the virgin queens
</p>
</li>
<li> <p><code>getDronesCsdGeno()</code>: Access csd genotypes of the virgin queens
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">founderGenomes &lt;- quickHaplo(nInd = 8, nChr = 1, segSites = 100)
SP &lt;- SimParamBee$new(founderGenomes)

basePop &lt;- createVirginQueens(founderGenomes)

drones &lt;- createDrones(x = basePop[1], nInd = 1000)
droneGroups &lt;- pullDroneGroupsFromDCA(drones, n = 10, nDrones = nFathersPoisson)

# Create a Colony and a MultiColony class
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = droneGroups[[1]])
colony &lt;- buildUp(x = colony, nWorkers = 6, nDrones = 3)
colony &lt;- addVirginQueens(x = colony, nInd = 4)

apiary &lt;- createMultiColony(basePop[3:4], n = 2)
apiary &lt;- cross(apiary, drones = droneGroups[c(2, 3)])
apiary &lt;- buildUp(x = apiary, nWorkers = 6, nDrones = 3)
apiary &lt;- addVirginQueens(x = apiary, nInd = 5)

# Use getCsdGeno on a Population
getCsdGeno(getQueen(colony))
getCsdGeno(getWorkers(colony))

# Using dronesHaploid = TRUE returns drones as haploids instead of double haploids
getCsdGeno(getDrones(colony), nInd = 3, dronesHaploid = TRUE)
# Using dronesHaploid = FALSE returns drones as double haploids
getCsdGeno(getDrones(colony), nInd = 3, dronesHaploid = FALSE)

# Use getCsdGeno on a Colony
getCsdGeno(colony)
getCsdGeno(colony, caste = "queen")
getQueenCsdGeno(colony)
getCsdGeno(colony, caste = "workers")
getWorkersCsdGeno(colony)
# Same aliases exist for all the castes!

# Use getCsdGeno on a MultiColony - same behaviour as for the Colony!
getCsdGeno(apiary)
getCsdGeno(apiary, nInd = 2)
</code></pre>


</div>