<div class="container">

<table style="width: 100%;"><tr>
<td>getPooledGeno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get a pooled genotype from true genotypes</h2>

<h3>Description</h3>

<p>Level 0 function that returns a pooled genotype from true
genotypes to mimic genotyping of a pool of colony members.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getPooledGeno(x, type = NULL, sex = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>matrix, true genotypes with individuals in rows and sites in columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>character, "mean" for average genotype or "count" for the counts
of reference and alternative alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sex</code></td>
<td>
<p>character, vector of "F" and "M" to denote the sex of individuals
in <code>x</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a numeric vector with average allele dosage when <code>type = "mean"</code>
and a two-row matrix with the counts of reference (1st row) and
alternative (2nd row) alleles
</p>


<h3>Examples</h3>

<pre><code class="language-R">founderGenomes &lt;- quickHaplo(nInd = 3, nChr = 1, segSites = 50)
SP &lt;- SimParamBee$new(founderGenomes)


basePop &lt;- createVirginQueens(founderGenomes)
drones &lt;- createDrones(x = basePop[1], nInd = 1000)
droneGroups &lt;- pullDroneGroupsFromDCA(drones, n = 10, nDrones = nFathersPoisson)
apiary &lt;- createMultiColony(basePop[2:3], n = 2)
apiary &lt;- cross(x = apiary, drones = droneGroups[c(2, 3)])
apiary &lt;- buildUp(x = apiary, nWorkers = 6, nDrones = 3)
apiary &lt;- addVirginQueens(x = apiary, nInd = 5)

genoQ &lt;- getQueenSegSiteGeno(apiary[[1]])
genoF &lt;- getFathersSegSiteGeno(apiary[[1]])
genoW &lt;- getWorkersSegSiteGeno(apiary[[1]])
genoD &lt;- getDronesSegSiteGeno(apiary[[1]])
genoV &lt;- getVirginQueensSegSiteGeno(apiary[[1]])

# Pool of drones
sexD &lt;- getCasteSex(apiary[[1]], caste = "drones")
getPooledGeno(x = genoD, type = "count", sex = sexD)[, 1:10]
(poolD &lt;- getPooledGeno(x = genoD, type = "mean", sex = sexD))[, 1:10]
# ... compare to queen's genotype
genoQ[, 1:10]
plot(
  y = poolD, x = genoQ, ylim = c(0, 2), xlim = c(0, 2),
  ylab = "Average allele dosage in drones",
  xlab = "Allele dosage in the queen"
)

# As an exercise you could repeat the above with different numbers of drones!

# Pool of workers
getPooledGeno(x = genoW, type = "count")[, 1:10]
(poolW &lt;- getPooledGeno(x = genoW, type = "mean"))[, 1:10]
# ... compare to fathers' and queen's avearage genotype
sexF &lt;- getCasteSex(apiary[[1]], caste = "fathers")
sexQ &lt;- rep(x = "F", times = nrow(genoF))
sexFQ &lt;- c(sexF, sexQ)
genoFQ &lt;- rbind(genoF, genoQ[rep(x = 1, times = nrow(genoF)), ])
(poolFQ &lt;- getPooledGeno(x = genoFQ, type = "mean", sex = sexFQ))[, 1:10]
plot(
  y = poolW, x = poolFQ, ylim = c(0, 2), xlim = c(0, 2),
  ylab = "Average allele dosage in workers",
  xlab = "Average allele dosage in the queen and fathers"
)

# As an exercise you could repeat the above with different numbers of workers!

</code></pre>


</div>