<div class="container">

<table style="width: 100%;"><tr>
<td>createCastePop</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates caste population individuals from the colony</h2>

<h3>Description</h3>

<p>Level 1 function that creates the specified number of caste
individuals from the colony with a mated queens. If csd
locus is active, it takes it into account and any csd homozygotes are
removed and counted towards homozygous brood.
</p>


<h3>Usage</h3>

<pre><code class="language-R">createCastePop(
  x,
  caste = NULL,
  nInd = NULL,
  exact = TRUE,
  year = NULL,
  editCsd = TRUE,
  csdAlleles = NULL,
  simParamBee = NULL,
  ...
)

createWorkers(x, nInd = NULL, exact = FALSE, simParamBee = NULL, ...)

createDrones(x, nInd = NULL, simParamBee = NULL, ...)

createVirginQueens(
  x,
  nInd = NULL,
  year = NULL,
  editCsd = TRUE,
  csdAlleles = NULL,
  simParamBee = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p><code>link[AlphaSimR]{MapPop-class}</code> (only if caste is "virginQueens"), or
<code>Pop</code> (only if caste is "drones") or <code>Colony-class</code>
or <code>MultiColony-class</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>caste</code></td>
<td>
<p>character, "workers", "drones", or "virginQueens"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nInd</code></td>
<td>
<p>numeric or function, number of caste individuals; if <code>NULL</code> then
<code>SimParamBee$nWorkers</code>,  <code>SimParamBee$nDrones</code>
or <code>SimParamBee$nVirginQueens</code> is used depending on the caste;
only used when <code>x</code> is <code>Colony-class</code> or
<code>MultiColony-class</code>, when <code>x</code> is <code>link[AlphaSimR]{MapPop-class}</code>
all individuals in <code>x</code> are converted into virgin queens</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exact</code></td>
<td>
<p>logical, only relevant when creating workers,
if the csd locus is active and exact is <code>TRUE</code>,
create the exactly specified number of viable workers (heterozygous on the
csd locus)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>year</code></td>
<td>
<p>numeric, year of birth for virgin queens</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>editCsd</code></td>
<td>
<p>logical (only active when <code>x</code> is <code>link[AlphaSimR]{MapPop-class}</code>),
whether the csd locus should be edited to ensure heterozygosity at the csd
locus (to get viable virgin queens); see <code>csdAlleles</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>csdAlleles</code></td>
<td>
<p><code>NULL</code> or list (only active when <code>x</code> is <code>link[AlphaSimR]{MapPop-class}</code>);
If <code>NULL</code>, then the function samples a heterozygous csd genotype for
each virgin queen from all possible csd alleles.
If not <code>NULL</code>, the user provides a list of length <code>nInd</code> with each
node holding a matrix or a data.frame, each having two rows and n columns.
Each row must hold one csd haplotype (allele) that will be assigned to a
virgin queen. The n columns span the length of the csd locus as specified
in <code>SimParamBee</code>. The two csd alleles must be different to
ensure heterozygosity at the csd locus.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simParamBee</code></td>
<td>
<p><code>SimParamBee</code>, global simulation parameters</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments passed to <code>nInd</code> when this argument is a function</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>when <code>x</code> is <code>MapPop-class</code> returns
<code>virginQueens</code> (a <code>Pop-class</code>);
when <code>x</code> is <code>Colony-class</code> returns
<code>virginQueens</code> (a <code>Pop-class</code>);
when <code>x</code> is <code>MultiColony-class</code>
return is a named list of <code>virginQueens</code> (a <code>Pop-class</code>);
named by colony ID
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>createWorkers()</code>: Create workers from a colony
</p>
</li>
<li> <p><code>createDrones()</code>: Create drones from a colony
</p>
</li>
<li> <p><code>createVirginQueens()</code>: Create virgin queens from a colony
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">founderGenomes &lt;- quickHaplo(nInd = 4, nChr = 1, segSites = 50)
SP &lt;- SimParamBee$new(founderGenomes)

SP$setTrackRec(TRUE)
SP$setTrackPed(isTrackPed = TRUE)

# Create virgin queens on a MapPop
basePop &lt;- createCastePop(founderGenomes, caste = "virginQueens")
# Or alias
createVirginQueens(founderGenomes)
# Same aliases exist for all the castes!!!

# Create drones on a Pop
drones &lt;- createDrones(x = basePop[1],  nInd = 200)
# Or create unequal number of drones from multiple virgin queens
drones &lt;- createDrones(basePop[1:2], nInd = c(100, 200))
droneGroups &lt;- pullDroneGroupsFromDCA(drones, n = 3, nDrones = nFathersPoisson)

# Create a Colony and a MultiColony class
colony &lt;- createColony(x = basePop[2])
colony &lt;- cross(colony, drones = droneGroups[[1]])
apiary &lt;- createMultiColony(basePop[3:4], n = 2)
apiary &lt;- cross(apiary, drones = droneGroups[c(2, 3)])

# Using default nInd in SP
colony@virginQueens &lt;- createVirginQueens(colony)
colony@workers &lt;- createWorkers(colony)$workers
colony@drones &lt;- createDrones(colony)
# Usually, you would use functions buildUp() or addCastePop()

# These populations hold individual information
# Example on the virgin queens (same holds for all castes!)
virginQueens &lt;- colony@virginQueens
virginQueens@id
virginQueens@sex
virginQueens@mother
virginQueens@father

# Specify own number
SP$nVirginQueens &lt;- 15
SP$nWorkers &lt;- 100
SP$nDrones &lt;- 10
createVirginQueens(colony)
createVirginQueens(apiary)
# Or creating unequal numbers
createVirginQueens(apiary, nInd = c(5, 10))
# nVirginQueens will NOT vary between function calls when a constant is used

# Specify a function that will give a number
createVirginQueens(colony, nInd = nVirginQueensPoisson)
createVirginQueens(apiary, nInd = nVirginQueensPoisson)
# No. of individuals will vary between function calls when a function is used

# Store a function or a value in the SP object
SP$nVirginQueens &lt;- nVirginQueensPoisson
createVirginQueens(colony)
createVirginQueens(colony)
createVirginQueens(apiary)
createVirginQueens(apiary)
# No. of individuals will vary between function calls when a function is used

# csd homozygosity - relevant when creating virgin queens
SP &lt;- SimParamBee$new(founderGenomes, csdChr = 1, nCsdAlleles = 8)

basePop &lt;- createVirginQueens(founderGenomes, editCsd = FALSE)
all(isCsdHeterozygous(basePop))

basePop &lt;- createVirginQueens(founderGenomes, editCsd = TRUE)
all(isCsdHeterozygous(basePop))
</code></pre>


</div>