<div class="container">

<table style="width: 100%;"><tr>
<td>simulN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Function to simulate a SNP matrix, a phenotypic trait and the effects of associated SNPs.
</h2>

<h3>Description</h3>

<p>This function simulate a SNP matrix (coded as 0, 1, 2) and traits with a selected number of QTLs and their effects that will be sampled from a Normal distribution.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulN(Nind, Nmarkers, Nqtl, Esigma, Pmean, Perror)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Nind</code></td>
<td>

<p>number of individuals to simulate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Nmarkers</code></td>
<td>

<p>number of SNP markers to generate.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Nqtl</code></td>
<td>

<p>number of QTLs controlling the trait.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Esigma</code></td>
<td>

<p>standard deviation of effects with distribution N~(0,Esigma^2).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pmean</code></td>
<td>

<p>phenotype mean.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Perror</code></td>
<td>

<p>standard deviation of error (portion of phenotype not explained by genomic information).
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Genotypic data is simulated as the round value sampled from an uniform distribution with interval (-.5,2.5). Phenotypic data are obtained as a linear function defined by:
</p>
<p style="text-align: center;"><code class="reqn">y = Pmean + \sum QTN*Meffects + Perror </code>
</p>



<h3>Value</h3>

<p>An object of class list containing the SNP matrix, the trait, the markers associated and their effects.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>geno </code></td>
<td>
<p>SNP matrix generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno </code></td>
<td>
<p>vector with the trait values simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTN </code></td>
<td>
<p>column in the SNP matrix with the SNP associated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Meffects </code></td>
<td>
<p>effects of the associated SNPs.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>The genotype is simulated in the same way of simGeno function.
The trait, QTLs and their effects are simulated in the same way of simPheno function.
</p>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>See Also</h3>

<p>simGeno, simPheno, simulU
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(123)
simulN(100, 1000, 50, .9, 12, .5)
#[1] "nsimout was generated"
str(nsimout)
#List of 4
#$ geno    : num [1:100, 1:1000] 0 2 1 2 2 0 1 2 1 1 ...
#$ pheno   : num [1:100, 1] 25.4 21.6 16 13.8 19.4 ...
#$ QTN     : int [1:50] 568 474 529 349 45 732 416 51 413 514 ...
#$ Meffects: num [1:50] 0.2696 -0.1552 1.0192 0.0209 1.2023 ...
</code></pre>


</div>