<div class="container">

<table style="width: 100%;"><tr>
<td>simPheno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Function to simulate phenotypes
</h2>

<h3>Description</h3>

<p>Simulate a phenotype from a genotype matrix with QTLs with random effects sampled from a Normal distribution.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simPheno(x, Nqtl, Esigma, Pmean, Perror)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>SNP matrix coded like 0 homozygous; 1 heterozygous; 2 homozygous
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Nqtl</code></td>
<td>

<p>number of QTLs to simulate
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Esigma</code></td>
<td>

<p>standard deviation of effects with distribution N~(0,Esigma^2)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pmean</code></td>
<td>

<p>phenotype mean
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Perror</code></td>
<td>

<p>standard deviation of error (portion of phenotype not explained by genomic information)
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class list containing the trait, the markers associated and their effects.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>pheno </code></td>
<td>
<p>vector with the trait values simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTN </code></td>
<td>
<p>column in the SNP matrix with the SNP associated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Meffects </code></td>
<td>
<p>effects of the associated SNPs.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(123)
simGeno(100, 1000)
#' #[1] "simG was generated"
simPheno(simG, 50, .8, 12, .5)
#[1] "simP was generated"
str(simP)
#List of 3
#$ pheno   : num [1:100, 1] 24 20.5 15.6 13.6 18.5 ...
#$ QTN     : int [1:50] 568 474 529 349 45 732 416 51 413 514 ...
#$ Meffects: num [1:50] 0.2396 -0.138 0.906 0.0186 1.0687 ...
</code></pre>


</div>