<div class="container">

<table style="width: 100%;"><tr>
<td>simulHS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Function to simulate half sib progeny genotypes
</h2>

<h3>Description</h3>

<p>Simulate half sib progeny from one genotyped parent assuming a random genotype for the other parental. We assume that these are diploid organisms.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulHS(x, Nprogeny)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>genotype matrix of a set of moms
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Nprogeny</code></td>
<td>

<p>number of progeny's genotypes to simulate for each mom
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function assume: a diploid organism; mendelian segregation of alleles; and independent segregation.
</p>


<h3>Value</h3>

<p>a matrix of dimensions (nrow(x)*Nprogeny) x ncol(x)
</p>


<h3>Author(s)</h3>

<p>Martin Nahuel Garcia &lt;orcid:0000-0001-5760-986X&gt;
</p>


<h3>References</h3>

<p>Wu, R., Ma, C., &amp; Casella, G. (2007). Statistical genetics of quantitative traits: linkage, maps and QTL. Springer Science &amp; Business Media.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#' #simulate 100 individuals and 1000 SNPs
set.seed(123)
simGeno(100, 1000)
#[1] "simG was generated"
#simulate the genotype of 3 sets 5 HS (one set by mom)
simulHS(simG[1:3,],5)
#[1] "simulatedHS was generated"
dim(simulatedHS)
#[1]   15 1000
</code></pre>


</div>