<div class="container">

<table style="width: 100%;"><tr>
<td>coverage_normalization</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Normalize feature coverages to estimate absolute abundance or relative coverage using MAG/contig coverage values with or without multiplying total DNA concentration of the fraction</h2>

<h3>Description</h3>

<p>Normalize feature coverages to estimate absolute abundance or relative coverage using MAG/contig coverage values with or without multiplying total DNA concentration of the fraction
</p>


<h3>Usage</h3>

<pre><code class="language-R">coverage_normalization(
  f_tibble,
  contig_coverage,
  sequencing_yield,
  fractions_df,
  approach = "relative_coverage"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>f_tibble</code></td>
<td>
<p>Can be either of
(1) a tibble with first column "Feature" that contains bin IDs, and the rest of the columns represent samples with bins' coverage values.
(2) a tibble as outputted by the program "checkm coverage" from the tool CheckM. Please check CheckM documentation - https://github.com/Ecogenomics/CheckM on the usage for "checkm coverage" program</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contig_coverage</code></td>
<td>
<p>tibble with contig ID names ("Feature" column), sample columns with <strong>same sample names as in f_tibble</strong> containing coverage values of each contig, contig length in bp ("contig_length" column), and the MAG the contig is associated ("MAG" column) <strong>with same MAGs as in Feature column of f_tibble dataset</strong>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequencing_yield</code></td>
<td>
<p>tibble containing sample ID ("sample" column) with <strong>same sample names as in f_tibble</strong> and number of reads in bp recovered in that sample ("yield" column).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fractions_df</code></td>
<td>
<p>fractions data frame
A fractions file with the following columns
</p>

<ul>
<li>
<p> Replicate: Depends on how many replicates the study has
</p>
</li>
<li>
<p> Fractions: Typically in the range of 2-24
</p>
</li>
<li>
<p> Buoyant_density: As calculated from the refractometer for each fraction and replicate
</p>
</li>
<li>
<p> Isotope: "12C", "13C", "14N", "15N" etc.
</p>
</li>
<li>
<p> DNA_concentration
</p>
</li>
<li>
<p> Sample: In the format "'isotope'<em>rep</em>#<em>fraction</em>#".
For instance, "12C_rep_1_fraction_1"
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>approach</code></td>
<td>
<p>Please choose the method for coverage normalization as "relative_coverage", "greenlon", "starr" to estimate only relative coverage without multiplying DNA concentration of fraction, or as per methods in Greenlon et al. - https://journals.asm.org/doi/full/10.1128/msystems.00417-22 or Starr et al. - https://journals.asm.org/doi/10.1128/mSphere.00085-21</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>tibble containing normalized coverage in required format with MAG name as first column and the normalized coverage values in each sample as the rest of the columns.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(f_tibble)

rel.cov = coverage_normalization(f_tibble)


</code></pre>


</div>