<div class="container">

<table style="width: 100%;"><tr>
<td>singleCell</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate single cell sequencing data</h2>

<h3>Description</h3>

<p>Simulate single cell sequencing data by selecting a cell at a specified position
</p>


<h3>Usage</h3>

<pre><code class="language-R">singleCell(tumor, pos, noise = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tumor</code></td>
<td>
<p>A list which is the output of <code>simulateTumor()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>A vector of length 3 giving the (x,y,z) coordinates of the cell to sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noise</code></td>
<td>
<p>The false negative rate.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function selects the cell at <code>pos</code> (error if no cell at specified position exists) and returns
the list of mutations present in the cell. Due to technological artifacts, the false negative rate can be quite higher
(10-20 percent). To account for this,
the <code>noise</code> parameter introduces false negatives into the data set at the specified rate.
</p>


<h3>Value</h3>

<p>A data frame with 1 row and columns corresponding to the mutations present in the cell. A 1 indicates that
the mutation is detected while a 0 indicates the mutation is not detected.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol &lt;philnicol740@gmail.com&gt;
</p>


<h3>References</h3>

<p>K. Jahn, J. Kupiers and N. Beerenwinkel. Tree inference for single-cell data. Genome Biology, volume 17, 2016. 
https://doi.org/10.1186/s13059-016-0936-x.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(1126490984)
out &lt;- simulateTumor(max_pop = 1000)
df &lt;- singleCell(tumor = out, pos = c(0,0,0), noise = 0.1)

</code></pre>


</div>