<div class="container">

<table style="width: 100%;"><tr>
<td>randomNeedles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate fine needle aspiration</h2>

<h3>Description</h3>

<p>Simulate a sampling procedure which takes a fine needle through the simulated tumor and
reports the mutation allele frequency of the sampled cells.
</p>


<h3>Usage</h3>

<pre><code class="language-R">randomNeedles(tumor, nsamples, threshold = 0.05, coverage = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>tumor</code></td>
<td>
<p>A list which is the output of <code>simulateTumor()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsamples</code></td>
<td>
<p>The number of samples to take.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>Only mutations with an allele frequency greater than the threshold will be included in the sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coverage</code></td>
<td>
<p>If nonzero then deep sequencing with specified coverage is performed.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This sampling procedure is inspired by Chkhaidze et. al. (2019) and simulates 
fine needle aspiration. A random one-dimensional cross-section 
of the tumor is chosen, and the cells within this cross section are sampled, reporting mutation allele frequency.
</p>


<h3>Author(s)</h3>

<p>Phillip B. Nicol
</p>


<h3>References</h3>

<p>K. Chkhaidze, T. Heide, B. Werner, M. Williams, W. Huang, G. Caravagna, T. Graham, and 
A. Sottoriva. Spatially con- strained tumour growth affects the 
patterns of clonal selection and neutral drift in cancer genomic data. PLOS Computational Biology, 2019.
https://doi.org/10.1371/journal.pcbi.1007243.
</p>


<h3>Examples</h3>

<pre><code class="language-R">out &lt;- simulateTumor(max_pop = 1000)
df &lt;- randomNeedles(tumor = out, nsamples = 5)

</code></pre>


</div>