<div class="container">

<table style="width: 100%;"><tr>
<td>SKAT_ChrX</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SNP-set (Sequence) Kernel Association Test for  X and Y chromosome variables</h2>

<h3>Description</h3>

<p>Test for association between a set of SNPS/genes in the X chromosome and continuous or dichotomous outcomes using the kernel machine.      
</p>


<h3>Usage</h3>

<pre><code class="language-R">
SKAT_ChrX(Z, obj, is_X.inact =TRUE
, kernel = "linear.weighted", method="davies", weights.beta=c(1,25)
, weights = NULL, impute.method = "fixed", r.corr=0, is_check_genotype=TRUE
, is_dosage = FALSE, missing_cutoff=0.15, max_maf=1, estimate_MAF=1, SetID=NULL)

SKAT_ChrY(Z, obj, kernel = "linear.weighted", method="davies", weights.beta=c(1,25)
, weights = NULL, impute.method = "fixed", r.corr=0, is_check_genotype=TRUE
, is_dosage = FALSE, missing_cutoff=0.15, max_maf=1, estimate_MAF=1, SetID=NULL)


 </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>a numeric genotype matrix with each row as a different individual and each column as a separate gene/snp. 
Each genotype should be coded as 0, 1, 2, and 9 (or NA) for AA, Aa, aa, and missing, where A is a major allele and a is a minor allele. 
Missing genotypes will be imputed by the simple Hardy-Weinberg equilibrium (HWE) based imputation. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>output object of the SKAT_Null_Model_ChrX function. For SKAT_ChrY, SKAT_Null_Model_ChrX function should be  used with Model.Y=TRUE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_X.inact</code></td>
<td>
<p>an indicator variable for the X-inactivation coding (default=TRUE). 
Male genotypes are coded as g=(0,2) when it is TRUE, and  g=(0,1) when it is false.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kernel</code></td>
<td>
<p>a type of kernel (default= "linear.weighted"). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a method to compute the p-value (default= "davies"). See SKAT page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.beta</code></td>
<td>
<p>a numeric vector of parameters of beta weights. See SKAT page for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>a numeric vector of weights for the weighted kernels. See SKAT page for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute.method</code></td>
<td>
<p>a method to impute missing genotypes (default= "fixed"). 
"fixed" imputes missing genotypes by assigning the mean genotype value (2p), and "bestguess" uses best guess genotype values. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.corr</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for the compound symmetric kernel. See  SKAT page for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_check_genotype</code></td>
<td>
<p>a logical value indicating whether to check the validity of the genotype matrix Z (default= TRUE). See SKAT page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_dosage</code></td>
<td>
<p>a logical value indicating whether the matrix Z is a dosage matrix. If it is TRUE, SKAT will ignore “is_check_genotype”. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing_cutoff</code></td>
<td>
<p>a cutoff of the missing rates of SNPs (default=0.15). Any SNPs with missing rates higher than the cutoff will be excluded from the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_maf</code></td>
<td>
<p>a cutoff of the maximum minor allele frequencies (MAF) (default=1, no cutoff). Any SNPs with MAF &gt; cutoff will be excluded from the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate_MAF</code></td>
<td>
<p>a numeric value indicating how to estimate MAFs for the weight calculation and 
the missing genotype imputation. See SKAT page for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SetID</code></td>
<td>
<p>Internal use only. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For details of parameters, please see SKAT page.  
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>p-value of SKAT. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value.resampling</code></td>
<td>
<p>p-values from resampled outcome. You can get it when you use obj from SKAT_Null_Model function with resampling. See the SKAT_Null_Model. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value.noadj</code></td>
<td>
<p>p-value of SKAT without the small sample adjustment. It only appears when small sample adjustment is applied.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value.noadj.resampling</code></td>
<td>
<p>p-values from resampled outcome without the small sample adjustment. It only appears when small sample adjustment is applied. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pval.zero.msg</code></td>
<td>
<p>(only when p.value=0) text message that shows how small the p.value is. ex. "Pvalue &lt; 1.000000e-60" when p.value is smaller than <code class="reqn">10^{-60}</code> </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Q</code></td>
<td>
<p>the test statistic of SKAT. It has NA when method="optimal.adj" or "optimal".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param</code></td>
<td>
<p>estimated parameters of each method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$Is_Converged</code></td>
<td>
<p> (only with method="davies") an indicator of the convergence. 
When 0 (not converged), "liu" method is used to compute p-value. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$n.marker</code></td>
<td>
<p>a number of SNPs in the genotype matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$n.marker.test</code></td>
<td>
<p>a number of SNPs used for the test. It can be different from param$n.marker when 
some markers are monomorphic or have higher missing rates than the missing_cutoff. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Clement Ma and Seunggeun Lee</p>


<h3>Examples</h3>

<pre><code class="language-R">


data(SKAT.example.ChrX)
Z&lt;-SKAT.example.ChrX$Z

#############################################################
#	Compute the P-value of SKAT 

# binary trait
obj.x&lt;-SKAT_Null_Model_ChrX(y ~ x1 +x2 + Gender, 
  SexVar="Gender", out_type="D", data=SKAT.example.ChrX)

# SKAT
SKAT_ChrX(Z, obj.x, kernel = "linear.weighted")

# Burden
SKAT_ChrX(Z, obj.x, kernel = "linear.weighted", r.corr=1)

# SKAT-O
SKAT_ChrX(Z, obj.x, kernel = "linear.weighted", method="SKATO")

#############################################################
# Fit the Y chromosome function
# In this example, since male has only one copy of X (and Y), we reuse X chromosome genotype matrix.

# binary trait
obj.x&lt;-SKAT_Null_Model_ChrX(y ~ x1 +x2 + Gender, 
  SexVar="Gender", out_type="D", Model.Y=TRUE, data=SKAT.example.ChrX)

SKAT_ChrY(Z, obj.x, kernel = "linear.weighted", method="SKATO")



</code></pre>


</div>