<div class="container">

<table style="width: 100%;"><tr>
<td>SKATBinary_Single</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Single variant tests for binary traits with Firth and efficient resampling methods</h2>

<h3>Description</h3>

<p>This function computes p-values of single variant test using the firth and efficient resampling methods.   
</p>


<h3>Usage</h3>

<pre><code class="language-R"> 
 
	SKATBinary_Single(Z, obj, method.bin="Hybrid"
	, impute.method = "bestguess", is_check_genotype=TRUE, is_dosage = FALSE
	, missing_cutoff=0.15, max_maf=1, estimate_MAF=1
	, N.Resampling=2*10^6, seednum=100, epsilon=10^-6)


 </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>a numeric genotype vector. Each genotype should be coded as 0, 1, 2, and 9 (or NA) for AA, Aa, aa, and missing, 
where A is a major allele and a is a minor allele. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>output object from SKAT_Null_Model. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.bin</code></td>
<td>
<p>a type of method to compute a p-value (default="Hybrid"). See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute.method</code></td>
<td>
<p>a method to impute missing genotypes (default= "bestguess"). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_check_genotype</code></td>
<td>
<p>a logical value indicating whether to check the validity of the genotype matrix Z (default= TRUE). See SKAT page for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_dosage</code></td>
<td>
<p>a logical value indicating whether the matrix Z is a dosage matrix. If it is TRUE, SKAT will ignore “is_check_genotype”. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing_cutoff</code></td>
<td>
<p>a cutoff of the missing rates of SNPs (default=0.15). Any SNPs with missing rates higher than the cutoff will be excluded from the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_maf</code></td>
<td>
<p>a cutoff of the maximum minor allele frequencies (MAF) (default=1, no cutoff). Any SNPs with MAF &gt; cutoff will be excluded from the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate_MAF</code></td>
<td>
<p>a numeric value indicating how to estimate MAFs for the weight calculation and 
the missing genotype imputation. See SKAT page for details </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N.Resampling</code></td>
<td>
<p>a number of resampling to be conducted to get p-values (default=2 *10^6).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seednum</code></td>
<td>
<p>a seed number for random number generation (default=100). If NULL, no seed number will be assigned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>epsilon</code></td>
<td>
<p>a precision level  (default=10^-6).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function implements three methods (method.bin) to compute p-values: 1) Efficient resampling (ER);
2) Firth biased adjusted likelihood ratio test (Firth);  and 3) Hybrid. 
"Hybrid" selects a method based on the total minor allele count (MAC), the number of individuals with minor 
alleles (m), and the degree of case-control imbalance. 
</p>
<p>Adaptive ER (ER.A) is not implemented yet. 
</p>
<p>If seednum is not NULL, set.seed(seednum) function is used to specify seeds to get the same p-values 
of ER based methods for different runs. Therefore, please set seednum=NULL, if you do not want to set seeds. 
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>p-value. It will be the mid p-value if ER is used to compute the p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value.standard</code></td>
<td>
<p>(ER only) standard p-value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value.resampling</code></td>
<td>
<p>p-values from resampled outcome. You can obtain it when n.Resampling in SKAT_Null_Model was &gt; 0. See the SKAT_Null_Model. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value.standard.resampling</code></td>
<td>
<p>(ER only) standard p-values from resampled outcome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>the number of individuals with minor alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAP</code></td>
<td>
<p>the minimum possible p-values. It is available when the method.bin="ER" and m is sufficiently small.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAC</code></td>
<td>
<p>the total minor allele count (MAC).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.total</code></td>
<td>
<p>(ER only) the number of resampling to be generated to get the p-value. 
It can be smaller than N.Resampling when the total number of configurations of case-controls among individuals with minor alleles are smaller than
N.Resampling.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.accurate</code></td>
<td>
<p>logical value for the accuracy of the p-value. If it is false, more resampling is needed to accurately estimate the p-value. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method.bin</code></td>
<td>
<p>a type of method to be used to compute the p-value.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Seunggeun Lee</p>


<h3>References</h3>

<p>Lee, S., Fuchsberger, C., Kim, S., Scott, L. (2015) 
An efficient resampling method for calibrating single and gene-based rare variant association analysis in case-control studies.
<em>Biostatistics</em>, in press.
</p>


<h3>Examples</h3>

<pre><code class="language-R">

data(SKATBinary.example)
Z&lt;-SKATBinary.example$Z


obj&lt;-SKAT_Null_Model(y ~ x1 + x2, out_type="D", data=SKATBinary.example)
out = SKATBinary_Single(Z[,1], obj)

# p-value
out$p.value

# MAP
out$MAP

# method used to compute p-value (method.bin)
out$method.bin


#
#	Use firth method to compute p-value
SKATBinary_Single(Z[,1], obj, method.bin="Firth")$p.value

</code></pre>


</div>