<div class="container">

<table style="width: 100%;"><tr>
<td>SKAT_CommonRare</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SKAT for the combined effect of common and rare variants</h2>

<h3>Description</h3>

<p>Sequence Kernel association test for the combined effect of common and rare variants.      
</p>


<h3>Usage</h3>

<pre><code class="language-R">
	SKAT_CommonRare(Z, obj, weights.beta.rare=c(1,25)
	, weights.beta.common=c(0.5,0.5), weights=NULL
	, method="C", r.corr.rare=0, r.corr.common=0, CommonRare_Cutoff=NULL
	, test.type="Joint", is_dosage=FALSE, missing_cutoff=0.15
	, estimate_MAF=1, SetID1=NULL)


	SKAT_CommonRare.SSD.OneSet(SSD.INFO
	, SetID, obj, ..., obj.SNPWeight=NULL)

	SKAT_CommonRare.SSD.OneSet_SetIndex(SSD.INFO
	, SetIndex, obj, ..., obj.SNPWeight=NULL )


	
 </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>a numeric genotype matrix with each row as a different individual and each column as a separate gene/snp. 
Each genotype should be coded as 0, 1, 2, and 9 (or NA) for AA, Aa, aa, and missing, where A is a major allele and a is a minor allele. 
Missing genotypes will be imputed by the simple Hardy-Weinberg equilibrium (HWE) based imputation. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>an output object of the SKAT_Null_Model function. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.beta.rare</code></td>
<td>
<p>a numeric vector of parameters of beta weights for rare variants (default=c(1,25)).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.beta.common</code></td>
<td>
<p>a numeric vector of parameters of beta weights for common variants (default=c(0.5,0.5)).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>a numeric vector of weights for both common and rare variants. When it is NULL, the beta weight with the “weights.beta.rare” and “weights.beta.common”  parameter are used. When method ="C", the coefficient to combine common and rare variants test statistics will be calculated after applying the weights to variants. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a method to combine common and rare variant effects (default="C"). "C" represents the combined sum test,
and "A" represents the adaptive sum test. "AR" represents a different type of adaptive test in which 
common variants are projected over rare variants. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.corr.rare</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for rare variants (default= 0). <code class="reqn">\rho</code> =0 and 1 indicate SKAT and Burden test, respectively</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.corr.common</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for common variants (default= 0). <code class="reqn">\rho</code> =0 and 1 indicate SKAT and Burden test, respectively</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CommonRare_Cutoff</code></td>
<td>
<p>MAF cutoff for common vs rare variants (default=NULL). It should be a numeric value between 
0 and 0.5, or NULL. When it is NULL, <code class="reqn">1/ \sqrt{2 SampleSize }</code> will be used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.type</code></td>
<td>
<p>a string to indicate test type (default="Joint"). "Joint" indicates the joint test of the 
combined effects of common and rare variants. "Rare.Only" and "Common.Only" will conduct test only with rare and common variants, 
respectively.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_dosage</code></td>
<td>
<p>see SKAT</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing_cutoff</code></td>
<td>
<p>see SKAT</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate_MAF</code></td>
<td>
<p>see SKAT</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SetID1</code></td>
<td>
<p>internal use only</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SSD.INFO</code></td>
<td>
<p>an SSD_INFO object returned from Open_SSD. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SetID</code></td>
<td>
<p>a character value of Set ID. You can find a set ID of each set from SetInfo object of SSD.INFO</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SetIndex</code></td>
<td>
<p>a numeric value of Set index. You can find a set index of each set from SetInfo object of SSD.INFO  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj.SNPWeight</code></td>
<td>
<p>output object from Read_SNP_WeightFile (default=NULL). 
If NULL, the beta weight with the “weights.beta” parameter will be used.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> furthuer arguments to be passed to “SKAT_CommonRare” </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The small sample adjustment for binary traits is not implemented for "A" and "AR". 
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>p-value. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value.resampling</code></td>
<td>
<p>p-values from resampled phenotypes. You can get it when you use obj from SKAT_Null_Model function with resampling. See the SKAT_Null_Model. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.rare</code></td>
<td>
<p>the number of rare variants used for the test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.common</code></td>
<td>
<p>the number of common variants used for the test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Cutoff</code></td>
<td>
<p>the MAF cut-off to divide common and rare variants</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Q</code></td>
<td>
<p>the test statistic of SKAT. It has NA when method="A" or "AR".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param</code></td>
<td>
<p>estimated parameters of each method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$Is_Converged</code></td>
<td>
<p>an indicator of the convergence. 1 indicates the method is converged, and 0 indicates the method is not converged. 
When 0 (not converged), "liu.mod" method is used to compute p-value. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$n.marker</code></td>
<td>
<p>a number of SNPs in the genotype matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$n.marker.test</code></td>
<td>
<p>a number of SNPs used for the test. It can be different from param$n.marker when 
some markers are monomorphic or have higher missing rates than the missing_cutoff. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test.snp.mac</code></td>
<td>
<p>a vector of minor allele count (MAC) of the snps tested. The name is SNP-ID. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Seunggeun Lee</p>


<h3>References</h3>

<p>Ionita-Laza, I.*, Lee, S.*, Makarov, V., Buxbaum, J. Lin, X. (2013). 
Sequence kernel association tests for the combined effect of rare and common variants.  
<em>American Journal of Human Genetics</em>, 92, 841-853. 
* contributed equally. 
</p>


<h3>Examples</h3>

<pre><code class="language-R">

data(SKAT.example)
Z&lt;-SKAT.example$Z


# continuous trait
obj&lt;-SKAT_Null_Model(y.c ~ X, out_type="C", data=SKAT.example)
SKAT_CommonRare(Z, obj)$p.value
SKAT_CommonRare(Z, obj, method="A")$p.value
SKAT_CommonRare(Z, obj, method="AR")$p.value


# dichotomous trait 
obj&lt;-SKAT_Null_Model(y.b ~ X, out_type="D", data=SKAT.example)

# Combined sum test in the manuscript (SKAT-C and Burden-C)
SKAT_CommonRare(Z, obj)$p.value
SKAT_CommonRare(Z, obj, r.corr.rare=1, r.corr.common=1 )$p.value


# Test only with common variant
SKAT_CommonRare(Z, obj, test.type="Common.Only")$p.value

# Test only with rare variant
SKAT_CommonRare(Z, obj, test.type="Rare.Only")$p.value


# Use CommonRare_Cutoff=0.01 instead of CommonRare_Cutoff = NULL
SKAT_CommonRare(Z, obj, CommonRare_Cutoff=0.01)$p.value

# Use custom weights; the first 10 variants have higher weights 
weights&lt;-rep(1,67); weights[1:10]&lt;-2
SKAT_CommonRare(Z, obj, weights=weights)$p.value

</code></pre>


</div>