<div class="container">

<table style="width: 100%;"><tr>
<td>read_ibd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read IBD segment file</h2>

<h3>Description</h3>

<p>Reads in the inferred IBD segments from <code>hapibd</code> (<a href="https://github.com/browning-lab/hap-ibd#output-files">documentation</a>) or IBD segment file generated by <code>ped-sim</code> (<a href="https://github.com/williamslab/ped-sim#output-ibd-segments-file">documentation</a>).
</p>
<p>If reading a <code>hapibd</code> segment file, the input data should have the following columns:
</p>

<ol>
<li>
<p> First sample identifier
</p>
</li>
<li>
<p> First sample haplotype index (1 or 2)
</p>
</li>
<li>
<p> Second sample identifier
</p>
</li>
<li>
<p> Second sample haplotype index (1 or 2)
</p>
</li>
<li>
<p> Chromosome
</p>
</li>
<li>
<p> Base coordinate of first marker in segment
</p>
</li>
<li>
<p> Base coordinate of last marker in segment
</p>
</li>
<li>
<p> cM length of IBD segment
</p>
</li>
</ol>
<p>If read a <code>pedsim</code> segment file, the input data should have the following columns:
</p>

<ol>
<li>
<p> First sample identifier
</p>
</li>
<li>
<p> Second sample identifer
</p>
</li>
<li>
<p> Chromosome
</p>
</li>
<li>
<p> Physical position start
</p>
</li>
<li>
<p> Physical position end
</p>
</li>
<li>
<p> IBD type
</p>
</li>
<li>
<p> Genetic position start
</p>
</li>
<li>
<p> Genetic position end
</p>
</li>
<li>
<p> Genetic length (end - start)
</p>
</li>
</ol>
<h3>Usage</h3>

<pre><code class="language-R">read_ibd(file, source)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>Input file path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>source</code></td>
<td>
<p>Source of the input file; must be one of <code>"hapibd"</code> or <code>"pedsim"</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>if <code>source="hapibd"</code>, a tibble is returned.
If <code>source="pedsim"</code>, a list with two tibble elements, <code>IBD1</code> and <code>IBD2</code> is returned.
Both the <code>hapibd</code> tibble, and the two <code>pedsim</code> tibbles contain six columns:
</p>

<ol>
<li>
<p> id1 (sample identifier 1)
</p>
</li>
<li>
<p> id2 (sample identifier 2)
</p>
</li>
<li>
<p> chr (chromosome)
</p>
</li>
<li>
<p> start (segment bp start coordinate)
</p>
</li>
<li>
<p> end (segment bp end coordinate)
</p>
</li>
<li>
<p> length (shared segment length in genetic units, cM)
</p>
</li>
</ol>
<h3>References</h3>

<p><a href="https://github.com/browning-lab/hap-ibd#output-files">https://github.com/browning-lab/hap-ibd#output-files</a>
</p>
<p><a href="https://github.com/williamslab/ped-sim#output-ibd-segments-file">https://github.com/williamslab/ped-sim#output-ibd-segments-file</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">hapibd_fp &lt;- system.file("extdata", "GBR.sim.ibd.gz", package="skater", mustWork=TRUE)
hapibd_seg &lt;- read_ibd(hapibd_fp, source = "hapibd")
pedsim_fp &lt;- system.file("extdata", "GBR.sim.seg.gz", package="skater", mustWork=TRUE)
pedsim_seg &lt;- read_ibd(pedsim_fp, source = "pedsim")
</code></pre>


</div>