<div class="container">

<table style="width: 100%;"><tr>
<td>Get_Genotypes_SSD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get Genotype data from SSD file</h2>

<h3>Description</h3>

<p>Read a SSD file and return a genotype matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">	Get_Genotypes_SSD(SSD_INFO, Set_Index, is_ID = TRUE)
	
	Get_Genotypes_SSD_Sparse(SSD_INFO, Set_Index)
 </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SSD_INFO</code></td>
<td>
<p>SSD_INFO object returned from Open_SSD.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Set_Index</code></td>
<td>
<p>a numeric value of Set index. The set index of each set can be found from SetInfo object of SSD.INFO. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_ID</code></td>
<td>
<p>a logical value indicating whether to read SNP ID (default=TRUE). If TRUE, it reads SNP IDs and use them as column names.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A genotype matrix with n rows and m columns, where n is the number of samples and m is the number of SNPs. Get_Genotypes_SSD_Sparse returns a sparse matrix. Get_Genotypes_SSD_Sparse always returns SNP IDs as the column names, so does not have is_ID parameter. 
</p>


<h3>Author(s)</h3>

<p>Seunggeun Lee, Larisa Miropolsky</p>


</div>