<div class="container">

<table style="width: 100%;"><tr>
<td>Power_Continuous</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Power calculation, continuous traits</h2>

<h3>Description</h3>

<p>Compute an average power of SKAT and SKAT-O for testing association between a genomic region
and continuous phenotypes with a given disease model. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">Power_Continuous(Haplotypes=NULL, SNP.Location=NULL, SubRegion.Length=-1
, Causal.Percent=5, Causal.MAF.Cutoff=0.03, alpha =c(0.01,10^(-3),10^(-6))
, N.Sample.ALL = 500 * (1:10), Weight.Param=c(1,25), N.Sim=100
, BetaType = "Log", MaxBeta=1.6, Negative.Percent=0)

 
Power_Continuous_R(Haplotypes=NULL, SNP.Location, SubRegion.Length=-1
, Causal.Percent=5, Causal.MAF.Cutoff=0.03, alpha =c(0.01,10^(-3),10^(-6))
, N.Sample.ALL = 500 * (1:10), Weight.Param=c(1,25), N.Sim=100
, BetaType = "Log", MaxBeta=1.6, Negative.Percent=0, r.corr=0)

 </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Haplotypes</code></td>
<td>
<p>a haplotype matrix with each row as a different individual and each column as a separate SNP (default= NULL). 
Each element of the matrix should be either 0 (major allel) or 1 (minor allele). 
If NULL, SKAT.haplotype dataset will be used to compute power.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNP.Location</code></td>
<td>
<p>a numeric vector of SNP locations that should be matched with the SNPs in the Haplotype matrix (default= NULL). 
It is used to obtain subregions. When Haplotype=NULL, it should be NULL. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SubRegion.Length</code></td>
<td>
<p>a value of the length of subregions (default= -1). 
Each subregion will be randomly selected, and then the average power will be calculated by taking the average 
over the estimated powers of all subregions. 
If SubRegion.Length=-1 (default), the length of the subregion will be the same as the length of the whole region, 
so there will no random selection of subregions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Causal.Percent</code></td>
<td>
<p>a value of the percentage of causal SNPs among rare SNPs (MAF &lt; Causal.MAF.Cutoff)(default= 5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Causal.MAF.Cutoff</code></td>
<td>
<p>a value of MAF cutoff for the causal SNPs. Only SNPs that have MAFs smaller than the cutoff will be considered as causal SNPs (default= 0.03).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>a vector of the significance levels (default= c(0.01,10^(-3),10^(-6))). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N.Sample.ALL</code></td>
<td>
<p>a vector of the sample sizes (default= 500 * (1:10)). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Weight.Param</code></td>
<td>
<p>a vector of parameters of beta weights (default= c(1,25)).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N.Sim</code></td>
<td>
<p> a value of number of causal SNP/SubRegion sets to be generated to compute the average power (default= 100). 
Power will be computed for each causal SNP/SubRegion set, and then the average power will be obtained by taking average over the computed powers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BetaType</code></td>
<td>
<p>a type of effect sizes (default= “Log”). “Log” indicates that effect sizes of causal variants equal to <code class="reqn">c|log10(MAF)|</code>, 
and “Fixed” indicates that effect sizes of all causal variants are the same.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MaxBeta</code></td>
<td>
<p>a numeric value of the maximum effect size (default= 1.6). When BetaType="Log", the maximum effect size is MaxBeta (when MAF=0.0001).
When BetaType="Fixed", all causal variants have the same effect size (= MaxBeta). See details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Negative.Percent</code></td>
<td>
<p>a numeric value of the percentage of coefficients of causal variants that are negative (default= 0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.corr</code></td>
<td>
<p>(Power_Continuous_R only) the <code class="reqn">\rho</code> parameter for the compound symmetric correlation kernel  (default= 0). See details.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>By default it uses the haplotype information in the SKAT.haplotypes dataset. 
So if you want to use the SKAT.haplotypes dataset, you can left Haplotypes and SNP.Location as NULL. 
</p>
<p>When BetaType=“Log”, MaxBeta is a coeffecient value (<code class="reqn">\beta</code>) of the causal SNP at MAF <code class="reqn">= 10^{-4}</code> 
and used to obtain c value of the function <code class="reqn">c|log10(MAF)|</code>. For example, if MaxBeta=1.6, <code class="reqn"> c = 1.6/4 = 0.4</code>. 
Then a variant with MAF=0.001 has <code class="reqn">\beta = 1.2</code> and a variant with MAF=0.01 has <code class="reqn">\beta = 0.8</code>.
</p>
<p>When SubRegion.Length is small such as 3kb or 5kb, 
it is possible that you can have different estimated power for each run with N.Sim = <code class="reqn">50 \sim 100</code>. 
Then, please increase the N.Sim to <code class="reqn">500 \sim 1000</code> to obtain stable results. 
</p>
<p>R.sq is computed under the no linkage disequilibrium assumption.  
</p>
<p>Power_Continuous_R computes power with new class of kernels with the compound symmetric correlation structure. 
It uses a slightly different approach, and thus 
Power_Continuous and Power_Continuous_R can produce slightly different results although r.corr=0.
</p>
<p>If you want to computer power of SKAT-O by estimating the optimal r.corr, use r.corr=2. 
The estimated optimal r.corr is 
<code class="reqn">r.corr = p_1^2 ( 2p_2-1)^2</code>,
where <code class="reqn">p_1</code> is a proportion of causal variants, and <code class="reqn">p_2</code> is a proportion of negatively associated causal variants
among the causal variants.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Power</code></td>
<td>
<p>A matrix with each row as a different 
sample size and each column as a different significance level. Each element of the matrix is the estimated power.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>R.sq</code></td>
<td>
<p>Proportion of phenotype variance explained by genetic variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.corr</code></td>
<td>
<p>r.corr value. When r.corr=2 is used, it provides the estimated r.corr value. See details.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Seunggeun Lee</p>


<h3>Examples</h3>

<pre><code class="language-R">
#
#	Calculate the average power of randomly selected 3kb regions 
#	with the following conditions.
#
#	Causal percent = 20%
#	Negative percent = 20%
#	Max effect size  = 2 at MAF = 10^-4
#
#	When you use this function, please increase N.Sim (more than 100)	
#

out.c&lt;-Power_Continuous(SubRegion.Length=3000, 
Causal.Percent= 20, N.Sim=5, MaxBeta=2,Negative.Percent=20)
out.c

#
#	Calculate the required sample sizes to achieve 80% power

Get_RequiredSampleSize(out.c, Power=0.8)

</code></pre>


</div>