<div class="container">

<table style="width: 100%;"><tr>
<td>SKAT_CommonRare_Robust</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SNP set test (both common and rare variants) for binary traits with robust region-based methods</h2>

<h3>Description</h3>

<p>This function computes p-values of robust burden test, SKAT, and SKAT-O for binary traits using SPA and ER.   
</p>


<h3>Usage</h3>

<pre><code class="language-R"> 
	SKAT_CommonRare_Robust(Z, obj, kernel = "linear.weighted"
	, method="SKAT", r.corr=NULL,weights.beta.rare=c(1,25)
	, weights.beta.common=c(0.5,0.5), weights = NULL
	, CommonRare_Cutoff=NULL, impute.method = "bestguess"
	, is_check_genotype=TRUE,is_dosage = FALSE, missing_cutoff=0.15
	, max_maf=1, estimate_MAF=1)

	SKAT_CommonRare_Robust.SSD.OneSet(SSD.INFO
	, SetID, obj, ..., obj.SNPWeight=NULL)

	SKAT_CommonRare_Robust.SSD.OneSet_SetIndex(SSD.INFO
	, SetIndex, obj, ..., obj.SNPWeight=NULL )
	

 </code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>a numeric genotype matrix with each row as a different individual and each column as a separate gene/snp. 
Each genotype should be coded as 0, 1, 2, and 9 (or NA) for AA, Aa, aa, and missing, 
where A is a major allele and a is a minor allele. Now we support both matrix and sparse matrix. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj</code></td>
<td>
<p>output object from SKAT_Null_Model. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kernel</code></td>
<td>
<p>type of kernel (default= "linear.weighted"). The possible choices are "linear" and "linear.weighted".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>type of gene based test (default= "SKAT"). The possible choices are
"SKAT", "Burden" and "SKATO", which represents robust SKAT, Burden and SKAT-O tests, respectively. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.corr</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for all variants. <code class="reqn">\rho</code> =0 and 1 indicate SKAT and Burden test, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.beta.rare</code></td>
<td>
<p>a numeric vector of parameters of beta weights for rare variants. It is only used for weighted kernels. 
If you want to use your own  weights, please specify the "weights" parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.beta.common</code></td>
<td>
<p>a numeric vector of parameters of beta weights for common variants. It is only used for weighted kernels. 
If you want to use your own  weights, please specify the "weights" parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>a numeric vector of weights for the weighted kernels. See SKAT page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CommonRare_Cutoff</code></td>
<td>
<p>MAF cutoff for common vs rare variants (default=NULL). It should be a numeric value between 
0 and 0.5, or NULL. When it is NULL, <code class="reqn">1/ \sqrt{2 SampleSize }</code> will be used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute.method</code></td>
<td>
<p>a method to impute missing genotypes (default= "bestguess"). "bestguess" imputes missing genotypes as most likely 
values (0,1,2), "random" imputes missing genotypes by generating binomial(2,p) random variables (p is the MAF), 
and "fixed" imputes missing genotypes by assigning the mean genotype value (2p).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_check_genotype</code></td>
<td>
<p>a logical value indicating whether to check the validity of the genotype matrix Z (default= TRUE). See SKAT page for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_dosage</code></td>
<td>
<p>a logical value indicating whether the matrix Z is a dosage matrix. If it is TRUE, SKAT will ignore “is_check_genotype”. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing_cutoff</code></td>
<td>
<p>a cutoff of the missing rates of SNPs (default=0.15). Any SNPs with missing rates higher than the cutoff will be excluded from the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_maf</code></td>
<td>
<p>a cutoff of the maximum minor allele frequencies (MAF) (default=1, no cutoff). Any SNPs with MAF &gt; cutoff will be excluded from the analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate_MAF</code></td>
<td>
<p>a numeric value indicating how to estimate MAFs for the weight calculation and 
the missing genotype imputation. See SKAT page for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SSD.INFO</code></td>
<td>
<p>an SSD_INFO object returned from Open_SSD. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SetID</code></td>
<td>
<p>a character value of Set ID. You can find a set ID of each set from SetInfo object of SSD.INFO. In SKATBinary_Robust function, this parameter is for the internal use only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SetIndex</code></td>
<td>
<p>a numeric value of Set index. You can find a set index of each set from SetInfo object of SSD.INFO  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj.SNPWeight</code></td>
<td>
<p>output object from Read_SNP_WeightFile (default=NULL). 
If NULL, the beta weight with the “weights.beta” parameter will be used.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to “SKATBinary_Robust” </p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>p-value. It will be the p-value based on robust methods. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value_singlevariant</code></td>
<td>
<p>p-value for each single variant in this region-based test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$n.marker</code></td>
<td>
<p>a number of SNPs in the genotype matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$n.marker.test</code></td>
<td>
<p>a number of SNPs used for the test. It can be different from param$n.marker when some markers are monomorphic or have higher missing rates than the missing_cutoff. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param$rho</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for all variants. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.common</code></td>
<td>
<p>A number of common markers used for the test. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mac.common</code></td>
<td>
<p>The total minor allele count (MAC) of common markers used for the test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.rare</code></td>
<td>
<p>A number of rare markers used for the test. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mac.rare</code></td>
<td>
<p>The total minor allele count (MAC) of rare markers used for the test.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Zhangchen Zhao</p>


<h3>References</h3>

<p>Zhao, Z., Bi, W., Zhou, W., VandeHaar, P., Fritsche, L. G., &amp; Lee, S. (2019). UK-Biobank Whole Exome Sequence Binary Phenome Analysis with Robust Region-based Rare Variant Test. <em>The American Journal of Human Genetics</em>, in press.
</p>


<h3>Examples</h3>

<pre><code class="language-R">

data(SKATBinary.example)
Z&lt;-SKATBinary.example$Z

obj&lt;-SKAT_Null_Model(y ~ x1 + x2, out_type="D", data=SKATBinary.example)

# run SKAT (default method) with Hybrid
out = SKAT_CommonRare_Robust(Z, obj)

# p-value
out$p.value

#
#	Run burden and SKAT

SKAT_CommonRare_Robust(Z, obj, method="Burden")$p.value
SKAT_CommonRare_Robust(Z, obj, method="SKAT")$p.value



</code></pre>


</div>