<div class="container">

<table style="width: 100%;"><tr>
<td>ts_eigenstrat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert genotypes to the EIGENSTRAT file format</h2>

<h3>Description</h3>

<p>EIGENSTRAT data produced by this function can be used by the admixr R package
(<a href="https://bodkan.net/admixr/">https://bodkan.net/admixr/</a>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">ts_eigenstrat(ts, prefix, chrom = "chr1", outgroup = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ts</code></td>
<td>
<p>Tree sequence object of the class <code>slendr_ts</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefix</code></td>
<td>
<p>EIGENSTRAT trio prefix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>The name of the chromosome in the EIGENSTRAT snp file
(default "chr1")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outgroup</code></td>
<td>
<p>Should a formal, artificial outgroup be added? If <code>NULL</code>
(default), no outgroup is added. A non-NULL character name will serve as
the name of the outgroup in an ind file.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In case an outgroup was not formally specified in a slendr model which
generated the tree sequence data, it is possible to artificially create an
outgroup sample with the name specified by the <code>outgroup</code> argument,
which will carry all ancestral alleles (i.e. value "2" in a geno file
for each position in a snp file).
</p>


<h3>Value</h3>

<p>Object of the class EIGENSTRAT created by the admixr package
</p>


</div>