<div class="container">

<table style="width: 100%;"><tr>
<td>smfishHmrf.hmrfem.multi</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Performs HMRF for multivariate normal distribution. Accepts R data structures as inputs. Accepts a single beta.</h2>

<h3>Description</h3>

<p>This function performs HMRF (Zhu et al. 2018-Dec-01) on multivariate normal distributions. Different from other variations, this function accepts R data structures directly as inputs, and only accepts a single value of beta.
</p>
<p>This function exists for legacy and compatibility reason. User should use <strong>smfishHmrf.hmrfem.multi.it.min</strong> function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">smfishHmrf.hmrfem.multi(
  y,
  neighbors,
  numnei,
  blocks,
  beta = 0.5,
  mu,
  sigma,
  err = 1e-07,
  maxit = 50,
  verbose,
  dampFactor = NULL,
  forceDetectDamp = FALSE,
  tolerance = 1e-60
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>gene expression matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>neighbors</code></td>
<td>
<p>adjacency matrix between cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numnei</code></td>
<td>
<p>a vector containing number of neighbors per cell</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blocks</code></td>
<td>
<p>a list of cell colors for deciding the order of cell update</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>the beta to try (smoothness parameter)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>a 2D matrix (i,j) of cluster mean (initialization)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>
<p>a 3D matrix (i,j,k) where (i,j) is the covariance of cluster k (initialization)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>err</code></td>
<td>
<p>the error that is allowed between successive iterations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxit</code></td>
<td>
<p>maximum number of iterations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>TRUE or FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dampFactor</code></td>
<td>
<p>the dampening factor</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forceDetectDamp</code></td>
<td>
<p>will auto detect a dampening factor instead of using the specified one</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tolerance</code></td>
<td>
<p>applicable when forceDetectDamp is set to TRUE</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of prob, new mu, new sigma, unnormalized prob after iterations finish
</p>


<h3>More information</h3>

<p>Arguments mu and sigma refer to the cluster centroids from running kmeans algorithm.
They serve as initialization of HMRF.
Users should refer to <strong>smfishHmrf.hmrfem.multi.it.min</strong> for more information about function parameters and the requirements.
</p>


<h3>References</h3>

<p>Zhu Q, Shah S, Dries R, Cai L, Yuan G (2018-Dec-01).
“Identification of spatially associated subpopulations by combining scRNAseq and sequential fluorescence in situ hybridization data.”
<em>Nature Biotechnology</em>, <b>36</b>, 1183–1190.
doi: <a href="https://doi.org/10.1038/nbt.4260">10.1038/nbt.4260</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">    data(seqfishplus)
    s &lt;- seqfishplus
    res&lt;-smfishHmrf.hmrfem.multi(s$y, s$nei, s$numnei, s$blocks, beta=28, 
    mu=s$mu, sigma=s$sigma, err=1e-7, maxit=50, verbose=TRUE, dampFactor=s$damp, 
    tolerance=1e-5)
</code></pre>


</div>