<div class="container">

<table style="width: 100%;"><tr>
<td>smfishHmrf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>smfishHmrf: A package for running hidden markov random field on smFISH and other spatial transcriptomic datasets</h2>

<h3>Description</h3>

<p>A package for running hidden markov random field (Zhu et al. 2018-Dec-01) on smFISH and other spatial transcriptomic datasets.
</p>


<h3>Input</h3>

<p>The inputs of HMRF are the following:
</p>

<ul>
<li>
<p> Gene expression matrix
</p>
</li>
<li>
<p> Cell neighborhood matrix
</p>
</li>
<li>
<p> Initial centroids of clusters
</p>
</li>
<li>
<p> Number of clusters
</p>
</li>
<li>
<p> beta
</p>
</li>
</ul>
<p>smfishHmrf has been tested to work on seqFISH, MERFISH, starMAP, 10X Visium and other datasets. See Giotto (Dries et al. 2020) for examples of such datasets and to learn about the technologies.
smfishHmrf is a general algorithm, and should probably work with other data types.
</p>


<h3>Running</h3>

<p>The first step is to calculate initial centroids on the gene expression matrix given k (the number of clusters). The function <strong>smfishHmrf.generate.centroid.it</strong> is used for this purpose.
</p>
<p>The next step is to run the HMRF algorithm given the expression matrix, and cell neighborhood matrix. The function <strong>smfishHmrf.hmrfem.multi.it.min</strong> is used for this purpose.
</p>


<h3>Variations</h3>

<p>You might notice several variations of the functions:
</p>

<ul>
<li> <p><code>smfishHmrf.hmrfem.multi.it.min</code>: supports multiple betas; supports file names as inputs. <strong>This is the recommended function.</strong>
</p>
</li>
<li> <p><code>smfishHmrf.hmrfem.multi.it</code>: supports multiple betas; supports R data structures as inputs.
</p>
</li>
<li> <p><code>smfishHmrf.hmrfem.multi</code>: supports a single beta; supports R data structures as inputs.
</p>
</li>
<li>
<p> Note: beta is the smoothness parameter of HMRF
</p>
</li>
</ul>
<p>Also:
</p>

<ul>
<li> <p><code>smfishHmrf.generate.centroid.it</code>: supports file names as inputs. <strong>This is the recommended function</strong>
</p>
</li>
<li> <p><code>smfishHmrf.generate.centroid</code>: supports R matrices as inputs. Assumes input files have been read into R matrices.
</p>
</li>
<li> <p><code>smfishHmrf.generate.centroid.use.exist</code>: loads existing centroids. Assumes that centroids have been generated previously and saved to disk.
</p>
</li>
</ul>
<h3>References</h3>

<p>Zhu Q, Shah S, Dries R, Cai L, Yuan G (2018-Dec-01).
“Identification of spatially associated subpopulations by combining scRNAseq and sequential fluorescence in situ hybridization data.”
<em>Nature Biotechnology</em>, <b>36</b>, 1183–1190.
doi: <a href="https://doi.org/10.1038/nbt.4260">10.1038/nbt.4260</a>.
</p>
<p>Dries R, Zhu Q, Dong R, Eng CL, Li H, Liu K, Fu Y, Zhao T, Sarkar A, Bao F, George RE, Pierson N, Cai L, Yuan G (2020).
“Giotto, a toolbox for integrative analysis and visualization of spatial expression data.”
<em>bioRxiv</em>.
doi: <a href="https://doi.org/10.1101/701680">10.1101/701680</a>.
</p>


</div>