<div class="container">

<table style="width: 100%;"><tr>
<td>seqfishplus</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SeqFISHplus dataset</h2>

<h3>Description</h3>

<p>Data from SeqFISH experiment on SS cortex. This is a dataset with 523 cells and the expression of about 500 spatial genes
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(seqfishplus)
</code></pre>


<h3>Format</h3>

<p>A list containing the following fields: y, nei, blocks, damp, mu, sigma
</p>

<dl>
<dt>y</dt>
<dd>
<p>gene expression matrix</p>
</dd>
<dt>nei</dt>
<dd>
<p>cell adjacency matrix</p>
</dd>
<dt>blocks</dt>
<dd>
<p>vertex (or cell) update order; a list of vertex colors; cells marked with the same color are updated at once</p>
</dd>
<dt>damp</dt>
<dd>
<p>dampening constants (length k, the number of clusters)</p>
</dd>
<dt>mu</dt>
<dd>
<p>initialization (means). Means is a (i,k) matrix</p>
</dd>
<dt>sigma</dt>
<dd>
<p>initialization (sigmas). Sigmas is a (i,j,k) 3D matrix. k is cluster id. (i,j) is covariance matrix</p>
</dd>
</dl>
<h3>References</h3>

<p>Eng CL, Lawson M, Zhu Q, Dries R, Koulena N, Takei Y, Yun J, Cronin C, Karp C, Yuan G, Cai L (2019-Apr-01).
“Transcriptome-scale super-resolved imaging in tissues by RNA seqFISH+.”
<em>Nature</em>, <b>568</b>, 235–239.
doi: <a href="https://doi.org/10.1038/s41586-019-1049-y">10.1038/s41586-019-1049-y</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(seqfishplus)
</code></pre>


</div>