<div class="container">

<table style="width: 100%;"><tr>
<td>SMUT</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Multi-SNP Mediation Intersection-Union Test
</h2>

<h3>Description</h3>

<p>Testing the mediation effect of multiple SNPs on an outcome through a mediator.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SMUT(G, mediator, outcome, 
outcome_type="continuous", method="score",approxi=TRUE, debug=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>

<p>n by p matrix (n rows and p columns). Each row is one individual; each column is one SNP.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mediator</code></td>
<td>

<p>a vector length of n. It is the mediator variable.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outcome</code></td>
<td>

<p>a vector length of n. It is the outcome variable.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outcome_type</code></td>
<td>

<p>Type of the outcome variable. For now, this package only deals with continuous outcome. Default is "continuous".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>The method of testing coefficient of mediator in the outcome model. The score test is used. Default is "score".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>approxi</code></td>
<td>

<p>a boolean value. This is an indicator whether the approximation of the score statistic is applied to save computing time. Default is TRUE.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>debug</code></td>
<td>

<p>a boolean value. If TRUE a lot of computing details is printed; otherwise the
function is completely silent. Default is FALSE.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p_value_IUT</code></td>
<td>

<p>The p value for testing the mediation effect (theta*beta) based on intersection-union test.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_value_theta</code></td>
<td>

<p>The p value for testing theta in the outcome model.
The outcome model is the following. <br>
outcome ~ intercept + G*gamma + mediator*theta + error.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_value_beta</code></td>
<td>

<p>The p value for testing beta in the mediator model.
The mediator model is the following. <br>
mediator ~ intercept + G*beta + error
</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Wujuan Zhong
</p>


<h3>References</h3>

<p>Zhong, W., Spracklen, C. N., Mohlke, K. L., Zheng, X., Fine, J., &amp; Li, Y. (2019). Multi-SNP mediation intersection-union test. <em>Bioinformatics</em>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(SMUT)
# load the Genotype data included in this R package
data("Genotype_data")

# generate one mediator and one outcome

# first example, the mediation effect is significant
set.seed(1)
beta=rnorm(ncol(Genotype_data),1,2)
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + e1

theta=0.8
gamma=rnorm(ncol(Genotype_data),0.5,2)
e2 = rnorm(nrow(Genotype_data), 0, 1)
outcome = 2 + eigenMapMatMult(Genotype_data,gamma) + theta*mediator + e2

p_value=SMUT(G=Genotype_data,mediator=mediator,outcome=outcome)
print(p_value)

# p_value_IUT is the p value for the mediation effect.
# we have significant(at alpha level 0.05) mediation effects (p_value_IUT = 0.001655787).

# second example, the mediation effect is non-significant
set.seed(1)
beta=rnorm(ncol(Genotype_data),1,2)
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + e1

theta=0
gamma=rnorm(ncol(Genotype_data),0.5,2)
e2 = rnorm(nrow(Genotype_data), 0, 1)
outcome = 2 + eigenMapMatMult(Genotype_data,gamma) + theta*mediator + e2

p_value=SMUT(G=Genotype_data,mediator=mediator,outcome=outcome)
print(p_value)

# p_value_IUT is the p value for the mediation effect.
# we have non-significant(at alpha level 0.05) mediation effects (p_value_IUT = 0.3281677).

# third example, the mediation effect is non-significant
set.seed(1)
beta=rep(0,ncol(Genotype_data))
e1 = rnorm(nrow(Genotype_data), 0, 1)
mediator = 1 + eigenMapMatMult(Genotype_data,beta) + e1

theta=0.8
gamma=rnorm(ncol(Genotype_data),0.5,2)
e2 = rnorm(nrow(Genotype_data), 0, 1)
outcome = 2 + eigenMapMatMult(Genotype_data,gamma) + theta*mediator + e2

p_value=SMUT(G=Genotype_data,mediator=mediator,outcome=outcome)
print(p_value)

# p_value_IUT is the p value for the mediation effect.
# we have non-significant(at alpha level 0.05) mediation effects (p_value_IUT = 0.5596977).

# Thanks for using our R package SMUT 
</code></pre>


</div>