<div class="container">

<table style="width: 100%;"><tr>
<td>SMVar-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Structural Model for Variances  </h2>

<h3>Description</h3>

<p>Package containing moderated t-tests to detect differentially expressed genes for paired and unpaired data</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> SMVar</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.3.3</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2011-08-03</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL </td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>SMVar.unpaired and SMVar.paired are the most important functions. 
</p>


<h3>Author(s)</h3>

<p>Guillemette Marot &lt;guillemette.marot@inria.fr&gt;
</p>


<h3>References</h3>

<p>F. Jaffrezic, Marot, G., Degrelle, S., Hue, I. and Foulley, J. L. (2007) A structural mixed model for variances in differential gene expression studies. Genetical Research (89) 19:25 
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(SMVar)
data(ApoAIdata)
attach(ApoAIdata)
SMVar.unpaired(ApoAIGeneId,list(ApoAICond1,ApoAICond2))
</code></pre>


</div>