<div class="container">

<table style="width: 100%;"><tr>
<td>annotate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run the annotation pipeline on a list of variants</h2>

<h3>Description</h3>

<p>This function receives a list of variants and checks their information on Ensembl website via
the Ensembl REST API server.
</p>


<h3>Usage</h3>

<pre><code class="language-R">annotate(
  rslist,
  server,
  db,
  outputPath,
  window_size = 500,
  r2 = 0.5,
  LDlist = TRUE,
  cadd = FALSE,
  geneNames.file = NULL,
  regulatoryType.file = NULL,
  cores = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>rslist</code></td>
<td>
<p>A vector of rs numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>server</code></td>
<td>
<p>Name of the server. "https://rest.ensembl.org" can be used for GRCh38
and "https://grch37.rest.ensembl.org" for GRCh37.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>The population database for calculating LD scores. This can be found using <code>listDatabases</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputPath</code></td>
<td>
<p>The report file will be saved in this path as an Excel file (.xlsx)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window_size</code></td>
<td>
<p>Number of base pairs around the variant for checking LD scores (max = 500kb)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r2</code></td>
<td>
<p>The LD threshold for selecting variants around the target SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LDlist</code></td>
<td>
<p>If set to TRUE, variants in high LD will be found and added to the output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cadd</code></td>
<td>
<p>If set to TRUE, the CADD scores will be added to variant information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geneNames.file</code></td>
<td>
<p>path the gene information file (*.rds). Default value is NULL and ENSEMBL website will be checked if no file is provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>regulatoryType.file</code></td>
<td>
<p>path the variants regulatory type information file (*.rds). Default value is NULL and this step will be skipped if no file is provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>set to a value above 0 for parallel processing.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data table with all variant information is returned.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# select the required server
server &lt;- "https://grch37.rest.ensembl.org"

# select the database for population data
# this can be selected from listDatabases() function
db &lt;- "1000GENOMES:phase_3:EUR"

# create a vector of required SNPs
rslist=c('rs236349')

output &lt;- annotate(rslist,server,db,
  outputPath = paste(tempdir(),'sampleOutput.xlsx',sep="/"),
  window_size = 500,
  r2 = .9,
  cadd = FALSE)
  
## End(Not run)

</code></pre>


</div>