<div class="container">

<table style="width: 100%;"><tr>
<td>odds</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extract odds ratios, 95% CI and pvalues</h2>

<h3>Description</h3>

<p>Extract odds ratios, 95
</p>


<h3>Usage</h3>

<pre><code class="language-R"> odds(x, model=c("log-additive", "dominant", "recessive", "overdominant", "codominant"), 
      sorted=c("no","p-value","or"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class 'WGassociation' output of WGassociation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>model to be extracted. Only first one is used. The first letter is enough, low or upper case.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sorted</code></td>
<td>
<p>Sort the output by P value or OR.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix with OR 95% CI (lower, upper) and P value for the selected model. For codominant model, the OR and 95%CI are given for heterozygous and homozigous.
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>Examples</h3>

<pre><code class="language-R"> data(SNPs)
 datSNP&lt;-setupSNP(SNPs,6:40,sep="")
 ans&lt;-WGassociation(casco~1,data=datSNP,model="all")
 odds(ans)
</code></pre>


</div>