<div class="container">

<table style="width: 100%;"><tr>
<td>make.geno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a group of locus objects from some SNPs, assign to 'model.matrix' class.
</h2>

<h3>Description</h3>

<p>This function prepares the CRITICAL element corresponding to matrix of genotypes necessary to be included in 'haplo.glm' function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">make.geno(data, SNPs.sel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>an object of class 'setupSNP' containing the the SNPs that will be used to estimate the haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNPs.sel</code></td>
<td>
<p> a vector indicating the names of SNPs that are used to estimate the haplotypes  </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>the same as 'setupGeno' function, from 'haplo.stats' library, returns
</p>


<h3>See Also</h3>

<p><code>snp</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run:
data(SNPs)
# first, we create an object of class 'setupSNP'
datSNP&lt;-setupSNP(SNPs,6:40,sep="")
geno&lt;-make.geno(datSNP,c("snp10001","snp10002","snp10003"))
## End(Not run)


</code></pre>


</div>