<div class="container">

<table style="width: 100%;"><tr>
<td>plotMissing</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Plot of missing genotypes</h2>

<h3>Description</h3>

<p>Plot a grid showing which genotypes are missing
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotMissing(x, print.labels.SNPs = TRUE, 
        main = "Genotype missing data", ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class 'setupSNP'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>print.labels.SNPs</code></td>
<td>
<p>should labels of SNPs be printed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>title to place on plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>extra arguments of 'image' function</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function uses 'image' function to plot a grid with black pixels where the genotypes
are missing.  
</p>


<h3>Value</h3>

<p>No return value, just the plot
</p>


<h3>See Also</h3>

 <p><code>setupSNP</code></p>


<h3>Examples</h3>

<pre><code class="language-R"> data(SNPs)
 data(SNPs.info.pos) 
 ans&lt;-setupSNP(SNPs,colSNPs=6:40,sep="")
 plotMissing(ans)
 
 # The same plot with the SNPs sorted by genomic position and 
 # showing the information about chromosomes

 ans&lt;-setupSNP(SNPs,colSNPs=6:40,sort=TRUE,SNPs.info.pos,sep="") 
 plotMissing(ans)
</code></pre>


</div>