<div class="container">

<table style="width: 100%;"><tr>
<td>inheritance</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Collapsing (or recoding) genotypes into different categories (generally two) depending on a given genetic mode of inheritance</h2>

<h3>Description</h3>

<p>codominant function recodifies a variable having genotypes depending on the allelic frequency
in descending order. <br></p>
<p>dominant, recessive, and overdominant functions collapse the three categories of a given SNP 
into two categories as follows: Let 'AA', 'Aa', and 'aa' be the three genotypes. After determining
the most frequent allele (let's suppose that 'A' is the major allele) the functions return a vector
with to categories as follows. dominant: 'AA' and 'Aa-aa'; recessive: 'AA-Aa' and 'aa'; 
overdominant: 'AA-aa' vs 'Aa'. <br></p>
<p>additive function creates a numerical variable, 1, 2, 3 corresponding to the three genotypes sorted
out by descending allelic frequency (this model is referred as log-additive).</p>


<h3>Usage</h3>

<pre><code class="language-R">codominant(o)
dominant(o)
recessive(o)
overdominant(o)
additive(o)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>o</code></td>
<td>
<p>categorical covariate having genotypes</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A snp object collapsing genotypes into different categories depending on a given genetic mode of inheritance</p>


<h3>Examples</h3>

<pre><code class="language-R">data(SNPs)
dominant(snp(SNPs$snp10001,sep=""))
overdominant(snp(SNPs$snp10001,sep=""))
</code></pre>


</div>