<div class="container">

<table style="width: 100%;"><tr>
<td>permTest</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Permutation test analysis</h2>

<h3>Description</h3>

<p>This function extract the p values for permutation approach performed using scanWGassociation function
</p>


<h3>Usage</h3>

<pre><code class="language-R">   permTest(x, method="minimum", K)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a required object of class 'WGassociation' with the attribute 'permTest'. See details</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>statistic used in the permutation test. The default is 'minimum' but 'rtp' (rank truncated product) is
also available.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>number of the K most significant p values from the total number of test performed (e.g number of SNPs) used
to compute the rank truncated product. This argument is only required when method='rtp'. See references </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function extract the p values from an object of class 'WGassociation'. This object migth be obtained using 
the funcion called 'scanWGassociation' indicating the number of permutations in the argument 'nperm'. 
</p>


<h3>Value</h3>

<p>An object of class 'permTest'. 
</p>
<p>'print' returns a summary indicating the number of SNPs analyzed, the number of valid SNPs (those non-Monomorphic and 
that pass the calling rate), the p value after Bonferroni correction, and the p values based on permutation approach. 
One of them is based on considering the empirical percentil for the minimum p values, and the another one on assuming 
that the minimum p values follow a beta distribution.
</p>
<p>'plot' produces a plot of the empirical distribution for the minimum p values (histogram) and the expected distribution
assuming a beta distribution. The corrected p value is also showed in the plot. 
</p>
<p>See examples for further illustration about all previous issues. 
</p>


<h3>References</h3>

<p>Dudbridge F, Gusnanto A and Koeleman BPC. Detecting multiple associations in genome-wide studies.  
Human Genomics, 2006;2:310-317.
</p>
<p>Dudbridge F and Koeleman BPC. Efficient computation of significance levels for multiple associations in large studies of 
correlated data, including genomewide association studies. Am J Hum Genet, 2004;75:424-435.
</p>
<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

 <p><code>scanWGassociation</code></p>


<h3>Examples</h3>

<pre><code class="language-R">
library(SNPassoc)

data(asthma, package = "SNPassoc")
asthma.s &lt;- setupSNP(data=asthma, colSNPs=7:ncol(asthma), sep="")

ans &lt;- WGassociation(casecontrol, data=asthma.s)

</code></pre>


</div>