<div class="container">

<table style="width: 100%;"><tr>
<td>snp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SNP object</h2>

<h3>Description</h3>

<p><code>snp</code> creates an snp object
</p>
<p><code>is</code> returns <code>TRUE</code> if <code>x</code> is of class 'snp'
</p>
<p><code>as</code> attempts to coerce its argument into an object of class 'snp'
</p>
<p><code>reorder</code> change the reference genotype
</p>
<p><code>summary</code> gives a summary for an object of class 'snp' including genotype
and allele frequencies and an exact thest of Hardy-Weinberg 
equilibrium 
</p>
<p><code>plot</code>    gives a summary for an object of class 'snp' including genotype
and allele frequencies and an exact thest of Hardy-Weinberg 
equilibrium in a plot. Barplot or pie are allowed
</p>
<p><code>[.snp</code>   is a copy of [.factor modified to preserve all attributes
</p>


<h3>Usage</h3>

<pre><code class="language-R">
  snp(x, sep = "/", name.genotypes, reorder="common", 
    remove.spaces = TRUE, allow.partial.missing = FALSE) 

  is.snp(x)
 
  as.snp(x, ...)

  ## S3 method for class 'snp'
additive(o)

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>either an object of class 'snp' or an object to be converted to class 'snp'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>character separator used to divide alleles when <code>x</code> is a vector of strings where 
each string holds both alleles. The default is "/". See below for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.genotypes</code></td>
<td>
<p>the codes for the genotypes. This argument may be useful when genotypes 
are coded using three different codes (e.g., 0,1,2 or hom1, het, hom2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reorder</code></td>
<td>
<p>how should genotypes within an individual be reordered. Possible values are
'common' or 'minor'. The default is 
<code>reorder="common"</code>. In that case, alleles are sorted within each individual by
common homozygous.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove.spaces</code></td>
<td>
<p>logical indicating whether spaces and tabs will be
removed from the genotypes before processing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allow.partial.missing</code></td>
<td>
<p>logical indicating whether one allele is
permitted to be missing.  When set to 'FALSE' both alleles
are set to 'NA' when either is missing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>o</code></td>
<td>
<p>an object of class 'snp' to be coded as a linear covariate: 0,1,2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>SNP objects hold information on which gene or marker alleles were
observed for different individuals.  For each individual, two alleles
are recorded.
</p>
<p>The snp class considers the stored alleles to be unordered , i.e., "C/T"
is equivalent to "T/C". It assumes that the order of the alleles is not important.
</p>
<p>When <code>snp</code> is called, <code>x</code> is a character vector, and it is
assumed that each element encodes both alleles. In this case, if
<code>sep</code> is a character string, <code>x</code> is assumed to be coded
as "Allele1&lt;sep&gt;Allele2".  If <code>sep</code> is a numeric value, it is
assumed that character locations <code>1:sep</code> contain allele 1 and
that remaining locations contain allele 2.
</p>
<p><code>additive.snp</code> recodes the SNPs for being analyzed as a linear covariate (codes 0,1,2) 
</p>


<h3>Value</h3>

<p>The snp class extends "factor" where the levels is a character vector of possible 
genotype values stored coded by <code>paste( allele1, "", allele2, sep="/")</code>
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

<p><code>association</code>  
</p>


<h3>Examples</h3>

<pre><code class="language-R"># some examples of snp data in different formats

dat1  &lt;- c("21", "21", "11", "22", "21",
                    "22", "22", "11", "11", NA)
ans1  &lt;- snp(dat1,sep="")
ans1

dat2 &lt;- c("A/A","A/G","G/G","A/G","G/G",
                    "A/A","A/A","G/G",NA)
ans2  &lt;- snp(dat2,sep="/")
ans2

dat3 &lt;- c("C-C","C-T","C-C","T-T","C-C",
                    "C-C","C-C","C-C","T-T",NA)
ans3 &lt;- snp(dat3,sep="-")
ans3


dat4 &lt;- c("het","het","het","hom1","hom2",
                    "het","het","hom1","hom1",NA)
ans4 &lt;- snp(dat4,name.genotypes=c("hom1","het","hom2"))
ans4


# summary 
summary(ans3)

# plots

plot(ans3)
plot(ans3,type=pie)
plot(ans3,type=pie,label="SNP 10045")

</code></pre>


</div>