<div class="container">

<table style="width: 100%;"><tr>
<td>setupSNP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert columns in a dataframe to class 'snp'</h2>

<h3>Description</h3>

<p><code>setupSNP</code> Convert columns in a dataframe to class 'snp'
</p>
<p><code>summary.setupSNP</code> gives a summary for an object of class 'setupSNP' including 
allele names, major allele frequencie, an exact thest of Hardy-Weinberg 
equilibrium and percentage of missing genotypes
</p>


<h3>Usage</h3>

<pre><code class="language-R">setupSNP(data, colSNPs, sort = FALSE, info, sep = "/", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>dataframe containing columns with the SNPs to be converted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colSNPs</code></td>
<td>
<p>Vector specifying which columns contain SNPs data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort</code></td>
<td>
<p>should SNPs be sorted. Default is FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>info</code></td>
<td>
<p>if sort is TRUE a dataframe containing information 
about the SNPs regarding their genomic position and the gene where 
they are located</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>
<p>character separator used to divide alleles in the genotypes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>optional arguments</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a dataframe of class 'setupSNP' containing converted SNP variables. 
All other variables will be unchanged.
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

 <p><code>snp</code></p>


<h3>Examples</h3>

<pre><code class="language-R">
 data(SNPs)
 myDat&lt;-setupSNP(SNPs,6:40,sep="")


#sorted SNPs and having genomic information
 data(SNPs.info.pos)
 myDat.o&lt;-setupSNP(SNPs,6:40,sep="",sort=TRUE, info=SNPs.info.pos)

# summary
 summary(myDat.o)

# plot one SNP
  plot(myDat,which=2)

</code></pre>


</div>