<div class="container">

<table style="width: 100%;"><tr>
<td>getGeneSymbol</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get gene symbol from a list of SNPs</h2>

<h3>Description</h3>

<p>Get gene symbol from a list of SNPs
</p>


<h3>Usage</h3>

<pre><code class="language-R">getGeneSymbol(
  x,
  snpCol = 1,
  chrCol = 2,
  posCol = 3,
  db = TxDb.Hsapiens.UCSC.hg19.knownGene
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>data.frame containing: SNP name, chromosome and genomic position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpCol</code></td>
<td>
<p>column of x having the SNP name. Default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrCol</code></td>
<td>
<p>column of x having the SNP chromosome. Default is 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posCol</code></td>
<td>
<p>column of x having the SNP position. Default is 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>reference genome. Default is 'TxDb.Hsapiens.UCSC.hg19.knownGene'</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a data.frame having initial information and gene symbol
</p>


<h3>Examples</h3>

<pre><code class="language-R">    
        snps = c('rs58108140','rs189107123','rs180734498','rs144762171')
        chr = c('chr1','chr1','chr1','chr1')
        pos = c(10583, 10611, 13302, 13327)
        
        x &lt;- data.frame(snps, chr, pos )
        
        getGeneSymbol(x)
    
</code></pre>


</div>