<div class="container">

<table style="width: 100%;"><tr>
<td>sortSNPs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sort a vector of SNPs by genomic position</h2>

<h3>Description</h3>

<p>This function sorts a vector with the position of SNPs in a data frame
using another data frame which contains information about SNPs, their 
chromosome, and their genomic position
</p>


<h3>Usage</h3>

<pre><code class="language-R">sortSNPs(data, colSNPs, info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a required data frame with the SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colSNPs</code></td>
<td>
<p>a required vector indicating which columns of 'data' contains genotype data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>info</code></td>
<td>
<p>a required data frame with genomic information for the SNPs (chromosome and position).
The first column must have the SNPs, the second one the chromosome and the third one the genomic 
position. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>First of all, the function obtains a vector with the SNPs sorted using the data frame with the genomic 
positions (see 'dataSNPs.pos' argument).
Then, the columns which indicate where the information about the 
genotypes is in our data frame, are sorted using this vector. 
</p>
<p>This information is useful when <code>WGassociation</code> function is called since it 
allow the user to summaryze the results with the SNPs sorted by genomic position
</p>


<h3>Value</h3>

<p>a vector indicating the colums where the SNPs are recorded in our data frame ('data' argument), sorted
using the genomic positions listed in 'dataSNPs.pos' argument)  
</p>


<h3>Examples</h3>

<pre><code class="language-R">#
# data(SNPs)
# data(SNPs.info.pos)
# colSNPs.order&lt;-sortSNPs(SNPs,c(6:40),SNPs.info.pos)
#
</code></pre>


</div>