<div class="container">

<table style="width: 100%;"><tr>
<td>getSignificantSNPs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extract significant SNPs from an object of class 'WGassociation' </h2>

<h3>Description</h3>

<p>Extract significant SNPs from an object of class 'WGassociation' when genomic 
information is available</p>


<h3>Usage</h3>

<pre><code class="language-R">getSignificantSNPs(x, chromosome, model, sig = 1e-15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class 'WGassociation'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosome</code></td>
<td>
<p> chromosome from which SNPs are extracted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p> genetic model from which SNPs are extracted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig</code></td>
<td>
<p> statistical significance level. The default is 1e-15</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>names</code></td>
<td>
<p>the name of SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>column</code></td>
<td>
<p>the columns corresponding to the SNPs in the original data frame</p>
</td>
</tr>
</table>
<p>...
</p>


<h3>See Also</h3>

  <p><code>WGassociation</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">data(resHapMap)
# resHapMap contains the results for a log-additive genetic model

# to get the significant SNPs for chromosome 12
getSignificantSNPs(resHapMap,chromosome=12)
# to get the significant SNPs for chromosome 5
getSignificantSNPs(resHapMap,5)
# to get the significant SNPs for chromosome X at level 1e-8
getSignificantSNPs(resHapMap,5,sig=1e-8)

</code></pre>


</div>