<div class="container">

<table style="width: 100%;"><tr>
<td>plot.WGassociation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function to plot -log p values from an object of class 'WGassociation'</h2>

<h3>Description</h3>

<p>Function to plot -log p values from an object of class 'WGassociation'
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'WGassociation'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> an object of class 'WGassociation'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other graphical parameters </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A panel with different plots (one for each mode of inheritance) are plotted. Each of them represents 
the -log(p value) for each SNP. Two horizontal lines are also plotted. One one them indicates the nominal 
statistical significance level whereas the other one indicates the statistical 
significance level after Bonferroni correction.
</p>


<h3>Value</h3>

<p>No return value, just the plot
</p>


<h3>References</h3>

<p>JR Gonzalez, L Armengol, X Sole, E Guino, JM Mercader, X Estivill, V Moreno.
SNPassoc: an R package to perform whole genome association studies. Bioinformatics,
2007;23(5):654-5.
</p>


<h3>See Also</h3>

 <p><code>association</code> <code>setupSNP</code> <code>WGassociation</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
library(SNPassoc)

data(asthma, package = "SNPassoc")
asthma.s &lt;- setupSNP(data=asthma, colSNPs=7:ncol(asthma), sep="")

ans &lt;- WGassociation(casecontrol, data=asthma.s)

plot(ans)

</code></pre>


</div>