<div class="container">

<table style="width: 100%;"><tr>
<td>min_mac</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Vizualise, filter based on Minor Allele Count (MAC)</h2>

<h3>Description</h3>

<p>This function can be run in two ways: 1) Without 'min.mac' specified. This will return
a folded site frequency spectrum (SFS), without performing any filtering on the vcf file. Or
2) With 'min.mac' specified. This will also print the folded SFS and
show you where your specified min. MAC count falls. It will then return your vcfR object
with SNPs falling below your min. MAC threshold removed.
Note: previous filtering steps (especially removing samples) may have resulted
in invariant SNPs (MAC =0). For this reason it's a good idea to run min_mac(vcfR, min.mac=1)
before using a SNP dataset in downstream analyses.
</p>


<h3>Usage</h3>

<pre><code class="language-R">min_mac(vcfR, min.mac = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcfR</code></td>
<td>
<p>a vcfR object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.mac</code></td>
<td>
<p>an integer specifying the minimum minor allele count for a
SNP to be retained (e.g. 'min.mac=3' would remove all SNPs with a MAC of 2 or less)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>if 'min.mac' is not specified, the allele frequency spectrum is returned.
If 'min.mac' is specified, SNPs falling below the MAC cutoff will be removed,
and the filtered vcfR object will be returned.
</p>


<h3>Examples</h3>

<pre><code class="language-R">min_mac(vcfR=SNPfiltR::vcfR.example)
</code></pre>


</div>