<div class="container">

<table style="width: 100%;"><tr>
<td>distance_thin</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Filter a vcf file based on distance between SNPs on a given scaffold</h2>

<h3>Description</h3>

<p>This function requires a vcfR object as input, and returns a vcfR object filtered
to retain only SNPs greater than a specified distance apart on each scaffold.
The function starts by automatically retaining the first SNP on a given scaffold,
and then subsequently keeping the next SNP that is greater than the specified distance away,
until it reaches the end of the scaffold/chromosome. This function scales well with
an increasing number of SNPs, but poorly with an increasing number of scaffolds/chromosomes.
For this reason, there is a built in progress bar,
to monitor potentially long-running executions with many scaffolds.
This type of filtering is often employed to reduce linkage among input SNPs,
especially for downstream input to programs like structure, which require unlinked SNPs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">distance_thin(vcfR, min.distance = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcfR</code></td>
<td>
<p>a vcfR object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.distance</code></td>
<td>
<p>a numeric value representing the smallest distance (in base-pairs)
allowed between SNPs after distance thinning</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An identical vcfR object, except that SNPs separated by less than the
specified distance have been removed from the file
</p>


<h3>Examples</h3>

<pre><code class="language-R">distance_thin(vcfR = SNPfiltR::vcfR.example, min.distance = 1000)
</code></pre>


</div>