<div class="container">

<table style="width: 100%;"><tr>
<td>hard_filter</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Hard filter a vcf file by depth and genotype quality (gq)</h2>

<h3>Description</h3>

<p>This function requires a vcfR object as input.
The user can then specify the minimum value for depth of coverage required to retain
a called genotype (must be numeric).
Additionally, the user can specify a minimum genotype quality required to retain
a called genotype (again, must be numeric).
</p>


<h3>Usage</h3>

<pre><code class="language-R">hard_filter(vcfR, depth = NULL, gq = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcfR</code></td>
<td>
<p>a vcfR object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depth</code></td>
<td>
<p>an integer representing the minimum depth for genotype calls that you
wish to retain
(e.g. 'depth = 5' would remove all genotypes with a sequencing depth of 4 reads or less)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gq</code></td>
<td>
<p>an integer representing the minimum genotype quality for
genotype calls that you wish to retain
(e.g. 'gq = 30' would remove all genotypes with a quality score of 29 or lower)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The vcfR object input, with the sites failing specified filters converted to 'NA'
</p>


<h3>Examples</h3>

<pre><code class="language-R">hard_filter(vcfR = SNPfiltR::vcfR.example, depth = 5)
hard_filter(vcfR = SNPfiltR::vcfR.example, depth = 5, gq = 30)
</code></pre>


</div>