<div class="container">

<table style="width: 100%;"><tr>
<td>knockoffGenotypes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Group-knockoffs of unphased genotypes</h2>

<h3>Description</h3>

<p>This function efficiently constructs group-knockoffs of 0,1,2 variables distributed
according to the Li and Stephens model for unphased genotypes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">knockoffGenotypes(X, r, alpha, theta, groups = NULL, seed = 123,
  cluster = NULL, display_progress = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>a 0,1,2 matrix of size n-by-p containing the original variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>
<p>a vector of length p containing the "r" parameters estimated by fastPHASE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>a matrix of size p-by-K containing the "alpha" parameters estimated by fastPHASE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>a matrix of size p-by-K containing the "theta" parameters estimated by fastPHASE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groups</code></td>
<td>
<p>a vector of length p containing group memberships for each variable. Indices
are assumed to be monotone increasing, starting from 1 (default: NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>an integer random seed (default: 123).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cluster</code></td>
<td>
<p>a computing cluster object created by makeCluster (default: NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>display_progress</code></td>
<td>
<p>whether to show progress bar (default: FALSE).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Generate group-knockoffs of unphased genotypes according to the Li and Stephens HMM.
The required model parameters can be obtained through fastPHASE and loaded with loadHMM.
This function is more efficient than knockoffHMM for haplotype data.
</p>


<h3>Value</h3>

<p>A 0,1,2 matrix of size n-by-p containing the knockoff variables.
</p>


<h3>References</h3>

<p>Sesia M, Katsevich E, Bates S, Candès E, Sabatti C (2019).
“Multi-resolution localization of causal variants across the genome.”
<em>bioRxiv</em>.
doi: <a href="http://doi.org/10.1101/631390">10.1101/631390</a>.
</p>


<h3>See Also</h3>

<p>Other knockoffs: <code>knockoffDMC</code>,
<code>knockoffHMM</code>,
<code>knockoffHaplotypes</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Problem size
p = 10
n = 100
# Load HMM to generate data
r_file = system.file("extdata", "haplotypes_rhat.txt", package = "SNPknock")
alpha_file = system.file("extdata", "haplotypes_alphahat.txt", package = "SNPknock")
theta_file = system.file("extdata", "haplotypes_thetahat.txt", package = "SNPknock")
char_file = system.file("extdata", "haplotypes_origchars", package = "SNPknock")
hmm.data = loadHMM(r_file, alpha_file, theta_file, char_file, compact=FALSE, phased=FALSE)
hmm.data$Q = hmm.data$Q[1:(p-1),,]
hmm.data$pEmit = hmm.data$pEmit[1:p,,]
# Sample X from this HMM
X = sampleHMM(hmm.data$pInit, hmm.data$Q, hmm.data$pEmit, n=n)
# Load HMM to generate knockoffs
hmm = loadHMM(r_file, alpha_file, theta_file, char_file)
hmm$r = hmm$r[1:p]
hmm$alpha = hmm$alpha[1:p,]
hmm$theta = hmm$theta[1:p,]
# Generate knockoffs
Xk = knockoffGenotypes(X, hmm$r, hmm$alpha, hmm$theta)
# Generate group-knockoffs for groups of size 3
groups = rep(seq(p), each=3, length.out=p)
Xk = knockoffGenotypes(X, hmm$r, hmm$alpha, hmm$theta, groups=groups)
</code></pre>


</div>