<div class="container">

<table style="width: 100%;"><tr>
<td>knockoffDMC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Group knockoffs of discrete Markov chains</h2>

<h3>Description</h3>

<p>This function constructs knockoffs of variables distributed as a discrete Markov chain.
</p>


<h3>Usage</h3>

<pre><code class="language-R">knockoffDMC(X, pInit, Q, groups = NULL, seed = 123, cluster = NULL,
  display_progress = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>an integer matrix of size n-by-p containing the original variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pInit</code></td>
<td>
<p>an array of length K, containing the marginal distribution of the states for the first variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Q</code></td>
<td>
<p>an array of size (p-1,K,K), containing a list of p-1 transition matrices between the K states of the Markov chain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groups</code></td>
<td>
<p>an array of length p, describing the group membership of each variable (default: NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>an integer random seed (default: 123).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cluster</code></td>
<td>
<p>a computing cluster object created by makeCluster (default: NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>display_progress</code></td>
<td>
<p>whether to show progress bar (default: FALSE).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Each element of the matrix X should be an integer value between 0 and K-1.
The transition matrices contained in Q are defined such that <code class="reqn">P[X_{j+1}=k|X_{j}=l]=Q[j,l,k]</code>.
</p>


<h3>Value</h3>

<p>An integer matrix of size n-by-p containing the knockoff variables.
</p>


<h3>References</h3>

<p>Sesia M, Sabatti C, Candès EJ (2019).
“Gene hunting with hidden Markov model knockoffs.”
<em>Biometrika</em>, <b>106</b>, 1–18.
doi: <a href="http://doi.org/10.1093/biomet/asy033">10.1093/biomet/asy033</a>.
Sesia M, Katsevich E, Bates S, Candès E, Sabatti C (2019).
“Multi-resolution localization of causal variants across the genome.”
<em>bioRxiv</em>.
doi: <a href="http://doi.org/10.1101/631390">10.1101/631390</a>.
</p>


<h3>See Also</h3>

<p>Other knockoffs: <code>knockoffGenotypes</code>,
<code>knockoffHMM</code>,
<code>knockoffHaplotypes</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Generate data
p = 10; K = 5;
pInit = rep(1/K,K)
Q = array(stats::runif((p-1)*K*K),c(p-1,K,K))
for(j in 1:(p-1)) { Q[j,,] = Q[j,,] / rowSums(Q[j,,]) }
X = sampleDMC(pInit, Q, n=20)
# Generate knockoffs
Xk = knockoffDMC(X, pInit, Q)
# Generate group-knockoffs for groups of size 3
groups = rep(seq(p), each=3, length.out=p)
Xk = knockoffDMC(X, pInit, Q, groups=groups)

</code></pre>


</div>