<div class="container">

<table style="width: 100%;"><tr>
<td>snplinkage-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>snplinkage: Single Nucleotide Polymorphisms Linkage Disequilibrium Visualizations</h2>

<h3>Description</h3>

<p>Linkage disequilibrium visualizations of up to several hundreds of single nucleotide polymorphisms (SNPs), annotated with chromosomic positions and gene names. Two types of plots are available for small numbers of SNPs (&lt;40) and for large numbers (tested up to 500). Both can be extended by combining other ggplots, e.g. association studies results, and functions enable to directly visualize the effect of SNP selection methods, as minor allele frequency filtering and TagSNP selection, with a second correlation heatmap. The SNPs correlations are computed on Genotype Data objects from the 'GWASTools' package using the 'SNPRelate' package, and the plots are customizable 'ggplot2' and 'gtable' objects and are annotated using the 'biomaRt' package. Usage is detailed in the vignette with example data and results from up to 500 SNPs of 1,200 scans are in Charlon T. (2019) <a href="https://doi.org/10.13097/archive-ouverte/unige%3A161795">doi:10.13097/archive-ouverte/unige:161795</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Thomas Charlon <a href="mailto:charlon@protonmail.com">charlon@protonmail.com</a> (<a href="https://orcid.org/0000-0001-7497-0470">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li>
<p> Karl Forner
</p>
</li>
<li>
<p> Alessandro Di Cara
</p>
</li>
<li>
<p> Jérôme Wojcik
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://gitlab.com/thomaschln/snplinkage">https://gitlab.com/thomaschln/snplinkage</a>
</p>
</li>
<li>
<p> Report bugs at <a href="https://gitlab.com/thomaschln/snplinkage/-/issues">https://gitlab.com/thomaschln/snplinkage/-/issues</a>
</p>
</li>
</ul>
</div>