<div class="container">

<table style="width: 100%;"><tr>
<td>snprelate_qc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>snprelate_qc</h2>

<h3>Description</h3>

<p>Quality control using SNPRelate functions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snprelate_qc(
  gdata,
  samples_nas = 0.03,
  ibs = 0.99,
  keep_ids = NULL,
  snps_nas = 0.01,
  maf = 0.05,
  tagsnp = 0.8,
  n_cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gdata</code></td>
<td>
<p>Genotype data object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samples_nas</code></td>
<td>
<p>NA threshold for samples, default 3 pct</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ibs</code></td>
<td>
<p>Samples identity by state threshold, default 99 pct</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_ids</code></td>
<td>
<p>Samples ids to keep even if IBS is higher than threshold.
Used for monozygotic twins.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps_nas</code></td>
<td>
<p>NA threshold for SNPs, default 1 pct</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>Minor allele frequency threshold, default 5 pct</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tagsnp</code></td>
<td>
<p>TagSNP r2 correlation threshold, default 0.8</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_cores</code></td>
<td>
<p>Number of cores</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List of gdata, Genotype data object, and df_qc, QC info data frame
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(snplinkage)
gds_path &lt;- save_hgdp_as_gds()
gdata &lt;- load_gds_as_genotype_data(gds_path)
qc &lt;- snprelate_qc(gdata, tagsnp = .99)

</code></pre>


</div>