<div class="container">

<table style="width: 100%;"><tr>
<td>gtable_ld_associations_gdata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gtable of linkage disequilibrium and associations using a GenotypeData object</h2>

<h3>Description</h3>

<p>Compute linkage disequilibrium using snprelate_ld on the set of SNPs in the
associations data frame and call gtable_ld_associations.
Creates a gtable of a linkage disequilibrium, chromosomic positions, and
association scores ggplots.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gtable_ld_associations_gdata(
  df_assocs,
  gdata,
  pvalue_colname = "pvalues",
  labels_colname = "probe_id",
  diamonds = nrow(df_assocs) &lt;= 40,
  window = 15,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df_assocs</code></td>
<td>
<p>SNP annotation data frame with columns chromosome,
position, and as specified by parameters pvalue_colname
and optionally labels_colname.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gdata</code></td>
<td>
<p>GenotypeData object, as returned by
load_gds_as_genotype_data</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue_colname</code></td>
<td>
<p>Column name of df_snp with association values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels_colname</code></td>
<td>
<p>Optional column name of df_snp with labels.
Set NULL to remove labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diamonds</code></td>
<td>
<p>Should the values be displayed as diamonds or points ?
Default is TRUE for up to 40 SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>window</code></td>
<td>
<p>Window size for snprelate_ld.
Forced to the total number of SNPs if diamonds is FALSE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Passed to gtable_ld_associations</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>gtable
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(snplinkage)
gds_path &lt;- save_hgdp_as_gds()
gdata &lt;- load_gds_as_genotype_data(gds_path)
qc &lt;- snprelate_qc(gdata, tagsnp = .99)

snp_idxs_mhc &lt;- select_region_idxs(qc$gdata,
  chromosome = 6, position_min = 29e6, position_max = 33e6)
df_assocs &lt;- chisq_pvalues_gdata(qc$gdata, snp_idxs_mhc)

df_top_aim &lt;- subset(df_assocs, rank(-pvalues, ties.method = 'first') &lt;= 20)

#qc$gdata &lt;- gdata_add_gene_annots(qc$gdata, rownames(df_top_aim))
qc$gdata &lt;- gdata_add_gene_annots_aim_example(qc$gdata, rownames(df_top_aim))

plt &lt;- gtable_ld_associations_gdata(df_top_aim, qc$gdata,
  labels_colname = 'gene')

</code></pre>


</div>