<div class="container">

<table style="width: 100%;"><tr>
<td>gtable_ld</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gtable of linkage disequilibrium and chromosomic positions</h2>

<h3>Description</h3>

<p>Creates a gtable of linkage disequilibrium and chromosomic positions
ggplots. A biplot_subset parameter is available to add a second linkage
disequibrium ggplot to visualize the effect of a SNP selection.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gtable_ld(
  df_ld,
  df_snp,
  biplot_subset = NULL,
  labels_colname = NULL,
  diamonds = length(unique(df_ld$SNP_A)) &lt; 40,
  point_size = ifelse(is.null(biplot_subset), 120, 80)/sqrt(nrow(df_ld)),
  title = "",
  title_biplot = "",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df_ld</code></td>
<td>
<p>Data frame returned by snprelate_ld</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df_snp</code></td>
<td>
<p>SNP annotations with columns snpID and position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>biplot_subset</code></td>
<td>
<p>SNP indexes of the subset for the second ld plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels_colname</code></td>
<td>
<p>Column name of df_snp to use as SNP labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diamonds</code></td>
<td>
<p>Display the values as diamonds or as points
Default is TRUE for less than 40 SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>point_size</code></td>
<td>
<p>Size for geom_point. Ignored if diamonds is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>Plot title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title_biplot</code></td>
<td>
<p>Optional biplot title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Passed to ggplot_ld</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>gtable of ggplots
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(snplinkage)
gds_path &lt;- save_hgdp_as_gds()
gdata &lt;- load_gds_as_genotype_data(gds_path)
qc &lt;- snprelate_qc(gdata, tagsnp = .99)
snp_idxs_8p23 &lt;- select_region_idxs(qc$gdata, chromosome = 8,
  position_min = 11e6, position_max = 12e6)

df_ld &lt;- snprelate_ld(qc$gdata, snps_idx = snp_idxs_8p23, quiet = TRUE)
plt &lt;- gtable_ld(df_ld, df_snp = gdata_snps_annots(qc$gdata))

</code></pre>


</div>