<div class="container">

<table style="width: 100%;"><tr>
<td>snprelate_allele_frequencies</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute allele frequencie and snp missing rate</h2>

<h3>Description</h3>

<p>Wrapper over SNPRelate::snpgdsSNPRateFreq
</p>


<h3>Usage</h3>

<pre><code class="language-R">snprelate_allele_frequencies(
  gdata,
  snps_idx = NULL,
  scans_idx = NULL,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gdata</code></td>
<td>
<p>A GenotypeData object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snps_idx</code></td>
<td>
<p>Vector of snps indices</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scans_idx</code></td>
<td>
<p>Vector of scans indices</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>Whether to be quiet</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame of snps_idx, snps_ids, allele1, allele2, maf, missing
where allele1 and allele2 are the rates of the alleles, and maf the minimum
of the 2. Missing is the missing rate. N.B: the allele rates are computed
on the non missing genotypes, i.e. their sum equals 1.
</p>


</div>