<div class="container">

<table style="width: 100%;"><tr>
<td>snpset_test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Set-based association tests</h2>

<h3>Description</h3>

<p>Perform set-based association tests between multiple sets of SNPs and a
phenotype using GWAS summary statistics. If the function encounters missing
genotypes in the reference data, they will be imputed with genotype means.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snpset_test(hsumstats, x, snp_sets, method = c("saddle", "davies"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>hsumstats</code></td>
<td>
<p>A data frame processed by <code>harmonize_sumstats</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>bed.matrix</code> object created from the reference data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp_sets</code></td>
<td>
<p>A named list where each index represents a separate set of
SNPs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>A method to compute a set-level p value. "saddle" uses
Kuonen's saddlepoint approximation (1999) and "davies" uses the algorithm
of Davies (1980). When "davies" method failed to produce a meaningful
result, "saddle" method is used as a fallback. Default is "saddle".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data.table with columns: "set.id", "pvalue", "n.snp", "top.snp.id"
and "top.snp.pvalue"
</p>

<ul>
<li>
<p> set.id = a name of SNP set
</p>
</li>
<li>
<p> tstat = a test statistic
</p>
</li>
<li>
<p> pvalue = a set-level p value
</p>
</li>
<li>
<p> n.snp = the number of SNPs used in a test
</p>
</li>
<li>
<p> top.snp.id = SNP ID with the smallest p-value within a set of SNPs
</p>
</li>
<li>
<p> top.snp.pvalue = The smallest p-value within a set of SNPs
</p>
</li>
</ul>
<h3>References</h3>

<p>Kuonen, D. Saddlepoint Approximations for Distributions of Quadratic Forms
in Normal Variables. Biometrika 86, 929–935 (1999).
</p>
<p>Davies, R. B. Algorithm AS 155: The Distribution of a Linear Combination of
Chi-Square Random Variables. Journal of the Royal Statistical Society.
Series C (Applied Statistics) 29, 323–333 (1980).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## GWAS summary statistics
head(exGWAS)

## Load reference genotype data
bfile &lt;- system.file("extdata", "example.bed", package = "snpsettest")
x &lt;- read_reference_bed(path = bfile)

## GWAS harmonization with reference data
hsumstats &lt;- harmonize_sumstats(exGWAS, x)

## Perform a set-based test with an arbitrary SNP set
snpset_test(hsumstats, x, list(test = c("SNP_880", "SNP_1533", "SNP_4189")))

## Gene information data
head(gene.curated.GRCh37)

## Map SNPs to genes
snp_sets &lt;- map_snp_to_gene(hsumstats, gene.curated.GRCh37)

## Perform gene-based association tests
out &lt;- snpset_test(hsumstats, x, snp_sets$sets)

</code></pre>


</div>