<div class="container">

<table style="width: 100%;"><tr>
<td>map_snp_to_gene</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Map SNPs to genes</h2>

<h3>Description</h3>

<p>Annotate SNPs onto their neighboring genes (or arbitrary genomic regions) to
perform set-based association tests.
</p>


<h3>Usage</h3>

<pre><code class="language-R">map_snp_to_gene(
  info_snp,
  info_gene,
  extend_start = 20L,
  extend_end = 20L,
  only_sets = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>info_snp</code></td>
<td>
<p>A data frame with columns: "id", "chr", and "pos".
</p>

<ul>
<li>
<p> id = SNP ID (e.g., rs numbers)
</p>
</li>
<li>
<p> chr = chromosome
</p>
</li>
<li>
<p> pos = base-pair position
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>info_gene</code></td>
<td>
<p>A data frame with columns: "gene.id", "chr", "start",
and "end".
</p>

<ul>
<li>
<p> gene.id = gene ID (or identifier for genomic regions)
</p>
</li>
<li>
<p> chr = chromosome (must be the same chromosome coding scheme in <code>info_snp</code>)
</p>
</li>
<li>
<p> start = genomic start position
</p>
</li>
<li>
<p> end = genomic end position
</p>
</li>
</ul>
<p>If a gene has multiple intervals, SNPs mapped to any of them will be merged
into a single set. Please assign unique IDs if you don't want this behavior.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extend_start</code></td>
<td>
<p>A single non-negative integer, allowing for a certain kb
window before the gene to be included. Default is 20 (= 20kb).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extend_end</code></td>
<td>
<p>A single non-negative integer, allowing for a certain kb
window after the gene to be included. Default is 20 (= 20kb).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>only_sets</code></td>
<td>
<p>If <code>TRUE</code>, only sets of SNPs for individual genes are
returned. Otherwise, both sets and mapping information are returned.
Default is <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A nested list containing following components:
</p>

<ul>
<li>
<p> sets: a named list where each index represents a separate set of SNPs
</p>
</li>
<li>
<p> map: a data frame containing SNP mapping information
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
## GWAS summary statistics
head(exGWAS)

## Gene information data
head(gene.curated.GRCh37)

## Map SNPs to genes
snp_sets &lt;- map_snp_to_gene(exGWAS, gene.curated.GRCh37)

## Better to use harmonized GWAS data for gene mapping
bfile &lt;- system.file("extdata", "example.bed", package = "snpsettest")
x &lt;- read_reference_bed(path = bfile)
hsumstats &lt;- harmonize_sumstats(exGWAS, x)
snp_sets &lt;- map_snp_to_gene(hsumstats, gene.curated.GRCh37)

</code></pre>


</div>