<div class="container">

<table style="width: 100%;"><tr>
<td>harmonize_sumstats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Harmonizing GWAS summary to reference data</h2>

<h3>Description</h3>

<p>Finds an intersection of variants between GWAS summary and reference data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">harmonize_sumstats(
  sumstats,
  x,
  match_by_id = TRUE,
  check_strand_flip = FALSE,
  return_indice = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sumstats</code></td>
<td>
<p>A data frame with two columns: "id" and "pvalue".
</p>

<ul>
<li>
<p> id = SNP ID (e.g., rs numbers)
</p>
</li>
<li>
<p> pvalue = SNP-level p value
</p>
</li>
</ul>
<p>If <code>match_by_id = FALSE</code>, it requires additional columns: "chr", "pos", "A1"
and "A2".
</p>

<ul>
<li>
<p> chr =  chromosome
</p>
</li>
<li>
<p> pos =  base-pair position (must be integer)
</p>
</li>
<li>
<p> A1, A2 = allele codes (allele order is not important)
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>bed.matrix</code> object created using the reference data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>match_by_id</code></td>
<td>
<p>If <code>TRUE</code>, SNP matching will be performed by SNP IDs
instead of genomic position and allele codes. Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_strand_flip</code></td>
<td>
<p>Only applies when <code>match_by_id = FALSE</code>. If <code>TRUE</code>,
the function 1) removes ambiguous A/T and G/C SNPs for which the strand is
not obvious, and 2) attempts to find additional matching entries by
flipping allele codes (i.e., A-&gt;T, T-&gt;A, C-&gt;G, G-&gt;A). If the GWAS genotype
data itself is used as the reference data, it would be safe to set
<code>FALSE</code>. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_indice</code></td>
<td>
<p>Only applied when <code>match_by_id = FALSE</code>. If <code>TRUE</code>, the
function provides an additional column indicating whether the match is
with swapped alleles. If <code>check_strand_flip = TRUE</code>, the function also
provides an additional column indicating whether the match is with flipped
strand. Unnecessary for gene-based tests in this package, but may be
useful for other purposes (e.g., harmonization for meta-analysis that
needs to flip the sign of beta for a match with swapped alleles).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Pre-processing of GWAS summary data is required because the sets of variants
available in a particular GWAS might be poorly matched to the variants in
reference data. SNP matching can be performed either 1) by SNP ID or 2) by
chromosome code, base-pair position, and allele codes, while taking into
account possible strand flips and reference allele swap. For matched
entries, the SNP IDs in GWAS summary data are replaced with the ones in the
reference data.
</p>


<h3>Value</h3>

<p>A data frame with columns: "id", "chr", "pos", "A1", "A2" and
"pvalue". If <code>return_indice = TRUE</code>, the data frame includes additional
columns <code>key_</code>, <code>swapped_</code>, and <code>flipped_</code>. <code>key_</code> is "chr_pos_A1_A2" in
<code>sumstat</code> (the original input before harmonization). <code>swapped_</code> contains a
logical vector indicating reference allele swap. <code>flipped_</code> contains a
logical vector indicating strand flip.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## GWAS summary statistics
head(exGWAS)

## Load reference genotype data
bfile &lt;- system.file("extdata", "example.bed", package = "snpsettest")
x &lt;- read_reference_bed(path = bfile)

## Harmonize by SNP IDs
hsumstats1 &lt;- harmonize_sumstats(exGWAS, x)

## Harmonize by genomic position and allele codes
## Reference allele swap will be taken into account
hsumstats2 &lt;- harmonize_sumstats(exGWAS, x, match_by_id = FALSE)

## Check matching entries by flipping allele codes
## Ambiguous SNPs will be excluded from harmonization
hsumstats3 &lt;- harmonize_sumstats(exGWAS, x, match_by_id = FALSE,
                                 check_strand_flip = TRUE)
</code></pre>


</div>