<div class="container">

<table style="width: 100%;"><tr>
<td>get_LFDRs_given_caller</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimates LFDR values per mutant site detected by a desired variant caller</h2>

<h3>Description</h3>

<p>Based on a given read count matrix and a list of calls made by a desired variant caller, estimates LFDRs that corresponds to each genomic site. It further
classifies sites to either mutant or non-mutant sites by comparing their estimated LFDRs with an LFDR cut-off value. The cut-off value as well as error rates can be defined by users.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_LFDRs_given_caller(bam_input, calls, LFDR.T, error)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bam_input</code></td>
<td>
<p>Path to an original BAM file used to call variants. The file must be in the format of a csv file generated by bam-readcount (https://github.com/genome/bam-readcount). See Examples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>calls</code></td>
<td>
<p>Path to a vcf file generated by a variant caller. The first and second columns of this file have to be CHR names and positions, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LFDR.T</code></td>
<td>
<p>A number between 0 and 1. It can be set to 0 to include all sites. Otherwise, this threshold excludes sites with an LFDR below the specified threshold.  If no value is specified, it will be set to 0.01 by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>error</code></td>
<td>
<p>Error rate between 0 and 1. If it is set to NULL, a weighted average of average base call quality and average mapping quality per site will be calculated. Otherwise, it may be set to 0.01 or a desired error vector can be introduced by the user.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list. Slot <strong>estimated.LFDRs</strong> returns estimated LFDRs for all sites in the input file.
Slot <strong>updated.bam</strong> adds estimated LFDRs, model errors and a mutant variable (indicating whether each site is detected to be a mutant (1) or non-mutant (0) site) to the input bam file.
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Perkins, T.J. (2024). Empirical Bayes single nucleotide variant calling for next-generation sequencing data. Scientific Reports 14,  1550, &lt;doi:10.1038/s41598-024-51958-z&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">bam_path &lt;- system.file("extdata", "bam_input.csv", package="SNVLFDR")
calls_path &lt;- system.file("extdata", "calls.vcf", package="SNVLFDR")
output=get_LFDRs_given_caller(bam_input=bam_path,calls=calls_path,LFDR.T=0.01,error=NULL)

</code></pre>


</div>