<div class="container">

<table style="width: 100%;"><tr>
<td>A.mat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Additive relationship matrix
</h2>

<h3>Description</h3>

<p>Calculates the realized additive relationship matrix. Currently is the C++ implementation of van Raden (2008). 
</p>


<h3>Usage</h3>

<pre><code class="language-R">A.mat(X,min.MAF=0,return.imputed=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>Matrix (<code class="reqn">n \times m</code>) of unphased genotypes for <code class="reqn">n</code> lines and <code class="reqn">m</code> biallelic markers, 
coded as {-1,0,1}. Fractional (imputed) and missing values (NA) are allowed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.MAF</code></td>
<td>

<p>Minimum minor allele frequency. The A matrix is not sensitive to rare alleles, so by default only monomorphic markers are removed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.imputed</code></td>
<td>

<p>When TRUE, the imputed marker matrix is returned.  
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For vanraden method: the marker matrix is centered by subtracting column means <code class="reqn">M= X - ms</code> where ms is the coumn means. Then <code class="reqn">A=M M'/c</code>, where <code class="reqn">c = \sum_k{d_k}/k</code>, the mean value of the diagonal values of the <code class="reqn">M M'</code> portion.
</p>


<h3>Value</h3>

<p>If return.imputed = FALSE, the <code class="reqn">n \times n</code> additive relationship matrix is returned.
</p>
<p>If return.imputed = TRUE, the function returns a list containing
</p>

<dl>
<dt>$A</dt>
<dd>
<p>the A matrix</p>
</dd>
<dt>$X</dt>
<dd>
<p>the imputed marker matrix</p>
</dd>
</dl>
<h3>References</h3>

<p>Endelman, J.B., and J.-L. Jannink. 2012. Shrinkage estimation of the realized relationship matrix. G3:Genes, Genomes, Genetics. 2:1405-1413. doi: 10.1534/g3.112.004259
</p>
<p>Covarrubias-Pazaran G (2016) Genome assisted prediction of quantitative traits using the R package sommer. PLoS ONE 11(6): doi:10.1371/journal.pone.0156744 
</p>


<h3>See Also</h3>

<p><code>mmer</code> â€“ the core function of the package </p>


<h3>Examples</h3>

<pre><code class="language-R">####=========================================####
#### random population of 200 lines with 1000 markers
####=========================================####
X &lt;- matrix(rep(0,200*1000),200,1000)
for (i in 1:200) {
  X[i,] &lt;- ifelse(runif(1000)&lt;0.5,-1,1)
}

A &lt;- A.mat(X)

####=========================================####
#### take a look at the Genomic relationship matrix 
#### (just a small part)
####=========================================####
# colfunc &lt;- colorRampPalette(c("steelblue4","springgreen","yellow"))
# hv &lt;- heatmap(A[1:15,1:15], col = colfunc(100),Colv = "Rowv")
# str(hv)
</code></pre>


</div>