<div class="container">

<table style="width: 100%;"><tr>
<td>E.mat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Epistatic relationship matrix
</h2>

<h3>Description</h3>

<p>Calculates the realized epistatic relationship matrix of second order (additive x additive, additive x dominance, or dominance x dominance) using hadamard products with the C++ Armadillo library.
</p>


<h3>Usage</h3>

<pre><code class="language-R">E.mat(X,nishio=TRUE,type="A#A",min.MAF=0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>Matrix (<code class="reqn">n \times m</code>) of unphased genotypes for <code class="reqn">n</code> lines and <code class="reqn">m</code> biallelic markers, 
coded as {-1,0,1}. Fractional (imputed) and missing values (NA) are allowed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nishio</code></td>
<td>

<p>If TRUE Nishio ans Satoh. (2014), otherwise Su et al. (2012) (see Details in the D.mat help page).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>

<p>An argument specifying the type of epistatic relationship matrix desired. The default is the second order epistasis (additive x additive) type="A#A". Other options are additive x dominant (type="A#D"), or dominant by dominant (type="D#D").  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.MAF</code></td>
<td>

<p>Minimum minor allele frequency. The A matrix is not sensitive to rare alleles, so by default only monomorphic markers are removed.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>it is computed as the Hadamard product of the epistatic relationship matrix; E=A#A, E=A#D, E=D#D. 
</p>


<h3>Value</h3>

<p>The epistatic relationship matrix is returned.
</p>


<h3>References</h3>

<p>Covarrubias-Pazaran G (2016) Genome assisted prediction of quantitative traits using the R package sommer. PLoS ONE 11(6): doi:10.1371/journal.pone.0156744 
</p>
<p>Endelman, J.B., and J.-L. Jannink. 2012. Shrinkage estimation of the realized relationship matrix. G3:Genes, Genomes, Genetics. 2:1405-1413. doi: 10.1534/g3.112.004259
</p>
<p>Nishio M and Satoh M. 2014. Including Dominance Effects in the Genomic BLUP Method for Genomic Evaluation. Plos One 9(1), doi:10.1371/journal.pone.0085792
</p>
<p>Su G, Christensen OF, Ostersen T, Henryon M, Lund MS. 2012. Estimating Additive and Non-Additive Genetic Variances and Predicting Genetic Merits Using Genome-Wide Dense Single Nucleotide Polymorphism Markers. PLoS ONE 7(9): e45293. doi:10.1371/journal.pone.0045293
</p>


<h3>See Also</h3>

<p>The core functions of the package <code>mmer</code></p>


<h3>Examples</h3>

<pre><code class="language-R">####=========================================####
####random population of 200 lines with 1000 markers
####=========================================####
X &lt;- matrix(rep(0,200*1000),200,1000)
for (i in 1:200) {
  X[i,] &lt;- sample(c(-1,0,0,1), size=1000, replace=TRUE)
}

E &lt;- E.mat(X, type="A#A") 
# if heterozygote markers are present can be used "A#D" or "D#D"
</code></pre>


</div>