<div class="container">

<table style="width: 100%;"><tr>
<td>sommer-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
<strong>So</strong>lving <strong>M</strong>ixed <strong>M</strong>odel <strong>E</strong>quations in <strong>R</strong>
<br><img src="../help/figures/mai.png" height="10%" alt="Figure: mai.png">
</h2>

<h3>Description</h3>

<p>Sommer is a structural multivariate-univariate linear mixed model solver for multiple random effects allowing the specification and/or estimation of variance covariance structures. REML estimates can be obtained using two major methods
</p>
<p>Direct-Inversion (Newton-Raphson and Average Information) &gt;&gt; <code>mmer</code> function
</p>
<p>Henderson's mixed model equations (Average Information) &gt;&gt; <code>mmec</code> function
</p>
<p>The algorithms are coded in C++ using the Armadillo library to optimize dense matrix operations common in genomic models. Sommer was designed to include complex covariance structures, e.g., unstructured, reduced-rank, diagonal. And also to model relationships between levels of a random effect, e.g., additive, dominance and epistatic relationship structures.  
</p>
<p>The <code>mmer</code> function can deal well with small and medium-size data sets (&lt; 10,000 observations/records for average computers given the computational burden carried by the direct-inversion algorithms) since it works in the c &gt; r problem and inverts an r x r matrix (being r the number of records and c the number of coefficients). On the other hand, the <code>mmec</code> function can deal with greater number of records (r&gt;250K) as long as the number of coefficients to estimate is &lt; 10,000 coefficients (c) since it works in the r &gt; c problem and inverts a c x c matrix (being c the number of coefficients). The <code>predict.mmer</code> and <code>predict.mmec</code> functions can be used to obtain adjusted means. This package returns variance-covariance components, BLUPs, BLUEs, residuals, fitted values, variances-covariances for fixed and random effects, etc.
</p>


<h3>Functions for genetic analysis</h3>

<p>The package provides kernels to estimate additive (<code>A.mat</code>), dominance (<code>D.mat</code>), epistatic (<code>E.mat</code>), single-step (<code>H.mat</code>) relationship matrices for diploid and polyploid organisms. It also provides flexibility to fit other genetic models such as full and half diallel models and random regression models.
</p>
<p>A good converter from letter code to numeric format is implemented in the function <code>atcg1234</code>, which supports higher ploidy levels than diploid. Additional functions for genetic analysis have been included such as build a genotypic hybrid marker matrix (<code>build.HMM</code>), plot of genetic maps (<code>map.plot</code>), creation of manhattan plots (<code>manhattan</code>). If you need to use pedigree you need to convert your pedigree into a relationship matrix (use the 'getA' function from the pedigreemm package).
</p>


<h3>Functions for statistical analysis and S3 methods</h3>

<p>The <code>vpredict</code> function can be used to estimate standard errors for linear combinations of variance components (e.g. ratios like h2). The <code>r2</code> function calculates reliability. S3 methods are available for some parameter extraction such as:
</p>
<p>+ <code>predict.mmer</code>, <code>predict.mmec</code>,
</p>
<p>+ <code>fitted.mmer</code>, <code>fitted.mmec</code>, 
</p>
<p>+ <code>residuals.mmer</code>,  <code>residuals.mmec</code>, 
</p>
<p>+ <code>summary.mmer</code>,  <code>summary.mmec</code>, 
</p>
<p>+ <code>coef.mmer</code>, <code>coef.mmec</code>, 
</p>
<p>+ <code>anova.mmer</code>, <code>anova.mmec</code>, 
</p>
<p>+ <code>plot.mmer</code>, <code>plot.mmec</code>.
</p>


<h3>Functions for trial analysis</h3>

<p>Recently, spatial modeling has been added added to sommer using the two-dimensional spline (<code>spl2Da</code> and <code>spl2Db</code> for <code>mmer</code> and <code>spl2Dc</code> for <code>mmec</code> ) functions.
</p>


<h3>Keeping sommer updated</h3>

<p>The sommer package is updated on CRAN every 4-months due to CRAN policies but you can find the latest source at https://github.com/covaruber/sommer. This can be easily installed typing the following in the R console:
</p>
<p>library(devtools)
</p>
<p>install_github("covaruber/sommer")
</p>
<p>This is recommended if you reported a bug, was fixed and was immediately pushed to GitHub but not in CRAN until the next update.
</p>


<h3>Tutorials</h3>

<p><strong>For tutorials</strong> on how to perform different analysis with sommer please look at the vignettes by typing in the terminal:
</p>
<p><strong>vignette("v1.sommer.quick.start")</strong> 
</p>
<p><strong>vignette("v2.sommer.changes.and.faqs")</strong>
</p>
<p><strong>vignette("v3.sommer.qg")</strong>
</p>
<p><strong>vignette("v4.sommer.gxe")</strong>
</p>
<p>or visit <strong>https://covaruber.github.io</strong>
</p>


<h3>Getting started</h3>

<p>The package has been equiped with several datasets to learn how to use the sommer package (and almost to learn all sort of quantitative genetic analysis): 
</p>
<p>* <code>DT_halfdiallel</code>, <code>DT_fulldiallel</code> and <code>DT_mohring</code> datasets have examples to fit half and full diallel designs. 
</p>
<p>* <code>DT_h2</code> to calculate heritability
</p>
<p>* <code>DT_cornhybrids</code> and <code>DT_technow</code> datasets to perform genomic prediction in hybrid single crosses 
</p>
<p>* <code>DT_wheat</code> dataset to do genomic prediction in single crosses in species displaying only additive effects.
</p>
<p>* <code>DT_cpdata</code> dataset to fit genomic prediction models within a biparental population coming from 2 highly heterozygous parents including additive, dominance and epistatic effects. 
</p>
<p>* <code>DT_polyploid</code> to fit genomic prediction and GWAS analysis in polyploids. 
</p>
<p>* <code>DT_gryphon</code> data contains an example of an animal model including pedigree information.
</p>
<p>* <code>DT_btdata</code> dataset contains an animal (birds) model.
</p>
<p>* <code>DT_legendre</code> simulated dataset for random regression model.
</p>
<p>* <code>DT_sleepstudy</code> dataset to know how to translate lme4 models to sommer models.
</p>


<h3>Differences of sommer &gt;= 4.1.7 with previous versions</h3>

<p>Since version 4.1.7 I have introduced the mme-based average information function 'mmec' which is much faster when dealing with the r &gt; c problem (more records than coefficients to estimate). This introduces its own covariance structure functons such as vsc(), usc(), dsc(), atc(), csc(). Please give it a try, although is in early phase of development.
</p>


<h3>Differences of sommer &gt;= 3.7.0 with previous versions</h3>

<p>Since version 3.7 I have completly redefined the specification of the variance-covariance structures to provide more flexibility to the user. This has particularly helped the residual covariance structures and the easier combination of custom random effects and overlay models. I think that although this will bring some uncomfortable situations at the beggining, in the long term this will help users to fit better models. In esence, I have abandoned the asreml formulation (not the structures available) given it's limitations to combine some of the sommer structures but all covariance structures can now be fitted using the 'vsr' functions. 
</p>


<h3>Differences of sommer &gt;= 3.0.0 with previous versions</h3>

<p>Since version 3.0 I have decided to focus in developing the multivariate solver and for doing this I have decided to remove the M argument (for GWAS analysis) from the mmer function and move it to it's own function <code>GWAS</code>.
</p>
<p>Before the mmer solver had implemented the usr(trait), diag(trait), at(trait) asreml formulation for multivariate models that allow to specify the structure of the trait in multivariate models. Therefore the MVM argument was no longer needed. After version 3.7 now the multi-trait structures can be specified in the <code>Gt</code> and <code>Gtc</code> arguments of the <code>vsr</code> function.
</p>
<p>The Average Information algorithm had been removed in the past from the package because of its instability to deal with very complex models without good initial values. Now after 3.7 I have brought it back after I noticed that starting with NR the first three iterations gives enough flexibility to the AI algorithm.
</p>
<p>Keep in mind that sommer uses direct inversion (DI) algorithm which can be very slow for datasets with many observations (big 'n'). The package is focused in problems of the type p &gt; n (more random effect(s) levels than observations) and models with dense covariance structures. For example, for experiment with dense covariance structures with low-replication (i.e. 2000 records from 1000 individuals replicated  twice with a covariance structure of 1000x1000) sommer will be faster than MME-based software. Also for genomic problems with large number of random effect levels, i.e. 300 individuals (n) with 100,000 genetic markers (p). On the other hand, for highly replicated trials with small covariance structures or n &gt; p (i.e. 2000 records from 200 individuals replicated 10 times with covariance structure of 200x200) asreml or other MME-based algorithms will be much faster and I recommend you to use that software.
</p>


<h3>Models Enabled</h3>

<p>The core of the package are the <code>mmer</code> and <code>mmec</code> (formula-based) functions which solve the mixed model equations. The functions are an interface to call the 'NR' Direct-Inversion Newton-Raphson, 'AI' Direct-Inversion Average Information or the mme-based Average Information (Tunnicliffe 1989; Gilmour et al. 1995; Lee et al. 2016). Since version 2.0 sommer can handle multivariate models. Following Maier et al. (2015), the multivariate (and by extension the univariate) mixed model implemented has the form:
</p>
<p><img src="../help/figures/form1.png" height="10%" alt="Figure: form1.png"></p>
<p>where y_i is a vector of trait phenotypes, <code class="reqn">\beta_i</code> is a vector of fixed effects, u_i is a vector of random effects for individuals and e_i are residuals for trait i (i = 1,..., t). The random effects (u_1 ... u_i and e_i) are assumed to be normally distributed with mean zero. X and Z are incidence matrices for fixed and random effects respectively. The distribution of the multivariate response and the phenotypic variance covariance (V) are:
</p>
<p><img src="../help/figures/form2.png" height="10%" alt="Figure: form2.png"></p>
<p>where K is the relationship or covariance matrix for the kth random effect (u=1,...,k), and R=I is an identity matrix for the residual term. The terms <code class="reqn">\sigma^2_{g_{i}}</code> and <code class="reqn">\sigma^2_{\epsilon_{i}}</code> denote the genetic (or any of the kth random terms) and residual variance of trait i, respectively and <code class="reqn">\sigma_{g_{_{ij}}}</code> and <code class="reqn">\sigma_{\epsilon_{_{ij}}}</code> the genetic (or any of the kth random terms) and residual covariance between traits i and j (i=1,...,t, and j=1,...,t). The algorithm implemented optimizes the log likelihood:
</p>
<p><img src="../help/figures/form3.png" height="10%" alt="Figure: form3.png"></p>
<p>where || is the determinant of a matrix. And the REML estimates are updated using a Newton optimization algorithm of the form:
</p>
<p><img src="../help/figures/form4.png" height="10%" alt="Figure: form4.png"></p>
<p>Where, theta is the vector of variance components for random effects and covariance components among traits, H^-1 is the inverse of the Hessian matrix of second derivatives for the kth cycle, dL/dsigma^2_i is the vector of first derivatives of the likelihood with respect to the variance-covariance components. The Eigen decomposition of the relationship matrix proposed by Lee and Van Der Werf (2016) was included in the Newton-Raphson algorithm to improve time efficiency. Additionally, the popular vpredict function to estimate standard errors for linear combinations of variance components (i.e. heritabilities and genetic correlations) was added to the package as well.
</p>


<h3>GWAS Models</h3>

<p>The GWAS models in the sommer package are enabled by using the M argument in the functions <code>GWAS</code>, which is expected to be a numeric marker matrix. Markers are treated as fixed effects according to the model proposed by Yu et al. (2006) for diploids, and Rosyara et al. (2016) (for polyploids). The matrices X and M are both fixed effects, but they are separated by 2 different arguments to distinguish factors such as environmental and design factors for the argument "X" and markers with "M".
</p>
<p>The genome-wide association analysis is based on the mixed model:
</p>
<p style="text-align: center;"><code class="reqn">y = X \beta + Z g + M \tau + e</code>
</p>

<p>where <code class="reqn">\beta</code> is a vector of fixed effects that can model both environmental factors and population structure.  
The variable <code class="reqn">g</code> models the genetic background of each line as a random effect with <code class="reqn">Var[g] = K \sigma^2</code>.  
The variable <code class="reqn">\tau</code> models the additive SNP effect as a fixed effect.  The residual variance is <code class="reqn">Var[\varepsilon] = I \sigma_e^2</code>
</p>
<p>When principal components are included (P+K model), the loadings are determined from an eigenvalue decomposition of the K matrix and are used in the fixed effect part.
</p>
<p>The argument "P3D" introduced by Zhang et al. (2010) can be used with the P3D argument.  When P3D=FALSE, this function is equivalent to AI/NR with REML where the variance components are estimated for each SNP or marker tested (Kang et al. 2008).  When P3D=TRUE, it is equivalent to NR (Kang et al. 2010) where the assumption is that variance components for all SNP/markers are the same and therefore the variance components are estimated only once (and markers are tested in a WLS framework being the the weight matrix (M) the inverse of the phenotypic variance matrix (V)).  Therefore, P3D=TRUE option is faster but can underestimate significance compared to P3D=FALSE.
</p>
<p>Multivariate GWAS are based in Covarrubias-Pazaran et al. (2018, In preparation), which adjusts betas for all response variables and then does the regular GWAS with such adjusted betas or marker effects.
</p>
<p>For extra details about the methods please read the canonical papers listed in the References section.
</p>


<h3>Bug report and contact</h3>

<p>If you have any questions or suggestions please post it in https://stackoverflow.com or https://stats.stackexchange.com
</p>
<p>I'll be glad to help or answer any question. I have spent a valuable amount of time developing this package. Please cite this package in your publication. Type 'citation("sommer")' to know how to cite it.
</p>


<h3>Author(s)</h3>

<p>Giovanny Covarrubias-Pazaran
</p>


<h3>References</h3>

<p>Covarrubias-Pazaran G. 2016. Genome assisted prediction of quantitative traits using the R package sommer. PLoS ONE 11(6): doi:10.1371/journal.pone.0156744 
</p>
<p>Covarrubias-Pazaran G. 2018. Software update: Moving the R package sommer to multivariate mixed models for genome-assisted prediction. doi: https://doi.org/10.1101/354639
</p>
<p>Bernardo Rex. 2010. Breeding for quantitative traits in plants. Second edition. Stemma Press. 390 pp.
</p>
<p>Gilmour et al. 1995. Average Information REML: An efficient algorithm for variance parameter estimation in linear mixed models. Biometrics 51(4):1440-1450.
</p>
<p>Henderson C.R. 1975. Best Linear Unbiased Estimation and Prediction under a Selection Model. Biometrics vol. 31(2):423-447.
</p>
<p>Kang et al. 2008. Efficient control of population structure in model organism association mapping. Genetics 178:1709-1723.
</p>
<p>Lee et al. 2015. MTG2: An efficient algorithm for multivariate linear mixed model analysis based on genomic information. Cold Spring Harbor. doi: http://dx.doi.org/10.1101/027201.
</p>
<p>Maier et al. 2015. Joint analysis of psychiatric disorders increases accuracy of risk prediction for schizophrenia, bipolar disorder, and major depressive disorder. Am J Hum Genet; 96(2):283-294.
</p>
<p>Searle. 1993. Applying the EM algorithm to calculating ML and REML estimates of variance components. Paper invited for the 1993 American Statistical Association Meeting, San Francisco.
</p>
<p>Yu et al. 2006. A unified mixed-model method for association mapping that accounts for multiple levels of relatedness. Genetics 38:203-208.
</p>
<p>Tunnicliffe W. 1989. On the use of marginal likelihood in time series model estimation. JRSS 51(1):15-27.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
####=========================================####
#### For CRAN time limitations most lines in the 
#### examples are silenced with one '#' mark, 
#### remove them and run the examples
####=========================================####

####=========================================####
#### EXAMPLES
#### Different models with sommer
####=========================================####

data(DT_example)

# DT &lt;- DT_example
# DT=DT[with(DT, order(Env)), ]
# head(DT)
# 
# ####=========================================####
# #### Univariate homogeneous variance models  ####
# ####=========================================####
# 
# ## Compound simmetry (CS) model
# ans1 &lt;- mmer(Yield~Env,
#              random= ~ Name + Env:Name,
#              rcov= ~ units,
#              data=DT)
# summary(ans1)
# 
# ans2 &lt;- mmec(Yield~Env,
#              random= ~ Name + Env:Name,
#              rcov= ~ units,
#              data=DT)
# summary(ans2)
# 
# ####===========================================####
# #### Univariate heterogeneous variance models  ####
# ####===========================================####
# ## Compound simmetry (CS) + Diagonal (DIAG) model
# ans3 &lt;- mmer(Yield~Env,
#              random= ~Name + vsr(dsr(Env),Name),
#              rcov= ~ vsr(dsr(Env),units),
#              data=DT)
# summary(ans3)
# 
# ans4 &lt;- mmec(Yield~Env,
#              random= ~Name + vsc(dsc(Env),isc(Name)),
#              rcov= ~ vsc(dsc(Env),isc(units)),
#              data=DT)
# summary(ans4)

</code></pre>


</div>