<div class="container">

<table style="width: 100%;"><tr>
<td>D.mat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Dominance relationship matrix
</h2>

<h3>Description</h3>

<p>C++ implementation of the dominance matrix. Calculates the realized dominance relationship matrix. Can help to increase the prediction accuracy when 2 conditions are met; 1) The trait has intermediate to high heritability, 2) The population contains a big number of individuals that are half or full sibs (HS &amp; FS).
</p>


<h3>Usage</h3>

<pre><code class="language-R">D.mat(X,nishio=TRUE,min.MAF=0,return.imputed=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>Matrix (<code class="reqn">n \times m</code>) of unphased genotypes for <code class="reqn">n</code> lines and <code class="reqn">m</code> biallelic markers, 
coded as {-1,0,1}. Fractional (imputed) and missing values (NA) are allowed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nishio</code></td>
<td>

<p>If TRUE Nishio ans Satoh. (2014), otherwise Su et al. (2012). See references.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.MAF</code></td>
<td>

<p>Minimum minor allele frequency. The D matrix is not sensitive to rare alleles, so by default only monomorphic markers are removed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.imputed</code></td>
<td>

<p>When TRUE, the imputed marker matrix is returned.  
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The additive marker coefficients will be used to compute dominance coefficients as: Xd = 1-abs(X) for diploids.
</p>
<p>For nishio method: the marker matrix is centered by subtracting column means <code class="reqn">M= Xd - ms</code> where ms is the column means. Then <code class="reqn">A=M M'/c</code>, where <code class="reqn">c = 2 \sum_k {p_k (1-p_k)}</code>.
</p>
<p>For su method: the marker matrix is normalized by subtracting row means <code class="reqn">M= Xd - 2pq</code> where 2pq is the product of allele frequencies times 2. Then <code class="reqn">A=M M'/c</code>, where <code class="reqn">c = 2 \sum_k {2pq_k (1-2pq_k)}</code>.
</p>


<h3>Value</h3>

<p>If return.imputed = FALSE, the <code class="reqn">n \times n</code> additive relationship matrix is returned.
</p>
<p>If return.imputed = TRUE, the function returns a list containing
</p>

<dl>
<dt>$D</dt>
<dd>
<p>the D matrix</p>
</dd>
<dt>$imputed</dt>
<dd>
<p>the imputed marker matrix</p>
</dd>
</dl>
<h3>References</h3>

<p>Covarrubias-Pazaran G (2016) Genome assisted prediction of quantitative traits using the R package sommer. PLoS ONE 11(6): doi:10.1371/journal.pone.0156744 
</p>
<p>Nishio M and Satoh M. 2014. Including Dominance Effects in the Genomic BLUP Method for Genomic Evaluation. Plos One 9(1), doi:10.1371/journal.pone.0085792
</p>
<p>Su G, Christensen OF, Ostersen T, Henryon M, Lund MS. 2012. Estimating Additive and Non-Additive Genetic Variances and Predicting Genetic Merits Using Genome-Wide Dense Single Nucleotide Polymorphism Markers. PLoS ONE 7(9): e45293. doi:10.1371/journal.pone.0045293
</p>


<h3>See Also</h3>

<p>The core functions of the package <code>mmer</code></p>


<h3>Examples</h3>

<pre><code class="language-R">####=========================================####
#### EXAMPLE 1
####=========================================####
####random population of 200 lines with 1000 markers
X &lt;- matrix(rep(0,200*1000),200,1000)
for (i in 1:200) {
  X[i,] &lt;- sample(c(-1,0,0,1), size=1000, replace=TRUE)
}

D &lt;- D.mat(X)

</code></pre>


</div>