<div class="container">

<table style="width: 100%;"><tr>
<td>SpaCCI_local</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Infer Cell-Cell Interactions on a Local Scale</h2>

<h3>Description</h3>

<p>This function infers cell-cell interactions on a local scale using spatial transcriptomics data. It utilizes permutation testing to identify significant ligand-receptor interactions within specified neighborhoods around randomly selected center spots.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SpaCCI_local(
  gene_spot_df,
  spot_cell_prop_df,
  spatial_coord,
  prop,
  radius,
  matching_L_R_pairs,
  matching_L_R_pairs_info
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gene_spot_df</code></td>
<td>
<p>A data frame where the rows are genes and the columns are spots (Spot_IDs), representing gene expression levels across spatial spots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spot_cell_prop_df</code></td>
<td>
<p>A data frame of cell type proportions for each spot. The rows represent spots (Spot_IDs), and the columns represent different cell types.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spatial_coord</code></td>
<td>
<p>A data frame of the spatial coordinates. The column names should include 'c("Spot_ID", "imagerow", "imagecol")', and the row names must be the Spot_IDs, which is the same as the row names in the cell type proportion data frame or the column names of the gene*spot expression data frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prop</code></td>
<td>
<p>A numeric value representing the proportion of spots to randomly sample as center spots for local neighborhood analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>radius</code></td>
<td>
<p>A numeric value specifying the radius of the spatial neighborhood around each center spot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>matching_L_R_pairs</code></td>
<td>
<p>A data frame containing matching ligand-receptor pairs. Each row corresponds to a ligand-receptor pair, with columns for <code>ligand_vector</code> and <code>receptor_vector</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>matching_L_R_pairs_info</code></td>
<td>
<p>A data frame providing additional information for each ligand-receptor pair, such as pathway information.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing:
</p>

<dl>
<dt>dataframelist</dt>
<dd>
<p>A list of data frames, each representing the inferred interactions for a specific center spot. Each data frame includes information on ligand and receptor cell types, P-values, and adjusted P-values.</p>
</dd>
<dt>RegionIDs_matrix</dt>
<dd>
<p>A list of matrices, each containing the IDs of the spots within the specified radius of each center spot.</p>
</dd>
</dl>
</div>