<div class="container">

<table style="width: 100%;"><tr>
<td>Global_Permutations</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform Global Permutations</h2>

<h3>Description</h3>

<p>This function performs global permutations on the spatial transcriptomics data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Global_Permutations(
  permutationMatrix,
  permut_null_regionMatrix,
  permut_col,
  cellPropMatrix,
  spotGeneMatrix,
  LigandVectorIndex,
  ReceptorVectorIndex,
  null_expression,
  nBoot
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>permutationMatrix</code></td>
<td>
<p>A matrix containing permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permut_null_regionMatrix</code></td>
<td>
<p>A matrix of null region permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permut_col</code></td>
<td>
<p>A column matrix of permutations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cellPropMatrix</code></td>
<td>
<p>A matrix of cell type proportions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spotGeneMatrix</code></td>
<td>
<p>A matrix of gene expressions at spots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LigandVectorIndex</code></td>
<td>
<p>A vector of ligand indices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ReceptorVectorIndex</code></td>
<td>
<p>A vector of receptor indices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null_expression</code></td>
<td>
<p>A matrix of null expression values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nBoot</code></td>
<td>
<p>Number of bootstrap iterations.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix with the results of the global permutations.
</p>


</div>