<div class="container">

<table style="width: 100%;"><tr>
<td>arabidopsis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Arabidopsis genetic and climatic data
</h2>

<h3>Description</h3>

<p>For 948 “accessions” from European Arabidopsis thaliana populations, this data set merges the genotypic information at 
four single nucleotide polymorphisms (SNP) putatively involved in adaptation to climate (Fournier-Level et al, 2011, Table 1), 
with 13 climatic variables from Hancock et al. (2011). 
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("arabidopsis")</code></pre>


<h3>Format</h3>

<p>The data frame includes 948 observations on the following variables:
</p>

<dl>
<dt>pos1046738, pos5510910, pos6235221, pos8132698</dt>
<dd>
<p>Genotypes at four SNP loci</p>
</dd>
<dt>LAT</dt>
<dd>
<p>latitude</p>
</dd>
<dt>LONG</dt>
<dd>
<p>longitude</p>
</dd>
<dt>seasonal, tempWarmest, tempColdest, preciWettest,
preciDriest, preciCV, PAR_SPRING,</dt>
<dd></dd> 
<dt>growingL, conseqCold, conseqFrFree, RelHumidSp, dayLSp, aridity</dt>
<dd>
<p>Thirteen climatic variables. 
See Hancock et al. (2011) for details about these variables.</p>
</dd>
</dl>
<h3>Details</h3>

<p>The  response is binary so <code>method="PQL/L"</code> seems warranted (see Rousset and Ferdy, 2014).  
</p>


<h3>Source</h3>

<p>The data were retrieved from <code>http://bergelson.uchicago.edu/regmap-data/climate-genome-scan</code> on 22 February 2013 (they may no longer be available from there).
</p>


<h3>References</h3>

<p>Fournier-Level A, Korte A., Cooper M. D., Nordborg M., Schmitt J., Wilczek AM (2011). A map of local adaptation in Arabidopsis thaliana. Science 334: 86-89.
</p>
<p>Hancock, A. M., Brachi, B., Faure, N., Horton, M. W., Jarymowycz, L. B., Sperone, F. G., Toomajian, C., Roux, F., and Bergelson, J. 2011. 
Adaptation to climate across the Arabidopsis thaliana genome, Science 334: 83-86.
</p>
<p>Rousset F., Ferdy, J.-B. (2014) Testing environmental and genetic effects in the presence of spatial autocorrelation. Ecography, 37: 781-790.
<a href="https://doi.org/10.1111/ecog.00566">doi:10.1111/ecog.00566</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data("arabidopsis")
if (spaMM.getOption("example_maxtime")&gt;2.5) {
  fitme(cbind(pos1046738,1-pos1046738)~seasonal+Matern(1|LAT+LONG),
        fixed=list(rho=0.119278,nu=0.236990,lambda=8.599),
        family=binomial(),method="PQL/L",data=arabidopsis)
}
## The above 'fixed' values are deduced from the following fit:
if (spaMM.getOption("example_maxtime")&gt;46) {
  SNPfit &lt;- fitme(cbind(pos1046738,1-pos1046738)~seasonal+Matern(1|LAT+LONG),
              verbose=c(TRACE=TRUE), 
              family=binomial(),method="PQL/L",data=arabidopsis)
  summary(SNPfit) # p_v=-125.0392
}
</code></pre>


</div>