<div class="container">

<table style="width: 100%;"><tr>
<td>lassoSEM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
The Lasso penalty for SEM
</h2>

<h3>Description</h3>

<p>Upon lambda_max to lambda_min in 20 step, the function compute 5 fold CV 
to determine the optimal lambda for the data. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">	lassoSEM(Y, X, Missing, B, verbose = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>

<p>gene expression M by N matrix
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>The network node attribute matrix with dimension of M by N. Theoretically, X can be L by N matrix, with L being the total
node attributes. In current implementation, each node only allows one and only one attribute. <br>
If you have more than one attributes for some nodes,  please consider selecting the top one by either
correlation or principal component methods.  <br>
If for some nodes there is no attribute available, fill in the rows with all zeros.  See the yeast data 'yeast.rda' for example. <br>
X is normalized inside the function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Missing</code></td>
<td>

<p>missing data in Y
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>

<p>true network topology if available
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>describe the information output from -1 - 10, larger number means more output
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>the function perform CV and parameter inference, calculate power and FDR
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Bout</code></td>
<td>
<p>the matrix B from SEM</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fout</code></td>
<td>
<p>f: the weight for matrix X</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat</code></td>
<td>
<p>compute the power and FDR statistics if the ture topology is provided</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simTime</code></td>
<td>
<p>computational time</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Anhui Huang</p>


<h3>References</h3>

<p>1. Cai, X., Bazerque, J.A., and Giannakis, G.B. (2013). Inference of Gene Regulatory Networks with Sparse Structural Equation Models Exploiting Genetic Perturbations. PLoS Comput Biol 9, e1003068. <br>
2. Huang, A. (2014). "Sparse model learning for inferring genotype and phenotype associations." Ph.D Dissertation. University of Miami(1186).<br><br></p>


<h3>Examples</h3>

<pre><code class="language-R">	library(sparseSEM)
	data(B);
	data(Y);
	data(X);
	data(Missing);
	## Not run: OUT &lt;- lassoSEM(Y, X, Missing, B, verbose = 0); 

</code></pre>


</div>