<div class="container">

<table style="width: 100%;"><tr>
<td>trainDeconvModel</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Train deconvolution model for spatial transcriptomics data</h2>

<h3>Description</h3>

<p>Train a deep neural network model using training data from the
<code>SpatialDDLS</code> object. This model will be used to
deconvolute spatial transcriptomics data from the same biological context as
the single-cell RNA-seq data used to train it. In addition, the trained
model is evaluated using test data, and prediction results are obtained to
determine its performance (see <code>?calculateEvalMetrics</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">trainDeconvModel(
  object,
  type.data.train = "mixed",
  type.data.test = "mixed",
  batch.size = 64,
  num.epochs = 60,
  num.hidden.layers = 2,
  num.units = c(200, 200),
  activation.fun = "relu",
  dropout.rate = 0.25,
  loss = "kullback_leibler_divergence",
  metrics = c("accuracy", "mean_absolute_error", "categorical_accuracy"),
  normalize = TRUE,
  scaling = "standardize",
  norm.batch.layers = TRUE,
  custom.model = NULL,
  shuffle = TRUE,
  sc.downsampling = NULL,
  use.generator = FALSE,
  on.the.fly = FALSE,
  agg.function = "AddRawCount",
  threads = 1,
  view.metrics.plot = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p><code>SpatialDDLS</code> object with
<code>single.cell.real</code>/<code>single.cell.simul</code>, <code>prob.cell.types</code>,
and <code>mixed.profiles</code> slots (the last only if <code>on.the.fly =
  FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type.data.train</code></td>
<td>
<p>Type of profiles to be used for training. It can be
<code>'both'</code>, <code>'single-cell'</code> or <code>'mixed'</code> (<code>'mixed'</code> by
default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type.data.test</code></td>
<td>
<p>Type of profiles to be used for evaluation. It can be
<code>'both'</code>, <code>'single-cell'</code> or <code>'mixed'</code> (<code>'mixed'</code> by
default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch.size</code></td>
<td>
<p>Number of samples per gradient update (64 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.epochs</code></td>
<td>
<p>Number of epochs to train the model (60 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.hidden.layers</code></td>
<td>
<p>Number of hidden layers of the neural network (2 by
default). This number must be equal to the length of <code>num.units</code>
argument.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.units</code></td>
<td>
<p>Vector indicating the number of neurons per hidden layer
(<code>c(200, 200)</code> by default). The length of this vector must be equal to
the <code>num.hidden.layers</code> argument.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>activation.fun</code></td>
<td>
<p>Activation function (<code>'relu'</code> by default). See
the
<a href="https://tensorflow.rstudio.com/reference/keras/activation_relu.html">keras
documentation</a> to know available activation functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dropout.rate</code></td>
<td>
<p>Float between 0 and 1 indicating the fraction of
input neurons to be dropped in layer dropouts (0.25 by default). By
default, <span class="pkg">SpatialDDLS</span> implements 1 dropout layer per hidden layer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loss</code></td>
<td>
<p>Character indicating loss function selected for model training
(<code>'kullback_leibler_divergence'</code> by default). See the
<a href="https://tensorflow.rstudio.com/reference/keras/loss-functions.html">keras
documentation</a> to know available loss functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metrics</code></td>
<td>
<p>Vector of metrics used to assess model performance during
training and evaluation (<code>c("accuracy", "mean_absolute_error",
  "categorical_accuracy")</code> by default). See the
<a href="https://tensorflow.rstudio.com/reference/keras/metric_binary_accuracy.html">keras
documentation</a> to know available performance metrics.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize</code></td>
<td>
<p>Whether to normalize data using logCPM (<code>TRUE</code> by
default). This parameter is only considered when the method used to
simulate mixed transcriptional profiles (<code>simMixedProfiles</code>
function) was <code>"AddRawCount"</code>. Otherwise, data were already
normalized.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scaling</code></td>
<td>
<p>How to scale data before training. It can be:
<code>"standardize"</code> (values are centered around the mean with a unit
standard deviation), <code>"rescale"</code> (values are shifted and rescaled so
that they end up ranging between 0 and 1) or <code>"none"</code> (no
scaling is performed). <code>"standardize"</code> by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm.batch.layers</code></td>
<td>
<p>Whether to include batch normalization layers
between each hidden dense layer (<code>TRUE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>custom.model</code></td>
<td>
<p>It allows to use a custom neural network architecture. It
must be a <code>keras.engine.sequential.Sequential</code> object in which the
number of input neurons is equal to the number of considered
features/genes, and the number of output neurons is equal to the number of
cell types considered (<code>NULL</code> by default). If provided, the arguments
related to the neural network architecture will be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shuffle</code></td>
<td>
<p>Boolean indicating whether data will be shuffled (<code>TRUE</code>
by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sc.downsampling</code></td>
<td>
<p>It is only used if <code>type.data.train</code> is equal to
<code>'both'</code> or <code>'single-cell'</code>. It allows to set a maximum number of
single-cell profiles of a specific cell type for training to avoid
an unbalanced representation of classes (<code>NULL</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.generator</code></td>
<td>
<p>Boolean indicating whether to use generators during
training and test. Generators are automatically used when <code>on.the.fly
  = TRUE</code> or HDF5 files are used, but it can be activated by the user on
demand (<code>FALSE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>on.the.fly</code></td>
<td>
<p>Boolean indicating whether simulated data will be generated
'on the fly' during training (<code>FALSE</code> by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>agg.function</code></td>
<td>
<p>If <code>on.the.fly == TRUE</code>, function used to build
mixed transcriptional profiles. It may be:
</p>
 <ul>
<li> <p><code>"AddRawCount"</code> (by default): single-cell
profiles (raw counts) are added up across cells. Then, log-CPMs are
calculated. </p>
</li>
<li> <p><code>"MeanCPM"</code>: single-cell profiles (raw counts) are
transformed into logCPM and cross-cell averages are calculated.
</p>
</li>
<li> <p><code>"AddCPM"</code>: single-cell profiles (raw counts) are transformed
into CPMs and are added up across cells. Then, log-CPMs are calculated.</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threads</code></td>
<td>
<p>Number of threads used during simulation of mixed
transcriptional profiles if <code>on.the.fly = TRUE</code> (1 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>view.metrics.plot</code></td>
<td>
<p>Boolean indicating whether to show plots of loss and
evaluation metrics during training (<code>TRUE</code> by default). <span class="pkg">keras</span>
for R allows to see model progression during training if you are working in
RStudio.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Boolean indicating whether to display model progression during
training and model architecture information (<code>TRUE</code> by default).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><strong>Simulation of mixed transcriptional profiles 'on the fly'</strong>
</p>
<p><code>trainDeconvModel</code> can avoid storing simulated mixed spot profiles by
using the <code>on.the.fly</code> argument. This functionality aims at reducing the
the <code>simMixedProfiles</code> function's memory usage: simulated profiles are
built in each batch during training/evaluation.
</p>
<p><strong>Neural network architecture</strong>
</p>
<p>It is possible to change the model's architecture: number of hidden layers,
number of neurons for each hidden layer, dropout rate, activation function,
and loss function. For more customized models, it is possible to provide a
pre-built model through the <code>custom.model</code> argument (a
<code>keras.engine.sequential.Sequential</code> object) where it is necessary that
the number of input neurons is equal to the number of considered
features/genes, and the number of output neurons is equal to the number of
considered cell types.
</p>


<h3>Value</h3>

<p>A <code>SpatialDDLS</code> object with <code>trained.model</code>
slot containing a <code>DeconvDLModel</code> object. For more
information about the structure of this class, see
<code>?DeconvDLModel</code>.
</p>


<h3>See Also</h3>

<p><code>plotTrainingHistory</code> <code>deconvSpatialDDLS</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
set.seed(123)
sce &lt;- SingleCellExperiment::SingleCellExperiment(
  assays = list(
    counts = matrix(
      rpois(30, lambda = 5), nrow = 15, ncol = 10,
      dimnames = list(paste0("Gene", seq(15)), paste0("RHC", seq(10)))
    )
  ),
  colData = data.frame(
    Cell_ID = paste0("RHC", seq(10)),
    Cell_Type = sample(x = paste0("CellType", seq(2)), size = 10,
                       replace = TRUE)
  ),
  rowData = data.frame(
    Gene_ID = paste0("Gene", seq(15))
  )
)
SDDLS &lt;- createSpatialDDLSobject(
  sc.data = sce,
  sc.cell.ID.column = "Cell_ID",
  sc.gene.ID.column = "Gene_ID",
  sc.filt.genes.cluster = FALSE
)
SDDLS &lt;- genMixedCellProp(
  object = SDDLS,
  cell.ID.column = "Cell_ID",
  cell.type.column = "Cell_Type",
  num.sim.spots = 50,
  train.freq.cells = 2/3,
  train.freq.spots = 2/3,
  verbose = TRUE
)
SDDLS &lt;- simMixedProfiles(SDDLS)
SDDLS &lt;- trainDeconvModel(
  object = SDDLS,
  batch.size = 12,
  num.epochs = 5
)

  
</code></pre>


</div>