<div class="container">

<table style="width: 100%;"><tr>
<td>SpatialDDLS-Rpackage</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SpatialDDLS: an R package to deconvolute spatial transcriptomics data using
deep neural networks</h2>

<h3>Description</h3>

<p><span class="pkg">SpatialDDLS</span> is an R package that provides a neural network-based
solution for cell type deconvolution of spatial transcriptomics data. The
package takes advantage of single-cell RNA sequencing (scRNA-seq) data to
simulate mixed transcriptional profiles with known cell composition and train
fully-connected neural networks to predict the cell type composition of
spatial transcriptomics spots. The resulting trained models can be applied to
new spatial transcriptomics data to predict cell type proportions, allowing
for more accurate cell type identification and characterization of
spatially-resolved transcriptomic data. Finally, predictions are forced to keep
spatial consistency through a process we refer to as spatial regularization.
Overall, <span class="pkg">SpatialDDLS</span> is a powerful tool for cell type deconvolution in
spatial transcriptomics data, providing a reliable, fast and flexible
solution for researchers in the field. See Ma√±anes et al. (2024)
(<a href="https://doi.org/10.1093/bioinformatics/btae072">doi:10.1093/bioinformatics/btae072</a>) and some examples
(<a href="https://diegommcc.github.io/SpatialDDLS/">https://diegommcc.github.io/SpatialDDLS/</a>) for more details.
</p>


</div>