<div class="container">

<table style="width: 100%;"><tr>
<td>SpatialDDLS-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The SpatialDDLS Class</h2>

<h3>Description</h3>

<p>The <code>SpatialDDLS</code> object is the core of the
<span class="pkg">SpatialDDLS</span> package. This object stores different intermediate data
needed for the construction of new deconvolution models, the spatial
transcriptomics profiles to be deconvoluted, and the predicted cell type
proportions.
</p>


<h3>Details</h3>

<p>This object uses other classes to store different types of data generated
during the workflow: </p>
 <ul>
<li>
<p><code>SingleCellExperiment</code> class for single-cell RNA-Seq data
storage, using sparse matrix from the <span class="pkg">Matrix</span> package
(<code>dgCMatrix</code> class) or <code>HDF5Array</code> class in case of
using HDF5 files as back-end (see below for more information). </p>
</li>
<li>
<p><code>SpatialExperiment</code> class for spatial transcriptomics data
storage. </p>
</li>
<li> <p><code>ZinbModel</code> class with estimated parameters
for the simulation of new single-cell profiles. </p>
</li>
<li>
<p><code>SummarizedExperiment</code> class for simulated mixed
transcriptional profiles storage. </p>
</li>
<li> <p><code>PropCellTypes</code>
class for composition cell type matrices. See
<code>?PropCellTypes</code> for details. </p>
</li>
<li>
<p><code>DeconvDLModel</code> class to store information related to
deep neural network models. See <code>?DeconvDLModel</code> for
details. </p>
</li>
</ul>
<p>In order to provide a way to work with large amounts of data in
RAM-constrained machines, we provide the possibility of using HDF5 files as
back-end to store count matrices of both real and simulated single-cell
profiles by using the <span class="pkg">HDF5Array</span> and <span class="pkg">DelayedArray</span> classes from the
homonymous packages.
</p>


<h3>Slots</h3>


<dl>
<dt><code>single.cell.real</code></dt>
<dd>
<p>Real single-cell data stored in a
<code>SingleCellExperiment</code> object. The count matrix is stored either as
<code>dgCMatrix</code> or <code>HDF5Array</code> objects.</p>
</dd>
<dt><code>spatial.experiments</code></dt>
<dd>
<p>List of <code>SpatialExperiment</code>
objects to be deconvoluted.</p>
</dd>
<dt><code>zinb.params</code></dt>
<dd>
<p><code>ZinbModel</code> object with estimated
parameters for the simulation of new single-cell expression profiles.</p>
</dd>
<dt><code>single.cell.simul</code></dt>
<dd>
<p>Simulated single-cell expression profiles using the
ZINB-WaVE model.</p>
</dd>
<dt><code>prob.cell.types</code></dt>
<dd>
<p><code>PropCellTypes</code> class with cell
composition matrices built for the simulation of mixed transcriptional
profiles with known cell composition.</p>
</dd>
<dt><code>mixed.profiles</code></dt>
<dd>
<p>List of simulated train and test mixed transcriptional
profiles. Each entry is a <code>SummarizedExperiment</code> object.
Count matrices can be stored as <code>HDF5Array</code> objects using HDF5 files
as back-end in case of RAM limitations.</p>
</dd>
<dt><code>trained.model</code></dt>
<dd>
<p><code>DeconvDLModel</code> object with
information related to the deconvolution model. See
<code>?DeconvDLModel</code> for more details.</p>
</dd>
<dt><code>deconv.spots</code></dt>
<dd>
<p>Deconvolution results. It consists of a list where each
element corresponds to the results for each
<code>SpatialExperiment</code> object contained in the
<code>spatial.experiments</code> slot.</p>
</dd>
<dt><code>project</code></dt>
<dd>
<p>Name of the project.</p>
</dd>
<dt><code>version</code></dt>
<dd>
<p>Version of <span class="pkg">SpatialDDLS</span> this object was built under.</p>
</dd>
</dl>
</div>