<div class="container">

<table style="width: 100%;"><tr>
<td>plotSpatialGeneExpr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot normalized gene expression data (logCPM) in spatial coordinates</h2>

<h3>Description</h3>

<p>Color spots on the spatial coordinates according to the logCPM values of a
particular gene.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotSpatialGeneExpr(
  object,
  index.st,
  gene,
  colors = "spectral",
  size.point = 1,
  title = NULL,
  theme = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A <code>SpatialDDLS</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index.st</code></td>
<td>
<p>Index of the spatial transcriptomics data to be plotted. It
can be either a position or a name if a named list of
<code>SpatialExperiment</code> objects was provided.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p>Gene to color spots by.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colors</code></td>
<td>
<p>Color scale to be used. It can be <code>"blues"</code> or
<code>"spectral"</code> (the latter by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size.point</code></td>
<td>
<p>Size of points (0.1 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>Title of plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theme</code></td>
<td>
<p><span class="pkg">ggplot2</span> theme.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>See Also</h3>

<p><code>interGradientsDL</code> <code>topGradientsCellType</code>
</p>


</div>