<div class="container">

<table style="width: 100%;"><tr>
<td>loadSTProfiles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Loads spatial transcriptomics data into a SpatialDDLS object</h2>

<h3>Description</h3>

<p>This function loads a <code>SpatialExperiment</code> object (or a
list with several <code>SpatialExperiment</code> objects) into a
<code>SpatialDDLS</code> object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">loadSTProfiles(
  object,
  st.data,
  st.spot.ID.column,
  st.gene.ID.column,
  st.min.counts = 0,
  st.min.spots = 0,
  st.n.slides = 3,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>A <code>SpatialDDLS</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>st.data</code></td>
<td>
<p>A <code>SpatialExperiment</code> object (or a list
with several <code>SpatialExperiment</code> objects) to be
deconvoluted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>st.spot.ID.column</code></td>
<td>
<p>Name or number of the column in spots metadata
corresponding to spot names in the expression matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>st.gene.ID.column</code></td>
<td>
<p>Name or number of the column in genes metadata
corresponding to names used for features/genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>st.min.counts</code></td>
<td>
<p>Minimum gene counts to filter (0 by default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>st.min.spots</code></td>
<td>
<p>Minimum of spots with more than <code>min.counts</code> (0 by
default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>st.n.slides</code></td>
<td>
<p>Minimum number of slides
(<code>SpatialExperiment</code> objects) in which a gene has to be
expressed in order to keep it. This parameter is applicable only when
multiple <code>SpatialExperiment</code> objects are provided. Genes
not present in at least <code>st.n.slides</code> will be discarded. If no
filtering is desired, set <code>st.n.slides = 1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Show informative messages during execution (<code>TRUE</code> by
default).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>It is recommended to perform this step when creating the
<code>SpatialDDLS</code> object using the
<code>createSpatialDDLSobject</code> function in order to only keep genes
shared between the spatial transcriptomics and the single-cell
transcriptomics data used as reference. In addition, please, make sure the
gene identifiers used the spatial and single-cell transcriptomics data are
consistent.
</p>


<h3>Value</h3>

<p>A <code>SpatialDDLS</code> object with the provided spatial
trainscriptomics data loaded into the <code>spatial.experiments</code> slot.
</p>


<h3>See Also</h3>

<p><code>createSpatialDDLSobject</code> <code>trainDeconvModel</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
set.seed(123)
sce &lt;- SingleCellExperiment::SingleCellExperiment(
  assays = list(
    counts = matrix(
      rpois(100, lambda = 5), nrow = 40, ncol = 30,
      dimnames = list(paste0("Gene", seq(40)), paste0("RHC", seq(30)))
    )
  ),
  colData = data.frame(
    Cell_ID = paste0("RHC", seq(30)),
    Cell_Type = sample(x = paste0("CellType", seq(4)), size = 30,
                       replace = TRUE)
  ),
  rowData = data.frame(
    Gene_ID = paste0("Gene", seq(40))
  )
)
SDDLS &lt;- createSpatialDDLSobject(
  sc.data = sce,
  sc.cell.ID.column = "Cell_ID",
  sc.gene.ID.column = "Gene_ID",
  sc.filt.genes.cluster = FALSE
)

## simulating a SpatialExperiment object
counts &lt;- matrix(rpois(30, lambda = 5), ncol = 6)
rownames(counts) &lt;- paste0("Gene", 1:5)
colnames(counts) &lt;- paste0("Spot", 1:6)
coordinates &lt;- matrix(
  c(1, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 3), ncol = 2
)
ste &lt;- SpatialExperiment::SpatialExperiment(
  assays = list(counts = as.matrix(counts)),
  rowData = data.frame(Gene_ID = paste0("Gene", 1:5)),
  colData = data.frame(Cell_ID = paste0("Spot", 1:6)),
  spatialCoords = coordinates
)

## previous SpatialDDLS object
SDDLS &lt;- loadSTProfiles(
  object = SDDLS,
  st.data = ste,
  st.spot.ID.column = "Cell_ID",
  st.gene.ID.column = "Gene_ID"
)

  
</code></pre>


</div>