<div class="container">

<table style="width: 100%;"><tr>
<td>knn_graph</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>K-nearest neighbors graph</h2>

<h3>Description</h3>

<p>Create a directed sparse graph with edges to each nodes <code>k</code> nearest
neighbors. Nearness is calculated as the inverse of the euclidean distance
between two columns.
</p>


<h3>Usage</h3>

<pre><code class="language-R">knn_graph(mat, k, weighted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>mat</code></td>
<td>
<p>A matrix to be compared column-by-column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>How many nearest neighbors to collect.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weighted</code></td>
<td>
<p>By default, a binary edge is made between a node and each of
it's <code>k</code> closest nodes. Set <code>weighted</code> to <code>TRUE</code> to weigh each edge by the
similarity (inverse of euclidean distance).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A directed sparse adjacency matrix with <code>k * ncol(mat)</code> nonzero
edges. Each column has k edges connected to the k closest columns (not
including itself).
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Simple random graph
mat &lt;- matrix(runif(100) &gt; 0.75, nrow = 5)
knn_graph(mat, 3)

## Don't run because loading data is slow.

if (requireNamespace("scRNAseq") &amp;&amp;
  requireNamespace("SummarizedExperiment")) {
  # Single Cell RNA data
  library(Matrix)

  expression &lt;- scRNAseq::FletcherOlfactoryData()
  cell_types &lt;- expression$cluster_id

  ## Filter genes with low expression. Remove any genes with less than 10
  ## cells with with any reads.
  counts &lt;- SummarizedExperiment::assay(expression, "counts")
  indices &lt;- rowSums(counts &gt; 0) &gt; 10
  counts &lt;- counts[indices, ]

  ## Normalize by shifted logarithm
  target &lt;- median(colSums(counts))
  size_factors &lt;- colSums(counts) / target
  counts_norm &lt;- log(t(t(counts) / size_factors + 1))

  ## Dimension reduction
  counts_norm &lt;- t(prcomp(t(counts_norm), scale. = FALSE)$x)[1:50, ]

  adj &lt;- knn_graph(counts_norm, 10)
}

</code></pre>


</div>