<div class="container">

<table style="width: 100%;"><tr>
<td>speck</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Abundance estimation for single cell RNA-sequencing (scRNA-seq) data.</h2>

<h3>Description</h3>

<p>Performs normalization, reduced rank reconstruction (RRR) and thresholding for a <code class="reqn">m x n</code> scRNA-seq matrix
with <code class="reqn">m</code> samples and <code class="reqn">n</code> genes. The <code>speck()</code> function calls the
<code>randomizedRRR()</code> function on the scRNA-seq matrix. Thresholding is next
applied to each gene from the <code class="reqn">m x n</code> RRR matrix using the <code>ckmeansThreshold()</code>
function, resulting in a <code class="reqn">m x n</code> thresholded matrix. See documentation for the <code>randomizedRRR()</code> and
<code>ckmeansThreshold()</code> functions for individual implementation details.
</p>


<h3>Usage</h3>

<pre><code class="language-R">speck(
  counts.matrix,
  rank.range.end = 100,
  min.consec.diff = 0.01,
  rep.consec.diff = 2,
  manual.rank = NULL,
  max.num.clusters = 4,
  seed.rsvd = 1,
  seed.ckmeans = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>counts.matrix</code></td>
<td>
<p><code class="reqn">m x n</code> scRNA-seq counts matrix with <code class="reqn">m</code> samples
and <code class="reqn">n</code> genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rank.range.end</code></td>
<td>
<p>Upper value of the rank for RRR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.consec.diff</code></td>
<td>
<p>Minimum difference in the rate of change between a pair of successive standard deviation estimate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep.consec.diff</code></td>
<td>
<p>Frequency of the minimum difference in the rate of change between a pair of successive standard deviation estimate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>manual.rank</code></td>
<td>
<p>Optional, user-specified upper value of the rank used
for RRR as an alternative to automatically computed rank.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.num.clusters</code></td>
<td>
<p>Maximum number of clusters for computation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed.rsvd</code></td>
<td>
<p>Seed specified to ensure reproducibility of the RRR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed.ckmeans</code></td>
<td>
<p>Seed specified to ensure reproducibility of the clustered thresholding.</p>
</td>
</tr>
</table>
<h3>Value</h3>


<ul>
<li>
<p> thresholded.mat - A <code class="reqn">m x n</code> thresholded RRR matrix with <code class="reqn">m</code> samples and <code class="reqn">n</code> genes.
</p>
</li>
<li>
<p> rrr.mat - A <code class="reqn">m x n</code> RRR matrix with <code class="reqn">m</code> samples and <code class="reqn">n</code> genes.
</p>
</li>
<li>
<p> rrr.rank - Automatically computed rank.
</p>
</li>
<li>
<p> component.stdev - A vector corresponding to standard deviations of non-centered sample principal components.
</p>
</li>
<li>
<p> clust.num - A vector of length <code class="reqn">n</code> indicating the number of clusters identified by the
<code>Ckmeans.1d.dp()</code> algorithm for each gene.
</p>
</li>
<li>
<p> clust.max.prop - A vector of length <code class="reqn">n</code> indicating the proportion of samples with the
specified maximum number of clusters for each gene.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">set.seed(10)
data.mat &lt;- matrix(data = rbinom(n = 18400, size = 230, prob = 0.01), nrow = 80)
speck.full &lt;- speck(counts.matrix = data.mat, rank.range.end = 60,
min.consec.diff = 0.01, rep.consec.diff = 2,
manual.rank = NULL, max.num.clusters = 4,
seed.rsvd = 1, seed.ckmeans = 2)
print(speck.full$component.stdev)
print(speck.full$rrr.rank)
head(speck.full$clust.num); table(speck.full$clust.num)
head(speck.full$clust.max.prop); table(speck.full$clust.max.prop)
speck.output &lt;- speck.full$thresholded.mat
dim(speck.output); str(speck.output)

</code></pre>


</div>