<div class="container">

<table style="width: 100%;"><tr>
<td>oligoProfile</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Oligo Profiles and Oligo Profile Correlation Plots of Nucleotide Sequences
</h2>

<h3>Description</h3>

<p>Construct a k-mer oligo profile of a nucleotide sequence and print such a 
profile or its reverse complement.  There is also a plot function for producing 
plots of the profile or its reverse complement and for comparing primary and 
complementary strand profiles.
</p>


<h3>Usage</h3>

<pre><code class="language-R">oligoProfile(x, k, content=c("dna", "rna"), 
case=c("lower", "upper", "as is"), circular=TRUE, disambiguate=TRUE, 
plot=TRUE, ...)
## S3 method for class 'OligoProfile'
plot(x, which=1L, units=c("percentage", "count", "proportion"),
 main=NULL, xlab=NULL, ylab=NULL, ...)
## S3 method for class 'OligoProfile'
print(x, which=1L, units=c("percentage", "count", "proportion"), 
digits=switch(units, percentage=3L, count=NULL, proportion=3L), ...)


</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>a character vector or an object that can be coersed to a character vector.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>

<p>the k-mer profile to produce.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>content</code></td>
<td>

<p>The content type (“<code>dna</code>” or “<code>rna</code>”) of the input 
sequence.  <code>oligoProfile</code> can often detect this automatically based on 
the presence/absence of <code>t</code>'s or <code>u</code>'s, but if neither is present, 
the <code>content</code> argument is consulted.  The default value is 
“<code>dna</code>”.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>case</code></td>
<td>

<p>determines how labels for the array should be generated: in
lowercase, in uppercase or left as is, in which case labels such as “b”
and “B” will be seen as distinct symbols and counted separately.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circular</code></td>
<td>

<p>Determines if the vector should be treated as circular or not.  The default is 
<code>TRUE</code>, meaning that the start and end of the sequence will be joined 
together for the purpose of counting.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disambiguate</code></td>
<td>

<p>if set to the default of <code>true</code>, makes the input sequence unambiguous 
before generating the profile.  Otherwise, ambiguous symbols are treated like 
any other symbols and k-mer counts including them will be computed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>

<p>should a plot of the profile be produced?  The default is <code>TRUE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>which</code></td>
<td>

<p>For <code>print</code>, specifies whether to display the profile for the sequence used 
to generate the OligoProfile object (<code>1</code>) or the profile of its reverse 
complement (<code>2</code>).
</p>
<p>For the <code>plot</code> method, <code>which</code> determines what should be plotted. 
Values <code>1</code> and <code>2</code> cause the profile for the original sequence 
(primary strand) or its reverse complement (complementary strand) to be plotted, 
respectively.  Specifying <code>which=3</code> will plot a comparison of the two 
profiles which can be used to assess compliance with Chargaff's second parity 
rule.
</p>
<p>the <code>which</code> argument may also be specified when calling 
<code>oligoProfile</code>, in which case it will be passed on to the <code>plot</code> 
method if the <code>plot</code> argument is set to <code>TRUE</code>. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>units</code></td>
<td>

<p>The oligo profiles can be scaled according to three different units for 
presentation on plots: “<code>percentage</code>”, “<code>count</code>” or 
“<code>proportion</code>”. The default is “<code>percentage</code>”.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>

<p>The title of the plot.  See <code>plot.default</code>.  If not specified, an 
appropriate title is automatically generated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>

<p>a label for the x-axis of the plot.  See <code>plot.default</code>.  If not specified, an 
appropriate label is automatically generated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>

<p>a label for the y-axis of the plot.  See <code>plot.default</code>.  If not specified, an 
appropriate label is automatically generated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digits</code></td>
<td>

<p>The number of significant digits to print.  The default is <code>0L</code> when <code>units</code> is set to “<code>count</code>” and <code>3L</code> otherwise.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>arguments to be passed from or to other functions



</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function returns the oligo profile for a sequence in an <code>OligoProfile</code> 
object, which is printed on screen if the <code>plot</code> parameter is <code>FALSE</code>. 
An oligo profile is simply the counts of all <code>k</code>-mers in a sequence for 
some specified value of <code>k</code>.  
</p>
<p>By default, <code>oligoProfile</code> produces a plot of the oligo profile expressed 
in terms of percentages.  The <code>plot</code> argument determines if the plot 
should be generated or not and plotting parameters such as <code>main</code>, 
<code>sub</code>, etc., may be passed as arguments to the function when <code>plot</code> is 
<code>TRUE</code>.  
</p>
<p>The <code>plot</code> method, either called directly or indirectly via the 
<code>oligoProfile</code> function, can produce either the oligo profile of <code>x</code> 
(<code>which = 1</code>), the oligo profile of its reverse complement (<code>which = 
2</code>), or an interstrand k-mer correlation plot comparing the k-oligo profile 
of<code>x</code> with that of its reverse complement (<code>which = 3)</code>.  Such 
</p>
<p>Correlation plots effectively show the relationship between k-mers on the 
primary and complementary strands in a DNA duplex and can be used to assess 
compliance with CSPR.  More precisely, one would conclude that a genomic 
sequence complies with CSPR if all the plotted points lie on a diagonal line 
running from the bottom-left corner to the top-right corner of the graph.
</p>


<h3>Value</h3>

<p>A list with class “OligoProfile” containing the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>a name to identify the source of the profile.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>wordLength</code></td>
<td>
<p>the value of k used to derive the k-mer profile.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>content</code></td>
<td>
<p>indicates if the profile pertains to a DNA or RNA sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>case</code></td>
<td>
<p>indicates how the case of letters was processed before 
producing the profile.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circular</code></td>
<td>
<p>indicates whether or not the sequence was considered circular for the purpose of producing the profile.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disambiguate</code></td>
<td>
<p>indicates if the sequence was made unambiguous before 
producing the profile.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>profile</code></td>
<td>
<p>a vector containing the raw counts (frequencies) of all k-mers.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Andrew Hart and Servet Martínez
</p>


<h3>References</h3>

<p>Albrecht-Buehler, G. (2006)  Asymptotically increasing compliance of genomes 
with Chargaff's second parity rules through inversions and inverted 
transpositions. <em>PNAS</em> <b>103(47)</b>, 17828–17833.
</p>


<h3>See Also</h3>

<p><code>pair.counts</code>, <code>triple.counts</code>, 
<code>quadruple.counts</code>, <code>cylinder.counts</code>,
<code>array2vector</code>, <code>table2vector</code>, <code>disambiguate</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(nanoarchaeum)
#Get the 3-oligo profile of Nanoarchaeum without plotting it
nano.prof &lt;- oligoProfile(nanoarchaeum, 3, plot=FALSE)
nano.prof #print oligo profile as percentages
print(nano.prof, units="count") #print oligo profile as counts
plot(nano.prof) #oligo profile plotted as percentages
plot(nano.prof, units="count") #plot it as counts

#plot the 2-oligo profile of Nanoarchaeum as proportions
oligoProfile(nanoarchaeum, k=3, units="proportion")
</code></pre>


</div>