<div class="container">

<table style="width: 100%;"><tr>
<td>ag.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Test of Purine-Pyrimidine Parity Based on Purine Count</h2>

<h3>Description</h3>

<p>Performs a test proposed by Hart and Martínez (2011) for the equivalence of the
relative frequencies of purines (<code class="reqn">A+G</code>) and pyrimidines (<code class="reqn">C+T</code>) in DNA
sequences.  It does this by checking whether or not the mononucleotide
frequencies of a DNA sequence satisfy the relationship A+G=C+T.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ag.test(x, type=c("interval", "simplex"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>either a vector containing the relative frequencies of each of the 4 nucleotides 
A, C, G, T, a character vector representing a DNA sequence in which each element contains a single nucleotide, or
a DNA sequence stored using the SeqFastadna class from the <span class="pkg">seqinr</span> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Specifies one of two possible tests to perform, both of which are
based on the same test statistic, but assuming different forms of the Dirichlet
distribution under the null. “‘<span class="samp">⁠simplex⁠</span>’” assumes a Dirichlet(1,1,1,1) distribution
on the 3-simplex while “‘<span class="samp">⁠interval⁠</span>’” assumes a Dirichlet(1,1) (uniform)
distribution on the unit interval.  The default is “‘<span class="samp">⁠interval⁠</span>’”.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The first argument may be a character vector representing a DNA sequence, a DNA
sequence represented using the SeqFastadna class from the <span class="pkg">seqinr</span> package,
or a vector containing the relative frequencies of the A, C, G and T nucleic
acids.
</p>
<p>Let A, C, G and T denote the relative frequencies of the nucleotide bases
appearing in  a DNA sequence. This function carries out a statistical hypothesis
test that the relative frequencies satisfy the relation <code class="reqn">A+G=C+T</code>, or that
purines <code class="reqn">\{A, G\}</code> occur equally as often as pyrimidines <code class="reqn">\{C,T\}</code> in a DNA sequence.
The relationship can be rewritten as <code class="reqn">A-T=C-G</code>, from which it is easy to see
that the property being tested is a generalisation of Chargaff's second parity
rule for mononucleotides, which states that <code class="reqn">A=T</code> and <code class="reqn">C=G</code>. The test is
set up as follows:
</p>
<p><code class="reqn">H_0</code>: <code class="reqn">A+G \neq C+T</code> <br><code class="reqn">H_1</code>: <code class="reqn">A+G = C+T</code>
</p>
<p>If ‘<span class="samp">⁠type⁠</span>’ is set to “‘<span class="samp">⁠simplex⁠</span>’”, the vector <code class="reqn">(A,C,G,T)</code> is
assumed to come from a Dirichlet(1,1,1,1) distribution on the 3-simplex under
the null hypothesis.  Otherwise, if ‘<span class="samp">⁠type⁠</span>’ is set to
“‘<span class="samp">⁠interval⁠</span>’”, it is assumed under the null hypothesis that
<code class="reqn">(A+G,C+T)</code> ~ Dirichlet(1,1) or, in other words, <code class="reqn">A+G</code> and <code class="reqn">C+T</code> are
uniformly distributed on the unit interval and satisfy <code class="reqn">A+G+C+T=1</code>.
</p>
<p>In both cases, the test statistic is <code class="reqn">\eta_V^* = |A+G-0.5|</code>.
</p>


<h3>Value</h3>

<p>A list with class "htest.ext" containing the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>statistic</code></td>
<td>
<p>the value of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>the p-value of the test. Only included if ‘<span class="samp">⁠no.p.value⁠</span>’ is ‘<span class="samp">⁠FALSE⁠</span>’.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a character string indicating what type of test was performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.name</code></td>
<td>
<p>a character string giving the name of the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate</code></td>
<td>
<p>the probability vector used to derive the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat.desc</code></td>
<td>
<p>a brief description of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null</code></td>
<td>
<p>the null hypothesis (<code class="reqn">H_0</code>) of the test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p>the alternative hypothesis (<code class="reqn">H_1</code>) of the test.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Andrew Hart and Servet Martínez
</p>


<h3>References</h3>

<p>Hart, A.G. and Martínez, S. (2011)
Statistical testing of Chargaff's second parity rule in bacterial genome sequences.
<em>Stoch. Models</em> <b>27(2)</b>, 1–46.
</p>


<h3>See Also</h3>

<p><code>chargaff0.test</code>, <code>chargaff1.test</code>, 
<code>chargaff2.test</code>, <code>agct.test</code>,
<code>chargaff.gibbs.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Demonstration on real viral sequence
data(pieris)
ag.test(pieris)
ag.test(pieris, type="simplex")

#Simulate synthetic DNA sequence that does not exhibit Purine-Pyrimidine parity
trans.mat &lt;- matrix(c(.4, .1, .4, .1, .2, .1, .6, .1, .4, .1, .3, .2, .1, .2, .4, .3), 
ncol=4, byrow=TRUE)
seq &lt;- simulateMarkovChain(500000, trans.mat, states=c("a", "c", "g", "t"))
ag.test(seq)
</code></pre>


</div>