<div class="container">

<table style="width: 100%;"><tr>
<td>chargaff1.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Matrix Test of CSPR for Mononucleotides</h2>

<h3>Description</h3>

<p>Performs a test of Chargaff's second parity rule (CSPR) for mononucleotides on a genomic sequence using a 4X4 stochastic matrix 
estimated from the sequence.  The test was proposed by
Hart and Martínez (2011).
</p>


<h3>Usage</h3>

<pre><code class="language-R">chargaff1.test(x, alg=c("table", "simulate", "upper"), n, no.p.value=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>either a vector containing the relative frequencies of each of the 4 nucleotides 
A, C, G, T, a character vector representing a DNA sequence in which each element contains a single nucleotide, or
a DNA sequence stored using the SeqFastadna class from the <span class="pkg">seqinr</span> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alg</code></td>
<td>
<p>the algorithm for computing the p-value.  If set to “‘<span class="samp">⁠simulate⁠</span>’”, the
p-value is obtained via Monte Carlo simulation.  If set to “‘<span class="samp">⁠upper⁠</span>’”, an analytic upper
bound on the p-value is computed.  “‘<span class="samp">⁠upper⁠</span>’” are based on formulae
in Hart and Martínez (2011).  If
‘<span class="samp">⁠type⁠</span>’ is specified as “‘<span class="samp">⁠table⁠</span>’” (the default value),the p-value for the test is
obtained from a linear interpolation of a look-up table.  See the note below for further details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The number of replications to use for Monte Carlo simulation.  
If computationally feasible, a value &gt;= 10000000 is recommended.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.p.value</code></td>
<td>
<p>If ‘<span class="samp">⁠TRUE⁠</span>’, do not compute the p-value.  
The default is ‘<span class="samp">⁠FALSE⁠</span>’.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The first argument may be a character vector representing a DNA sequence, a DNA
sequence represented using the SeqFastadna class from the <span class="pkg">seqinr</span> package,
or a vector containing the relative frequencies of the A, C, G and T nucleic
acids.
</p>
<p>This function performs a test of Chargaff's second parity rule for
mononucleotides based on a 4X4 stochastic matrix <code class="reqn">P</code> estimated from the
empirical dinucleotide distribution of a genomic sequence .  The <code class="reqn">a,b)</code>
entry of <code class="reqn">P</code> gives the empirical probability (relative frequency) that a
nucleotide <code class="reqn">a</code> is followed by a nucleotide <code class="reqn">b</code> in the sequence. The test
is set up as follows:
</p>
<p><code class="reqn">H_0</code>:  the sequence (or matrix <code class="reqn">P</code>) does not comply with CSPR for mononucleotides <br><code class="reqn">H_1</code>:  the sequence (or matrix <code class="reqn">P</code>) complies with CSPR for mononucleotides
</p>


<h3>Value</h3>

<p>A list with class "htest.ext" containing the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>statistic</code></td>
<td>
<p>the value of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>the p-value of the test. Only included if no.p.value is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a character string indicating what type of test was performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.name</code></td>
<td>
<p>a character string giving the name of the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f</code></td>
<td>
<p>the 2-element vector used in calculating the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate</code></td>
<td>
<p>the stochastic matrix <code class="reqn">P</code> used to derive the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat.desc</code></td>
<td>
<p>a brief description of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null</code></td>
<td>
<p>the null hypothesis (<code class="reqn">H_0</code>) of the test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p>the alternative hypothesis (<code class="reqn">H_1</code>) of the test.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Currently, the look-up table that is employed when ‘<span class="samp">⁠alg⁠</span>’ is set to
“‘<span class="samp">⁠table⁠</span>’” does not provide an accurate p-value when the statistic is
smaller than 0.00806.  Care should be taken when adjusting p-values for multiple testing.
</p>
<p>The algebraically computed upper bound on the p-value obtained when ‘<span class="samp">⁠alg⁠</span>’
is set to “‘<span class="samp">⁠upper⁠</span>’” is not very tight and not suitable for real-
world applications.
</p>
<p>‘<span class="samp">⁠no.p.value⁠</span>’ suppresses computation of the p-value when it is set to
‘<span class="samp">⁠TRUE⁠</span>’.  This may be useful wen using this function to help simulate the
test statistic.
</p>


<h3>Author(s)</h3>

<p>Andrew Hart and Servet Martínez
</p>


<h3>References</h3>

<p>Hart, A.G. and Martínez, S. (2011)
Statistical testing of Chargaff's second parity rule in bacterial genome sequences.
<em>Stoch. Models</em> <b>27(2)</b>, 1–46.
</p>


<h3>See Also</h3>

<p><code>chargaff0.test</code>, <code>chargaff2.test</code>, 
<code>agct.test</code>, <code>ag.test</code>,
<code>chargaff.gibbs.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Demonstration on real bacterial sequence
data(nanoarchaeum)
chargaff1.test(nanoarchaeum)

#Simulate synthetic DNA sequence that does not satisfy Chargaff's second parity rule
trans.mat &lt;- matrix(c(.4, .1, .4, .1, .2, .1, .6, .1, .4, .1, .3, .2, .1, .2, .4, .3), 
ncol=4, byrow=TRUE)
seq &lt;- simulateMarkovChain(500000, trans.mat, states=c("a", "c", "g", "t"))
chargaff1.test(seq)
</code></pre>


</div>