<div class="container">

<table style="width: 100%;"><tr>
<td>agct.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Test of Purine-Pyrimidine Parity Based on Euclidean distance</h2>

<h3>Description</h3>

<p>Performs a test proposed by Hart and Martínez (2011) for the equivalence of the
relative frequencies of purines (<code class="reqn">A+G</code>) and pyrimidines (<code class="reqn">C+T</code>) in DNA
sequences.  It does this by checking whether or not the mononucleotide
frequencies of a DNA sequence satisfy the relationship A+G=C+T.
</p>


<h3>Usage</h3>

<pre><code class="language-R">agct.test(x, alg=c("exact", "simulate", "lower", "Lower", "upper"), n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>either a vector containing the relative frequencies of each of the 4 nucleotides 
A, C, G, T, a character vector representing a DNA sequence in which each element contains a single nucleotide, or
a DNA sequence stored using the SeqFastadna class from the <span class="pkg">seqinr</span> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alg</code></td>
<td>
<p>the algorithm for computing the p-value.  If set to “‘<span class="samp">⁠simulate⁠</span>’”, the
p-value is obtained via Monte Carlo simulation.  If set to “‘<span class="samp">⁠lower⁠</span>’”, an analytic
lower bound on the p-value is computed.  If set to “‘<span class="samp">⁠upper⁠</span>’”, an analytic upper
bound on the p-value is computed.  “‘<span class="samp">⁠lower⁠</span>’” and “‘<span class="samp">⁠upper⁠</span>’” are based on formulae
in Hart and Martínez (2011).  a Tighter (though unpublished) lower bound on the p-value  may be obtained by specifying “‘<span class="samp">⁠Lower⁠</span>’”. If
‘<span class="samp">⁠alg⁠</span>’ is specified as “‘<span class="samp">⁠exact⁠</span>’” (the default value), the p-value for the test is
computed exactly.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The number of replications to use for Monte Carlo simulation.  
If computationally feasible, a value &gt;= 10000000 is recommended.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The first argument may be a character vector representing a DNA sequence, a DNA
sequence represented using the SeqFastadna class from the <span class="pkg">seqinr</span> package,
or a vector containing the relative frequencies of the A, C, G and T nucleic
acids.
</p>
<p>Let A, C, G and T denote the relative frequencies of the nucleotide bases
appearing in  a DNA sequence. This function carries out a statistical hypothesis
test that the relative frequencies satisfy the relation <code class="reqn">A+G=C+T</code>, or that
purines <code class="reqn">\{A, G\}</code> occur equally as often as pyrimidines <code class="reqn">\{C,T\}</code> in a DNA sequence.
The relationship can be rewritten as <code class="reqn">A-T=C-G</code>, from which it is easy to see
that the property being tested is a generalisation of Chargaff's second parity
rule for mononucleotides, which states that <code class="reqn">A=T</code> and <code class="reqn">C=G</code>. The test is
set up as follows:
</p>
<p><code class="reqn">H_0</code>: <code class="reqn">A+G \neq C+T</code> <br><code class="reqn">H_1</code>: <code class="reqn">A+G = C+T</code>
</p>
<p>The vector <code class="reqn">(A,C,G,T)</code> is assumed to come from a Dirichlet(1,1,1,1)
distribution on the 3-simplex under the null hypothesis.
</p>
<p>The test statistic <code class="reqn">\eta_V</code> is the Euclidean distance from the
relative frequency vector <code class="reqn">(A,C,G,T)</code> to the closest point in the square set
<code class="reqn">\theta_V=\{(x,y,1/2-x,1/2-y) : 0 &lt;= x,y &lt;= 1/2\}</code>, which divides the 3-simplex into two equal parts.
<code class="reqn">\eta_V</code> lies in the range <code class="reqn">[0,\sqrt{3/8}]</code>.
</p>


<h3>Value</h3>

<p>A list with class "htest.ext" containing the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>statistic</code></td>
<td>
<p>the value of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>the p-value of the test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a character string indicating what type of test was performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.name</code></td>
<td>
<p>a character string giving the name of the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate</code></td>
<td>
<p>the probability vector used to derive the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat.desc</code></td>
<td>
<p>a brief description of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null</code></td>
<td>
<p>the null hypothesis (<code class="reqn">H_0</code>) of the test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p>the alternative hypothesis (<code class="reqn">H_1</code>) of the test.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p><code>agct.test(x, alg="upper")</code> is equivalent to <code>ag.test(x,
alg="simplex")</code> except that the p-value computed using the formula for
‘<span class="samp">⁠alg="upper"⁠</span>’ is exact for the test statistic <code class="reqn">\eta_V^*</code> used in
<code>ag.test</code>, whereas it is merely an upper bound on the p-value for
<code class="reqn">\eta_V</code>.
</p>


<h3>Author(s)</h3>

<p>Andrew Hart and Servet Martínez
</p>


<h3>References</h3>

<p>Hart, A.G. and Martínez, S. (2011)
Statistical testing of Chargaff's second parity rule in bacterial genome sequences.
<em>Stoch. Models</em> <b>27(2)</b>, 1–46.
</p>


<h3>See Also</h3>

<p><code>chargaff0.test</code>, <code>chargaff1.test</code>, 
<code>chargaff2.test</code>, <code>ag.test</code>,
<code>chargaff.gibbs.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Demonstration on real viral sequence
data(pieris)
agct.test(pieris)

#Simulate synthetic DNA sequence that does not exhibit Purine-Pyrimidine parity
trans.mat &lt;- matrix(c(.4, .1, .4, .1, .2, .1, .6, .1, .4, .1, .3, .2, .1, .2, .4, .3), 
ncol=4, byrow=TRUE)
seq &lt;- simulateMarkovChain(500000, trans.mat, states=c("a", "c", "g", "t"))
agct.test(seq)
</code></pre>


</div>