<div class="container">

<table style="width: 100%;"><tr>
<td>chargaff0.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Vector Test of Chargaff's Second Parity Rule (CSPR) for Mononucleotides</h2>

<h3>Description</h3>

<p>Performs the vector test of Chargaff's second parity rule (CSPR) for mononucleotides proposed in Hart and Martínez (2001).
</p>


<h3>Usage</h3>

<pre><code class="language-R">chargaff0.test(x, alg=c("exact", "simulate", "lower", "upper", "Lower", "Upper"), n, 
no.p.value=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>either a vector containing the relative frequencies of each of the 4 nucleotides 
A, C, G, T, a character vector representing a DNA sequence in which each element contains a single nucleotide, or
a DNA sequence stored using the SeqFastadna class from the <span class="pkg">seqinr</span> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alg</code></td>
<td>
<p>the algorithm for computing the p-value.  If set to “‘<span class="samp">⁠simulate⁠</span>’”, the
p-value is obtained via Monte Carlo simulation.  If set to “‘<span class="samp">⁠lower⁠</span>’”, an analytic
lower bound on the p-value is computed.  If set to “‘<span class="samp">⁠upper⁠</span>’”, an analytic upper
bound on the p-value is computed.  “‘<span class="samp">⁠lower⁠</span>’” and “‘<span class="samp">⁠upper⁠</span>’” are based on formulae
in Hart and Martínez (2011).  a Tighter (though unpublished) lower
/upper bound on the p-value  may be obtained by specifying “‘<span class="samp">⁠Lower⁠</span>’”/“‘<span class="samp">⁠Upper⁠</span>’”. If
‘<span class="samp">⁠type⁠</span>’ is specified as “‘<span class="samp">⁠exact⁠</span>’” (the default value),the p-value for the test is
computed exactly for small values of the test statistic and crudely approximated for large values.  See the note below for further details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>The number of replications to use for Monte Carlo simulation.  
If computationally feasible, a value &gt;= 10000000 is recommended.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.p.value</code></td>
<td>
<p>If ‘<span class="samp">⁠TRUE⁠</span>’, do not compute the p-value.  
The default is ‘<span class="samp">⁠FALSE⁠</span>’.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The first argument may be a character vector representing a DNA sequence, a DNA
sequence represented using the SeqFastadna class from the <span class="pkg">seqinr</span> package,
or a vector containing the relative frequencies of the A, C, G and T nucleic
acids.
</p>
<p>Letting A, C, G and T denote the relative frequencies of their corresponding nucleic acids, this function performs the following hypothesis test:
</p>
<p><code class="reqn">H_0</code>: <code class="reqn">A\neq T</code> or <code class="reqn">C\neq G</code> <br><code class="reqn">H_1</code>: <code class="reqn">A=T</code> and <code class="reqn">C=G</code>
</p>
<p>The vector <code class="reqn">(A,C,G,T)</code> is assumed to come from a Dirichlet(1,1,1,1)
distribution on the 3-simplex under the null hypothesis.
</p>
<p>The test statistic is <code class="reqn">\eta_0 = \sqrt{(A-T)^2/2+(C-G)^2/2}</code>.
</p>


<h3>Value</h3>

<p>A list with class "htest.ext" containing the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>statistic</code></td>
<td>
<p>the value of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>the p-value of the test. Only included if no.p.value is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>a character string indicating what type of test was performed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.name</code></td>
<td>
<p>a character string giving the name of the data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate</code></td>
<td>
<p>the probability vector used to derive the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stat.desc</code></td>
<td>
<p>a brief description of the test statistic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>null</code></td>
<td>
<p>the null hypothesis (<code class="reqn">H_0</code>) of the test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alternative</code></td>
<td>
<p>the alternative hypothesis (<code class="reqn">H_1</code>) of the test.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Currently, regardless of the algorithm (<code>alg</code>) selected, the p-value or
bound is only computed correctly when the test statistic is smaller than or
equal to <code class="reqn">\sqrt{2}/4</code>.  A value of 1 is returned when 
the test statistic is greater than <code class="reqn">\sqrt{2}/4</code>.  This is not
accurate, but shouldn't matter as it is well within the acceptance region of the
null hypothesis.
</p>
<p>The algebraically computed bounds on the p-value obtained when ‘<span class="samp">⁠alg⁠</span>’ is set
to either “‘<span class="samp">⁠lower⁠</span>’”or  “‘<span class="samp">⁠upper⁠</span>’” are not as tight as
those corresponding to “‘<span class="samp">⁠Lower⁠</span>’” and “‘<span class="samp">⁠Upper⁠</span>’”, which  should be
generally preferred.  However, “‘<span class="samp">⁠exact⁠</span>’” or “‘<span class="samp">⁠simulate⁠</span>’”
should be employed for real- world analysis.
</p>
<p>‘<span class="samp">⁠no.p.value⁠</span>’ suppresses computation of the p-value when it is set to
‘<span class="samp">⁠TRUE⁠</span>’.  This may be useful wen using this function to help simulate the
test statistic.
</p>


<h3>Author(s)</h3>

<p>Andrew Hart and Servet Martínez
</p>


<h3>References</h3>

<p>Hart, A.G. and Martínez, S. (2011)
Statistical testing of Chargaff's second parity rule in bacterial genome sequences.
<em>Stoch. Models</em> <b>27(2)</b>, 1–46.
</p>


<h3>See Also</h3>

<p><code>chargaff1.test</code>, <code>chargaff2.test</code>, 
<code>agct.test</code>, <code>ag.test</code>, 
<code>chargaff.gibbs.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Demonstration on real bacterial sequence
data(nanoarchaeum)
chargaff0.test(nanoarchaeum)

#Simulate synthetic DNA sequence that does not satisfy Chargaff's second parity rule
trans.mat &lt;- matrix(c(.4, .1, .4, .1, .2, .1, .6, .1, .4, .1, .3, .2, .1, .2, .4, .3), 
ncol=4, byrow=TRUE)
seq &lt;- simulateMarkovChain(500000, trans.mat, states=c("a", "c", "g", "t"))
chargaff0.test(seq)
</code></pre>


</div>