<div class="container">

<table style="width: 100%;"><tr>
<td>spgs-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Statistical Patterns in Genomic Sequences
</h2>

<h3>Description</h3>

<p>provides functions for exploring and testing statistical properties and 
patterns in DNA sequences.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> spgs </td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>This package provides a range of statistical tests for various properties of DNA
and/or other genomic sequences.  There are eight groups of functions:
</p>

<dl>
<dt>Testing for Chargaff's second parity rule in bacteria and other DNA sequences</dt>
<dd>
<p><code>chargaff0.test</code>, <code>chargaff1.test</code>, <code>chargaff2.test</code>,
<code>chargaff.gibbs.test</code>, <code>oligoProfile</code>
</p>
</dd>
<dt>Testing for purine-pyrimidine parity in viruses and other DNA sequences</dt>
<dd>
<p><code>ag.test</code>, <code>agct.test</code>
</p>
</dd>
<dt>Testing for Bernoulli/Markov processes</dt>
<dd>
<p><code>markov.test</code>, <code>diid.test</code>
</p>
</dd>
<dt>Independence tests</dt>
<dd>
<p><code>diffsign.test</code>, <code>turningpoint.test</code>, 
<code>rank.test</code>, <code>lb.test</code>
</p>
</dd>
<dt>Tests for uniform distribution</dt>
<dd>
<p><code>ks.unif.test</code>, <code>chisq.unif.test</code>
</p>
</dd>
<dt>Simulation of random vectors, stochastic matrices, Bernoulli processes and Markov chains </dt>
<dd>
<p><code>simulateMarkovChain</code>, <code>estimateMarkovChain</code>, 
<code>rstochvec</code>, <code>rstochmat</code>, <code>rcspr2mat</code>
</p>
</dd>
<dt>Functions for obtaining the complement or reverse complement of a DNA sequence</dt>
<dd>
<p><code>complement</code>, <code>reverseComplement</code>
</p>
</dd>
<dt>Functions for counting words/k-mers and cylinders in symbolic sequences</dt>
<dd>
<p><code>pair.counts</code>, <code>triple.counts</code>, <code>quadruple.counts</code>, <code>cylinder.counts</code>
</p>
</dd>
</dl>
<p>The word/k-mer counting functions are general and can deal with arbitrary 
symbolic sequences, not only DNA sequences.
</p>
<p>Functions which take a DNA sequence as input are able to work with sequences 
stored as <code>SeqFastadna</code> objects generated by the <span class="pkg">seqinr</span> package.
</p>


<h3>Author(s)</h3>

<p>Andrew Hart and Servet Martínez
</p>
<p>Maintainer: Andrew Hart &lt;ahart@dim.uchile.cl&gt;
</p>


<h3>References</h3>

<p>Hart, A.G. and Martínez, S. (2011)
Statistical testing of Chargaff's second parity rule in bacterial genome sequences.
<em>Stoch. Models</em> <b>27(2)</b>, 1–46.
</p>
<p>Hart, A.G. and Martínez, S. (2014) 
Markovianness and Conditional Independence in Annotated Bacterial DNA.  
<em>Stat. Appl. Genet. Mol. Biol.</em> <b>13(6)</b>, 693-716.  arXiv:1311.4411 [q-bio.QM].
</p>
<p>Hart, A.G. and Martínez, S. (2012)
A Gibbs approach to Chargaff's second parity rule.
<em>J. Stat. Phys.</em> <b>146(2)</b>, 408-422.  arXiv:1105.0685 [math.pr].
</p>


<h3>See Also</h3>

<p><code>seqinr</code>
</p>


</div>