<div class="container">

<table style="width: 100%;"><tr>
<td>slideNucDiag</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sliding nucleotide diagnostics</h2>

<h3>Description</h3>

<p>Calculates the number of diagnostic nucleotides in sliding windows.
</p>


<h3>Usage</h3>

<pre><code class="language-R">slideNucDiag(DNAbin, sppVector, width, interval = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DNAbin</code></td>
<td>
<p>A DNA alignment of class ‘DNAbin’.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sppVector</code></td>
<td>
<p>Species vector (see <code>sppVector</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>width</code></td>
<td>
<p>Desired width of windows in number of base pairs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interval</code></td>
<td>
<p>Distance between each window in number of base pairs.
Default of 1. Giving the option of <code>"codons"</code> sets the size to 3.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Determines the number of diagnostic nucleotides for each species in each
window.
</p>


<h3>Value</h3>

<p>A matrix giving the number of diagnostic nucleotides for each
species (rows) in each window (columns).
</p>


<h3>Author(s)</h3>

<p>Samuel Brown &lt;s_d_j_brown@hotmail.com&gt;
</p>


<h3>See Also</h3>

<p><code>slideAnalyses</code>, <code>slideBoxplots</code>,
<code>slidingWindow</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(dolomedes)
doloSpp &lt;- substr(dimnames(dolomedes)[[1]], 1, 5)

slideNucDiag(dolomedes, doloSpp, 200, interval = 3)

slidND &lt;- slideNucDiag(dolomedes, doloSpp, 200, interval = 3)

#Number of basepairs for each species
graphics::matplot(t(slidND), type = "l")

#Number of basepairs for a single species
graphics::plot(slidND[4, ], type = "l")

#Total number of basepairs per window
graphics::plot(colSums(slidND), type = "l")


</code></pre>


</div>