<div class="container">

<table style="width: 100%;"><tr>
<td>dat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>simulated data for demonstrating the usage of springer</h2>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the usage of springer.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("dat")
</code></pre>


<h3>Format</h3>

<p>The dat file consists of five components: e, g, y, clin and coeff. The coefficients are the true values of parameters used for generating Y.
</p>


<h3>Details</h3>

<p><strong>The data model for generating Y</strong>
</p>
<p>Consider a longitudinal case study with <code class="reqn">n</code> subjects and <code class="reqn">k_i</code> measurements over time for the <code class="reqn">i</code>th subject (<code class="reqn">i=1,\ldots,n</code>).
Let <code class="reqn">Y_{ij}</code> be the response of the <code class="reqn">j</code>th observation for the <code class="reqn">i</code>th subject (<code class="reqn">i=1,\ldots,n</code>, <code class="reqn">j=1,\ldots,k_i</code>),
<code class="reqn">X_{ij}=(X_{ij1},...,X_{ijp})^\top</code> be a <code class="reqn">p</code>-dimensional vector of covariates denoting <code class="reqn">p</code> genetic factors, <code class="reqn">E_{ij}=(E_{ij1},...,E_{ijq})^\top</code>
be a <code class="reqn">q</code>-dimensional environmental factor and <code class="reqn">Clin_{ij}=(Clin_{ij1},...,Clin_{ijt})^\top</code> be a <code class="reqn">t</code>-dimensional clinical factor.  There is time dependence among measurements on the same subject, but we assume that the measurements
between different subjects are independent.  The model we used for hierarchical variable selection for geneâ€“environment interactions is given as:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij}= \alpha_0 + \sum_{m=1}^{t}\theta_m Clin_{ijm} + \sum_{u=1}^{q}\alpha_u E_{iju} + \sum_{v=1}^{p}(\gamma_v X_{ijv} + \sum_{u=1}^{q}h_{uv} E_{iju} X_{ijv})+\epsilon_{ij},</code>
</p>

<p>where <code class="reqn">\alpha_{0}</code> is the intercept and the marginal density of <code class="reqn">Y_{ij}</code> belongs to a canonical exponential family defined in Liang and Zeger (1986).
Define <code class="reqn">\eta_v=(\gamma_v, h_{1v}, ..., h_{qv})^\top</code>, which is a vector of length q+1 and <code class="reqn">Z_{ijv}=(X_{ijv}, E_{ij1}X_{ijv}, ..., E_{ijq}X_{ijv})^\top</code>,
which contains the main genetic effect of the <code class="reqn">v</code>th SNP from the <code class="reqn">j</code>th measurement on the <code class="reqn">i</code>th subject and its interactions with all the <code class="reqn">q</code>
environmental factors. The model can be written as:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij}= \alpha_0 + \sum_{m=1}^{t}\theta_m Clin_{ijm} + \sum_{u=1}^{q}\alpha_u E_{iju} + \sum_{v=1}^{p}\eta_v^\top Z_{ijv}+\epsilon_{ij},</code>
</p>

<p>where <code class="reqn">Z_{ijv}</code> is the <code class="reqn">v</code>th genetic factor and its interactions with the <code class="reqn">q</code> environment factors for the <code class="reqn">j</code>th measurement on the <code class="reqn">i</code>th subject,
and <code class="reqn">\eta_{v}</code> is the corresponding coefficient vector of length <code class="reqn">1+q</code>. The random error <code class="reqn">\epsilon_{i}=(\epsilon_{i1},...,\epsilon_{ik_i})^{T}</code>, which is assumed to follow a multivariate normal distribution with <code class="reqn">\Sigma_i</code>
as the covariance matrix for the repeated measurements of the <code class="reqn">ith</code> subject among the <code class="reqn">k_i</code> time points.
</p>


<h3>See Also</h3>

<p><code>springer</code>
</p>


</div>