<div class="container">

<table style="width: 100%;"><tr>
<td>USiCGs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Universal Single-Copy Genes</h2>

<h3>Description</h3>

<p>Lists of Universal Single Copy Genes for Bacteria and Archaea.
These are useful for transforming coverages or tpms into copy numbers.
This is an alternative way of normalizing data in order to be able to
compare functional profiles in samples with different sequencing depths.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(USiCGs)
</code></pre>


<h3>Format</h3>

<p>Character vector with the KEGG identifiers for 15 Universal Single Copy Genes.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3798274/bin/pcbi.1003292.s016.xls">Carr <em>et al.</em>, 2013. Table S1</a>.
</p>


<h3>References</h3>

<p>Carr, Shen-Orr &amp; Borenstein (2013). 
Reconstructing the Genomic Content of Microbiome Taxa through Shotgun Metagenomic Deconvolution
<em>PLoS Comput. Biol.</em> <b>9</b>:e1003292.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3798274/">PubMed</a>).
</p>


<h3>See Also</h3>

<p><code>MGOGs</code> and <code>MGKOs</code> for an alternative set of single copy genes, and for examples on how to generate copy numbers.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Hadza)
data(USiCGs)
### Let's look at the Universal Single Copy Gene distribution in our samples.
KEGG.tpm = Hadza$functions$KEGG$tpm
all(USiCGs %in% rownames(KEGG.tpm)) # Are all the USiCGs present in our dataset?
# Plot a boxplot of USiCGs tpms and calculate median USiCGs tpm.
# This looks weird in the test dataset because it contains only a small subset of the metagenomes.
# In a set of complete metagenomes USiCGs should have fairly similar TPM averages
# and low dispersion across samples.
boxplot(t(KEGG.tpm[USiCGs,]), names=USiCGs, ylab="TPM", col="slateblue2")
 
### Now let's calculate the average copy numbers of each function.
# We do it for KEGG annotations here, but we could also do it for COGs or PFAMs.
USiCGs.cov = apply(Hadza$functions$KEGG$cov[USiCGs,], 2, median)
# Sample-wise division by the median USiCG coverage.
KEGG.copynumber = t(t(Hadza$functions$KEGG$cov) / USiCGs.cov)
</code></pre>


</div>