<div class="container">

<table style="width: 100%;"><tr>
<td>MGKOs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Single Copy Phylogenetic Marker Genes from Sunagawa's group (KOs)</h2>

<h3>Description</h3>

<p>Lists of Single Copy Phylogenetic Marker Genes.
These are useful for transforming coverages or tpms into copy numbers.
This is an alternative way of normalizing data in order to be able to
compare functional profiles in samples with different sequencing depths.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(MGKOs)
</code></pre>


<h3>Format</h3>

<p>Character vector with the KEGG identifiers for 10 Single Copy Phylogenetic Marker Genes.
</p>


<h3>References</h3>

<p>Salazar, G <em>et al.</em> (2019). 
Gene Expression Changes and Community Turnover Differentially Shape the Global Ocean Metatranscriptome
<em>Cell</em> <b>179</b>:1068-1083.
(<a href="https://pubmed.ncbi.nlm.nih.gov/31730850/">PubMed</a>).
</p>


<h3>See Also</h3>

<p><code>MGOGs</code> for an equivalent list using OGs instead of KOs; <code>USiCGs</code> for an alternative set of single copy genes, and for examples on how to generate copy numbers.
</p>


</div>