<div class="container">

<table style="width: 100%;"><tr>
<td>exportTable</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Export results in tabular format</h2>

<h3>Description</h3>

<p>This function is a wrapper for R's write.table function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">exportTable(table, output_name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>table</code></td>
<td>
<p>vector, matrix or data.frame. The table to be written.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_name</code></td>
<td>
<p>character. Name of the output file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>No return value, but a table is produced in the current working directory.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(Hadza)
Hadza.iron = subsetFun(Hadza, "iron")
# Write the taxonomic distribution at the genus level of all the genes related to iron.
exportTable(Hadza.iron$taxa$genus$percent, "Hadza.ironGenes.genus.tsv")
# Now write the distribution of the different iron-related COGs
#  (Clusters of Orthologous Groups) across samples.
exportTable(Hadza.iron$functions$COG$tpm, "Hadza.ironGenes.COG.tsv")
# Now write all the information contained in the ORF table.
exportTable(Hadza.iron$orfs$table, "Hadza.ironGenes.orftable.tsv")

</code></pre>


</div>