<div class="container">

<table style="width: 100%;"><tr>
<td>plotTaxonomy</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Barplot of the most abundant taxa in a SQM object</h2>

<h3>Description</h3>

<p>This function selects the most abundant taxa across all samples in a SQM object and represents their abundances in a barplot. Alternatively, a custom set of taxa can be represented.
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotTaxonomy(
  SQM,
  rank = "phylum",
  count = "percent",
  N = 15,
  tax = NULL,
  others = TRUE,
  samples = NULL,
  nocds = "treat_separately",
  ignore_unmapped = FALSE,
  ignore_unclassified = FALSE,
  no_partial_classifications = FALSE,
  rescale = FALSE,
  color = NULL,
  base_size = 11,
  max_scale_value = NULL,
  metadata_groups = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SQM</code></td>
<td>
<p>A SQM or a SQMlite object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rank</code></td>
<td>
<p>Taxonomic rank to plot (default <code>phylum</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>count</code></td>
<td>
<p>character. Either <code>"percent"</code> for percentages, or <code>"abund"</code> for raw abundances (default <code>"percent"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>integer Plot the <code>N</code> most abundant taxa (default <code>15</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tax</code></td>
<td>
<p>character. Custom taxa to plot. If provided, it will override <code>N</code> (default <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>others</code></td>
<td>
<p>logical. Collapse the abundances of least abundant taxa, and include the result in the plot (default <code>TRUE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samples</code></td>
<td>
<p>character. Character vector with the names of the samples to include in the plot. Can also be used to plot the samples in a custom order. If not provided, all samples will be plotted (default <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nocds</code></td>
<td>
<p>character. Either <code>"treat_separately"</code> to treat reads annotated as No CDS separately, <code>"treat_as_unclassified"</code> to treat them as Unclassified or <code>"ignore"</code> to ignore them in the plot (default <code>"treat_separately"</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_unmapped</code></td>
<td>
<p>logical. Don't include unmapped reads in the plot (default <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ignore_unclassified</code></td>
<td>
<p>logical. Don't include unclassified reads in the plot (default <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no_partial_classifications</code></td>
<td>
<p>logical. Treat reads not fully classified at the requested level (e.g. "Unclassified bacteroidetes" at the class level or below) as fully unclassified. This takes effect before <code>ignore_unclassified</code>, so if both are <code>TRUE</code> the plot will only contain fully classified contigs (default <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rescale</code></td>
<td>
<p>logical. Re-scale results to percentages (default <code>FALSE</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>Vector with custom colors for the different features. If empty, we will use our own hand-picked pallete if N&lt;=15, and the default ggplot2 palette otherwise (default <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base_size</code></td>
<td>
<p>numeric. Base font size (default <code>11</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_scale_value</code></td>
<td>
<p>numeric. Maximum value to include in the y axis. By default it is handled automatically by ggplot2 (default <code>NULL</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metadata_groups</code></td>
<td>
<p>list. Split the plot into groups defined by the user: list('G1' = c('sample1', sample2'), 'G2' = c('sample3', 'sample4')) default <code>NULL</code>).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a ggplot2 plot object.
</p>


<h3>See Also</h3>

<p><code>plotFunctions</code> for plotting the most abundant functions of a SQM object; <code>plotBars</code> and <code>plotHeatmap</code> for plotting barplots or heatmaps with arbitrary data.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Hadza)
Hadza.amin = subsetFun(Hadza, "Amino acid metabolism")
# Taxonomic distribution of amino acid metabolism ORFs at the family level.
plotTaxonomy(Hadza.amin, "family")
</code></pre>


</div>