<div class="container">

<table style="width: 100%;"><tr>
<td>PALM</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Population Allele Locating Mapmaker</h2>

<h3>Description</h3>

<p>Produces a frequency heatmap for a specified amino-acid motif, based on the allele frequency data in the Solberg dataset.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PALM(
  motif,
  filename = SSHAARP::solberg_dataset,
  direct = getwd(),
  color = TRUE,
  filterMigrant = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>The filename of the local copy of the Solberg dataset - the defaulted filename is the solberg_dataset in the SSHAARP package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>direct</code></td>
<td>
<p>The directory into which the map produced is written. The default directory is the user's working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>A logical parameter that identifies if the heat maps should be made in color (TRUE) or gray scale (FALSE). The default option is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filterMigrant</code></td>
<td>
<p>A logical parameter that determines if admixed populations (OTH) and migrant populations (i.e. any complexities with the 'mig') should be excluded from the dataset. The default option is TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The specified motif and the directory into which the heatmap was written are returned in an invisible character vector. If the user enters a motif that is not found in the Solberg dataset, or that does not exist, a warning message is returned. If an incorrectly formatted motif is entered, or the user does not have the GMT software installed on their operating system, a vector with a warning message is returned. The produced heatmap is written to the user's specified directory (default is user's working directory) as a .jpg file, where the filename is "'motif'.jpg".
</p>


<h3>Note</h3>

<p>The produced frequency heatmap is generated by using the Generic Mapping Tools (GMT) R Package, which is an interface between R and the GMT Map-Making software.
</p>
<p>The Solberg dataset is the tab-delimited ‘1-locus-alleles.dat’ text file in the results.zip archive at http://pypop.org/popdata/.
</p>
<p>The Solberg dataset is also prepackaged into SSHAARP as 'solberg_dataset'.
</p>
<p>While the map legend identifies the highest frequency value, values in this range may not be represented on the map due to frequency averaging over neighboring populations.
</p>


<h3>References</h3>

<p>Solberg et.al. (2008) &lt;doi: 10.1016/j.humimm.2008.05.001&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R">#example to produce a color frequency heat map where migrant populations are filtered out
PALM("DRB1*26F~28E~30Y",filename = solberg_dataset[85:100,], filterMigrant=TRUE)
#example to produce a greyscale heat map where migrant populations are not filtered out
PALM("DRB1*26F~28E~30Y", filename = solberg_dataset[85:100,], color=FALSE, filterMigrant=FALSE)

</code></pre>


</div>