<div class="container">

<table style="width: 100%;"><tr>
<td>BLAASD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>BLAASD - Build Loci Amino Acid Specific Dataframe</h2>

<h3>Description</h3>

<p>Extracts alignment sequence information for a given locus from the ANHIG/IMGTHLA GitHub repository to produce a dataframe of individual amino acid data for each amino acid position for all alleles, for a user-defined HLA locus or loci. The first 4 columns are locus, allele, trimmed allele, and allele_name.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BLAASD(loci)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>
<p>A vector of un-prefixed HLA locus names</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A list object of data frames for each specified locus. Each list element is a data frame of allele names and the corresponding peptide sequence for each amino acid position. An error message is return if the loci input is not a locus for which petpide alignments are available in the ANHIG/IMGTHLA Github Repository.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#BLAASD with one locus as input
BLAASD("C")

#BLAASD with multiple loci as input
BLAASD(c("A", "B", "C"))
</code></pre>


</div>