<div class="container">

<table style="width: 100%;"><tr>
<td>findMotif</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Returns an alignment data frame of alleles that share a specific amino acid motif</h2>

<h3>Description</h3>

<p>Consumes the alignment data frame produced by BLAASD() and returns an alignment data frame of alleles that share a specific amino acid motif.
</p>


<h3>Usage</h3>

<pre><code class="language-R">findMotif(motif)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>An amino acid alignment dataframe of alleles that share the specified motif. If the motif is not found in any alleles, or the motif has formatting errors, a warning message is returned.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#example with actual motif
findMotif("DRB1*26F~28E~30Y")
("DRB1*26F~28E")

#example with non-existent motif
findMotif("DRB1*26F~28E~30Z")

#extracting names of alleles with user-defined motif
findMotif("DRB1*26F~28E~30Y")[,4]
</code></pre>


</div>