<div class="container">

<table style="width: 100%;"><tr>
<td>checkMotif</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Syntactic and semantic validation of HLA amino acid motifs</h2>

<h3>Description</h3>

<p>Checks input motif for errors in format and amino acid positions not present in the locus alignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">checkMotif(motif)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>motif</code></td>
<td>
<p>An amino acid motif in the following format: Locus*##$~##$~##$, where ## identifies a peptide position, and $ identifies an amino acid residue. Motifs can include any number of amino acids.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A warning message if the input motif is formatted incorrectly, or contains an amino acid position not present in the alignment. Otherwise, a list object with extracted locus information, a correctly formatted motif, and locus specific amino acid dataframe are returned. Note checkMotif() does not check amino acid variants in a specified motif; that is done by findMotif().
</p>


<h3>Note</h3>

<p>For internal SSHAARP use only.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Example where a motif is formatted correctly
checkMotif("DRB1*26F~28E~30Y")

#Example where format is incorrect
checkMotif("DRB1**26F~28E~30Y")

#Example where an amino acid position does not exist
checkMotif("DRB1**26F~28E~300000Y")
</code></pre>


</div>