<div class="container">

<table style="width: 100%;"><tr>
<td>ssizeRNA_single</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sample Size Calculations for Two-Sample RNA-seq Experiments 
with Single Set of Parameters</h2>

<h3>Description</h3>

<p>This function calculates appropriate sample sizes for two-sample 
RNA-seq experiments for a desired power in which mean and 
dispersion parameters are identical for all genes. 
Sample size calculations are performed at controlled false discovery rates, 
user-specified proportions of non-differentially expressed genes, 
mean counts in control group, dispersion, and fold change. 
A plot of power versus sample size is generated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ssizeRNA_single(nGenes = 10000, pi0 = 0.8, m = 200, mu, disp, fc,
  up = 0.5, replace = TRUE, fdr = 0.05, power = 0.8, maxN = 35,
  side = "two-sided", cex.title = 1.15, cex.legend = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nGenes</code></td>
<td>
<p>total number of genes, the default value is <code>10000</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>
<p>proportion of non-differentially expressed genes, 
the default value is <code>0.8</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>pseudo sample size for generated data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>a vector (or scalar) of mean counts in control group 
from which to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disp</code></td>
<td>
<p>a vector (or scalar) of dispersion parameter 
from which to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc</code></td>
<td>
<p>a vector (or scalar, or a function that takes an integer n 
and generates a vector of length n)
of fold change for differentially expressed (DE) genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>up</code></td>
<td>
<p>proportion of up-regulated genes among all DE genes, 
the default value is <code>0.5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replace</code></td>
<td>
<p>sample with or without replacement from given parameters. 
See Details for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>the false discovery rate to be controlled.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>the desired power to be achieved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxN</code></td>
<td>
<p>the maximum sample size used for power calculations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>side</code></td>
<td>
<p>options are "two-sided", "upper", or "lower".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.title</code></td>
<td>
<p>controls size of chart titles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.legend</code></td>
<td>
<p>controls size of chart legend.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If a vector is input for <code>pi0</code>, sample size calculations 
are performed for each proportion.
</p>
<p>If the total number of genes is larger than length of <code>mu</code> or 
<code>disp</code>, <code>replace</code> always equals <code>TRUE</code>.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which 
desired power is first reached.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crit.vals</code></td>
<td>
<p>critical value calculations with 
corresponding sample sizes.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>References</h3>

<p>Liu, P. and Hwang, J. T. G. (2007) Quick calculation for 
sample size while controlling false discovery rate with application 
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. 
</p>
<p>Orr, M. and Liu, P. (2009) Sample size estimation while controlling 
false discovery rate for microarray experiments using ssize.fdr package. 
<em>The R Journal</em>, 1, 1, May 2009, 47-53. 
</p>
<p>Law, C. W., Chen, Y., Shi, W., Smyth, G. K. (2014). Voom: precision weights 
unlock linear model analysis tools for RNA-seq read counts. 
<em>Genome Biology</em> 15, R29.
</p>


<h3>See Also</h3>

<p><code>ssizeRNA_vary</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">mu &lt;- 10                ## mean counts in control group for all genes
disp &lt;- 0.1             ## dispersion for all genes
fc &lt;- 2                 ## 2-fold change for DE genes

size &lt;- ssizeRNA_single(m = 30, mu = mu, disp = disp, fc = fc, 
                        maxN = 20)
size$ssize              ## first sample size to reach desired power
size$power              ## calculated power for each sample size
size$crit.vals          ## calculated critical value for each sample size

</code></pre>


</div>