<div class="container">

<table style="width: 100%;"><tr>
<td>check.power</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Average Power and True FDR Based on limma/voom RNAseq Analysis Pipeline</h2>

<h3>Description</h3>

<p>For the limma/voom RNAseq analysis pipeline, when we control false discovery
rate by using the Benjamini and Hochberg step-up procedure (1995) 
and/or Storey and Tibshirani's q-value procedure (Storey et al, 2004),
<code>check.power</code> calculates average power and true FDR for given sample 
size, user-specified proportions of non-differentially expressed genes, 
number of iterations, FDR level to control, mean counts in control group, 
dispersion, and fold change.
</p>


<h3>Usage</h3>

<pre><code class="language-R">check.power(nGenes = 10000, pi0 = 0.8, m, mu, disp, fc, up = 0.5,
  replace = TRUE, fdr = 0.05, sims = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nGenes</code></td>
<td>
<p>total number of genes, the default value is <code>10000</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>
<p>proportion of non-differentially expressed genes, 
the default value is <code>0.8</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>sample size per treatment group.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>a vector (or scalar) of mean counts in control group 
from which to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disp</code></td>
<td>
<p>a vector (or scalar) of dispersion parameter 
from which to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc</code></td>
<td>
<p>a vector (or scalar, or a function that takes an integer n 
and generates a vector of length n)
of fold change for differentially expressed (DE) genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>up</code></td>
<td>
<p>proportion of up-regulated genes among all DE genes, 
the default value is <code>0.5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replace</code></td>
<td>
<p>sample with or without replacement from given parameters. 
See Details for more information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>the false discovery rate to be controlled.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sims</code></td>
<td>
<p>number of simulations to run when computing power and FDR.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pow_bh_ave</code></td>
<td>
<p>average power when controlling FDR 
by Benjamini and Hochberg (1995) method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr_bh_ave</code></td>
<td>
<p>true false discovery rate when controlling FDR 
by Benjamini and Hochberg (1995) method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pow_bh_ave</code></td>
<td>
<p>average power when controlling FDR 
by q-value procedure (Storey et al., 2004).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr_bh_ave</code></td>
<td>
<p>true false discovery rate when controlling FDR
by q-value procedure (Storey et al., 2004).</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>References</h3>

<p>Benjamini, Y. and Hochberg, Y. (1995) 
Controlling the false discovery rate: a practical and 
powerful approach to multiple testing. 
<em>J. R. Stat. Soc. B</em>, 57, 289-300.
</p>
<p>Storey, J. D., Taylor, J. E. and Siegmund, D. (2004)
Strong control, conservative point estimation and 
simultaneous rates: a unified approach. 
<em>J. R. Stat. Soc. B</em>, 66, 187- 205.
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(limma)
library(qvalue)
m &lt;- 14                      ## sample size per treatment group
mu &lt;- 10                     ## mean read counts in control group
disp &lt;- 0.1                  ## dispersion for all genes
fc &lt;- 2                      ## 2-fold change for DE genes

check.power(m = m, mu = mu, disp = disp, fc = fc, sims = 2)

</code></pre>


</div>