<div class="container">

<table style="width: 100%;"><tr>
<td>ssize.oneSampVary</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sample Size Calculations for One-Sample Microarray Experiments with
Differing Mean Expressions and Standard Deviations Among Genes</h2>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for two-sample microarray
experiments in which effect sizes as well as variances vary among genes.
Sample sizes are determined based on a desired power, a controlled false
discovery rate, and user-specified proportions of non-differentially
expressed genes. Outputs a graph of power versus sample size. A graph of
power versus sample size is created.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ssize.oneSampVary(deltaMean, deltaSE, a, b, fdr = 0.05, power = 0.8, pi0 = 0.95,
maxN = 35, side = "two-sided", cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>deltaMean</code></td>
<td>
<p>mean of normal distribution followed
by effect sizes among genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deltaSE</code></td>
<td>
<p>standard deviation of normal distribution
followed by effect sizes among genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p>shape parameter of inverse gamma distribution followed by variances
of genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>scale parameter of inverse gamma distribution followed by variances
of genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>the desired power to be achieved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially expressed
genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>side</code></td>
<td>
<p>options are "two-sided", "upper", or "lower"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The effect sizes among genes are assumed to follow a Normal distribution
with mean specified by <code>deltaMean</code> and standard deviation specified by
<code>deltaSE</code>. The variances among genes are assumed to follow an
Inverse Gamma distribution with shape parameter <code>a</code> and scale parameter
<code>b</code>.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which desired power is first
reached</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crit.vals</code></td>
<td>
<p>critical value calculations with corresponding sample sizes</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Numerical integration used in calculations performed by the function
<code>integrate</code>, which uses adaptive quadrature of functions.
</p>
<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as ‘NA’ are values &gt;20.
</p>
<p>Running this function may result in many warnings. Probabilities under
different t-distributions with non-zero non-centrality parameters are
calculated many times while the function runs.  If these probabilities
are virtually zero, the function <code>pt</code> outputs a value &lt;1e-8
and outputs a warning of “full precision not achieved”.  These values
have no impact on the accuracy of the resulting calculations.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p>Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for sample
size while controlling false discovery rate with application to microarray
analysis. <em>Bioinformatics</em> 23(6): 739-746. </p>


<h3>See Also</h3>

<p><code>ssize.twoSampVary</code>, <code>ssize.oneSamp</code>,
<code>ssize.oneSampVary</code>, <code>ssize.F</code>,
<code>ssize.Fvary</code></p>


<h3>Examples</h3>

<pre><code class="language-R"> dm&lt;-2;  ds&lt;-1	##the effect sizes of the genes follow a Normal(2,1) distribution
 alph&lt;-3;  beta&lt;-1	##the variances of the genes follow an Inverse Gamma(3,1) distribution.
 a2&lt;-0.05	##false discovery rate to be controlled
 pwr2&lt;-0.8	##desired power
 p0&lt;-c(0.90,0.95,0.995)	##proportions of non-differentially expressed genes
 N1&lt;-35		##maximum sample size to be used in calculations

 osv&lt;-ssize.oneSampVary(deltaMean=dm,deltaSE=ds,a=alph,b=beta,fdr=a2,power=pwr2,pi0=p0,
 maxN=N1,side="two-sided")
 osv$ssize	##first sample sizes to reach desired power
 osv$power	##calculated power for each sample size
 osv$crit.vals	##calculated critical value for each sample size
</code></pre>


</div>