<div class="container">

<table style="width: 100%;"><tr>
<td>ssize.twoSampVaryDelta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sample Size Calculations for Two-Sample Microarray Experiments with 
Differing Mean Expressions but fixed Standard Deviations Among Genes</h2>

<h3>Description</h3>

<p>For given desired power, controlled false discovery rate, 
and user-specified proportions of non-differentially expressed genes, 
<code>ssize.twoSampVaryDelta</code> calculates appropriate sample sizes for 
two-sample microarray experiments in which the differences between mean 
treatment expression levels (<em>delta.g</em> for gene <em>g</em>) 
vary among genes. 
A plot of power versus sample size is generated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ssize.twoSampVaryDelta(deltaMean, deltaSE, sigma, fdr = 0.05,
  power = 0.8, pi0 = 0.95, maxN = 35, side = "two-sided",
  cex.title = 1.15, cex.legend = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>deltaMean</code></td>
<td>
<p>location (mean) parameter of normal distribution 
followed by each <em>delta.g</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deltaSE</code></td>
<td>
<p>scale (standard deviation) parameter of normal distribution 
followed by each <em>delta.g</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>
<p>the common standard deviation of expressions for all genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>the false discovery rate to be controlled.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>the desired power to be achieved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially 
expressed genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxN</code></td>
<td>
<p>the maximum sample size used for power calculations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>side</code></td>
<td>
<p>options are "two-sided", "upper", or "lower".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.title</code></td>
<td>
<p>controls size of chart titles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.legend</code></td>
<td>
<p>controls size of chart legend.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Each <em>delta.g</em> is assumed to follow a Normal distribution 
with mean <code>deltaMean</code> and standard deviation <code>deltaSE</code>. 
The standard deviations of expressions are assumed identical for all genes.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations 
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at 
which desired power is first reached.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>power calculations with corresponding 
sample sizes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crit.vals</code></td>
<td>
<p>critical value calculations with 
corresponding sample sizes.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>References</h3>

<p>Liu, P. and Hwang, J. T. G. (2007) Quick calculation for 
sample size while controlling false discovery rate with application 
to microarray analysis. <em>Bioinformatics</em> 23(6): 739-746. 
</p>
<p>Orr, M. and Liu, P. (2009) Sample size estimation while controlling 
false discovery rate for microarray experiments using ssize.fdr package. 
<em>The R Journal</em>, 1, 1, May 2009, 47-53.
</p>


<h3>See Also</h3>

<p><code>ssize.twoSamp</code>, <code>ssize.twoSampVary</code>, 
<code>ssize.oneSamp</code>, <code>ssize.oneSampVary</code>, 
<code>ssize.F</code>, <code>ssize.Fvary</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">dm &lt;- 1.2; ds &lt;- 0.1  ## the delta.g's follow a Normal(1.2, 0.1) distribution
s &lt;- 1                ## common standard deviation
fdr &lt;- 0.05           ## false discovery rate to be controlled
pwr &lt;- 0.8            ## desired power
pi0 &lt;- c(0.5, 0.8, 0.99) ## proportions of non-differentially expressed genes
N &lt;- 35               ## maximum sample size for calculations

size &lt;- ssize.twoSampVaryDelta(deltaMean = dm, deltaSE = ds, sigma = s, 
                               fdr = fdr, power = pwr, pi0 = pi0, 
                               maxN = N, side = "two-sided")
size$ssize                ## first sample size(s) to reach desired power
size$power                ## calculated power for each sample size
size$crit.vals            ## calculated critical value for each sample size

</code></pre>


</div>