<div class="container">

<table style="width: 100%;"><tr>
<td>sim.counts</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>RNA-seq Count Data Simulation from Negative-Binomial Distribution</h2>

<h3>Description</h3>

<p>This function simulates count data from Negative-Binomial distribution
for two-sample RNA-seq experiments with given mean, dispersion 
and fold change. 
A count data matrix is generated.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sim.counts(nGenes = 10000, pi0 = 0.8, m, mu, disp, fc, up = 0.5,
  replace = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nGenes</code></td>
<td>
<p>total number of genes, the default value is <code>10000</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>
<p>proportion of non-differentially expressed genes, 
the default value is <code>0.8</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>sample size per treatment group.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>a vector (or scalar) of mean counts in control group 
from which to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disp</code></td>
<td>
<p>a vector (or scalar) of dispersion parameter 
from which to simulate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fc</code></td>
<td>
<p>a vector (or scalar, or a function that takes an integer n 
and generates a vector of length n)
of fold change for differentially expressed (DE) genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>up</code></td>
<td>
<p>proportion of up-regulated genes among all DE genes, 
the default value is <code>0.5</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replace</code></td>
<td>
<p>sample with or without replacement from given parameters. 
See Details for more information.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If the total number of genes <code>nGenes</code> is larger 
than length of <code>mu</code> or <code>disp</code>, 
<code>replace</code> always equals <code>TRUE</code>.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>RNA-seq count data matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>treatment group vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda0</code></td>
<td>
<p>mean counts in control group for each gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi0</code></td>
<td>
<p>dispersion parameter for each gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de</code></td>
<td>
<p>differentially expressed genes indicator: 
<code>0</code> for non-differentially expressed genes, 
<code>1</code> for up-regulated genes, 
<code>-1</code> for down-regulated genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delta</code></td>
<td>
<p>log2 fold change for each gene between 
treatment group and control group.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Ran Bi <a href="mailto:biranpier@gmail.com">biranpier@gmail.com</a>, 
Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">m &lt;- 3                    ## sample size per treatment group
mu &lt;- 10                  ## mean counts in control group for all genes 
disp &lt;- 0.1               ## dispersion for all genes
fc &lt;- 2                   ## 2-fold change for DE genes

sim &lt;- sim.counts(m = m, mu = mu, disp = disp, fc = fc)
sim$counts                ## count data matrix


## varying fold change
fc1 &lt;- function(x){exp(rnorm(x, log(2), 0.5*log(2)))}
sim1 &lt;- sim.counts(m = m, mu = mu, disp = disp, fc = fc1)

</code></pre>


</div>