<div class="container">

<table style="width: 100%;"><tr>
<td>ssize.twoSamp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Sample Size Calculations for Two-Sample Microarray Experiments</h2>

<h3>Description</h3>

<p>Calculates appropriate sample sizes for two-sample
microarray experiments for a desired power. Sample size calculations
are performed at controlled false discovery rates, user-specified
proportions of non-differentially expressed genes, effect size, and
standard deviation.  A graph of power versus sample size is created.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ssize.twoSamp(delta, sigma, fdr = 0.05, power = 0.8, pi0 = 0.95, maxN = 35,
side = "two-sided", cex.title=1.15, cex.legend=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>delta</code></td>
<td>
<p>the common difference in mean expressions between the two
samples	for all genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>
<p>the common standard deviation of expressions for all genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>
<p>the false discovery rate to be controlled</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>the desired power to be achieved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi0</code></td>
<td>
<p>a vector (or scalar) of proportions of non-differentially
expressed genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxN</code></td>
<td>
<p>the maximum sample size used for power calculations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>side</code></td>
<td>
<p>options are "two-sided", "upper", or "lower"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.title</code></td>
<td>
<p>controls size of chart titles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.legend</code></td>
<td>
<p>controls size of chart legend</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The true difference between mean expressions of the two samples
as well as the standard deviations of expressions are assumed
identical for all genes.  See the function
<code>ssize.twoSampVary</code> for sample size calculations
with varying differences between sample mean expressions and
standard deviations among genes.
</p>
<p>If a vector is input for <code>pi0</code>, sample size calculations
are performed for each proportion.
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ssize</code></td>
<td>
<p>sample sizes (for each treatment) at which desired power
is first reached</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power</code></td>
<td>
<p>power calculations with corresponding sample sizes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crit.vals</code></td>
<td>
<p>critical value calculations of two-sample t-test
with corresponding sample sizes</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Powers calculated to be 0 may be negligibly conservative.
</p>
<p>Critical values calculated as ‘NA’ are values &gt;20.
</p>
<p>Running this function with the <code>side</code> option of "lower" will
possibly result in multiple warnings. Calculating the probability
that an observation is less than the negative critical value under
a t-distribution with non-centrality parameter <em>delta/sigma</em>
(see argument section above) and the appropriate degrees of freedom
is a calculation that is performed many times while the function
runs. When the difference between the critical value and
<em>delta/sigma</em> is large, this probability is virtually zero.
This happens repeatedly while the function <code>optimize</code>
finds the appropriate critical value for each sample size.  Because
of this, the function <code>pt</code> outputs a value &lt;1e-8 in
addition to a warning of “full precision not achieved”. This has no
impact on the accuracy of the resulting calculations of sample size.
</p>


<h3>Author(s)</h3>

<p>Megan Orr <a href="mailto:megan.orr@ndsu.edu">megan.orr@ndsu.edu</a>, Peng Liu <a href="mailto:pliu@iastate.edu">pliu@iastate.edu</a></p>


<h3>References</h3>

<p> Liu, Peng and J. T. Gene Hwang. 2007. Quick calculation for
sample size while controlling false discovery rate with application
to microarray analysis. Bioinformatics 23(6): 739-746. </p>


<h3>See Also</h3>

<p><code>ssize.twoSampVary</code>, <code>ssize.oneSamp</code>,
<code>ssize.oneSampVary</code>, <code>ssize.F</code>,
<code>ssize.Fvary</code></p>


<h3>Examples</h3>

<pre><code class="language-R"> ##See Figure 1.(a) of Liu &amp; Hwang (2007)
 d&lt;-1		##difference in differentially expressed genes to be detected
 s&lt;-0.5  	##standard deviation
 a&lt;-0.05	##false discovery rate to be controlled
 pwr&lt;-0.8	##desired power
 p0&lt;-c(0.5,0.9,0.95)	##proportions of non-differentially expressed genes
 N&lt;-20		##maximum sample size for calculations

 ts&lt;-ssize.twoSamp(delta=d,sigma=s,fdr=a,power=pwr,pi0=p0,maxN=N,side="two-sided")
 ts$ssize	##first sample sizes to reach desired power for each proportion of
			 ##non-differentially expressed genes
 ts$power	##calculated power for each sample size
 ts$crit.vals	##calculated critical value for each sample size

</code></pre>


</div>