<div class="container">

<table style="width: 100%;"><tr>
<td>ssw-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>ssw: Striped Smith-Waterman Algorithm for Sequence Alignment using SIMD</h2>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style="float: right" alt="logo" width="120"></p>
<p>Provides an R interface for 'SSW' (Striped Smith-Waterman) via its 'Python' binding 'ssw-py'. 'SSW' is a fast 'C' and 'C++' implementation of the Smith-Waterman algorithm for pairwise sequence alignment using Single-Instruction-Multiple-Data (SIMD) instructions. 'SSW' enhances the standard algorithm by efficiently returning alignment information and suboptimal alignment scores. The core 'SSW' library offers performance improvements for various bioinformatics tasks, including protein database searches, short-read alignments, primary and split-read mapping, structural variant detection, and read-overlap graph generation. These features make 'SSW' particularly useful for genomic applications. Zhao et al. (2013) <a href="https://doi.org/10.1371/journal.pone.0082138">doi:10.1371/journal.pone.0082138</a> developed the original 'C' and 'C++' implementation.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Nan Xiao <a href="mailto:me@nanx.me">me@nanx.me</a> (<a href="https://orcid.org/0000-0002-0250-5673">ORCID</a>) [copyright holder]
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://nanx.me/ssw-r/">https://nanx.me/ssw-r/</a>
</p>
</li>
<li> <p><a href="https://github.com/nanxstats/ssw-r">https://github.com/nanxstats/ssw-r</a>
</p>
</li>
<li>
<p> Report bugs at <a href="https://github.com/nanxstats/ssw-r/issues">https://github.com/nanxstats/ssw-r/issues</a>
</p>
</li>
</ul>
</div>