<div class="container">

<table style="width: 100%;"><tr>
<td>choedata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Subset of the Choe et al. (2005) “Golden Spike” Experiment</h2>

<h3>Description</h3>

<p>These data are expression levels for a subset of the genes
investigated in the Choe et al. (2005) “Golden Spike”
Affymetrix case-control experiment.  
</p>
<p>From the original data the 2,535 probe sets for <em>spike-ins with 
ratio 1:1 were removed</em>, leaving in total 11,475 genes with 
3 replicates per group, and  1,331 known differentially 
expressed genes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(choedata)
</code></pre>


<h3>Format</h3>

<p><code>choe2.mat</code> is a matrix of dimension  6 times 11,475. It contains
the samples in its rows and the genes in its columns.
</p>
<p><code>choe2.L</code> describes the case control-structure of the experiment,
and  <code>choe2.degenes</code> indicates the known differentially expressed genes.
<code>choe2.symbol.name</code>,
<code>choe2.probe.name</code>, and
<code>choe2.mapping</code> provide additional information on the investigated genes.
</p>


<h3>References</h3>

<p>Choe, S. E., M. Boutros, A. M. Michelson, G. M. Church, and M. ~S. Halfon.
2005. Preferred analysis methods for Affymetrix GeneChips revealed
by a wholly defined control data set.
<em>Genome Biology </em> <b>6</b>, R16.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># load st library
library("st")

# load data set
data(choedata)

# 6 samples, 11,475 genes
dim(choe2.mat)

# two groups (case vs. control
choe2.L

# 1,331 differentially expressed genes
sum(choe2.degenes)

# further information on genes
choe2.symbol.name
choe2.probe.name
choe2.mapping 
</code></pre>


</div>