<div class="container">

<table style="width: 100%;"><tr>
<td>stamppAmova</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Analysis of Molecular Variance</h2>

<h3>Description</h3>

<p>Calculates an AMOVA based on the genetic distance matrix from stamppNeisD() using the amova() function from the package PEGAS for exploring within and between population variation
</p>


<h3>Usage</h3>

<pre><code class="language-R">stamppAmova(dist.mat, geno, perm = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dist.mat</code></td>
<td>
<p>the matrix of genetic distances between individuals generated from stamppNeisD()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>a data frame containing allele frequency data generated from stamppConvert, or a genlight object containing genotype data, individual IDs, population IDs and ploidy levels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm</code></td>
<td>
<p>the number of permutations for the tests of hypotheses</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Uses the formula distance ~ populations, to calculate an AMOVA for population differentiation and within &amp; between population variation.
This function uses the amova function from the PEGAS package.
</p>


<h3>Value</h3>

<p>An object of class "amova" which is a list containing a table of sum of square deviations (SSD), mean square deviations (MSD) and the number of degrees of freedom as well as the variance components
</p>


<h3>Author(s)</h3>

<p>Luke Pembleton &lt;luke.pembleton at agriculture.vic.gov.au&gt;
</p>


<h3>References</h3>

<p>Paradis E (2010) pegas: an R package for population genetics with an integrated-modular approach. Bioinformatics 26, 419-420. &lt;doi:10.1093/bioinformatics/btp696&gt;
</p>


<h3>Examples</h3>

<pre><code class="language-R"># import genotype data and convert to allele frequecies
data(potato.mini, package="StAMPP")
potato.freq &lt;- stamppConvert(potato.mini, "r")
# Calculate genetic distance between individuals
potato.D.ind &lt;- stamppNeisD(potato.freq, FALSE)
# Calculate AMOVA
stamppAmova(potato.D.ind, potato.freq, 100)
</code></pre>


</div>