<div class="container">

<table style="width: 100%;"><tr>
<td>esSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
A Simulated Gene Expression Dataset
</h2>

<h3>Description</h3>

<p>A simulated gene expression dataset for differential expression analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data("esSim")</code></pre>


<h3>Format</h3>

<p>The format is:
Formal class 'ExpressionSet' [package "Biobase"] with expression levels
of 100 probes for 20 samples.
</p>
<p>The phenotype data contain 2 phenotype variables:
sid (subject id) and grp (group indicator: 1 stands for case; 0 stands for control).
</p>
<p>The feature data contain 4 feature variables:
probeid (probe id), gene (fake gene symbol), chr (fake chromosome number),
and memProbes (probe significance indicator: 1 stands for probes over-expressed (OE) in cases; -1 stands for probes under-expressed (UE) in cases; and 0 stands for non-differentially expressed (NE) probes). There are 3 OE probes,
2 UE probes, and 95 NE probes.
</p>


<h3>Details</h3>

<p>The dataset was generated based on the R code in the manual
of the function <code>lmFit</code> of the R Bioconductor package <code>limma</code>.
There are 100 probes and 20 samples (10 controls and 10 cases). The first 3 probes are over-expressed in cases. The 4-th and 5-th probes are under-expressed in cases. The remaining 95 probes are non-differentially expressed between cases and controls. Expression levels for 100 probes were first generated from normal distribution with mean 0 and standard deviation varying between probes (<code class="reqn">sd=0.3\sqrt{4/\chi^2_4}</code>). For the 3 OE probes, we add 2 to the expression levels of the 10 cases. For the 2 UE probes, we subtract 2 from the expression levels of the 10 cases.
</p>


<h3>References</h3>

<p>Please see the example in the manual for the function <code>lmFit</code> in the R Bioconductor package <code>limma</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(esSim)

print(esSim)

###
dat=exprs(esSim)
print(dim(dat))
print(dat[1:2,])

###
pDat=pData(esSim)
print(dim(pDat))
print(pDat)

# subject group status
print(table(esSim$grp))

###
fDat = fData(esSim)
print(dim(fDat))
print(fDat[1:2, ])

# probe's status of differential expression
print(table(fDat$memProbes))

</code></pre>


</div>