<div class="container">

<table style="width: 100%;"><tr>
<td>selQTLMPP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multi round genome scans for QTL detection</h2>

<h3>Description</h3>

<p>Multi round genome scans for QTL detection.<br><br>
Several rounds of QTL detection are performed. First a model is fitted
without cofactors. If for at least one marker the <code class="reqn">-log10(p)</code> value is
above the threshold the marker with the lowest p-Value is added as cofactor
in the next round of QTL detection. This process continues until there are
no new markers with a <code class="reqn">-log10(p)</code> value above the threshold or until
the maximum number of cofactors is reached.
</p>


<h3>Usage</h3>

<pre><code class="language-R">selQTLMPP(
  MPPobj,
  trait = NULL,
  QTLwindow = 10,
  threshold = 3,
  maxCofactors = NULL,
  K = NULL,
  computeKin = FALSE,
  parallel = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>MPPobj</code></td>
<td>
<p>An object of class gDataMPP, typically the output of either
<code>calcIBDMPP</code> or <code>readRABBITMPP</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait</code></td>
<td>
<p>A character string indicating the trait QTL mapping is done for.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>QTLwindow</code></td>
<td>
<p>A numerical value indicating the window around a QTL that
is considered as part of that QTL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>A numerical value indicating the threshold for the
<code class="reqn">-log10p</code> value of a marker to be considered a QTL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxCofactors</code></td>
<td>
<p>A numerical value, the maximum number of cofactors to
include in the model. If <code>NULL</code> cofactors are added until no new
cofactors are found.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>A list of chromosome specific kinship matrices. If
<code>NULL</code> and <code>computeKin = FALSE</code> no kinship matrix is included in
the models.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>computeKin</code></td>
<td>
<p>Should chromosome specific kinship matrices be computed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>Should the computation of variance components be done in
parallel? This requires a parallel back-end to be registered. See examples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Should progress and intermediate plots be output?</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>By default only family specific effects and residual variances and no
kinship relations are included in the model. It is possible to include
kinship relations by either specifying <code>computeKin = TRUE</code>. When doing
so the kinship matrix is computed by averaging <code class="reqn">Z Z^t</code> over all markers,
where <code class="reqn">Z</code> is the genotype x parents matrix for the marker. It is also
possible to specify a list of precomputed chromosome
specific kinship matrices in <code>K</code>. Note that adding a kinship matrix
to the model increases the computation time a lot, especially for large
populations.
</p>


<h3>Value</h3>

<p>An object of class <code>QTLMPP</code>
</p>


<h3>See Also</h3>

<p><code>kinshipIBD</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Read phenotypic data.
pheno &lt;- read.delim(system.file("extdata/multipop", "AxBxCpheno.txt",
                               package = "statgenMPP"))
## Rename first column to genotype.
colnames(pheno)[1] &lt;- "genotype"


## Compute IBD probabilities for simulated population - AxB, AxC.
ABC &lt;- calcIBDMPP(crossNames = c("AxB", "AxC"),
                  markerFiles = c(system.file("extdata/multipop", "AxB.txt",
                                              package = "statgenMPP"),
                                  system.file("extdata/multipop", "AxC.txt",
                                              package = "statgenMPP")),
                  pheno = pheno,
                  popType = "F4DH",
                  mapFile = system.file("extdata/multipop", "mapfile.txt",
                                        package = "statgenMPP"),
                  evalDist = 5)

## Single-QTL Mapping.
ABC_SQM &lt;- selQTLMPP(ABC, trait = "yield", maxCofactors = 0)

## Multi-QTL Mapping.
## Not run: 
## Register parallel back-end with 2 cores.
doParallel::registerDoParallel(cores = 2)

## Run multi-QTL mapping.
ABC_MQM &lt;- selQTLMPP(ABC, trait = "yield", parallel = TRUE)

## Run multi-QTL mapping - include kinship matrix.
ABC_MQM_kin &lt;- selQTLMPP(ABC, trait = "yield", parallel = TRUE,
                        computeKin = TRUE)

## End(Not run)

</code></pre>


</div>