<div class="container">

<table style="width: 100%;"><tr>
<td>codeMarkers</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Code and impute markers</h2>

<h3>Description</h3>

<p><code>codeMarkers</code> codes markers in a <code>gData</code> object and optionally
performs imputation of missing values as well.<br>
The function performs the following steps:<br></p>

<ol>
<li>
<p>replace strings in <code>naStrings</code> by <code>NA</code>.
</p>
</li>
<li>
<p>remove genotypes with a fraction of missing values higher than
<code>nMissGeno</code>.
</p>
</li>
<li>
<p>remove SNPs with a fraction of missing values higher than
<code>nMiss</code>.
</p>
</li>
<li>
<p>recode SNPs to numerical values.
</p>
</li>
<li>
<p>remove SNPs with a minor allele frequency lower than <code>MAF</code>.
</p>
</li>
<li>
<p>optionally remove duplicate SNPs.
</p>
</li>
<li>
<p>optionally impute missing values.
</p>
</li>
<li>
<p>repeat steps 5. and 6. if missing values are imputed.
</p>
</li>
</ol>
<h3>Usage</h3>

<pre><code class="language-R">codeMarkers(
  gData,
  refAll = "minor",
  nMissGeno = 1,
  nMiss = 1,
  MAF = NULL,
  MAC = NULL,
  removeDuplicates = TRUE,
  keep = NULL,
  impute = TRUE,
  imputeType = c("random", "fixed", "beagle"),
  fixedValue = NULL,
  naStrings = NA,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gData</code></td>
<td>
<p>An object of class <code>gData</code> containing at least
<code>markers</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refAll</code></td>
<td>
<p>A character string indicating the reference allele used when
recoding markers.<br>
If "minor", then the recoding is done using the minor allele as reference
allele. Alternatively a single character can be supplied as a reference
allele for the whole set of SNPs, or a character vector with a reference
allele per SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nMissGeno</code></td>
<td>
<p>A numerical value between 0 and 1. Genotypes with a
fraction of missing values higher than <code>nMissGeno</code> will be removed.
Genotypes with only missing values will always be removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nMiss</code></td>
<td>
<p>A numerical value between 0 and 1. SNPs with a fraction of
missing values higher than <code>nMiss</code> will be removed. SNPs with only
missing values will always be removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>A numerical value between 0 and 1. SNPs with a Minor Allele
Frequency (MAF) below this value will be removed. Only one of <code>MAF</code>
and <code>MAC</code> may be specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAC</code></td>
<td>
<p>A numerical value. SNPs with Minor Allele Count (MAC) below this
value will be removed. Only one of <code>MAF</code> and <code>MAC</code> may be
specified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>removeDuplicates</code></td>
<td>
<p>Should duplicate SNPs be removed?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep</code></td>
<td>
<p>A vector of SNPs that should never be removed in the whole
process.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute</code></td>
<td>
<p>Should imputation of missing values be done?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imputeType</code></td>
<td>
<p>A character string indicating what kind of imputation of
values should be done.<br></p>

<ul>
<li>
<p>fixed - missing values will be replaced by a given fixed value.
</p>
</li>
<li>
<p>random - missing values will be replaced by a random value calculated
using allele frequencies per SNP.
</p>
</li>
<li>
<p>beagle - missing values will be imputed using beagle software, version
5.2. Beagle only accepts integers as map positions. If you use this option,
please cite the original papers in your publication (see references).
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fixedValue</code></td>
<td>
<p>A numerical value used for replacing missing values in
case <code>inputType</code> is fixed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>naStrings</code></td>
<td>
<p>A character vector of strings to be treated as NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Should a summary of the performed steps be printed?</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A copy of the input <code>gData</code> object with markers replaced by
coded and imputed markers.
</p>


<h3>References</h3>

<p>S R Browning and B L Browning (2007) Rapid and accurate haplotype
phasing and missing data inference for whole genome association studies by
use of localized haplotype clustering. Am J Hum Genet 81:1084-1097.
<a href="https://doi.org/10.1086/521987">doi:10.1086/521987</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Create markers
markers &lt;- matrix(c(
"AA",   "AB",   "AA",   "BB",   "BA",   "AB",   "AA",   "AA",   NA,  "AA",
"AA",   "AA",   "BB",   "BB",   "AA",   "AA",   "BB",   "AA",   NA,  "AA",
"AA",   "BA",   "AB",   "BB",   "AB",   "AB",   "AA",   "BB",   NA,  "AA",
"AA",   "AA",   "BB",   "BB",   "AA",   "AA",   "AA",   "AA",   NA,  "AA",
"AA",   "AA",   "BB",   "BB",   "AA",   "BB",   "BB",   "BB",  "AB", "AA",
"AA",   "AA",   "BB",   "BB",   "AA",    NA,    "BB",   "AA",   NA,  "AA",
"AB",   "AB",   "BB",   "BB",   "BB",   "AA",   "BB",   "BB",   NA,  "AB",
"AA",   "AA",    NA,    "BB",    NA,    "AA",   "AA",   "AA",  "AA", "AA",
"AA",    NA,     NA,    "BB",   "BB",   "BB",   "BB",   "BB",  "AA", "AA",
"AA",    NA,    "AA",   "BB",   "BB",   "BB",   "AA",   "AA",   NA,  "AA"),
ncol = 10, byrow = TRUE, dimnames = list(paste0("IND", 1:10),
paste0("SNP", 1:10)))

## create object of class 'gData'.
gData &lt;- createGData(geno = markers)

## Code markers by minor allele, no imputation.
gDataCoded1 &lt;- codeMarkers(gData = gData, impute = FALSE)

## Code markers by reference alleles, impute missings by fixed value.
gDataCoded2 &lt;- codeMarkers(gData = gData,
                           refAll = rep(x = c("A", "B"), times =  5),
                           impute = TRUE, imputeType = "fixed",
                           fixedValue = 1)

## Code markers by minor allele, impute by random value.
gDataCoded3 &lt;- codeMarkers(gData = gData, impute = TRUE,
                           imputeType = "random")

</code></pre>


</div>