<div class="container">

<table style="width: 100%;"><tr>
<td>kinship</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Functions for calculating kinship matrices</h2>

<h3>Description</h3>

<p>A collection of functions for calculating kinship matrices using different
algorithms. The following algorithms are included: astle (Astle and Balding,
2009), Identity By State (IBS) and VanRaden (VanRaden, 2008) for
marker matrices. For method identity an identity kinship matrix is returned.
</p>


<h3>Usage</h3>

<pre><code class="language-R">kinship(
  X,
  method = c("astle", "IBS", "vanRaden", "identity"),
  MAF = NULL,
  denominator = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>An n x m marker matrix with genotypes in the rows (n) and markers in
the columns (m).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>The method used for computing the kinship matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MAF</code></td>
<td>
<p>The minor allele frequency (MAF) threshold used in kinship
computation. A numerical value between 0 and 1. SNPs with MAF below this
value are not taken into account when computing the kinship. If NULL all
markers are used regardless of their allele frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>denominator</code></td>
<td>
<p>A numerical value. See details.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An n x n kinship matrix.
</p>


<h3>Marker matrices</h3>

<p>In all algorithms the input matrix <code>X</code> is first cleaned, i.e. markers
with a variance of 0 are excluded from the calculation of the kinship matrix.
Then some form of scaling is done which differs per algorithm. This gives a
scaled matrix <code>Z</code>. The matrix <code class="reqn">ZZ^t / denominator</code> is returned.
By default the denominator is equal to the number of columns in <code>Z</code> for
<code>astle</code> and <code>IBS</code> and <code class="reqn">2 * p * (1-p)</code> where
<code class="reqn">p = colSums(X) / (2 * nrow(X))</code> for <code>vanRaden</code>. This denominator
can be overwritten by the user, e.g. when computing kinship matrices by
splitting <code>X</code> in smaller matrices and then adding the results together
in the end.
</p>


<h3>References</h3>

<p>Astle, William, and David J. Balding. 2009. “Population Structure
and Cryptic Relatedness in Genetic Association Studies.” Statistical Science
24 (4): 451–71. <a href="https://doi.org/10.1214/09-sts307">doi:10.1214/09-sts307</a>.
</p>
<p>VanRaden P.M. (2008) Efficient methods to compute genomic
predictions. Journal of Dairy Science 91 (11): 4414–23.
<a href="https://doi.org/10.3168/jds.2007-0980">doi:10.3168/jds.2007-0980</a>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Create example matrix.
M &lt;- matrix(c(1, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1), nrow = 4)

## Compute kinship matrices using different methods.
kinship(M, method = "astle")
kinship(M, method = "IBS")
kinship(M, method = "vanRaden")

## Only use markers with a Minor Allele Frequency of 0.3 or more.
kinship(M, method = "astle", MAF = 0.3)

## Compute kinship matrix using astle and balding method with denominator 2.
kinship(M, method = "astle", denominator = 2)

</code></pre>


</div>