<div class="container">

<table style="width: 100%;"><tr>
<td>plot.GWAS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot function for the class <code>GWAS</code>
</h2>

<h3>Description</h3>

<p>Creates a plot of an object of S3 class <code>GWAS</code>. The following types of
plot can be made:
</p>

<ul>
<li>
<p>a manhattan plot, i.e. a plot of LOD-scores per SNP
</p>
</li>
<li>
<p>a QQ plot of observed LOD-scores versus expected LOD-scores
</p>
</li>
<li>
<p>a qtl plot of effect sizes and directions for multiple traits
</p>
</li>
</ul>
<p>Manhattan plots and QQ plots are made for a single trait which
should be indicated using the parameter <code>trait</code>. If the analysis was
done for only one trait, it is detected automatically. The qtl plot will plot
all traits analyzed.<br>
See details for a detailed description of the plots and the plot options
specific to the different plots.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'GWAS'
plot(
  x,
  ...,
  plotType = c("manhattan", "qq", "qtl"),
  trial = NULL,
  trait = NULL,
  title = NULL,
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>GWAS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments to be passed on to the actual plotting
functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotType</code></td>
<td>
<p>A character string indicating the type of plot to be made.
One of "manhattan", "qq" and "qtl".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trial</code></td>
<td>
<p>A character string or numeric index indicating for which
trial the plot should be made. If <code>x</code> only contains results for
one trial, <code>trial</code> may be <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trait</code></td>
<td>
<p>A character string indicating for which trait the results
should be plotted. For <code>type</code> "qtl" all traits are plotted. If <code>x</code>
only contains results for one trait, <code>trait</code> may be <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>A character string, the title of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a list of ggplot objects is invisibly returned.</p>
</td>
</tr>
</table>
<h3>Manhattan Plot</h3>

<p>A LOD-profile of all marker positions and corresponding LOD-scores is
plotted. Significant markers are highlighted with red dots. By default these
are taken from the result of the GWAS analysis however the LOD-threshold for
significant parameters may be modified using the parameter <code>yThr</code>. The
threshold is plotted as a horizontal line. If there are previously known
marker effect, false positives and true negatives can also be marked.<br>
Extra parameter options:
</p>

<dl>
<dt><code>xLab</code></dt>
<dd>
<p>A character string, the x-axis label. Default =
<code>"Chromosomes"</code></p>
</dd>
<dt><code>yLab</code></dt>
<dd>
<p>A character string, the y-axis label. Default =
<code>-log10(p)</code></p>
</dd>
<dt><code>effects</code></dt>
<dd>
<p>A character vector, indicating which SNPs correspond
to a real (known) effect. Used for determining true/false positives and
false negatives. True positives are colored green, false positives orange and
false negatives yellow.</p>
</dd>
<dt><code>colPalette</code></dt>
<dd>
<p>A color palette used for plotting. Default
coloring is done by chromosome, using black and grey.</p>
</dd>
<dt><code>yThr</code></dt>
<dd>
<p>A numerical value for the LOD-threshold. The value from
the GWAS analysis is used as default.</p>
</dd>
<dt><code>signLwd</code></dt>
<dd>
<p>A numerical value giving the thickness of the
points that are false/true positives/negatives. Default = 0.6</p>
</dd>
<dt><code>lod</code></dt>
<dd>
<p>A positive numerical value. For the SNPs with a LOD-value
below this value, only 5% is plotted. The chance of a SNP being plotted is
proportional to its LOD-score. This option can be useful when plotting a
large number of SNPs. The 5% of SNPs plotted is selected randomly. For
reproducible results use set.seed before calling the function.</p>
</dd>
<dt><code>chr</code></dt>
<dd>
<p>A vector of chromosomes to be plotted. By default, all
chromosomes are plotted. Using this option allows restricting the plot to a
subset of chromosomes.</p>
</dd>
<dt><code>startPos</code></dt>
<dd>
<p>A numerical value indicating the start position for
the plot. Using this option allows restricting the plot to a part of a
selected chromosome. Only used if exactly one chromosome is specified in
<code>chr</code>.</p>
</dd>
<dt><code>endPos</code></dt>
<dd>
<p>A numerical value indicating the end position for
the plot. Using this option allows restricting the plot to a part of a
selected chromosome. Only used if exactly one chromosome is specified in
<code>chr</code>.</p>
</dd>
</dl>
<h3>QQ-Plot</h3>

<p>From the LOD-scores calculated in the GWAS analysis, a QQ-plot is generated
with observed LOD-scores versus expected LOD-scores. Code is adapted from
Segura et al. (2012).
</p>


<h3>QTL Plot</h3>

<p>A plot of effect sizes for the significant SNPs found in the GWAS analysis
is created. Each horizontal line contains QTLs of one trait,
phenotypic trait or trial. Optionally, vertical white lines can indicate
chromosome subdivision, genes of interest, known QTL, etc. Circle diameters
are proportional to the absolute value of allelic effect. Colors indicate the
direction of the effect: green when the allele increases the trait value,
and blue when it decreases the value.<br>
Extra parameter options:
</p>

<dl>
<dt><code>normalize</code></dt>
<dd>
<p>Should the snpEffect be normalized? Default =
<code>FALSE</code></p>
</dd>
<dt><code>sortData</code></dt>
<dd>
<p>Should the data be sorted before plotting? Either
<code>FALSE</code>, if no sorting should be done, or a character string indicating
the data column to use for sorting. This should be a numerical column.
Default = <code>FALSE</code></p>
</dd>
<dt><code>binPositions</code></dt>
<dd>
<p>An optional data.frame containing at leasts two
columns, chr(omosome) and pos(ition). Vertical lines are plotted at those
positions. Default = <code>NULL</code></p>
</dd>
<dt><code>printVertGrid</code></dt>
<dd>
<p>Should default vertical grid lines be plotted.
Default = <code>TRUE</code></p>
</dd>
<dt><code>yLab</code></dt>
<dd>
<p>A character string, the y-axis label. Default =
<code>"Traits"</code></p>
</dd>
<dt><code>yThr</code></dt>
<dd>
<p>A numerical value for the LOD-threshold. The value from
the GWAS analysis is used as default.</p>
</dd>
<dt><code>chr</code></dt>
<dd>
<p>A vector of chromosomes to be plotted. By default all
chromosomes are plotted. Using this option this can be restricted to a
subset of chromosomes.</p>
</dd>
<dt><code>exportPptx</code></dt>
<dd>
<p>Should the plot be exported to a .pptx file?
Default = <code>FALSE</code></p>
</dd>
<dt><code>pptxName</code></dt>
<dd>
<p>A character string, the name of the .pptx file to
which the plot is exported. Ignored if exportPptx = <code>FALSE</code>.</p>
</dd>
</dl>
<h3>References</h3>

<p>Millet et al. (2016) Genome-wide analysis of yield in Europe:
Allelic effects vary with drought and heat scenarios. Plant Physiology,
October 2016, Vol. 172, p. 749–764
</p>
<p>Segura et al. (2012) An efficient multi-locus mixed-model
approach for genome-wide association studies in structured populations.
Nature Genetics, June 2012, Vol. 44, p. 825–830.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Create a gData object Using the data from the DROPS project.
## See the included vignette for a more extensive description on the steps.
data(dropsMarkers)
data(dropsMap)
data(dropsPheno)

## Add genotypes as row names of dropsMarkers and drop Ind column.
rownames(dropsMarkers) &lt;- dropsMarkers[["Ind"]]
dropsMarkers &lt;- dropsMarkers[colnames(dropsMarkers) != "Ind"]

## Add genotypes as row names of dropsMap.
rownames(dropsMap) &lt;- dropsMap[["SNP.names"]]

## Rename Chomosome and Position columns.
colnames(dropsMap)[match(c("Chromosome", "Position"), 
                   colnames(dropsMap))] &lt;- c("chr", "pos")
                   
## Convert phenotypic data to a list.
dropsPhenoList &lt;- split(x = dropsPheno, f = dropsPheno[["Experiment"]])

## Rename Variety_ID to genotype and select relevant columns.
dropsPhenoList &lt;- lapply(X = dropsPhenoList, FUN = function(trial) {
  colnames(trial)[colnames(trial) == "Variety_ID"] &lt;- "genotype"
  trial &lt;- trial[c("genotype", "grain.yield", "grain.number", "seed.size",
                 "anthesis", "silking", "plant.height", "tassel.height",
                 "ear.height")]
return(trial)
}) 

## Create gData object.
gDataDrops &lt;- createGData(geno = dropsMarkers, map = dropsMap, 
                          pheno = dropsPhenoList)
                          
## Run single trait GWAS for trait 'grain.yield' for trial Mur13W.

GWASDrops &lt;- runSingleTraitGwas(gData = gDataDrops,
                               trials = "Mur13W",
                               traits = "grain.yield")
                               
## Create a manhattan plot.
plot(GWASDrops)

## Manually set a threshold for significant snps and add a title.
plot(GWASDrops, 
    yThr = 3.5, 
    title = "Manhattan plot for Mur13W")
    
## Restrict plot to part of chr 6.
plot(GWASDrops, 
    yThr = 3.5, 
    chr = 6,
    startPos = 0,
    endPos = 110000000)
    
## Create a qq plot.
plot(GWASDrops, 
    plotType = "qq",
    title = "QQ plot for Mur13W")
    
## Create a QTL plot.
plot(GWASDrops,
    plotType = "qtl",
    title = "QTL plot for Mur13W")
    
## Manually set a threshold and don't show vertical lines.
plot(GWASDrops,     
    plotType = "qtl",
    yThr = 3.5,
    printVertGrid = FALSE,
    title = "QTL plot for Mur13W")          


</code></pre>


</div>