<div class="container">

<table style="width: 100%;"><tr>
<td>estEffsCPP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimates for covariates</h2>

<h3>Description</h3>

<p>Compute the estimates and standard errors for the covariates in the input
matrix W.
</p>


<h3>Usage</h3>

<pre><code class="language-R">estEffsCPP(
  y0,
  w0,
  x0,
  vg,
  ve,
  k,
  returnSe = TRUE,
  estCom = FALSE,
  nCores = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y0</code></td>
<td>
<p>An n x p matrix of observed phenotypes, on p traits or environments
for n genotypes. No missing values are allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w0</code></td>
<td>
<p>An n x c covariate matrix, c being the number of covariates and n
being the number of genotypes. c has to be at least one (typically an
intercept). No missing values are allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x0</code></td>
<td>
<p>An n x ns matrix of marker scores. Neither missing values nor
non-segregating markers are allowed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vg</code></td>
<td>
<p>A p x p matrix of genetic covariances.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ve</code></td>
<td>
<p>A p x p matrix of environmental covariances.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>An n x n genetic relatedness matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>returnSe</code></td>
<td>
<p>Should standard errors and p-values be returned?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estCom</code></td>
<td>
<p>Should the common SNP-effect model be fitted?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCores</code></td>
<td>
<p>An integer indicating the number of cores used for parallel
computation.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing the estimates, optionally the standard errors of
the estimates and corresponding p-values. If <code>estCom = TRUE</code> also
common SNP-effects, their standard errors and corresponding p-values and
the p-values for QtlxE are output.
</p>


<h3>References</h3>

<p>Zhou, X. and Stephens, M. (2014). Efficient multivariate linear
mixed model algorithms for genome-wide association studies. Nature Methods,
February 2014, Vol. 11, p. 407â€“409
</p>


</div>