<div class="container">

<table style="width: 100%;"><tr>
<td>readRABBITMPP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read IBD probabilities</h2>

<h3>Description</h3>

<p>Read a file with IBD probabilities computed by the RABBIT software package.
It is possible to additionally read the pedigree file that is also used by
RABBIT. Reading this file allows for plotting the pedigree. Phenotypic data
can be added from a data.frame.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readRABBITMPP(infile, pedFile = NULL, pheno = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>infile</code></td>
<td>
<p>A character string, a link to a .csv file with IBD
probabilities.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pedFile</code></td>
<td>
<p>A character string, a link to a .csv file with pedigree
information as used by RABBIT as input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>A data frame with at least columns "genotype" for the
"genotype" and one or more numerical columns containing phenotypic
information. A column "cross" can be used for indicating the cross the
genotype comes from.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class <code>gDataMPP</code> with map and markers
corresponding to the imported information in the imported .csv file.
</p>


<h3>References</h3>

<p>Zheng, Chaozhi, Martin P Boer, and Fred A Van Eeuwijk.
“Recursive Algorithms for Modeling Genomic Ancestral Origins in a Fixed
Pedigree.” G3 Genes|Genomes|Genetics 8 (10): 3231–45.
https://doi.org/10.1534/G3.118.200340.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Read RABBIT data for barley.
genoFile &lt;- system.file("extdata/barley", "barley_magicReconstruct.zip",
                       package = "statgenMPP")
barleyMPMPP &lt;- readRABBITMPP(unzip(genoFile, exdir = tempdir()))

</code></pre>


</div>