<div class="container">

<table style="width: 100%;"><tr>
<td>plot.IBDprob</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot function for objects of class IBDprob</h2>

<h3>Description</h3>

<p>Creates a plot for an object of class <code>IBDprob</code>. Six types of plot can
be made:
</p>

<ul>
<li> <p><code>singleGeno</code> A plot for a single genotype showing the IBD
probabilities for all parents across the genome.
</p>
</li>
<li> <p><code>allGeno</code> A plot showing for all genotypes the IBD
probabilities of the parent with the highest probability per marker.
</p>
</li>
<li> <p><code>pedigree</code> A plot showing the structure of the pedigree of
the population.
</p>
</li>
<li> <p><code>map</code> A plot of the genetic map showing the length of the
chromosomes and the positions of the markers.
</p>
</li>
<li> <p><code>meanProbs</code> A plot showing the coverage of each parent across
the population.
</p>
</li>
<li> <p><code>totalCoverage</code> A plot showing the total coverage of each
parent.
</p>
</li>
</ul>
<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'IBDprob'
plot(
  x,
  ...,
  plotType = c("singleGeno", "allGeno", "pedigree", "map", "meanProbs", "totalCoverage"),
  genotype,
  chr = NULL,
  title = NULL,
  output = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>IBDprob</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments. Unused.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotType</code></td>
<td>
<p>A character string indicating the type of plot that should
be made.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype</code></td>
<td>
<p>A character string indicating the genotype for which the
plot should be made. Only for <code>plotType = "singleGeno"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>A character vector indicating the chromosomes to which the
coverage should be restricted. Only for <code>plotType = "meanProbs"</code> and
<code>plotType = "totalCoverage"</code>. If <code>NULL</code> all chromosomes are
included.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>A character string, the title of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>Should the plot be output to the current device? If
<code>FALSE</code>, only a ggplot object is invisibly returned.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot object is invisibly returned.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## Compute IBD probabilities for Steptoe Morex.
## Add extra evaluation positions in between exiting marker positions
## to assure evaluation positions are at most 2 cM apart.
SxMIBD_Ext &lt;- calcIBD(popType = "DH",
                      markerFile = system.file("extdata/SxM", "SxM_geno.txt",
                                              package = "statgenIBD"),
                      mapFile = system.file("extdata/SxM", "SxM_map.txt",
                                           package = "statgenIBD"),
                      evalDist = 2)

## Plot results for genotype dh005.
plot(SxMIBD_Ext,
     plotType = "singleGeno",
     genotype = "dh005")

## Plot results for all genotypes.
plot(SxMIBD_Ext,
     plotType = "allGeno")

## Plot structure of the pedigree.
plot(SxMIBD_Ext,
     plotType = "pedigree")

## Plot genetic map.
plot(SxMIBD_Ext,
     plotType = "map")

## Plot coverage across population.
plot(SxMIBD_Ext,
     plotType = "meanProbs")

## Plot total coverage.
plot(SxMIBD_Ext,
     plotType = "totalCoverage")

## End(Not run)

</code></pre>


</div>