<div class="container">

<table style="width: 100%;"><tr>
<td>receptorAbundanceEstimation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Receptor abundance estimation for single cell RNA-sequencing (scRNA-seq) data using gene set scoring and thresholding.</h2>

<h3>Description</h3>

<p>Performs receptor abundance estimation for <code class="reqn">m x n</code> scRNA-seq target data using gene set scoring and thresholding. scRNA-seq target counts are normalized and
reduced rank reconstructed (RRR) using the <code>SPECK::randomizedRRR()</code> function. Gene set scoring is next performed leveraging expression from
the top most weighted genes based on the gene sets weights membership matrix with the <code>VAM::vam()</code> function. The resulting cell-specific gene set scores
are then thresholded utilizing the <code>Ckmeans.1d.dp::Ckmeans.1d.dp()</code> function. Note that this function only performs normalization
and does not perform any quality control (QC) checks on the inputted target scRNA-seq counts matrix. Any QC needed can be performed on the target matrix before passing it
as an input to the function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">receptorAbundanceEstimation(
  target.rnaseq,
  receptor.geneset.matrix,
  num.genes = 10,
  rank.range.end = 100,
  min.consec.diff = 0.01,
  rep.consec.diff = 2,
  manual.rank = NULL,
  seed.rsvd = 1,
  max.num.clusters = 4,
  seed.ckmeans = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>target.rnaseq</code></td>
<td>
<p><code class="reqn">m x n</code> scRNA-seq counts matrix for <code class="reqn">m</code> cells and <code class="reqn">n</code> genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>receptor.geneset.matrix</code></td>
<td>
<p><code class="reqn">n x h</code> Gene sets weights membership matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.genes</code></td>
<td>
<p>Number of top most weighted genes for subsequent gene set scoring and thresholding.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rank.range.end</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>manual.rank</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed.rsvd</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.num.clusters</code></td>
<td>
<p>See documentation for the <code>ckmeansThreshold</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed.ckmeans</code></td>
<td>
<p>See documentation for the <code>ckmeansThreshold</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
</table>
<h3>Value</h3>


<ul><li> <p><code>receptor.abundance.estimates</code> - A <code class="reqn">m x h</code> matrix consisting of abundance estimates for <code class="reqn">m</code> cells in <code class="reqn">h</code> receptors.
</p>
</li></ul>
<h3>Examples</h3>

<pre><code class="language-R">data("train.malt.rna.mat")
data("train.malt.adt.mat")
receptor.geneset.matrix.out &lt;- receptorGeneSetConstruction(train.rnaseq =
                                         train.malt.rna.mat[1:100,1:80],
                                         train.citeseq =
                                         train.malt.adt.mat[1:100,1:2],
                                         rank.range.end = 70,
                                         min.consec.diff = 0.01,
                                         rep.consec.diff = 2,
                                         manual.rank = NULL,
                                         seed.rsvd = 1)
dim(receptor.geneset.matrix.out)
head(receptor.geneset.matrix.out)
data("target.malt.rna.mat")
receptor.abundance.estimates.out &lt;- receptorAbundanceEstimation(target.rnaseq =
                                    target.malt.rna.mat[1:200,1:80],
                                    receptor.geneset.matrix =
                                    receptor.geneset.matrix.out,
                                    num.genes = 10, rank.range.end = 70,
                                    min.consec.diff = 0.01,
                                    rep.consec.diff = 2,
                                    manual.rank = NULL, seed.rsvd = 1,
                                    max.num.clusters = 4, seed.ckmeans = 2)
dim(receptor.abundance.estimates.out)
head(receptor.abundance.estimates.out)
</code></pre>


</div>