<div class="container">

<table style="width: 100%;"><tr>
<td>receptorGeneSetConstruction</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene sets weights membership matrix construction for receptor abundance estimation.</h2>

<h3>Description</h3>

<p>Computes <code class="reqn">n x h</code> gene sets weights membership matrix using associations learned between log-normalized and reduced rank reconstructed (RRR)
<code class="reqn">m x n</code> scRNA-seq training data and <code class="reqn">m x h</code> CITE-seq ADT training counts normalized using the centered log ratio (CLR) transformation. scRNA-seq counts are normalized and
RRR using the <code>SPECK::randomizedRRR()</code> function while CITE-seq counts are normalized using the
<code>Seurat::NormalizeData()</code> function with the <code>normalization.method</code> parameter set to <code>CLR</code>. Spearman rank correlations are computed between the
normalized CITE-seq data and the normalized and RRR scRNA-seq data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">receptorGeneSetConstruction(
  train.rnaseq,
  train.citeseq,
  rank.range.end = 100,
  min.consec.diff = 0.01,
  rep.consec.diff = 2,
  manual.rank = NULL,
  seed.rsvd = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>train.rnaseq</code></td>
<td>
<p><code class="reqn">m x n</code> scRNA-seq counts matrix for <code class="reqn">m</code> cells
and <code class="reqn">n</code> genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>train.citeseq</code></td>
<td>
<p><code class="reqn">m x h</code> CITE-seq ADT counts matrix for <code class="reqn">m</code> cells (same cells as the <code>train.rnaseq matrix</code>) and <code class="reqn">h</code> cell-surface proteins.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rank.range.end</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep.consec.diff</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>manual.rank</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed.rsvd</code></td>
<td>
<p>See documentation for the <code>randomizedRRR</code> function from the <code>SPECK</code> package.</p>
</td>
</tr>
</table>
<h3>Value</h3>


<ul><li> <p><code>receptor.geneset.matrix</code> - A <code class="reqn">n x h</code> gene sets weights membership matrix where a column <code class="reqn">i</code> from <code class="reqn">h</code> corresponds to the weights for <code class="reqn">n</code> genes from the scRNA-seq matrix trained against the corresponding CITE-seq ADT transcript <code class="reqn">h</code>.
</p>
</li></ul>
<h3>Examples</h3>

<pre><code class="language-R">data("train.malt.rna.mat")
data("train.malt.adt.mat")
receptor.geneset.matrix.out &lt;- receptorGeneSetConstruction(train.rnaseq =
                                         train.malt.rna.mat[1:100,1:80],
                                         train.citeseq =
                                         train.malt.adt.mat[1:100,1:2],
                                         rank.range.end = 70,
                                         min.consec.diff = 0.01,
                                         rep.consec.diff = 2,
                                         manual.rank = NULL, seed.rsvd = 1)
dim(receptor.geneset.matrix.out)
head(receptor.geneset.matrix.out)
</code></pre>


</div>