<div class="container">

<table style="width: 100%;"><tr>
<td>SimulateData</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>

Data Simulation Function to Study the Performance of TreeFDR.
</h2>

<h3>Description</h3>


<p>We include five scenarios ('S1-S5'). S1-S3 are phylogeny-informative/clade-consistent scenarios while S4-S5 are
phylogeny-noninformative/clade-inconsistent scenarios. In 'S1', we simulate two large clusters of differentially
abundant OTUs. The fold changes (effect sizes) for OTUs from the same cluster are the same. In 'S2', we weaken the
assumption, and generate variable fold changes for OTUs from the same cluster. In 'S3', we simulate many small
clusters (10) of differentially abundant OTUs with the same effect sizes. In 'S4', we still simulate two large 
clusters of differentially abundant OTUs but we allow opposite effects for OTUs from the same cluster. This violates
the assumption of similar effects for closely related OTUs. In 'S5', we pick 10% random OTUs without respect to 
the underlying phyologeny. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">SimulateData(nCases = 50, nControls = 50, nOTU = 400, nCluster = 20, depth = 10000,
		     p.est, theta, scene, signal.strength = 4, otu.no.min = 40, 
		     otu.no.max = 80, zero.pct = 0, balanced = FALSE) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>nCases, nControls</code></td>
<td>


<p>the number of case and control samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nOTU</code></td>
<td>


<p>the number of OTUs simulated.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCluster</code></td>
<td>


<p>the number of clusters to be clustered. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>depth</code></td>
<td>


<p>mean library sizes/sequencing depth. The library sizes are simulated from negative binomial distribution of size=25.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.est, theta</code></td>
<td>


<p>the parameters (proportion vector and dispersion parameter) of the Dirichlet distribution.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scene</code></td>
<td>


<p>simulation scenarios. 'S1', 'S2', 'S3', 'S4', 'S5' denote five scenarios, respectively. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>signal.strength</code></td>
<td>


<p>the strength of signal (related to the mean and sd of the log fold change). 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>otu.no.min, otu.no.max</code></td>
<td>


<p>the minimum and maximum numbers of differentially abundant OTUs. Defaults are 40 and 80.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zero.pct</code></td>
<td>


<p>the percentage of non-differential OTUs within the cluster/clade. Applicable to 'S1' and 'S2'
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>balanced</code></td>
<td>


<p>a logical value indicating whether the fold changes should be multiplied to cases samples (FALSE, increase/decrease in cases, 
no change for controls) only or to both case and control samples (TRUE, increase in case and control samples). Balanced design will
have similar power for all OTUs.
</p>
</td>
</tr>
</table>
<h3>Value</h3>






<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a vector of group membership. 0 = control, 1 = case.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>a matrix of normalized OTU counts. row: OTUs; column: samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.true</code></td>
<td>
<p>a vector of the true log fold changes for all OTUs. 0s for non-differential OTUs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>D</code></td>
<td>
<p>a matrix of the cophenetic distances among the simulated OTUs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>a simulated coalescent tree of the 'phylo' class.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clustering</code></td>
<td>
<p>a vector of cluster memberships for the OTUs based on PAM.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>


<p>Jun Chen
</p>


<h3>References</h3>

<p>Xiao, J., Cao, H., Chen, J. (2017). False discovery rate control incorporating phylogenetic tree increases detection power in microbiome-wide multiple testing. Bioinformatics, 33(18), 2873-2881.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Generate data set for different scenarios S1-S5
require(StructFDR)
data(throat.parameter)
# Scene 1
data.obj &lt;- SimulateData(nCases = 50, nControls = 50, nOTU = 400, nCluster = 20, 
		depth = 10000, p.est = throat.parameter$p.est, theta = throat.parameter$theta,
        scene = 'S1', signal.strength = 4)
# Scene 2
data.obj &lt;- SimulateData(nCases = 50, nControls = 50, nOTU = 400, nCluster = 20, 
		depth = 10000, p.est = throat.parameter$p.est, theta = throat.parameter$theta,
        scene = 'S2', signal.strength = 4)
# Scene 3
data.obj &lt;- SimulateData(nCases = 50, nControls = 50, nOTU = 400, nCluster = 100, 
		depth = 10000, p.est = throat.parameter$p.est, theta = throat.parameter$theta,
        scene = 'S3', signal.strength = 4)
# Scene 4
data.obj &lt;- SimulateData(nCases = 50, nControls = 50, nOTU = 400, nCluster = 20, 
		depth = 10000, p.est = throat.parameter$p.est, theta = throat.parameter$theta,
        scene = 'S4', signal.strength = 2)
# Scene 5
data.obj &lt;- SimulateData(nCases = 50, nControls = 50, nOTU = 400, nCluster = 20, 
		depth = 10000, p.est = throat.parameter$p.est, theta = throat.parameter$theta,
        scene = 'S5', signal.strength = 4)       
</code></pre>


</div>