<div class="container">

<table style="width: 100%;"><tr>
<td>AdjStats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Prior Structure-Adjusted Statistic</h2>

<h3>Description</h3>

<p>Produce the prior structure-adjusted statistic.</p>


<h3>Usage</h3>

<pre><code class="language-R">AdjStats(y, V, k, mu, fudge=0.005)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a vector or a matrix of unadjusted z-values</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>V</code></td>
<td>
<p>a correlation matrix defined based on the prior structure</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>a numeric value representing the ratio of variance components</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>a numeric value of the mean of the prior distribution</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fudge</code></td>
<td>
<p>a small numeric value added to the diagonal of the correlation matrix to improve stability</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a vector or a matrix of adjusted z-values</p>


<h3>Author(s)</h3>


<p>Jun Chen
</p>


<h3>References</h3>


<p>Xiao, J., Cao, H., Chen, J. (2017). False discovery rate control incorporating phylogenetic tree increases detection power in microbiome-wide multiple testing. Bioinformatics, 33(18), 2873-2881.
</p>


</div>