<div class="container">

<table style="width: 100%;"><tr>
<td>MicrobiomeSeqTreeFDR</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>

False Discovery Rate (FDR) Control Integrating Prior Tree Structure for Microbiome Data Based on F-test and Residual Permutation.
</h2>

<h3>Description</h3>


<p>The function is a wrapper of <code>TreeFDR</code> with a pre-defined test based on F-statistic and residual permutation. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">MicrobiomeSeqTreeFDR (otu.tab, tree, meta.dat, grp.name, adj.name = NULL, 
raw.count = FALSE, B = 100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>otu.tab</code></td>
<td>


<p>a data matrix, rows are the features and columns are the samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>


<p>an object of "phylo" class. The tree of OTUs.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>meta.dat</code></td>
<td>


<p>a data frame containing the variables specified by <code>grp.name</code> and <code>adj.name</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp.name</code></td>
<td>


<p>a character string indicating the variable of major interest. Can be categorical or numerical.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>adj.name</code></td>
<td>


<p>a character vector indicating the variables to be adjusted. Can be categorical or numerical.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>raw.count</code></td>
<td>


<p>a logical value indicating whether <code>X</code> are raw counts. If raw counts are supplied, internal normalization/transformation (GMPR/sqrt) will be performed. 
The default is FALSE. The user should be responsible for selecting the appropriate normalization/transformation methods.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>


<p>the number of permutations.  The default is 100 since the permutation test is very fast.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>


<p>further arguments such as covariates to be passed to <code>TreeFDR</code>
</p>
</td>
</tr>
</table>
<h3>Value</h3>






<table>
<tr style="vertical-align: top;">
<td><code>p.adj </code></td>
<td>
<p>TreeFDR adjusted p-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.unadj </code></td>
<td>
<p>raw p-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>z.adj </code></td>
<td>
<p>moderated z-values. The scale may be different from the raw z-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>z.unadj </code></td>
<td>
<p>raw z-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k, rho </code></td>
<td>
<p>the estimates of the hyperparameters.  The values indicate the informativeness of the prior structure.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>


<p>Jun Chen
</p>


<h3>References</h3>

<p>Xiao, J., Cao, H., Chen, J. (2017). False discovery rate control incorporating phylogenetic tree increases detection power in microbiome-wide multiple testing. Bioinformatics, 33(18), 2873-2881.
</p>


<h3>See Also</h3>


<p><code>TreeFDR</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">require(StructFDR)

# Generate data
data(throat.parameter)
data.obj &lt;- SimulateData(nCases = 50, nControls = 50, nOTU = 400, nCluster = 20, 
		depth = 10000, p.est = throat.parameter$p.est, theta = throat.parameter$theta,
        scene = 'S2', signal.strength = 4)
meta.dat &lt;- data.frame(group = factor(data.obj$y), sex = sample(gl(2, 50)))
beta.true &lt;- data.obj$beta.true

# Call TreeFDR
tree.fdr.obj &lt;- MicrobiomeSeqTreeFDR(data.obj$X, data.obj$tree, meta.dat, 'group', 'sex', B = 20)
tree.p.adj &lt;- tree.fdr.obj$p.adj

# Empirical power and FDR
(tree.emp.pwr &lt;- sum(tree.p.adj &lt;= 0.05 &amp; beta.true != 0) / sum(beta.true != 0)) 
(tree.emp.fdr &lt;- sum(tree.p.adj &lt;= 0.05 &amp; beta.true == 0) / sum(tree.p.adj &lt;= 0.05)) 

</code></pre>


</div>