<div class="container">

<table style="width: 100%;"><tr>
<td>Ztransform</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>

Transform P-values to Z-values
</h2>

<h3>Description</h3>


<p>Transform the p-values to z-values. Both two-sided and one-sided transformations are implemented.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Ztransform(p.value, e.sign, eff.sign = TRUE, tol = 1e-15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>


<p>a vector of p-values
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>e.sign</code></td>
<td>


<p>a vector of signs of the effects, taken on value -1 and 1. (In effects when 'eff.sign = TRUE')
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eff.sign</code></td>
<td>


<p>a logical value indicating whether the direction/sign of the effect should be considered
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>


<p>a numeric value at which the p-value (both ends) will truncate. 
</p>
</td>
</tr>
</table>
<h3>Value</h3>






<p>a vector of z-values
</p>


<h3>Author(s)</h3>


<p>Jun Chen
</p>


<h3>References</h3>


<p>Xiao, J., Cao, H., Chen, J. (2017). False discovery rate control incorporating phylogenetic tree increases detection power in microbiome-wide multiple testing. Bioinformatics, 33(18), 2873-2881.
</p>


</div>