<div class="container">

<table style="width: 100%;"><tr>
<td>EstHyper</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimate the Hyper-parameter Using Generalized Least Squares</h2>

<h3>Description</h3>

<p>Estimate the hyper-parameter by calling the <code>gls</code> function. The errors are allowed to be correlated and/or have unequal variances.</p>


<h3>Usage</h3>

<pre><code class="language-R">EstHyper(y, D, init.val)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a vector of z-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>D</code></td>
<td>
<p>a distance matrix defined based on the prior structure. Diagonal have to be zeros.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init.val</code></td>
<td>
<p>initial values for the transformed hyper-parameter. Default is 0.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a vector of estimated hyper-parameter values plus log likelihood.
</p>


<h3>Author(s)</h3>


<p>Jun Chen
</p>


<h3>References</h3>


<p>Xiao, J., Cao, H., Chen, J. (2017). False discovery rate control incorporating phylogenetic tree increases detection power in microbiome-wide multiple testing. Bioinformatics, 33(18), 2873-2881.
</p>


</div>