<div class="container">

<table style="width: 100%;"><tr>
<td>gene-based test functions</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene-based tests on summary statistics</h2>

<h3>Description</h3>

<p>A set of tests for gene-based association analysis on GWAS summary statistics:
sequence kernel association tests ('SKAT', 'SKATO'), sum of chi-squares ('sumchi'),
aggregated Cauchy association test ('ACAT'), burden test ('BT'), principal component
analysis-based test ('PCA'), functional linear model-based test ('FLM'),
Bonferroni correction test ('simpleM'), minimum P value test ('minp').
</p>


<h3>Usage</h3>

<pre><code class="language-R">SKAT(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", beta.par = c(1, 25),
weights.function = NULL, user.weights = FALSE, gen.var.weights = "se.beta",
method = "kuonen", acc = 1e-8, lim = 1e+6, rho = FALSE, p.threshold = 0.8,
write.file = FALSE, quiet = FALSE)

SKATO(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", beta.par = c(1, 25),
weights.function = NULL, user.weights = FALSE, method = "kuonen",
acc = 1e-8, lim = 1e+6, rho = TRUE, p.threshold = 0.8, write.file = FALSE,
quiet = FALSE)

sumchi(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", method = "kuonen", acc = 1e-8,
lim = 1e+6, write.file = FALSE, quiet = FALSE)

ACAT(score.file, gene.file, genes = "all", anno.type = "",
beta.par = c(1, 1), weights.function = NULL, user.weights = FALSE,
gen.var.weights = "none", mac.threshold = NA, n = NA,
write.file = FALSE, quiet = FALSE)

BT(score.file, gene.file, genes = "all", cor.path = "cor/",
anno.type = "", beta.par = c(1, 25), weights.function = NULL,
user.weights = FALSE, write.file = FALSE, quiet = FALSE)

PCA(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", n, beta.par = c(1, 1),
weights.function = NULL, user.weights = FALSE, reference.matrix = TRUE,
fun = "LH", var.fraction = 0.85, write.file = FALSE, quiet = FALSE)

FLM(score.file, gene.file, genes = "all", cor.path = "cor/",
approximation = TRUE, anno.type = "", n, beta.par = c(1, 1),
weights.function = NULL, user.weights = FALSE, basis.function = "fourier",
k = 25, order = 4, flip.genotypes = FALSE, Fan = TRUE,
reference.matrix = TRUE, fun = "LH", write.file = FALSE, quiet = FALSE)

simpleM(score.file, gene.file, genes = "all", cor.path = "cor/",
anno.type = "", var.fraction = .995, write.file = FALSE, quiet = FALSE)

minp(score.file, gene.file, genes = "all", cor.path = "cor/",
anno.type = "", write.file = FALSE, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>score.file</code></td>
<td>
<p>name of data file generated by <code>prep.score.files()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene.file</code></td>
<td>
<p>can be a name of a custom text file listing genes in refFlat format. Values "hg19" and "hg38"
can be set to use default gene files containing protein-coding genes with start and stop positions from corresponding build.
Mind that the same build should be used in <code>score.file</code> and <code>gene.file</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>character vector of gene names to be analyzed. Can be "chr1", "chr2" etc. to analyze
all genes on a corresponding chromosome. If not set, function will
attempt to analyze all genes listed in <code>gene.file</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cor.path</code></td>
<td>
<p>path to a folder with correlation matrix files (one file per each gene to be analyzed).
Correlation matrices in text format are allowed, though ".RData" is preferable as computationally efficient.
Each file should contain a square matrix with correlation coefficients (r) between genetic variants
of a gene. An example of correlation file format:<br>
"snpname1" "snpname2" "snpname3" ...<br>
"snpname1" 1 0.018 -0.003 ...<br>
"snpname2" 0.018 1 0.081 ...<br>
"snpname3" -0.003 0.081 1 ...<br>
...<br>
If genotypes are available, matrices can be generated as follows:<br><code>cor.matrix &lt;- cor(g)</code><br><code>save(cor.matrix, file = paste0(geneName, ".RData"))</code><br>
where <code>g</code> is a genotype matrix (nsample x nvariants) for a given gene with genotypes coded as 0, 1, 2
(coding should be exactly the same that was used to generate GWAS statistics).<br>
If genotypes are not available, correlations can be approximated through reference samples.
Reference matrices from 1KG European sample can be downloaded at http://mga.bionet.nsc.ru/sumFREGAT/.<br>
Names of correlation files should be constructed as "geneName.RData" (e.g. "ABCG1.RData", "ADAMTS1.RData", etc.)
for ".RData" format or "geneName.txt" for text format.<br>
Example corfiles can be found as:<br>
system.file("testfiles/CFH.cor", package = "sumFREGAT")<br>
system.file("testfiles/CFH.RData", package = "sumFREGAT")<br></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>approximation</code></td>
<td>
<p>a logical value indicating whether approximation for large genes (&gt;= 500 SNPs) should be used.
Applicable for SKAT, SKATO, sumchi, PCA, FLM (default = TRUE for these methods).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anno.type</code></td>
<td>
<p>given (functional) annotations provided by user (see) <code>prep.score.files()</code>,
a character (or character vector) indicating annotation types to be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>size of the sample on which summary statistics were obtained.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta.par</code></td>
<td>
<p>two positive numeric shape parameters in the beta distribution to assign weights 
for each genetic variant as a function of minor allele frequency (MAF) in the default weights function (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights.function</code></td>
<td>
<p>a function of MAF to assign weights
for each genetic variant. By default, the weights will be calculated using the beta distribution (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>user.weights</code></td>
<td>
<p>a logical value indicating whether weights from the input file should be applied. Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen.var.weights</code></td>
<td>
<p>a character indicating whether scores should be weighted by the variance of genotypes:
"none" - no weights applied, resulting in a sum chi-square test.
"se.beta" - scores weighted by variance of genotypes estimated from P values and effect sizes (regression coefficients)
if provided by user.
"af" - scores weighted by variance of genotypes calculated as AF * (1 - AF), where AF is allele frequency. 
In <code>SKAT()</code>, "se.beta" weighting results in a test analogous to a standard SKAT, while "af" is a way to approximate
the standard test in case when betas are not available. Unweighted sum chi-square test gives more weights to common variants
compared with SKAT-like tests.
In <code>ACAT()</code>, the simplest default unweighted test combines P values (Z scores) without any additional info.
Scores can be weighted by via gen.var.weights = 'se.beta' or 'af', similarly to SKAT.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mac.threshold</code></td>
<td>
<p> an integer number. In ACAT, scores with MAC &lt;= 10 will be combined using Burden test. MACs are calculated
from MAFs, <code>n</code> must be set to apply <code>mac.threshold</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>the method for computing P value in kernel-based tests. Available methods are "kuonen", "davies" and "hybrid"
(see Details). Default = "kuonen".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>acc</code></td>
<td>
<p>accuracy parameter for "davies" method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lim</code></td>
<td>
<p>limit parameter for "davies" method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho</code></td>
<td>
<p>if TRUE the optimal test (SKAT-O) is performed [Lee et al., 2012]. <code>rho</code> can be a vector of grid
values from 0 to 1. The default grid is c(0, 0.1^2, 0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.threshold</code></td>
<td>
<p>the largest P value that will be considered as important when performing computational optimization in SKAT-O.
All P values larger than p.threshold will be processed via burden test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>basis.function</code></td>
<td>
<p>a basis function type for beta-smooth. Can be set to "bspline" (B-spline basis) or
"fourier" (Fourier basis, default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>the number of basis functions to be used for beta-smooth (default = 25).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>order</code></td>
<td>
<p>a polynomial order to be used in "bspline". Default = 4 corresponds to the cubic B-splines.
as no effect if only Fourier bases are used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flip.genotypes</code></td>
<td>
<p>a logical value indicating whether the genotypes of some genetic variants should be
flipped (relabelled) for their better functional representation [Vsevolozhskaya et al., 2014]. Default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Fan</code></td>
<td>
<p>if TRUE (default) then linearly dependent genetic variants will be omitted, as it
was done in the original realization of FLM test by Fan et al. (2013).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference.matrix</code></td>
<td>
<p>logical indicating whether the correlation matrices were generated using reference matrix.
If TRUE, regularization algorithms will be applied in order to ensure invertibility and numerical stability of the matrices.
Use 'reference.matrix = FALSE' ONLY IF you are sure that correlation matrices were generated using the same genotype data
as for GWAS summary statistics in input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fun</code></td>
<td>
<p>one of two regularization algorithms, 'LH' (default) or 'derivLH'. Currently both give similar results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.fraction</code></td>
<td>
<p>minimal proportion of genetic variance within region that should be explained by principal
components used (see Details for more info).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>write.file</code></td>
<td>
<p>output file name. If specified, output (as it proceeds) will be written 
to the file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>quiet = TRUE suppresses excessive output from reading vcf file genewise.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>'SKAT' uses the linear weighted kernel function to set the inter-individual
similarity matrix <code class="reqn">K = GWWG^T</code> for SKAT and <code class="reqn">K = GW(I\rho + (1 - \rho)ee^T)WG^T</code> for SKAT-O,
where <code class="reqn">\mathit{G}</code> is the <code class="reqn">\mathit{n\times p}</code>
genotype matrix for <code class="reqn">\mathit{n}</code> individuals and <code class="reqn">\mathit{p}</code> genetic variants in the region, 
<code class="reqn">\mathit{W}</code> is the <code class="reqn">\mathit{p\times p}</code> diagonal weight matrix, <code class="reqn">I</code> is the <code class="reqn">\mathit{p\times p}</code>
identity matrix, <code class="reqn">\rho</code> is pairwise correlation
coefficient between genetic effects (which will be adaptively selected from given <code>rho</code>),
and <code class="reqn">e</code> is the <code class="reqn">\mathit{p\times 1}</code> vector of ones. Given the shape parameters
of the beta function, <code>beta.par = c(a, b)</code>, 
the weights are defined using probability density function of the beta distribution:<br><br><code class="reqn">W_{i}=(B(a,b))^{^{-1}}MAF_{i}^{a-1}(1-MAF_{i})^{b-1} </code>,<br><br>
where <code class="reqn">MAF_{i}</code> is a minor allelic frequency for the <code class="reqn">i^{th}</code> genetic variant in the region,
which is estimated from genotypes, and <code class="reqn">B(a,b)</code> is the beta function. This way of defining weights
is the same as in original SKAT (see [Wu et al., 2011] for details). <code>beta.par = c(1, 1)</code> corresponds
to the unweighted SKAT. The same weighting principle can be applied in 'BT' (burden test, default weights <code>beta.par = c(1, 25)</code>),
as well as in 'PCA' and 'FLM' tests (unweighted by default).<br></p>
<p>Depending on the method option chosen, either Kuonen or Davies method is used to calculate P values from the
score statistics in SKAT. Both an Applied Statistics algorithm that inverts the characteristic function
of the mixture chisq [Davies, 1980] and a saddlepoint approximation [Kuonen, 1999] are nearly exact,
with the latter usually being a bit faster.<br></p>
<p>A hybrid approach has been proposed by Wu et al. [2016]. It uses the Davies' method with high accuracy,
and then switches to the saddlepoint approximation method when the Davies' method fails to converge.
This approach yields more accurate results in terms of type I errors, especially for small significance levels.
However, 'hybrid' method runs several times slower than the saddlepoint approximation method itself (i.e. 'kuonen' method).
We therefore recommend using the hybrid approach only for those regions that show significant (or nearly significant)
P values to ensure their accuracy.<br></p>
<p>'ACAT' is an adaptation of a set-based aggregated Cauchy association test (ACAT-V). It has been recently proposed
by Liu, Y. et al. (2019). This in the only gene-based test statistic that can be calculated from P values (Z scores) only,
and does not require SNP-SNP correlation info.
</p>
<p>Burden test ('BT', collapsing technique) suggests that the effects of causal genetic variants within a region have
the same direction and the majority of variants are causal. If this is not the case, other regional tests (SKAT and FLM)
are shown to have higher power compared to burden test [Svishcheva et al., 2015]. By default, 'BT' assigns weights
calculated using the beta distribution with shape parameters <code>beta.par = c(1, 25)</code>.<br></p>
<p>'PCA' test is based on the spectral decomposition of
correlation matrix among genetic variants. The number of top principal components will be chosen
in such a way that &gt;= <code>var.fraction</code> of region variance can be explained by these PCs.
By default, <code>var.fraction</code> = 0.85, i.e. PCs explain &gt;= 85% of region variance.<br></p>
<p>A similar principle is used in 'simpleM' to calculate the effective number of independent tests.<br></p>
<p>'FLM' test assumes that the effects of multiple genetic variants
can be described as a continuous function, which can be modelled through B-spline
or Fourier basis functions. When the number of basis functions
(set by <code class="reqn">k</code>) is less than the number of variants
within the region, the FLM test has an advantage of using
less degrees of freedom [Svishcheva, et al., 2015].<br></p>
<p>For genes with <code class="reqn">m &lt;= k</code>, functional linear models are equivalent to a
standard multiple linear regression, and the latter is used for these cases.
The ultimate model name is returned in output (the "model" column).<br></p>


<h3>Value</h3>

<p>A data frame with map info, P values, numbers of variants
and filtered variants for each of analyzed genes.
<br>
In addition:
<br>
- BT() returns gene-level estimates of effect sizes (betas) and their standard errors.
<br>
- FLM() returns the names of the functional models used for each region.
Names shortly describe the functional basis and the number of basis functions used.
E.g., "F25" means 25 Fourier basis functions, "B15" means 15 B-spline basis functions.
<br>
- PCA() returns the number of the principal components
used for each region and the proportion of genetic variance
they make up.
</p>


<h3>References</h3>

<p>Davies R.B. (1980) Algorithm AS 155: The Distribution of a Linear Combination of chi-2 Random Variables, Journal of the Royal Statistical Society. Series C (Applied Statistics), Vol. 29, N 3, P. 323-333.<br>
Kuonen D. (1999) Saddlepoint Approximations for Distributions of Quadratic Forms in Normal Variables. Biometrika, Vol. 86, No. 4, P. 929-935.<br>
Wu M.C., et al. (2011) Rare-variant association testing for sequencing data with the sequence kernel association test. Am. J. Hum. Genet., Vol. 89, P. 82-93.<br>
Lee S., et al. (2012) Optimal unified approach for rare variant association testing with application to small sample case-control whole-exome sequencing studies. American Journal of Human Genetics, 91, 224-237.<br>
Liu Y. et al. (2019) ACAT: a fast and powerful p value combination method for rare-variant analysis in sequencing studies. Am. J. Hum. Genet. 104, 410-421.
Svishcheva G.R., Belonogova N.M. and Axenovich T.I. (2015) Region-based association test for familial data under functional linear models. PLoS ONE 10(6): e0128999.<br>
Wu B., et al. (2016) On efficient and accurate calculation of significance p-values for sequence kernel association testing of variant set. Ann Hum Genet, 80(2): 123-135.<br>
Vsevolozhskaya O.A., et al. (2014) Functional Analysis of Variance for Association Studies. PLoS ONE 9(9): e105074.<br>
Fan R, Wang Y, Mills JL, Wilson AF, Bailey-Wilson JE, et al. (2013) Functional linear models for association analysis of quantitative traits. Genet Epidemiol 37: 726-42.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Using example score files "CFH.vcf.gz" and "CFH.full.vcf.gz"
# generated by prep.score.files() function (see examples for prep.score.files())

# Test available with minimal input (P values only):

score.file &lt;- system.file("testfiles/CFH.vcf.gz",
	package = "sumFREGAT")
ACAT(score.file, gene.file = "hg19", genes = "CFH")

# Tests that require P values, rsID, and SNP-SNP correlation info:

score.file &lt;- system.file("testfiles/CFH.vcf.gz",
	package = "sumFREGAT")
cor.path &lt;- system.file("testfiles/", package = "sumFREGAT")

sumchi(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
simpleM(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
minp(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)

# Tests available with full input including P values, rsID,
# betas, effect allele, allele frequencies (for default weighting),
# and correlation matrices:

score.file &lt;- system.file("testfiles/CFH.full.vcf.gz",
	package = "sumFREGAT")
cor.path &lt;- system.file("testfiles/", package = "sumFREGAT")

BT(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
SKAT(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)
SKATO(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path)

# Tests that require sample size to be provided as "n" argument:

PCA(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path, n = 85)
FLM(score.file, gene.file = "hg19", genes = "CFH", cor.path = cor.path, n = 85)

</code></pre>


</div>