<div class="container">

<table style="width: 100%;"><tr>
<td>leukemia</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A part of the Golub's famous AML/ALL-leukemia dataset</h2>

<h3>Description</h3>

<p>Part of the training set of the famous AML/ALL-leukemia
dataset from the Whitehead Institute.  It has been reduced to 250
genes, about half of which are very informative for classification,
whereas the other half was chosen randomly.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(leukemia)
</code></pre>


<h3>Format</h3>

<p>Contains three <span style="font-family: Courier New, Courier; color: #666666;"><b>R</b></span>-objects:
The expression (38 x 250) matrix <code>leukemia.x</code>,
the associated binary (<code>0,1</code>) response variable <code>leukemia.y</code>,
and the associated 3-class response variable <code>leukemia.z</code> with
values in <code>0,1,2</code>.
</p>


<h3>Author(s)</h3>

<p>Marcel Dettling</p>


<h3>Source</h3>

<p>originally at <code>http://www.genome.wi.mit.edu/MPR/</code>, (which is
not a valid URL any more).
</p>


<h3>References</h3>

<p>First published in<br>
Golub et al. (1999)
Molecular Classification of Cancer: Class Discovery and Class
Prediction by Gene Expression Monitoring.
<em>Science</em> <b>286</b>, 531â€“538.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(leukemia, package="supclust")
str(leukemia.x)
str(leukemia.y)
str(leukemia.z)
op &lt;- par(mfrow= 1:2)
plot(leukemia.x[,56], leukemia.y)
plot(leukemia.x[,174],leukemia.z)
par(op)
</code></pre>


</div>