<div class="container">

<table style="width: 100%;"><tr>
<td>supercell_GE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simplification of scRNA-seq dataset</h2>

<h3>Description</h3>

<p>This function converts (i.e., averages or sums up) gene-expression matrix of single-cell data into a gene expression
matrix of metacells
</p>


<h3>Usage</h3>

<pre><code class="language-R">supercell_GE(
  ge,
  groups,
  mode = c("average", "sum"),
  weights = NULL,
  do.median.norm = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ge</code></td>
<td>
<p>gene expression matrix (or any coordinate matrix) with genes as rows and cells as cols</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groups</code></td>
<td>
<p>vector of membership (assignment of single-cell to metacells)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>string indicating whether to average or sum up 'ge' within metacells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>vector of a cell weight (NULL by default), used for computing average gene expression withing cluster of metaells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.median.norm</code></td>
<td>
<p>whether to normalize by median value (FALSE by default)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a matrix of simplified (averaged withing groups) data with ncol equal to number of groups and nrows as in the initial dataset
</p>


</div>