<div class="container">

<table style="width: 100%;"><tr>
<td>supercell_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot metacell NW</h2>

<h3>Description</h3>

<p>Plot metacell NW
</p>


<h3>Usage</h3>

<pre><code class="language-R">supercell_plot(
  SC.nw,
  group = NULL,
  color.use = NULL,
  lay.method = c("nicely", "fr", "components", "drl", "graphopt"),
  lay = NULL,
  alpha = 0,
  seed = 12345,
  main = NA,
  do.frames = TRUE,
  do.extra.log.rescale = FALSE,
  do.directed = FALSE,
  log.base = 2,
  do.extra.sqtr.rescale = FALSE,
  frame.color = "black",
  weights = NULL,
  min.cell.size = 0,
  return.meta = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SC.nw</code></td>
<td>
<p>a super-cell (metacell) network (a field <code>supercell_network</code> of the output of SCimplify)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>an assigment of metacells to any group (for ploting in different colors)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color.use</code></td>
<td>
<p>colros to use for groups, if <code>NULL</code>, an automatic palette of colors will be applied</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lay.method</code></td>
<td>
<p>method to compute layout of the network (for the moment there several available: "nicely"
for layout_nicely and "fr" for layout_with_fr, "components" for layout_components,
"drl" for layout_with_drl, "graphopt" for layout_with_graphopt). If your dataset has clear clusters, use "components"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lay</code></td>
<td>
<p>a particular layout of a graph to plot (in is not <code>NULL</code>, <code>lay.method</code> is ignored and new layout is not computed)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>a rotation of the layout (either provided or computed)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>a random seed used to compute graph layout</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>a title of a plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.frames</code></td>
<td>
<p>whether to keep vertex.frames in the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.extra.log.rescale</code></td>
<td>
<p>whether to log-scale node size (to balance plot if some metacells are large and covers smaller metacells)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.directed</code></td>
<td>
<p>whether to plot edge direction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.base</code></td>
<td>
<p>base with thich to log-scale node size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.extra.sqtr.rescale</code></td>
<td>
<p>whether to sqrt-scale node size (to balance plot if some metacells are large and covers smaller metacells)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frame.color</code></td>
<td>
<p>color of node frames, black by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>
<p>edge weights used for some layout algorithms</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.cell.size</code></td>
<td>
<p>do not plot cells with smaller size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.meta</code></td>
<td>
<p>whether to return all the meta data</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>plot of a super-cell network
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(cell_lines) # list with GE - gene expression matrix (logcounts), meta - cell meta data
GE &lt;- cell_lines$GE
cell.meta &lt;- cell_lines$meta

SC &lt;- SCimplify(GE,  # gene expression matrix
                gamma = 20) # graining level

# Assign metacell to a cell line
SC2cellline  &lt;- supercell_assign(
    clusters = cell.meta, # single-cell assignment to cell lines
    supercell_membership = SC$membership) # single-cell assignment to metacells

# Plot metacell network colored by cell line
supercell_plot(SC$graph.supercells, # network
               group = SC2cellline, # group assignment
               main = "Metacell colored by cell line assignment",
               lay.method = 'nicely')


</code></pre>


</div>