<div class="container">

<table style="width: 100%;"><tr>
<td>SCimplify</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Detection of metacells with the SuperCell approach</h2>

<h3>Description</h3>

<p>This function detects metacells (former super-cells) from single-cell gene expression matrix
</p>


<h3>Usage</h3>

<pre><code class="language-R">SCimplify(
  X,
  genes.use = NULL,
  genes.exclude = NULL,
  cell.annotation = NULL,
  cell.split.condition = NULL,
  n.var.genes = min(1000, nrow(X)),
  gamma = 10,
  k.knn = 5,
  do.scale = TRUE,
  n.pc = 10,
  fast.pca = TRUE,
  do.approx = FALSE,
  approx.N = 20000,
  block.size = 10000,
  seed = 12345,
  igraph.clustering = c("walktrap", "louvain"),
  return.singlecell.NW = TRUE,
  return.hierarchical.structure = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>log-normalized gene expression matrix with rows to be genes and cols to be cells</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes.use</code></td>
<td>
<p>a vector of genes used to compute PCA</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes.exclude</code></td>
<td>
<p>a vector of genes to be excluded when computing PCA</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell.annotation</code></td>
<td>
<p>a vector of cell type annotation, if provided, metacells that contain single cells of different cell type annotation will be split in multiple pure metacell (may result in slightly larger numbe of metacells than expected with a given gamma)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cell.split.condition</code></td>
<td>
<p>a vector of cell conditions that must not be mixed in one metacell. If provided, metacells will be split in condition-pure metacell (may result in significantly(!) larger number of metacells than expected)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.var.genes</code></td>
<td>
<p>if <code>"genes.use"</code> is not provided, <code>"n.var.genes"</code> genes with the largest variation are used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>
<p>graining level of data (proportion of number of single cells in the initial dataset to the number of metacells in the final dataset)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k.knn</code></td>
<td>
<p>parameter to compute single-cell kNN network</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.scale</code></td>
<td>
<p>whether to scale gene expression matrix when computing PCA</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.pc</code></td>
<td>
<p>number of principal components to use for construction of single-cell kNN network</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fast.pca</code></td>
<td>
<p>use irlba as a faster version of prcomp (one used in Seurat package)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.approx</code></td>
<td>
<p>compute approximate kNN in case of a large dataset (&gt;50'000)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>approx.N</code></td>
<td>
<p>number of cells to subsample for an approximate approach</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block.size</code></td>
<td>
<p>number of cells to map to the nearest metacell at the time (for approx coarse-graining)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>seed to use to subsample cells for an approximate approach</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>igraph.clustering</code></td>
<td>
<p>clustering method to identify metacells (available methods "walktrap" (default) and "louvain" (not recommended, gamma is ignored)).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.singlecell.NW</code></td>
<td>
<p>whether return single-cell network (which consists of approx.N if <code>"do.approx"</code> or all cells otherwise)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.hierarchical.structure</code></td>
<td>
<p>whether return hierarchical structure of metacell</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>other parameters of build_knn_graph function</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list with components
</p>

<ul>
<li>
<p> graph.supercells - igraph object of a simplified network (number of nodes corresponds to number of metacells)
</p>
</li>
<li>
<p> membership - assigmnent of each single cell to a particular metacell
</p>
</li>
<li>
<p> graph.singlecells - igraph object (kNN network) of single-cell data
</p>
</li>
<li>
<p> supercell_size - size of metacells (former super-cells)
</p>
</li>
<li>
<p> gamma - requested graining level
</p>
</li>
<li>
<p> N.SC - number of obtained metacells
</p>
</li>
<li>
<p> genes.use - used genes
</p>
</li>
<li>
<p> do.approx - whether approximate coarse-graining was perfirmed
</p>
</li>
<li>
<p> n.pc - number of principal components used for metacells construction
</p>
</li>
<li>
<p> k.knn - number of neighbors to build single-cell graph
</p>
</li>
<li>
<p> sc.cell.annotation. - single-cell cell type annotation (if provided)
</p>
</li>
<li>
<p> sc.cell.split.condition. - single-cell split condition (if provided)
</p>
</li>
<li>
<p> SC.cell.annotation. - super-cell cell type annotation (if was provided for single cells)
</p>
</li>
<li>
<p> SC.cell.split.condition. - super-cell split condition (if was provided for single cells)
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
data(cell_lines) # list with GE - gene expression matrix (logcounts), meta - cell meta data
GE &lt;- cell_lines$GE

SC &lt;- SCimplify(GE,  # log-normalized gene expression matrix
                gamma = 20, # graining level
                n.var.genes = 1000,
                k.knn = 5, # k for kNN algorithm
                n.pc = 10, # number of principal components to use
                do.approx = TRUE) #


</code></pre>


</div>