<div class="container">

<table style="width: 100%;"><tr>
<td>supercell_2_Seurat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Super-cells to Seurat object</h2>

<h3>Description</h3>

<p>This function transforms super-cell gene expression and super-cell partition into Seurat object
</p>


<h3>Usage</h3>

<pre><code class="language-R">supercell_2_Seurat(
  SC.GE,
  SC,
  fields = c(),
  var.genes = NULL,
  do.preproc = TRUE,
  is.log.normalized = TRUE,
  do.center = TRUE,
  do.scale = TRUE,
  N.comp = NULL,
  output.assay.version = "v4"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SC.GE</code></td>
<td>
<p>gene expression matrix with genes as rows and cells as columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SC</code></td>
<td>
<p>super-cell (output of SCimplify function)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fields</code></td>
<td>
<p>which fields of <code>SC</code> to use as cell metadata</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var.genes</code></td>
<td>
<p>set of genes used as a set of variable features of Seurat (by default is the set of genes used to generate super-cells), ignored if <code>!do.preproc</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.preproc</code></td>
<td>
<p>whether to do prepocessing, including data normalization, scaling, HVG, PCA, nearest neighbors, <code>TRUE</code> by default, change to <code>FALSE</code> to speed up conversion</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.log.normalized</code></td>
<td>
<p>whether <code>SC.GE</code> is log-normalized counts. If yes, then Seurat field <code>data</code> is replaced with <code>counts</code> after normalization (see 'Details' section), ignored if <code>!do.preproc</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.center</code></td>
<td>
<p>whether to center gene expression matrix to compute PCA, ignored if <code>!do.preproc</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>do.scale</code></td>
<td>
<p>whether to scale gene expression matrix to compute PCA, ignored if <code>!do.preproc</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N.comp</code></td>
<td>
<p>number of principal components to use for construction of single-cell kNN network, ignored if <code>!do.preproc</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.assay.version</code></td>
<td>
<p>version of the seurat assay in output, <code>`"v4"`</code> by default, <code>`"v5"`</code> requires Seurat v5 installed.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Since the input of CreateSeuratObject should be unnormalized count matrix (UMIs or TPMs, see CreateSeuratObject).
Thus, we manually set field <code>`assays$RNA@data`</code> to <code>SC.GE</code> if <code>is.log.normalized == TRUE</code>.
Avoid running NormalizeData for the obtained Seurat object, otherwise this will overwrite field <code>`assays$RNA@data`</code>.
If you have run NormalizeData, then make sure to replace <code>`assays$RNA@data`</code> with correct matrix by running
<code>`your_seurat@assays$RNA@data &lt;- your_seurat@assays$RNA@counts`</code>.
</p>
<p>Since super-cells have different size (consist of different number of single cells), we use sample-weighted algorithms for all
possible steps of the downstream analysis, including scaling and dimensionality reduction. Thus, generated Seurat object  comes
with the results of sample-wighted scaling (available as <code>`your_seurat@assays$RNA@scale.data`</code> or
<code>`your_seurat@assays$RNA@misc[["scale.data.weighted"]]`</code> to reproduce if the first one has been overwritten) and PCA (available as
<code>`your_seurat@reductions$pca`</code> or <code>`your_seurat@reductions$pca_weighted`</code> to reproduce if the first one has been overwritten).
</p>


<h3>Value</h3>

<p>Seurat object
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(cell_lines)
SC           &lt;- SCimplify(X=cell_lines$GE, gamma = 20)
SC$ident     &lt;- supercell_assign(clusters = cell_lines$meta, supercell_membership = SC$membership)
SC.GE        &lt;- supercell_GE(cell_lines$GE, SC$membership)
m.seurat     &lt;- supercell_2_Seurat(SC.GE = SC.GE, SC = SC, fields = c("ident"))

</code></pre>


</div>