<div class="container">

<table style="width: 100%;"><tr>
<td>supercell_merge</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Merging independent SuperCell objects</h2>

<h3>Description</h3>

<p>This function merges independent SuperCell objects
</p>


<h3>Usage</h3>

<pre><code class="language-R">supercell_merge(SCs, fields = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>SCs</code></td>
<td>
<p>list of SuperCell objects (results of SCimplify )</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fields</code></td>
<td>
<p>which additional fields (e.g., metadata) of the the SuperCell objects to keep when merging</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list with components
</p>

<ul>
<li>
<p> membership - assignment of each single cell to a particular metacell
</p>
</li>
<li>
<p> cell.ids - the original ids of single-cells
</p>
</li>
<li>
<p> supercell_size - size of metacells (former super-cells)
</p>
</li>
<li>
<p> gamma -  graining level of the merged object (estimated as an average size of metacells as the independent SuperCell objects might have different graining levels)
</p>
</li>
<li>
<p> N.SC - number of obtained metacells
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">
data(cell_lines) # list with GE - gene expression matrix (logcounts), meta - cell meta data
GE &lt;- cell_lines$GE
cell.meta &lt;- cell_lines$meta

cell.idx.HCC827 &lt;- which(cell.meta == "HCC827")
cell.idx.H838   &lt;- which(cell.meta == "H838")

SC.HCC827 &lt;- SCimplify(GE[,cell.idx.HCC827],  # log-normalized gene expression matrix
                gamma = 20, # graining level
                n.var.genes = 1000,
                k.knn = 5, # k for kNN algorithm
                n.pc = 10) # number of principal components to use
SC.HCC827$cell.line &lt;- supercell_assign(
    cell.meta[cell.idx.HCC827],
    supercell_membership = SC.HCC827$membership)

SC.H838 &lt;- SCimplify(GE[,cell.idx.H838],  # log-normalized gene expression matrix
                gamma = 30, # graining level
                n.var.genes = 1000, # number of top var genes to use for the dim reduction
                k.knn = 5, # k for kNN algorithm
                n.pc = 15) # number of proncipal components to use
SC.H838$cell.line &lt;- supercell_assign(
    cell.meta[cell.idx.H838],
    supercell_membership = SC.H838$membership)

SC.merged &lt;- supercell_merge(list(SC.HCC827, SC.H838), fields = c("cell.line"))

# compute metacell gene expression for SC.HCC827
SC.GE.HCC827 &lt;- supercell_GE(GE[, cell.idx.HCC827], groups = SC.HCC827$membership)
# compute metacell gene expression for SC.H838
SC.GE.H838 &lt;- supercell_GE(GE[, cell.idx.H838], groups = SC.H838$membership)
# merge GE matricies
SC.GE.merged &lt;- supercell_mergeGE(list(SC.GE.HCC827, SC.GE.H838))


</code></pre>


</div>