<div class="container">

<table style="width: 100%;"><tr>
<td>survSNP.power.table</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Table for Asymptotic and Empirical Power
</h2>

<h3>Description</h3>

<p>This function produces a table with the resulting empirical and
asymptotic power over a given range of Genotype Hazard Ratios,
relative allelic frequencies, sample sizes, and event rates.
</p>


<h3>Usage</h3>

<pre><code class="language-R">survSNP.power.table(GHRs, ns, rafs, erates, pilm, lm, model, test, alpha, 
                    exactvar = FALSE, B = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>GHRs</code></td>
<td>

<p>A vector of Genotype Hazard Ratios
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ns</code></td>
<td>

<p>A vector of sample sizes
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rafs</code></td>
<td>

<p>A vector of relative allelic frequencies (for the risk allele) 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>erates</code></td>
<td>

<p>A vector of event rates
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pilm</code></td>
<td>

<p>Probability that the time-to-event is greater than lm
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lm</code></td>
<td>

<p>Landmark time used for powering the study
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>

<p>True genetic risk model (choices are "additive", "recessive" or
"dominant". For the asymptotic calculations only the "additive"
should be used until further notice)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>

<p>Hypothesized genetic risk model (choices are "additive", "recessive"
or "dominant". For the asymptotic calculations only the "additive" should
be used until further notice)
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>

<p>Nominal two-sided type I error rate
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exactvar</code></td>
<td>

<p>Indicator for using the exact variance formula
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>

<p>Number of simulation replicates (set to 0 if no empirical
calculations are desired)
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This version only supports additive models.
</p>


<h3>Value</h3>

<p>See output of sim.snp.expsurv.power.
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">GRRs&lt;-seq(1.5,2,by=0.25)
ns&lt;-c(100,500,1000)
rafs&lt;-c(0.3,0.5,0.7)
erates=c(0.5,0.7,0.9)
res&lt;-survSNP.power.table(GRRs,ns,rafs,erates,pilm=0.5,lm=1,model="additive",
                         test="additive",alpha=0.05)

# Create key for illustration
KEY=paste("q=",levels(factor(res$raf)),sep="")
KEY&lt;-list(lines=list(col=1:length(KEY),lty=1:length(KEY)),
          text=list(labels=paste("q=",levels(factor(res$raf)),sep="")),
          column=3)


# Illustrate Power
print(xyplot(pow0~GHR|factor(erate)*factor(n),group=factor(raf),
             data=res,type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power"))

# Illustrate Power (restricted to n=100)

print(xyplot(pow0~GHR|factor(erate),group=factor(raf),
             data=subset(res,n==ns[1]),
             type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power",
             sub=paste("n=",ns[1],", alpha=",round(unique(res$alpha),2))))


</code></pre>


</div>