<div class="container">

<table style="width: 100%;"><tr>
<td>survSNP-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Power Calculations for SNP Studies with Censored Outcomes
</h2>

<h3>Description</h3>

<p>This package can be used to conduct asymptotic and
empirical power and sample size calculations for Single-nucleotide
Polymorphism (SNP) association studies with right censored time to
event outcomes.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> survSNP</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 0.26</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2023-01-20</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
<td style="text-align: left;">
LazyLoad: </td>
<td style="text-align: left;"> yes</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>The functions sim.snp.expsurv.power and survSNP.power.table
can be used to conduct power and sample size calculations.
The package vignette serves as a tutorial for using this package. The
technical details are provided in the reference cited below.
It is highly recommended that the vignette along with this
are reviewed before using this package.
Currently, this package only supports additive risk models.
</p>


<h3>Author(s)</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox, Sin-Ho Jung and Chanhee Yi
</p>


<h3>References</h3>

<p>Kouros Owzar, Zhiguo Li, Nancy Cox and Sin-Ho Jung.
Power and Sample Size Calculations for SNP Association
Studies with Censored Time-to-Event Outcomes.
<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645">https://onlinelibrary.wiley.com/doi/full/10.1002/gepi.21645</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># See vignette for examples including details on the example
# considered below

results&lt;-sim.snp.expsurv.power(GHR=1.25, B=0, n=500, raf=0.1, erate=0.75, pilm=0.5, 
                               lm=1, model="additive", test="additive", alpha=0.05)
results[,c("n","erate","alpha","pow0")]


GHRs&lt;-seq(1.05,1.5,by=0.05)
ns&lt;-c(100,500,700)
rafs&lt;-c(0.1,0.3,0.5)
erates&lt;-c(0.5,0.7,0.9)
res&lt;-survSNP.power.table(GHRs,ns,rafs,erates,pilm=0.5,lm=1,model="additive",
                         test="additive",alpha=0.05)

# Create key for illustration
KEY=paste("q=",levels(factor(res$raf)),sep="")
KEY&lt;-list(lines=list(col=1:length(KEY),lty=1:length(KEY)),
          text=list(labels=paste("q=",levels(factor(res$raf)),sep="")),
          column=3)


# Illustrate Power
print(xyplot(pow0~GHR|factor(erate)*factor(n),group=factor(raf),
             data=res,type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power"))

# Illustrate Power (restricted to n=100)

print(xyplot(pow0~GHR|factor(erate),group=factor(raf),
             data=subset(res,n==ns[1]),
             type="l",lty=KEY$lines$lty,col=KEY$lines$col,
             key=KEY,
             xlab="Genotype Hazard Ratio",ylab="Power",
             sub=paste("n=",ns[1],", alpha=",round(unique(res$alpha),2))))

</code></pre>


</div>