<div class="container">

<table style="width: 100%;"><tr>
<td>draw.linear</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Linear synteny plot</h2>

<h3>Description</h3>

<p>Draw Linear Synteny Plots
</p>


<h3>Usage</h3>

<pre><code class="language-R">draw.linear(
  output,
  sizefile,
  ...,
  directory = NULL,
  fileformat = "png",
  colours = colours.default,
  w = 13,
  h = 5,
  opacity = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>output file name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sizefile</code></td>
<td>
<p>Chromosome Size file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>synteny files (any number of alignment files can be entered)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>directory</code></td>
<td>
<p>string containing file path to chosen directory to save image file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileformat</code></td>
<td>
<p>output file format specified using the format <code>fileformat = "pdf"</code> (the default is "png")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colours</code></td>
<td>
<p>concatenated string of chromosome IDs and assigned colours if desired using the format <code>colours = c("1" = "red", "2" = "blue", "3" = "green", "X" = "grey")</code> if the no colours are assigned default values will be used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w</code></td>
<td>
<p>width of output image using the format <code>w = 13</code> (default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>h</code></td>
<td>
<p>height of output image using the format <code>h = 5</code> (default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opacity</code></td>
<td>
<p>opacity of syntenic bands using the format <code>opacity = .5</code> (default)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function draws linear synteny plots.
</p>
<p>It requires:
</p>

<ol>
<li>
<p> The desired output file name;
</p>
</li>
<li>
<p> Tab separated file of all chromosome, scaffold, or contig lengths and the species identifier,
in order from first target species in the alignment files followed by the first reference species in the alignment files
– top of file – to the last target species and reference species in the alignment files – end of file.
in this format:
chromosome ID, chromosome length, species identifier
</p>
</li>
<li>
<p> files containing the syntenic blocks - one file per alignment, in order from first target/reference
(most recent species pairwise alignment in ancestral reconstruction data) alignment file
to last target/reference (ancestor pairwise alignment in ancestral reconstruction data) alignment file
following this format:
reference chromosome, reference start position, reference end position, target chromosome,
target start position, target end position, orient, reference species identifier, target species identifier
</p>
</li>
</ol>
<p>Please separate files by tab and ensure any species identifiers used between length and alignment files are matching (same identifiers and caseing)
</p>
<p>There are optional parameters for some customization of this function:
</p>

<ol>
<li>
<p> The format for saving the image i.e. png or pdf can be altered by inputting: <code>fileformat = "pdf"</code> (the default value is "png")
</p>
</li>
<li>
<p> The colour of the synteny bands can be altered by inputting a concatenated string of chromosome IDs with assigned colour values which can be found with R colour Pallette
e.g. <code>colours = c("1" = "red", "2" = "blue", "3" = "green","4" = "orange", "5" = "purple","X" = "grey")</code> if no colours are assigned default values will be used but colours MUST be assigned to all chromosomes
</p>
</li>
<li>
<p> The width of the image created can be changed by inputting: <code>w = 13</code> (default)
</p>
</li>
<li>
<p> The height of the image created can be changed by inputting: <code>h = 5</code> (default)
</p>
</li>
<li>
<p> The opacity of the ribbons can be changed by inputting: <code>opacity = .5</code> (default)
</p>
</li>
<li>
<p> The directory where the image file should be saved, as default the image is saved to temporary directory, change by inputting: <code>directory = "path/to/directory"</code>
</p>
</li>
</ol>
<p>The function works using the chromosome length file to order the Y axis and provide chromosome lengths to draw chromosome ideograms and the alignment files provides coordinates to draw the alignment bands between ideograms
</p>
<p>Example: <code>draw.linear("outputname", "example_lengths.txt", "example_alignment_1.txt", "example_alignment_2.txt", "example_alignment_3.txt", directory = "path/to/directory", fileformat = "pdf")</code>
</p>


<h3>Value</h3>

<p>An image file showing the linear comparison drawings
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Create objects containing file paths to external dataset
# (see vignette to follow examples with personal data)

length.file &lt;- system.file("extdata", "example_lengths.txt", package = "syntenyPlotteR")
file1 &lt;- system.file("extdata", "example_alignment_1.txt", package = "syntenyPlotteR")
file2 &lt;- system.file("extdata", "example_alignment_2.txt", package = "syntenyPlotteR")
file3 &lt;- system.file("extdata", "example_alignment_3.txt", package = "syntenyPlotteR")

# -----------------------------------------------------------------------------------

# Run draw.linear function
# To run example and save file to working directory
# add directory parameter and set working directory
# To run example with personal data see vignette

draw.linear("outputName", length.file, file1, file2, file3, fileformat = "pdf")
</code></pre>


</div>